<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Clustering Cusanovich et al (2018) data using topic proportions with K = 13 topics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Clustering Cusanovich et al (2018) data using topic proportions with K = 13 topics</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree98f5f51bfa7f5e2e6f91a6eaac19668dc687bdf6targetblank98f5f51a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/98f5f51bfa7f5e2e6f91a6eaac19668dc687bdf6" target="_blank">98f5f51</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree98f5f51bfa7f5e2e6f91a6eaac19668dc687bdf6targetblank98f5f51a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/98f5f51bfa7f5e2e6f91a6eaac19668dc687bdf6" target="_blank">98f5f51</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/plotly/Buenrostro_2018_Chen2019pipeline/

Untracked files:
    Untracked:  analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/fit_models_Cusanovich2018_tissues.sh

Unstaged changes:
    Modified:   analysis/assess_fits_Cusanovich2018.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   analysis/process_data_Cusanovich2018.Rmd
    Modified:   scripts/fit_poisson_nmf.sbatch
    Modified:   scripts/prefit_poisson_nmf.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/clusters_Cusanovich2018_k13.Rmd</code>) and HTML (<code>docs/clusters_Cusanovich2018_k13.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/98f5f51bfa7f5e2e6f91a6eaac19668dc687bdf6/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">98f5f51</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
<td>
minor change on the knit settings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/clusters_Cusanovich2018_k13.html" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/91cab4be6aa39a7bf027a7280cc7a1acc932e98e/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">91cab4b</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
<td>
added k-means results on PCA of topic proportions, and merged two clusters, and moved PCA plots to the end
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/f8ec9d713c982870cac2da589006d0058d4f973d/docs/clusters_Cusanovich2018_k13.html" target="_blank">f8ec9d7</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b30b5c6404eb0279fd9906ff365db595986ee4d1/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">b30b5c6</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
print outlier cells in c15
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/24c6d8b44c56fb081679e8bd901c75c4a9c268f6/docs/clusters_Cusanovich2018_k13.html" target="_blank">24c6d8b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/586af90be0b53f3c537e043570a352acaccf9b2e/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">586af90</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
wflow_publish(&quot;analysis/clusters_Cusanovich2018_k13.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/019f436a076fb4fa8cd57c16d939fd718823c649/docs/clusters_Cusanovich2018_k13.html" target="_blank">019f436</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/06efc2ba6f42acd94b7b871a496cd7c8efe1b5ca/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">06efc2b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
zoom into a few clusters
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/clusters_Cusanovich2018_k13.html" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/8db261b331997dba3f6b2d35570dc3a51f770c5a/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">8db261b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
wflow_publish(&quot;analysis/clusters_Cusanovich2018_k13.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/deead06d87929eb5b5f35792b463b3257513d2b4/docs/clusters_Cusanovich2018_k13.html" target="_blank">deead06</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/2af43a81eb6052e61a85589f8d34d87398611a7a/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">2af43a8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
update tissue colors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/06b9bc9fb151b60f9580d704c3ec85314ab4a120/docs/clusters_Cusanovich2018_k13.html" target="_blank">06b9bc9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/bc7e77076cfad5ea602e9aace950bf6a94af3250/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">bc7e770</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
update tissue colors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/6c97fa964f13b229dbb6321d162a33ae59fa52a0/docs/clusters_Cusanovich2018_k13.html" target="_blank">6c97fa9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ab321de6a609e3a8d406a6079e1ce26d269082ae/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">ab321de</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
<td>
refine kmeans clusters and add cell label distributions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/e57dfd4474bd37d8c38a83cc50cefd969c3e704b/docs/clusters_Cusanovich2018_k13.html" target="_blank">e57dfd4</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a88649de20247ee325ab0d6ca22a45220d366e71/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">a88649d</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
wflow_publish(&quot;analysis/clusters_Cusanovich2018_k13.Rmd&quot;)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/137c252303ee779516da145a637555b8ff6b8be4/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">137c252</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
set toc = yes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/4c0ab9566a79fb17ef309d48d98c39fce0680650/docs/clusters_Cusanovich2018_k13.html" target="_blank">4c0ab95</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/9d31a9fba6cedf2b9b3ee1072364767c4156ad73/docs/clusters_Cusanovich2018_k13.html" target="_blank">9d31a9f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c00f12be96ead245dc0fdf28f03d7af246a06d42/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">c00f12b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
update colors and interpret clusters with tissue labels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/32ed5aea68036d63ab78e7d417a8067660328456/docs/clusters_Cusanovich2018_k13.html" target="_blank">32ed5ae</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/82292d5a2fb36d3db7da2959b1f5707af8b70dd3/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">82292d5</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
update colors and interpret clusters with tissue labels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/b8a48b9eaed38e1ad24382a36f106baee2bcca05/docs/clusters_Cusanovich2018_k13.html" target="_blank">b8a48b9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a4daf6de54a5dd39090a26672f73e5ae5d8a7601/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">a4daf6d</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
clustering with k = 13 topics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ac8ca65b1c246ccc29d024c45489bd18e1682210/docs/clusters_Cusanovich2018_k13.html" target="_blank">ac8ca65</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/98920edc38fb52a44e5e232aacfee04fd46847dc/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">98920ed</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
clustering with k = 13 topics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/clusters_Cusanovich2018_k13.html" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f8bea9604d079077c2a13d1d705864c42c024a9c/analysis/clusters_Cusanovich2018_k13.Rmd" target="_blank">f8bea96</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
clustering with k = 13 topics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we explore the structure in the Cusanovich <em>et al</em> (2018) ATAC-seq data inferred from the multinomial topic model with <span class="math inline">\(k = 13\)</span>.</p>
<p>Load packages and some functions used in this analysis.</p>
<pre class="r"><code>library(Matrix)
library(dplyr)
library(ggplot2)
library(cowplot)
library(fastTopics)
source(&quot;code/plots.R&quot;)</code></pre>
<p>Load the data. The counts are no longer needed at this stage of the analysis.</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
rm(counts)</code></pre>
<p>Load the model fit</p>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<p>About the samples: The study measured single cell chromatin accessibility for 17 samples spanning 13 different tissues in 8-week old mice.</p>
<pre class="r"><code>cat(nrow(samples), &quot;samples (cells). \n&quot;)</code></pre>
<pre><code># 81173 samples (cells).</code></pre>
<p>Tissues:</p>
<pre class="r"><code>samples$tissue &lt;- as.factor(samples$tissue)
cat(length(levels(samples$tissue)), &quot;tissues. \n&quot;)
# table(samples$tissue)

colors_tissues &lt;- c(&quot;darkblue&quot;,    # BoneMarrow
                    &quot;gray30&quot;,      # Cerebellum
                    &quot;red&quot;,         # Heart
                    &quot;springgreen&quot;, # Kidney
                    &quot;brown&quot;,       # LargeIntestine
                    &quot;purple&quot;,      # Liver
                    &quot;deepskyblue&quot;, # Lung
                    &quot;black&quot;,       # PreFrontalCortex
                    &quot;darkgreen&quot;,   # SmallIntestine
                    &quot;plum&quot;,        # Spleen
                    &quot;gold&quot;,        # Testes
                    &quot;darkorange&quot;,  # Thymus
                    &quot;gray&quot;)        # WholeBrain</code></pre>
<pre><code># 13 tissues.</code></pre>
<p>Cell types labels:</p>
<pre class="r"><code>samples$cell_label &lt;- as.factor(samples$cell_label)
cat(length(levels(samples$cell_label)), &quot;cell types \n&quot;)
# table(samples$cell_label)</code></pre>
<pre><code># 40 cell types</code></pre>
<div id="structure-plots" class="section level2">
<h2>Structure plots</h2>
<p>The structure plots below summarize the topic proportions in the samples grouped by different tissues.</p>
</div>
<div id="visualize-by-structure-plot-grouped-by-tissues" class="section level2">
<h2>Visualize by Structure plot grouped by tissues</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
samples$tissue &lt;- as.factor(samples$tissue)

rows &lt;- sample(nrow(fit$L),4000)

p.structure.1 &lt;- structure_plot(select(fit_multinom,loadings = rows),
                     grouping = samples[rows, &quot;tissue&quot;],n = Inf,gap = 40,
                     perplexity = 50,topics = 1:13,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)

print(p.structure.1)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-tissues-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-tissues-1">
Past versions of structure-plot-tissues-1.png
</button>
</p>
<div id="fig-structure-plot-tissues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-tissues-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="perform-k-means-clustering-on-topic-proportions" class="section level2">
<h2>Perform k-means clustering on topic proportions</h2>
<p>Define clusters using k-means, and then create structure plot based on the clusters from k-means.</p>
<p>k-means clustering (using 15 clusters) on topic proportions</p>
<pre class="r"><code>set.seed(10)
clusters &lt;- factor(kmeans(fit_multinom$L,centers = 15,iter.max = 100)$cluster)
summary(clusters)

rows &lt;- sample(nrow(fit$L),4000)

p.structure.kmeans &lt;- structure_plot(select(fit_multinom,loadings = rows),
                                        grouping = clusters[rows],n = Inf,gap = 20,
                                        perplexity = 50,topics = 1:13,colors = colors_topics,
                                        num_threads = 4,verbose = FALSE)
print(p.structure.kmeans)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-kmeans-1-1">
Past versions of structure-plot-kmeans-1-1.png
</button>
</p>
<div id="fig-structure-plot-kmeans-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/6c97fa964f13b229dbb6321d162a33ae59fa52a0/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">6c97fa9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/9d31a9fba6cedf2b9b3ee1072364767c4156ad73/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">9d31a9f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/32ed5aea68036d63ab78e7d417a8067660328456/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">32ed5ae</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-1-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>#     1     2     3     4     5     6     7     8     9    10    11    12    13 
#  2740  3868  3775  3879  3645  3651 12037  2955  6045 12026  6068  3582  9319 
#    14    15 
#  3411  4172</code></pre>
<p>First do PCA on topic proportions and then do k-means clustering (using 15 clusters). Results are the same as the results from running k-means directly on the topic proportions.</p>
<pre class="r"><code>set.seed(10)
pca &lt;- prcomp(fit_multinom$L)$x
clusters2 &lt;- factor(kmeans(pca,centers = 15,iter.max = 100)$cluster)
summary(clusters2)

rows &lt;- sample(nrow(fit$L),4000)

p.structure.kmeans &lt;- structure_plot(select(fit_multinom,loadings = rows),
                                        grouping = clusters2[rows],n = Inf,gap = 20,
                                        perplexity = 50,topics = 1:13,colors = colors_topics,
                                        num_threads = 4,verbose = FALSE)
print(p.structure.kmeans)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-pca-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-kmeans-pca-1-1">
Past versions of structure-plot-kmeans-pca-1-1.png
</button>
</p>
<div id="fig-structure-plot-kmeans-pca-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-pca-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>length(which(clusters != clusters2))</code></pre>
<pre><code>#     1     2     3     4     5     6     7     8     9    10    11    12    13 
#  2740  3868  3775  3879  3645  3651 12037  2955  6045 12026  6068  3582  9319 
#    14    15 
#  3411  4172 
# [1] 0</code></pre>
<p>Refine k-means clusters:</p>
<ul>
<li>Split cluster 10 into two clusters.</li>
<li>merge cluster 4 and 9.</li>
</ul>
<p>Structure plot based on refined clusters</p>
<pre class="r"><code>set.seed(10)
clusters.refined &lt;- as.numeric(clusters)
clusters.refined[which(clusters == 9)] &lt;- 4

# clusters.refined[which(clusters == 10 &amp; fit_multinom$L[,8] &gt;= 0.2)] &lt;- 16
# clusters.refined[which(clusters == 10 &amp; fit_multinom$L[,8] &lt; 0.2)] &lt;- 17

clusters.refined[which(clusters == 10)] &lt;- kmeans(fit_multinom$L[which(clusters == 10), ],centers = 2)$cluster+15
clusters.refined &lt;- factor(clusters.refined, labels = paste0(&quot;c&quot;, c(1:length(unique(clusters.refined)))))

samples$cluster_kmeans &lt;- clusters.refined

rows &lt;- sample(nrow(fit$L),4000)

p.structure.kmeans.refined &lt;- structure_plot(select(fit_multinom,loadings = rows),
                                             grouping = clusters.refined[rows],n = Inf,gap = 40,
                                             perplexity = 50,topics = 1:13,colors = colors_topics,
                                             num_threads = 4,verbose = FALSE)
print(p.structure.kmeans.refined)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-refined-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-kmeans-refined-1-1">
Past versions of structure-plot-kmeans-refined-1-1.png
</button>
</p>
<div id="fig-structure-plot-kmeans-refined-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-refined-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-refined-1-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/6c97fa964f13b229dbb6321d162a33ae59fa52a0/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-kmeans-refined-1-1.png" target="_blank">6c97fa9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Save the clustering results to an RDS file</p>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
saveRDS(samples, paste0(out.dir, &quot;/samples-clustering-Cusanovich2018.rds&quot;))

samples &lt;- readRDS(&quot;output/samples-clustering-Cusanovich2018.rds&quot;)
clusters.refined &lt;- samples$cluster_kmeans
print(table(samples$cluster_kmeans))</code></pre>
</div>
<div id="plot-the-distribution-of-tissues-in-each-cluster" class="section level2">
<h2>Plot the distribution of tissues in each cluster</h2>
<p>Tissue composition in each cluster:</p>
<pre class="r"><code>library(plyr);library(dplyr)
library(RColorBrewer)

freq_cluster_tissue &lt;- count(samples, vars=c(&quot;cluster_kmeans&quot;,&quot;tissue&quot;)) 

# stacked barplot for the counts of tissues by clusters
p.barplot &lt;- ggplot(freq_cluster_tissue, aes(fill=tissue, y=freq, x=cluster_kmeans)) + 
  geom_bar(position=&quot;fill&quot;, stat=&quot;identity&quot;) +
  theme_classic() + xlab(&quot;Cluster&quot;) + ylab(&quot;Proportion of cells&quot;) +
  scale_fill_manual(values = colors_tissues) +
  guides(fill=guide_legend(ncol=2)) +
  theme(
  legend.title = element_text(size = 10),
  legend.text = element_text(size = 8)
  )
print(p.barplot)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/freq-cluster-tissue-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-freq-cluster-tissue-1">
Past versions of freq-cluster-tissue-1.png
</button>
</p>
<div id="fig-freq-cluster-tissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/freq-cluster-tissue-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b8a48b9eaed38e1ad24382a36f106baee2bcca05/docs/figure/clusters_Cusanovich2018_k13.Rmd/freq-cluster-tissue-1.png" target="_blank">b8a48b9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/freq-cluster-tissue-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see a few clusters are tissue specific:</p>
<ul>
<li>cluster c3 is heart specific;</li>
<li>cluster c8 is kidney specific;</li>
<li>cluster c15 is liver specific;</li>
<li>cluster c1 is primarily bone marrow;</li>
<li>cluster c11 is primarily thymus.</li>
</ul>
<p>Some clusters are combinations of related tissues:</p>
<ul>
<li>cluster c4 is half lung and half spleen;</li>
<li>cluster c5 and c14 are mainly from pre-frontal cortex, whole brain (and cerebellum) -- all neuron related.</li>
<li>cluster c14 is also from whole brain and cerebellum.</li>
<li>cluster c10 is mainly from Kidney, LargeIntestine, and Lung.</li>
</ul>
<p>Some clusters are more heterogeneous mixtures of different tissues.</p>
<p>Distribution of tissue labels by cluster.</p>
<pre class="r"><code>freq_table_cluster_tissue &lt;- with(samples,table(tissue,cluster_kmeans))
print(freq_table_cluster_tissue)</code></pre>
<pre><code>#                   cluster_kmeans
# tissue               c1   c2   c3   c4   c5   c6   c7   c8   c9  c10  c11  c12
#   BoneMarrow       2455   15    0 1675    0   33 1060    0 3063    0  102    0
#   Cerebellum          0   51    0   37  525   44  191    0   42    0    0   15
#   Heart               0 1374 3768  194    3 1225  803    5  259    3   13    3
#   Kidney              0  757    0   95    0  402   98 2950  280 1614   19  215
#   LargeIntestine      0   73    1  178    4  665 2778    0  172  441   94 2680
#   Liver               0  268    0  136    0   84   31    0   50    1   34    4
#   Lung                2 1093    5 4002    2  659 1010    0  465 1521  773  434
#   PreFrontalCortex    0   75    0  215 1169   48  118    0   49    0    1   19
#   SmallIntestine      0   22    0   84    0  341 2112    0 1457    1   27   18
#   Spleen             43    3    0 2937    0    5   24    0  116    0  892    0
#   Testes            240   19    0    3    0   37 2379    0   38    0    0    4
#   Thymus              0    2    1  172    0    9   32    0   30    1 7363    7
#   WholeBrain          0  116    0  196 1942   99 1401    0   47    0    1   12
#                   cluster_kmeans
# tissue              c13  c14  c15
#   BoneMarrow          0    0    0
#   Cerebellum       1177  196    0
#   Heart               0    0    0
#   Kidney              0    0    1
#   LargeIntestine      0    0    0
#   Liver               1    0 5558
#   Lung                2    2   26
#   PreFrontalCortex    4 4261    0
#   SmallIntestine      0    0   15
#   Spleen              0    0    0
#   Testes              0    0    3
#   Thymus              0    0    0
#   WholeBrain       2988 1964    0</code></pre>
<p>Distribution of cell labels by cluster.</p>
<pre class="r"><code>freq_table_cluster_celllabel &lt;- with(samples,table(cell_label,cluster_kmeans))
print(freq_table_cluster_celllabel)</code></pre>
<pre><code>#                             cluster_kmeans
# cell_label                     c1   c2   c3   c4   c5   c6   c7   c8   c9  c10
#   Activated B cells             0    0    0  477    0    0    0    0   23    0
#   Alveolar macrophages          0    0    0  507    0    2   16    0   34    0
#   Astrocytes                    0    0    0    0 1574   11   58    0   20    0
#   B cells                       0    0    0 5086    0    3  648    0   34    0
#   Cardiomyocytes                0   15 3773    1    0   42  169    0   76    0
#   Cerebellar granule cells      0    0    0    9    0    3  306    0   19    0
#   Collecting duct               0    0    0    0    0   10    0    0    5  124
#   Collisions                    6    2    0  340  423   15  166    0   64    0
#   DCT/CD                        0    0    0    0    0    1    0    1    6  441
#   Dendritic cells               1    0    0  840    0    2   12    0  102    0
#   Distal convoluted tubule      0    0    0    0    0    0    0    0    6  301
#   Endothelial I (glomerular)    0  386    0    0    0  158    0    0    8    0
#   Endothelial I cells           0  512    0    0    0  438    1    0    1    0
#   Endothelial II cells          0 2405    0   15    0  574    7    0   15    0
#   Enterocytes                   0    0    0    0    1   11 1585    0   54  442
#   Erythroblasts              2420    0    0   37    0    0   79    0  275    0
#   Ex. neurons CPN               0    0    0    0    0    0   10    0    0    0
#   Ex. neurons CThPN             0    0    0    0    0    0   91    0    0    0
#   Ex. neurons SCPN              0    0    0    0    0    0   40    0   32    0
#   Hematopoietic progenitors     0    0    0   26    0    1  899    0 2499    0
#   Hepatocytes                   0    0    0    2    0    5   11    0   44    0
#   Immature B cells              0    0    0  353    0    0    0    0  218    0
#   Inhibitory neurons            0    0    0    3   68   27  754    0    8    0
#   Loop of henle                 0    1    0    0    0    8    2    1   11  718
#   Macrophages                   0    0    0  464    0   14  191    0   42    0
#   Microglia                     0    1    0  385    1    3   31    0    1    0
#   Monocytes                    69    0    0 1019    0    1   50    0   32    0
#   NK cells                      0    0    0  231    0    1    2    0   16    0
#   Oligodendrocytes              0    0    0    1 1506    1   30    0   17    0
#   Podocytes                     0  369    0    0    0  113    0    0   16    0
#   Proximal tubule               0    1    0    1    0   16   33 2361  132    7
#   Proximal tubule S3            0    0    0    0    0    0    2  592    0    0
#   Purkinje cells                0    0    0    0    1    2   50    0    0    0
#   Regulatory T cells            3    0    0   17    0    1    7    0    4    0
#   SOM+ Interneurons             0    0    0    0    7    8   66    0    5    0
#   Sperm                       240    0    0    0    0    1 1814    0   32    0
#   T cells                       1    1    0   35    1    7  104    0   50    0
#   Type I pneumocytes            0    0    0    1    0    3    0    0   27 1415
#   Type II pneumocytes           0    8    0    0    0   14    0    0    1   66
#   Unknown                       0  167    2   74   63 2155 4803    0 2139   68
#                             cluster_kmeans
# cell_label                    c11  c12  c13  c14  c15
#   Activated B cells             0    0    0    0    0
#   Alveolar macrophages          0    0    0    0    0
#   Astrocytes                    0    0    0    3    0
#   B cells                       0    1    0    0    0
#   Cardiomyocytes                0    0    0    0    0
#   Cerebellar granule cells      0    0 3762    0    0
#   Collecting duct               0   25    0    0    0
#   Collisions                   11    8   97   86    0
#   DCT/CD                        0   57    0    0    0
#   Dendritic cells               0    1    0    0    0
#   Distal convoluted tubule      0   12    0    0    0
#   Endothelial I (glomerular)    0    0    0    0    0
#   Endothelial I cells           0    0    0    0    0
#   Endothelial II cells          1    1    1    0    0
#   Enterocytes                   0 2690    0    0    0
#   Erythroblasts                 0    0    0    0    0
#   Ex. neurons CPN               0    0    0 1822    0
#   Ex. neurons CThPN             0    0    0 1449    0
#   Ex. neurons SCPN              0    0    0 1394    0
#   Hematopoietic progenitors     0    0    0    0    0
#   Hepatocytes                   0    0    0    0 5602
#   Immature B cells              0    0    0    0    0
#   Inhibitory neurons            0    0    3  965    0
#   Loop of henle                 0   74    0    0    0
#   Macrophages                   0    0    0    0    0
#   Microglia                     0    0    0    0    0
#   Monocytes                     2    0    0    0    0
#   NK cells                     71    0    0    0    0
#   Oligodendrocytes              0    0    0    3    0
#   Podocytes                     0    0    0    0    0
#   Proximal tubule               0   19    0    0    0
#   Proximal tubule S3            0    0    0    0    0
#   Purkinje cells                0    0  262    5    0
#   Regulatory T cells          475    0    0    0    0
#   SOM+ Interneurons             0    0    1  466    0
#   Sperm                         0    2    0    0    0
#   T cells                    8753    2    0    0    0
#   Type I pneumocytes            6  170    0    0    0
#   Type II pneumocytes           0   68    0    0    0
#   Unknown                       0  281   46  230    1</code></pre>
<div id="zoom-into-a-few-clusters" class="section level3">
<h3>zoom into a few clusters:</h3>
<p>Structure plot for c2 cluster: Group samples by tissue labels first, then by cell labels</p>
<pre class="r"><code>rows.c2 &lt;- which(samples$cluster_kmeans == &quot;c2&quot;)

tissue_labels_cluster &lt;- as.factor(as.character(samples[rows.c2, &quot;tissue&quot;]))
cell_labels_cluster &lt;- as.factor(as.character(samples[rows.c2, &quot;cell_label&quot;]))

sort(table(tissue_labels_cluster), decreasing = T)
sort(table(cell_labels_cluster), decreasing = T)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = rows.c2),
                     grouping = tissue_labels_cluster,
                     rows = order(cell_labels_cluster),
                     n = Inf,gap = 40,
                     perplexity = 50,topics = 1:13,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)
print(p.structure)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c2-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-c2-1-1">
Past versions of structure-plot-c2-1-1.png
</button>
</p>
<div id="fig-structure-plot-c2-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c2-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/24c6d8b44c56fb081679e8bd901c75c4a9c268f6/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c2-1-1.png" target="_blank">24c6d8b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c2-1-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># tissue_labels_cluster
#            Heart             Lung           Kidney            Liver 
#             1374             1093              757              268 
#       WholeBrain PreFrontalCortex   LargeIntestine       Cerebellum 
#              116               75               73               51 
#   SmallIntestine           Testes       BoneMarrow           Spleen 
#               22               19               15                3 
#           Thymus 
#                2 
# cell_labels_cluster
#       Endothelial II cells        Endothelial I cells 
#                       2405                        512 
# Endothelial I (glomerular)                  Podocytes 
#                        386                        369 
#                    Unknown             Cardiomyocytes 
#                        167                         15 
#        Type II pneumocytes                 Collisions 
#                          8                          2 
#              Loop of henle                  Microglia 
#                          1                          1 
#            Proximal tubule                    T cells 
#                          1                          1</code></pre>
<pre class="r"><code>breaks &lt;- c(0,1,5,10,100,Inf)

fit.c2 &lt;- select(fit_multinom,loadings = rows.c2)
p.pca.c2 &lt;- pca_plot(fit.c2,fill = &quot;none&quot;)
p.pca.hexbin.c2 &lt;- pca_hexbin_plot(fit.c2,breaks = breaks) + guides(fill = &quot;none&quot;)
plot_grid(p.pca.c2, p.pca.hexbin.c2)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-c2-1">
Past versions of pca-c2-1.png
</button>
</p>
<div id="fig-pca-c2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/019f436a076fb4fa8cd57c16d939fd718823c649/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-1.png" target="_blank">019f436</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The variation in PCs 1 and 2 is mostly produced by topics 3,13</p>
<pre class="r"><code>p.pca.c2.topics &lt;- pca_plot(fit.c2,k = c(3,13))
print(p.pca.c2.topics)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-topics-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-c2-topics-1">
Past versions of pca-c2-topics-1.png
</button>
</p>
<div id="fig-pca-c2-topics-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f8ec9d713c982870cac2da589006d0058d4f973d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-topics-1.png" target="_blank">f8ec9d7</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/019f436a076fb4fa8cd57c16d939fd718823c649/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c2-topics-1.png" target="_blank">019f436</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colors_celllabels_c2 &lt;- c(&quot;firebrick&quot;,&quot;dodgerblue&quot;,&quot;forestgreen&quot;,
                           &quot;darkmagenta&quot;,&quot;darkorange&quot;,&quot;gold&quot;,&quot;darkblue&quot;,
                           &quot;peru&quot;,&quot;greenyellow&quot;,&quot;olivedrab&quot;, 
                           &quot;darkgray&quot;, &quot;black&quot;)

p.pca.c2.tissue &lt;- labeled_pca_plot(fit.c2,1:2,samples[rows.c2, &quot;tissue&quot;],font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca.c2.cell &lt;- labeled_pca_plot(fit.c2,1:2,samples[rows.c2, &quot;cell_label&quot;],font_size = 7,
                        colors = colors_celllabels_c2, legend_label = &quot;cell_label&quot;)
plot_grid(p.pca.c2.tissue,p.pca.c2.cell,rel_widths = c(8,9))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-c2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-c2-1">
Past versions of pca-labels-c2-1.png
</button>
</p>
<div id="fig-pca-labels-c2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-c2-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-c2-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="zoom-into-cluster-c14" class="section level4">
<h4>zoom into cluster c14</h4>
<p>Structure plot of c14 cluster: Group samples by tissue labels first, then by cell labels</p>
<pre class="r"><code>rows.c14 &lt;- which(samples$cluster_kmeans == &quot;c14&quot;)

tissue_labels_cluster &lt;- as.factor(as.character(samples[rows.c14, &quot;tissue&quot;]))
cell_labels_cluster &lt;- as.factor(as.character(samples[rows.c14, &quot;cell_label&quot;]))

sort(table(tissue_labels_cluster), decreasing = T)
sort(table(cell_labels_cluster), decreasing = T)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = rows.c14),
                     grouping = tissue_labels_cluster,
                     rows = order(cell_labels_cluster),
                     n = Inf,gap = 40,
                     perplexity = 50,topics = 1:13,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)
print(p.structure)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c14-1-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-c14-1-1">
Past versions of structure-plot-c14-1-1.png
</button>
</p>
<div id="fig-structure-plot-c14-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/structure-plot-c14-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># tissue_labels_cluster
# PreFrontalCortex       WholeBrain       Cerebellum             Lung 
#             4261             1964              196                2 
# cell_labels_cluster
#    Ex. neurons CPN  Ex. neurons CThPN   Ex. neurons SCPN Inhibitory neurons 
#               1822               1449               1394                965 
#  SOM+ Interneurons            Unknown         Collisions     Purkinje cells 
#                466                230                 86                  5 
#         Astrocytes   Oligodendrocytes 
#                  3                  3</code></pre>
<pre class="r"><code>breaks &lt;- c(0,1,5,10,100,Inf)

fit.c14 &lt;- select(fit_multinom,loadings = rows.c14)
p.pca.c14 &lt;- pca_plot(fit.c14,fill = &quot;none&quot;)
p.pca.hexbin.c14 &lt;- pca_hexbin_plot(fit.c14,breaks = breaks) + guides(fill = &quot;none&quot;)
plot_grid(p.pca.c14, p.pca.hexbin.c14)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-c14-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-c14-1">
Past versions of pca-c14-1.png
</button>
</p>
<div id="fig-pca-c14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c14-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The variation in PCs 1 and 2 is mostly produced by topics 9,10,11</p>
<pre class="r"><code>p.pca.c14.topics &lt;- pca_plot(fit.c14,k = c(9,10,11))
print(p.pca.c14.topics)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-c14-topics-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-c14-topics-1">
Past versions of pca-c14-topics-1.png
</button>
</p>
<div id="fig-pca-c14-topics-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-c14-topics-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>colors_celllabels_c14 &lt;- c(&quot;firebrick&quot;,&quot;dodgerblue&quot;,&quot;forestgreen&quot;,
                           &quot;darkmagenta&quot;,&quot;darkorange&quot;,&quot;gold&quot;,&quot;darkblue&quot;,
                           &quot;peru&quot;,&quot;greenyellow&quot;,&quot;black&quot;)

p.pca.c14.tissue &lt;- labeled_pca_plot(fit.c14,1:2,samples[rows.c14, &quot;tissue&quot;],font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca.c14.cell &lt;- labeled_pca_plot(fit.c14,1:2,samples[rows.c14, &quot;cell_label&quot;],font_size = 7,
                        colors = colors_celllabels_c14, legend_label = &quot;cell_label&quot;)
plot_grid(p.pca.c14.tissue,p.pca.c14.cell,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-c14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-c14-1">
Past versions of pca-labels-c14-1.png
</button>
</p>
<div id="fig-pca-labels-c14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-c14-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>print(samples[intersect(rows.c14, which(samples$tissue == &quot;Lung&quot;)),])</code></pre>
<pre><code>#                                       cell tissue tissue.replicate cluster
# 22109 TCCGCGAATAGGTAACTTACTGAGCGACGGCTCTGA   Lung      Lung2_62216      15
# 61277 TCTCGCGCTATTGCTGGACCTCCGACGGGTACTGAC   Lung      Lung2_62216       5
#       subset_cluster   tsne_1    tsne_2 subset_tsne1 subset_tsne2
# 22109              2 -1.07335 -12.29793    -8.494056    -4.764539
# 61277              2 -1.77437 -23.62355     4.220333     2.583797
#                          id         cell_label cluster_kmeans
# 22109 clusters_15.cluster_2 Inhibitory neurons            c14
# 61277  clusters_5.cluster_2   Ex. neurons SCPN            c14</code></pre>
</div>
</div>
</div>
<div id="pca-plots" class="section level2">
<h2>PCA plots</h2>
<p>Plot PCs of the topic proportions.</p>
<pre class="r"><code>p.pca1.1 &lt;- pca_plot(fit_multinom,pcs = 1:2,fill = &quot;none&quot;)
p.pca1.2 &lt;- pca_plot(fit_multinom,pcs = 3:4,fill = &quot;none&quot;)
p.pca1.3 &lt;- pca_plot(fit_multinom,pcs = 5:6,fill = &quot;none&quot;)
p.pca1.4 &lt;- pca_plot(fit_multinom,pcs = 7:8,fill = &quot;none&quot;)
p.pca1.5 &lt;- pca_plot(fit_multinom,pcs = 9:10,fill = &quot;none&quot;)
p.pca1.6 &lt;- pca_plot(fit_multinom,pcs = 11:12,fill = &quot;none&quot;)

plot_grid(p.pca1.1,p.pca1.2,p.pca1.3,p.pca1.4,p.pca1.5,p.pca1.6)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-k13-1-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-k13-1-1">
Past versions of pca-k13-1-1.png
</button>
</p>
<div id="fig-pca-k13-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-k13-1-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Some of the structure is more evident from “hexbin” plots showing the density of the points.</p>
<pre class="r"><code>breaks &lt;- c(0,1,5,10,100,Inf)
p.pca2.1 &lt;- pca_hexbin_plot(fit_multinom, pcs = 1:2, breaks = breaks) + guides(fill = &quot;none&quot;)
p.pca2.2 &lt;- pca_hexbin_plot(fit_multinom, pcs = 3:4, breaks = breaks) + guides(fill = &quot;none&quot;)
p.pca2.3 &lt;- pca_hexbin_plot(fit_multinom, pcs = 5:6, breaks = breaks) + guides(fill = &quot;none&quot;)
p.pca2.4 &lt;- pca_hexbin_plot(fit_multinom, pcs = 7:8, breaks = breaks) + guides(fill = &quot;none&quot;)
p.pca2.5 &lt;- pca_hexbin_plot(fit_multinom, pcs = 9:10, breaks = breaks) + guides(fill = &quot;none&quot;)
p.pca2.6 &lt;- pca_hexbin_plot(fit_multinom, pcs = 11:12, breaks = breaks) + guides(fill = &quot;none&quot;)

plot_grid(p.pca2.1,p.pca2.2,p.pca2.3,p.pca2.4,p.pca2.5,p.pca2.6)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-k13-2-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-k13-2-1">
Past versions of pca-k13-2-1.png
</button>
</p>
<div id="fig-pca-k13-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-k13-2-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="layer-the-tissue-and-cell-labels-onto-the-pc-plots" class="section level2">
<h2>Layer the tissue and cell labels onto the PC plots</h2>
<p>PCs 1 and 2:</p>
<pre class="r"><code>p.pca3.1 &lt;- labeled_pca_plot(fit,1:2,samples$tissue,font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca3.2 &lt;- labeled_pca_plot(fit,1:2,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.1,p.pca3.2,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-1-1">
Past versions of pca-labels-k13-1-1.png
</button>
</p>
<div id="fig-pca-labels-k13-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-1-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-1-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCs 3 and 4:</p>
<pre class="r"><code>p.pca3.3 &lt;- labeled_pca_plot(fit,3:4,samples$tissue,font_size = 7,
                        colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca3.4 &lt;- labeled_pca_plot(fit,3:4,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.3,p.pca3.4,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-2-1">
Past versions of pca-labels-k13-2-1.png
</button>
</p>
<div id="fig-pca-labels-k13-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-2-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-2-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-2-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCs 5 and 6:</p>
<pre class="r"><code>p.pca3.5 &lt;- labeled_pca_plot(fit,5:6,samples$tissue,font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca3.6 &lt;- labeled_pca_plot(fit,5:6,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.5,p.pca3.6,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-3-1">
Past versions of pca-labels-k13-3-1.png
</button>
</p>
<div id="fig-pca-labels-k13-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-3-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-3-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-3-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCs 7 and 8:</p>
<pre class="r"><code>p.pca3.7 &lt;- labeled_pca_plot(fit,7:8,samples$tissue,font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca3.8 &lt;- labeled_pca_plot(fit,7:8,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.7,p.pca3.8,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-4-1">
Past versions of pca-labels-k13-4-1.png
</button>
</p>
<div id="fig-pca-labels-k13-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-4-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-4-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-4-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCs 9 and 10:</p>
<pre class="r"><code>p.pca3.9 &lt;- labeled_pca_plot(fit,9:10,samples$tissue,font_size = 7,
                       colors = colors_tissues, legend_label = &quot;tissue&quot;)
p.pca3.10 &lt;- labeled_pca_plot(fit,9:10,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.9,p.pca3.10,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-5-1">
Past versions of pca-labels-k13-5-1.png
</button>
</p>
<div id="fig-pca-labels-k13-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-5-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/96392bfdfa9ac41a77985ae18763ba93a6d2f497/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-5-1.png" target="_blank">96392bf</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-5-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCs 11 and 12:</p>
<pre class="r"><code>p.pca3.11 &lt;- labeled_pca_plot(fit,11:12,samples$tissue,font_size = 7,
                       legend_label = &quot;tissue&quot;)
p.pca3.12 &lt;- labeled_pca_plot(fit,11:12,samples$cell_label,font_size = 7,
                        legend_label = &quot;cell_label&quot;)
plot_grid(p.pca3.11,p.pca3.12,rel_widths = c(8,11))</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-labels-k13-6-1">
Past versions of pca-labels-k13-6-1.png
</button>
</p>
<div id="fig-pca-labels-k13-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-labels-k13-6-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The clusters defined by k-means on topic proportions reasonably identify the clusters shown in the PCA hexbin plots (below).</p>
<pre class="r"><code>p.pca.4.1 &lt;- labeled_pca_plot(fit,1:2,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
p.pca.4.2 &lt;- labeled_pca_plot(fit,3:4,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
p.pca.4.3 &lt;- labeled_pca_plot(fit,5:6,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
p.pca.4.4 &lt;- labeled_pca_plot(fit,7:8,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
p.pca.4.5 &lt;- labeled_pca_plot(fit,9:10,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
p.pca.4.6 &lt;- labeled_pca_plot(fit,11:12,samples$cluster_kmeans,font_size = 7,
                       legend_label = &quot;cluster_kmeans&quot;)
plot_grid(p.pca.4.1,p.pca.4.2,p.pca.4.3,p.pca.4.4,p.pca.4.5,p.pca.4.6)</code></pre>
<p><img src="figure/clusters_Cusanovich2018_k13.Rmd/pca-kmeans-1-1.png" width="1080" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-kmeans-1-1">
Past versions of pca-kmeans-1-1.png
</button>
</p>
<div id="fig-pca-kmeans-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3770d7e15ee6d57512cd3d1ea0165787ad08736e/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-kmeans-1-1.png" target="_blank">3770d7e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/6c97fa964f13b229dbb6321d162a33ae59fa52a0/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-kmeans-1-1.png" target="_blank">6c97fa9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/32ed5aea68036d63ab78e7d417a8067660328456/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-kmeans-1-1.png" target="_blank">32ed5ae</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a38788b2e6ab19366c8d3bc9e45dc9844fde044d/docs/figure/clusters_Cusanovich2018_k13.Rmd/pca-kmeans-1-1.png" target="_blank">a38788b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] RColorBrewer_1.1-2 plyr_1.8.6         fastTopics_0.4-29  cowplot_1.1.1     
# [5] ggplot2_3.3.3      dplyr_1.0.3        Matrix_1.2-18      workflowr_1.6.2   
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.1      Rcpp_1.0.6         lattice_0.20-41    tidyr_1.1.2       
#  [5] prettyunits_1.1.1  rprojroot_2.0.2    digest_0.6.27      R6_2.5.0          
#  [9] MatrixModels_0.4-1 evaluate_0.14      coda_0.19-4        httr_1.4.2        
# [13] pillar_1.4.7       rlang_0.4.10       progress_1.2.2     lazyeval_0.2.2    
# [17] data.table_1.13.6  irlba_2.3.3        SparseM_1.78       hexbin_1.28.1     
# [21] whisker_0.4        rmarkdown_2.6      labeling_0.4.2     Rtsne_0.15        
# [25] stringr_1.4.0      htmlwidgets_1.5.3  munsell_0.5.0      compiler_3.6.1    
# [29] httpuv_1.5.4       xfun_0.19          pkgconfig_2.0.3    mcmc_0.9-7        
# [33] htmltools_0.5.1.1  tidyselect_1.1.0   tibble_3.0.6       quadprog_1.5-8    
# [37] matrixStats_0.58.0 viridisLite_0.3.0  crayon_1.4.0       conquer_1.0.2     
# [41] withr_2.4.1        later_1.1.0.1      MASS_7.3-53        grid_3.6.1        
# [45] jsonlite_1.7.2     gtable_0.3.0       lifecycle_0.2.0    DBI_1.1.0         
# [49] git2r_0.27.1       magrittr_2.0.1     scales_1.1.1       RcppParallel_5.0.2
# [53] stringi_1.5.3      farver_2.0.3       fs_1.3.1           promises_1.1.1    
# [57] ellipsis_0.3.1     generics_0.1.0     vctrs_0.3.6        tools_3.6.1       
# [61] glue_1.4.2         purrr_0.3.4        hms_1.0.0          yaml_2.2.1        
# [65] colorspace_2.0-0   plotly_4.9.3       knitr_1.30         quantreg_5.83     
# [69] MCMCpack_1.5-0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
