<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Compare results from different number of topics (k) in Cusanovich2018 data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Compare results from different number of topics (k) in Cusanovich2018 data</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree7d7de435434ca0a10f0180d2979892becab49cfbtargetblank7d7de43a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/7d7de435434ca0a10f0180d2979892becab49cfb" target="_blank">7d7de43</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree7d7de435434ca0a10f0180d2979892becab49cfbtargetblank7d7de43a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/7d7de435434ca0a10f0180d2979892becab49cfb" target="_blank">7d7de43</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/plotly/Buenrostro_2018_Chen2019pipeline/

Untracked files:
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-main-1.57.1/
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/fit_models_Cusanovich2018_tissues.sh

Unstaged changes:
    Modified:   analysis/assess_fits_Cusanovich2018.Rmd
    Modified:   analysis/clusters_Cusanovich2018_k13.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   analysis/process_data_Cusanovich2018.Rmd
    Modified:   code/plots.R
    Modified:   output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_10_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_11_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_12_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_13_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_1_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_2_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_3_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_4_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_5_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_6_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_7_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_8_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_9_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_10_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_11_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_12_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_13_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_1_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_2_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_3_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_4_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_5_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_6_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_7_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_8_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_9_tss-sum.html
    Modified:   scripts/fit_all_models_Cusanovich2018.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/compare_different_k_Cusanovich2018.Rmd</code>) and HTML (<code>docs/compare_different_k_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7d7de435434ca0a10f0180d2979892becab49cfb/analysis/compare_different_k_Cusanovich2018.Rmd" target="_blank">7d7de43</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-26
</td>
<td>
modified structure plots (k20) by merging topics, and added k = 40 result
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ab8c17f9def4ea1654bd04af8f698761ae8d660a/docs/compare_different_k_Cusanovich2018.html" target="_blank">ab8c17f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/88e1f8d9fda9c4b6f46a1180e70d055e39f6f57c/analysis/compare_different_k_Cusanovich2018.Rmd" target="_blank">88e1f8d</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
<td>
added tissue distribution plots with k = 20
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/51a34ba74cc7ca9872a65214967f9ca45d71fd6f/docs/compare_different_k_Cusanovich2018.html" target="_blank">51a34ba</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f7a84f554f0aa3953de9f693ce3c25ca6f83ccbe/analysis/compare_different_k_Cusanovich2018.Rmd" target="_blank">f7a84f5</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
reordered the tissues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/842c681de6f4a38c5b793cc40c341d2589c843a5/docs/compare_different_k_Cusanovich2018.html" target="_blank">842c681</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d9c3e5a8a29187531d13ceb8c8dc92ac7a5563dd/analysis/compare_different_k_Cusanovich2018.Rmd" target="_blank">d9c3e5a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
created structure plots for different k's
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we examine and compare the topic modeling results for the scATAC-seq dataset from the mouse single-cell atlas paper <a href="scATACseq_data_Cusanovich2018.html">Cusanovich <em>et al</em> (2018)</a></p>
<p>Load the packages used in the analysis below, as well as additional functions that will be used to generate some of the plots.</p>
<pre class="r"><code>library(fastTopics)
library(ggplot2)
library(cowplot)
library(Matrix)
library(tools)
library(reshape2)
library(dplyr)
library(RColorBrewer)
source(&quot;code/plots.R&quot;)</code></pre>
<p>colors</p>
<pre class="r"><code>col_13 &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
            &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
            &quot;gray&quot;)

col_15 &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
            &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;gold1&quot;,
            &quot;#b15928&quot;, &quot;black&quot;, &quot;gray&quot;)

# from https://stackoverflow.com/questions/9563711/r-color-palettes-for-many-data-classes/41230685#41230685
col_25 &lt;- c(
  &quot;dodgerblue2&quot;, &quot;#E31A1C&quot;, # red
  &quot;green4&quot;,
  &quot;#6A3D9A&quot;, # purple
  &quot;#FF7F00&quot;, # orange
  &quot;black&quot;, &quot;gold1&quot;,
  &quot;skyblue2&quot;, &quot;#FB9A99&quot;, # lt pink
  &quot;palegreen2&quot;,
  &quot;#CAB2D6&quot;, # lt purple
  &quot;#FDBF6F&quot;, # lt orange
  &quot;gray70&quot;, &quot;khaki2&quot;,
  &quot;maroon&quot;, &quot;orchid1&quot;, &quot;deeppink1&quot;, &quot;blue1&quot;, &quot;steelblue4&quot;,
  &quot;darkturquoise&quot;, &quot;green1&quot;, &quot;yellow4&quot;, &quot;yellow3&quot;,
  &quot;darkorange4&quot;, &quot;brown&quot;
)

col_20 &lt;- c(
  &quot;dodgerblue2&quot;, &quot;#E31A1C&quot;, # red
  &quot;green4&quot;,
  &quot;#6A3D9A&quot;, # purple
  &quot;#FF7F00&quot;, # orange
  &quot;gold1&quot;,
  &quot;skyblue2&quot;, &quot;#FB9A99&quot;, # lt pink
  &quot;palegreen2&quot;,
  &quot;#CAB2D6&quot;, # lt purple
  &quot;#FDBF6F&quot;, # lt orange
  &quot;khaki2&quot;,
  &quot;maroon&quot;, &quot;orchid1&quot;, &quot;deeppink1&quot;, &quot;blue1&quot;, &quot;steelblue4&quot;,
  &quot;darkturquoise&quot;, &quot;green1&quot;, &quot;brown&quot;
)

# from https://stackoverflow.com/questions/15282580/how-to-generate-a-number-of-most-distinctive-colors-in-r
qual_col_pals &lt;- brewer.pal.info[brewer.pal.info$category == &#39;qual&#39;,]
col_74 &lt;- unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))

# colors for tissues as shown in the original paper
colors_tissues &lt;- c(&quot;darkblue&quot;,    # BoneMarrow
                    &quot;gray30&quot;,      # Cerebellum
                    &quot;red&quot;,         # Heart
                    &quot;springgreen&quot;, # Kidney
                    &quot;brown&quot;,       # LargeIntestine
                    &quot;purple&quot;,      # Liver
                    &quot;deepskyblue&quot;, # Lung
                    &quot;black&quot;,       # PreFrontalCortex
                    &quot;darkgreen&quot;,   # SmallIntestine
                    &quot;plum&quot;,        # Spleen
                    &quot;gold&quot;,        # Testes
                    &quot;darkorange&quot;,  # Thymus
                    &quot;gray&quot;)        # WholeBrain</code></pre>
<div id="load-cusanovich-2018-mouse-scatac-seq-data" class="section level2">
<h2>Load Cusanovich 2018 mouse scATAC-seq data</h2>
<div id="load-the-data-and-poisson-nmf-model-fit" class="section level3">
<h3>Load the data and Poisson NMF model fit</h3>
<p>Load the data</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
rm(counts)
# 81173 x 436206 counts matrix.</code></pre>
<p>About the samples:</p>
<p>The study measured single cell chromatin accessibility for 17 samples spanning 13 different tissues in 8-week old mice.</p>
<pre class="r"><code>cat(nrow(samples), &quot;samples (cells). \n&quot;)
# 81173 samples (cells).</code></pre>
<p>Tissues:</p>
<pre class="r"><code>samples$tissue &lt;- factor(samples$tissue, levels = c(&quot;Heart&quot;, &quot;Liver&quot;, &quot;Thymus&quot;, 
                                                    &quot;Spleen&quot;,&quot;Lung&quot;,  &quot;Kidney&quot;, 
                                                    &quot;LargeIntestine&quot;, &quot;SmallIntestine&quot;, &quot;Testes&quot;, 
                                                    &quot;BoneMarrow&quot;, 
                                                    &quot;Cerebellum&quot;, &quot;PreFrontalCortex&quot;, &quot;WholeBrain&quot;))
cat(length(levels(samples$tissue)), &quot;tissues. \n&quot;)
table(samples$tissue)
# 13 tissues. 
# 
#            Heart            Liver           Thymus           Spleen 
#             7650             6167             7617             4020 
#             Lung           Kidney   LargeIntestine   SmallIntestine 
#             9996             6431             7086             4077 
#           Testes       BoneMarrow       Cerebellum PreFrontalCortex 
#             2723             8403             2278             5959 
#       WholeBrain 
#             8766</code></pre>
<p>Cell types labels:</p>
<pre class="r"><code>samples$cell_label &lt;- as.factor(samples$cell_label)
cat(length(levels(samples$cell_label)), &quot;cell types \n&quot;)
table(samples$cell_label)
# 40 cell types 
# 
#          Activated B cells       Alveolar macrophages 
#                        500                        559 
#                 Astrocytes                    B cells 
#                       1666                       5772 
#             Cardiomyocytes   Cerebellar granule cells 
#                       4076                       4099 
#            Collecting duct                 Collisions 
#                        164                       1218 
#                     DCT/CD            Dendritic cells 
#                        506                        958 
#   Distal convoluted tubule Endothelial I (glomerular) 
#                        319                        552 
#        Endothelial I cells       Endothelial II cells 
#                        952                       3019 
#                Enterocytes              Erythroblasts 
#                       4783                       2811 
#            Ex. neurons CPN          Ex. neurons CThPN 
#                       1832                       1540 
#           Ex. neurons SCPN  Hematopoietic progenitors 
#                       1466                       3425 
#                Hepatocytes           Immature B cells 
#                       5664                        571 
#         Inhibitory neurons              Loop of henle 
#                       1828                        815 
#                Macrophages                  Microglia 
#                        711                        422 
#                  Monocytes                   NK cells 
#                       1173                        321 
#           Oligodendrocytes                  Podocytes 
#                       1558                        498 
#            Proximal tubule         Proximal tubule S3 
#                       2570                        594 
#             Purkinje cells         Regulatory T cells 
#                        320                        507 
#          SOM+ Interneurons                      Sperm 
#                        553                       2089 
#                    T cells         Type I pneumocytes 
#                       8954                       1622 
#        Type II pneumocytes                    Unknown 
#                        157                      10029</code></pre>
</div>
</div>
<div id="structure-plots-using-different-number-of-k" class="section level2">
<h2>Structure plots using different number of <span class="math inline">\(k\)</span></h2>
<p>The structure plots below summarize the topic proportions in the samples grouped by different tissues.</p>
<div id="k-13" class="section level3">
<h3><span class="math inline">\(k = 13\)</span></h3>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<pre class="r"><code>set.seed(10)

colors_topics &lt;- col_13
idx_selected_samples &lt;- sample(nrow(fit_multinom$L),4000)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:13,colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-k13-1">
Past versions of structure-plot-k13-1.png
</button>
</p>
<div id="fig-structure-plot-k13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/51a34ba74cc7ca9872a65214967f9ca45d71fd6f/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k13-1.png" target="_blank">51a34ba</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/842c681de6f4a38c5b793cc40c341d2589c843a5/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k13-1.png" target="_blank">842c681</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Count the number of tissues has membership &gt; 1% in each topic</p>
<pre class="r"><code>topic_proportions &lt;- fit_multinom$L
topic_appearance &lt;- as.matrix((fit_multinom$L &gt; 0.01) + 0)
samples_topics &lt;- samples[,c(&quot;cell&quot;, &quot;tissue&quot;, &quot;cell_label&quot;)]
samples_topics &lt;- cbind(samples_topics, topic_appearance)
samples_topics$num_topics &lt;- rowSums(topic_appearance)

freq_table_tissue_in_topic &lt;- matrix(0, nrow = length(levels(samples_topics$tissue)), ncol = ncol(topic_appearance))
rownames(freq_table_tissue_in_topic) &lt;- levels(samples_topics$tissue)
colnames(freq_table_tissue_in_topic) &lt;- colnames(topic_appearance)

for ( k in 1:ncol(topic_appearance)) {
  freq_table_tissue_in_topic[,k] &lt;- table(samples_topics[samples_topics[,paste0(&quot;k&quot;,k)] == 1, &quot;tissue&quot;])
}
print(freq_table_tissue_in_topic)
#                    k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11  k12
# Heart            3189 2241 4936 2477 1483 2095 5960 1581 3092 5728 4526 1918
# Liver            2425  949 1635 1891 1430 1110 1437 5900 1708 3700 1605  999
# Thymus           2138  398  916  954  308 3062  617  352 1138 3283 1349 7542
# Spleen           1162   72  306  320  174 3956  154  239  183 1334  284 2571
# Lung             5148 1802 5619 4624 1118 7180 3634 2894 2819 4750 3804 4308
# Kidney           1607 1090 2345 3524 5541 1457 1549 1095 1993 2352 1394 1153
# LargeIntestine   2307 1643 2751 5941 2908 2265 2385 5252 1902 7026 4237 2265
# SmallIntestine   1176 1263 2944 1677 1850 1758 2184 2050 1519 3825 3742 1493
# Testes           2359 1634  676  901  766  379 1321  590 2092 2698 2318  644
# BoneMarrow       6425 1149 2125 1160  823 5635 1967 1200 1522 5115 1578 3282
# Cerebellum        586 1875  923  750  524  288  859  441 1401 1090 1126  526
# PreFrontalCortex 2262 3170 2331 1691  605 1018 1869  633 4002 3973 5036 1451
# WholeBrain       3244 6643 3313 2605 1856 1579 4108 1700 5938 6177 5947 2316
#                   k13
# Heart            6613
# Liver            4759
# Thymus           5977
# Spleen           2429
# Lung             7772
# Kidney           4587
# LargeIntestine   6894
# SmallIntestine   4055
# Testes           2478
# BoneMarrow       7791
# Cerebellum       1530
# PreFrontalCortex 3523
# WholeBrain       5914</code></pre>
</div>
<div id="k-20" class="section level3">
<h3><span class="math inline">\(k = 20\)</span></h3>
<pre class="r"><code>k &lt;- 20
fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(sprintf(&quot;%s/fit-Cusanovich2018-scd-ex-k=%d.rds&quot;, fit.dir, k))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<pre class="r"><code>set.seed(10)

colors_topics &lt;- col_20
idx_selected_samples &lt;- sample(nrow(fit_multinom$L),4000)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:k,colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-k20-1">
Past versions of structure-plot-k20-1.png
</button>
</p>
<div id="fig-structure-plot-k20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/51a34ba74cc7ca9872a65214967f9ca45d71fd6f/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k20-1.png" target="_blank">51a34ba</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/842c681de6f4a38c5b793cc40c341d2589c843a5/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k20-1.png" target="_blank">842c681</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Count the number of tissues has membership &gt; 5% in each topic</p>
<pre class="r"><code>topic_proportions &lt;- fit_multinom$L
topic_appearance &lt;- as.matrix((fit_multinom$L &gt; 0.05) + 0)
samples_topics &lt;- samples[,c(&quot;cell&quot;, &quot;tissue&quot;, &quot;cell_label&quot;)]
samples_topics &lt;- cbind(samples_topics, topic_appearance)
samples_topics$count_topics &lt;- rowSums(topic_appearance)

freq_table_tissue_in_topic &lt;- matrix(0, nrow = length(levels(samples_topics$tissue)), ncol = ncol(topic_appearance))
rownames(freq_table_tissue_in_topic) &lt;- levels(samples_topics$tissue)
colnames(freq_table_tissue_in_topic) &lt;- colnames(topic_appearance)

for ( k in 1:ncol(topic_appearance)) {
  freq_table_tissue_in_topic[,k] &lt;- table(samples_topics[samples_topics[,paste0(&quot;k&quot;,k)] == 1, &quot;tissue&quot;])
}
print(freq_table_tissue_in_topic)
#                    k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11  k12
# Heart              46    9 2334   89   13 3456  358   71  133  102    9   18
# Liver              11    1  854   25   11  517  347    8   62  175   25    8
# Thymus              5    0 1415   19 7309   21  205    1  173 3220   10    1
# Spleen              0    0  578    4  276   14  549    2 2858 1274   82    1
# Lung              272    9 2515 2692  248 2501 2040   25 3127 1495   64    3
# Kidney            119    0  927 2228    7 1507  152   22   85  149    4 4043
# LargeIntestine   4952   12 4545  233   47 1692  143  146  332  281    6   39
# SmallIntestine    309    7 2167   93   39 1699   94  110  356  275   10   45
# Testes              3    0 2694   49    1   69    5   16    1    4   12    9
# BoneMarrow          2    0 2229    3  416   86 3851   10 1317  717 3217    1
# Cerebellum         15 1390  229   94    2  182   39  662   13    9    0    8
# PreFrontalCortex    8  356  714   77    5  213  276 1661   64   10    0    5
# WholeBrain         25 4026 2932  191    4  401  289 2523   40   14    4    5
#                   k13  k14  k15  k16  k17  k18  k19  k20
# Heart            4305  103  436 3640  355 4831   22 4845
# Liver              10    3   23  545   15 1992 5658 3470
# Thymus              5    2  167  510    6 3484    1 6568
# Spleen              1    0   10  150   10 1252    1 3190
# Lung              102   56  278 2743  262 5657  138 7809
# Kidney             29    3  156  752   32 3164   30 4374
# LargeIntestine    258  163  765 5861  644 5012   81 2934
# SmallIntestine     67  116  532 3911  942 3916   95  603
# Testes             50   20  316 1816  240 2441   42 1646
# BoneMarrow          3    9   85 4310   26 7185    4 3499
# Cerebellum         36   56  644  594  134  621    4 1604
# PreFrontalCortex   36 3466 2731 1335 3740  920    5 3566
# WholeBrain        104 1524 3232 3594 2268 1694   11 5429</code></pre>
<p>Plot the distribution of tissues in each topic, ordered by entropy in each topic</p>
<pre class="r"><code># compute Shannon entropy
# from https://rstudio-pubs-static.s3.amazonaws.com/455435_30729e265f7a4d049400d03a18e218db.html
entropy &lt;- function(target) {
  freq &lt;- table(target)/length(target)
  # vectorize
  vec &lt;- as.data.frame(freq)[,2]
  #drop 0 to avoid NaN resulting from log2
  vec &lt;- vec[vec&gt;0]
  #compute entropy
  -sum(vec * log2(vec))
}</code></pre>
<pre class="r"><code>freq_table_tissue_in_topic.df &lt;- melt(freq_table_tissue_in_topic)
colnames(freq_table_tissue_in_topic.df) &lt;- c(&quot;tissue&quot;, &quot;topic&quot;, &quot;freq&quot;)

topic_entropy &lt;- sort(apply(freq_table_tissue_in_topic, 2, entropy))
freq_table_tissue_in_topic.df$topic &lt;- factor(freq_table_tissue_in_topic.df$topic, levels = names(topic_entropy))

# stacked barplot for the counts of tissues by clusters
p.barplot &lt;- ggplot(freq_table_tissue_in_topic.df, aes(fill=tissue, y=freq, x=topic)) + 
  geom_bar(position=&quot;fill&quot;, stat=&quot;identity&quot;) +
  theme_classic() + xlab(&quot;Topic&quot;) + ylab(&quot;Proportion of cells&quot;) +
  scale_fill_manual(values = colors_tissues) +
  guides(fill=guide_legend(ncol=2)) +
  theme(
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )

print(p.barplot)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/barplot-tissues-topics-k20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-barplot-tissues-topics-k20-1">
Past versions of barplot-tissues-topics-k20-1.png
</button>
</p>
<div id="fig-barplot-tissues-topics-k20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ab8c17f9def4ea1654bd04af8f698761ae8d660a/docs/figure/compare_different_k_Cusanovich2018.Rmd/barplot-tissues-topics-k20-1.png" target="_blank">ab8c17f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Simple heatmap of the proportions by hierarchical clustering

create_simple_heatmap(freq_table_tissue_in_topic, normalize_by = &quot;row&quot;, title = &quot;proportions normalized by rows (each row sums to 1)&quot;, cluster_rows = TRUE)

create_simple_heatmap(freq_table_tissue_in_topic, normalize_by = &quot;column&quot;, title = &quot;proportions normalized by columns (each column sums to 1)&quot;, cluster_rows = TRUE)
</code></pre>
<pre class="r"><code># Cells with membership &gt; 5% in topic k20
data &lt;- as.data.frame(table(samples_topics[samples_topics$k20 == 1, &quot;cell_label&quot;]))
colnames(data) &lt;- c(&quot;cell_label&quot;, &quot;count&quot;)
data &lt;- data[order(data$count, decreasing = T), ]
data$cell_label &lt;- factor(data$cell_label, levels = unique(data$cell_label))

# piechart
p_pie &lt;- ggplot(data, aes(x=&quot;&quot;, y=count, fill=cell_label)) +
  geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  scale_fill_manual(values = col_74) +
  coord_polar(&quot;y&quot;, start=0) +
  ggtitle(&quot;Topic k20&quot;) +
  theme_void() # remove background, grid, numeric labels
print(p_pie)
</code></pre>
<p>Structure plot for Cerebellum, PreFrontalCortex, WholeBrain.</p>
<p>Merge topics with cumulative average proportion &lt; 0.1. i.e. the topics included cover &gt; 90% of the area.</p>
<pre class="r"><code>set.seed(10)

idx_selected_samples &lt;- which(samples$tissue %in% c(&quot;Cerebellum&quot;, &quot;PreFrontalCortex&quot;, &quot;WholeBrain&quot;))

fit_tissue &lt;- select(fit_multinom,loadings = idx_selected_samples)
grouping_tissue &lt;- as.factor(as.character(samples[idx_selected_samples, &quot;tissue&quot;]))

# merge topics with cumulative average proportion &lt; 0.1
topic_mean_proportion &lt;- sort(colMeans(fit_tissue$L))
topics_merged &lt;- names(which(cumsum(topic_mean_proportion) &lt; 0.1))
fit_tissue_merged &lt;- merge_topics(fit_tissue, k = topics_merged)
colnames(fit_tissue_merged$L)[ncol(fit_tissue_merged$L)] &lt;- &quot;others&quot;
colnames(fit_tissue_merged$F)[ncol(fit_tissue_merged$F)] &lt;- &quot;others&quot;

colors_topics &lt;- col_20
names(colors_topics) &lt;- paste0(&quot;k&quot;, 1:20)
colors_topics &lt;- colors_topics[colnames(fit_tissue_merged$L)]
colors_topics[is.na(colors_topics)] &lt;- &quot;gray&quot;

p.structure &lt;- structure_plot(fit_tissue_merged,
                              grouping = grouping_tissue,
                              n = min(3000, length(idx_selected_samples)), 
                              gap = 40, perplexity = 50,
                              topics = colnames(fit_tissue_merged$L),
                              colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-brain-related-k20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-brain-related-k20-1">
Past versions of structure-plot-brain-related-k20-1.png
</button>
</p>
<div id="fig-structure-plot-brain-related-k20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ab8c17f9def4ea1654bd04af8f698761ae8d660a/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-brain-related-k20-1.png" target="_blank">ab8c17f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Structure plot for Heart, Lung, Kidney, Spleen.</p>
<pre class="r"><code>set.seed(10)

idx_selected_samples &lt;- which(samples$tissue %in% c(&quot;Heart&quot;, &quot;Lung&quot;, &quot;Kidney&quot;, &quot;Spleen&quot;))

fit_tissue &lt;- select(fit_multinom,loadings = idx_selected_samples)
grouping_tissue &lt;- as.factor(as.character(samples[idx_selected_samples, &quot;tissue&quot;]))

# merge topics with cumulative average proportion &lt; 0.1
topic_mean_proportion &lt;- sort(colMeans(fit_tissue$L))
topics_merged &lt;- names(which(cumsum(topic_mean_proportion) &lt; 0.1))
fit_tissue_merged &lt;- merge_topics(fit_tissue, k = topics_merged)
colnames(fit_tissue_merged$L)[ncol(fit_tissue_merged$L)] &lt;- &quot;others&quot;
colnames(fit_tissue_merged$F)[ncol(fit_tissue_merged$F)] &lt;- &quot;others&quot;

colors_topics &lt;- col_20
names(colors_topics) &lt;- paste0(&quot;k&quot;, 1:20)
colors_topics &lt;- colors_topics[colnames(fit_tissue_merged$L)]
colors_topics[is.na(colors_topics)] &lt;- &quot;gray&quot;

p.structure &lt;- structure_plot(fit_tissue_merged,
                              grouping = grouping_tissue,
                              n = min(3000, length(idx_selected_samples)), 
                              gap = 40, perplexity = 50,
                              topics = colnames(fit_tissue_merged$L),
                              colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-heart-lung-kidney-k20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-heart-lung-kidney-k20-1">
Past versions of structure-plot-heart-lung-kidney-k20-1.png
</button>
</p>
<div id="fig-structure-plot-heart-lung-kidney-k20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ab8c17f9def4ea1654bd04af8f698761ae8d660a/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-heart-lung-kidney-k20-1.png" target="_blank">ab8c17f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Structure plot for Heart only.</p>
<pre class="r"><code>set.seed(10)

idx_selected_samples &lt;- which(samples$tissue %in% &quot;Heart&quot;)

fit_tissue &lt;- select(fit_multinom,loadings = idx_selected_samples)
grouping_tissue &lt;- as.factor(as.character(samples[idx_selected_samples, &quot;tissue&quot;]))

# merge topics with cumulative average proportion &lt; 0.1
topic_mean_proportion &lt;- sort(colMeans(fit_tissue$L))
topics_merged &lt;- names(which(cumsum(topic_mean_proportion) &lt; 0.1))
fit_tissue_merged &lt;- merge_topics(fit_tissue, k = topics_merged)
colnames(fit_tissue_merged$L)[ncol(fit_tissue_merged$L)] &lt;- &quot;others&quot;
colnames(fit_tissue_merged$F)[ncol(fit_tissue_merged$F)] &lt;- &quot;others&quot;

colors_topics &lt;- col_20
names(colors_topics) &lt;- paste0(&quot;k&quot;, 1:20)
colors_topics &lt;- colors_topics[colnames(fit_tissue_merged$L)]
colors_topics[is.na(colors_topics)] &lt;- &quot;gray&quot;

p.structure &lt;- structure_plot(fit_tissue_merged,
                              grouping = grouping_tissue,
                              n = min(3000, length(idx_selected_samples)), 
                              gap = 40, perplexity = 50,
                              topics = colnames(fit_tissue_merged$L),
                              colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-heart-k20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-heart-k20-1">
Past versions of structure-plot-heart-k20-1.png
</button>
</p>
<div id="fig-structure-plot-heart-k20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ab8c17f9def4ea1654bd04af8f698761ae8d660a/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-heart-k20-1.png" target="_blank">ab8c17f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-30" class="section level3">
<h3><span class="math inline">\(k = 30\)</span></h3>
<pre class="r"><code>k &lt;- 30
fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(sprintf(&quot;%s/fit-Cusanovich2018-scd-ex-k=%d.rds&quot;, fit.dir, k))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<pre class="r"><code>set.seed(10)

colors_topics &lt;- sample(col_74, k)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:k,colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k30-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-k30-1">
Past versions of structure-plot-k30-1.png
</button>
</p>
<div id="fig-structure-plot-k30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/51a34ba74cc7ca9872a65214967f9ca45d71fd6f/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k30-1.png" target="_blank">51a34ba</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/842c681de6f4a38c5b793cc40c341d2589c843a5/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k30-1.png" target="_blank">842c681</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-35" class="section level3">
<h3><span class="math inline">\(k = 35\)</span></h3>
<pre class="r"><code>k &lt;- 35
fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(sprintf(&quot;%s/fit-Cusanovich2018-scd-ex-k=%d.rds&quot;, fit.dir, k))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- sample(col_74, k)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:k,colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k35-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-k35-1">
Past versions of structure-plot-k35-1.png
</button>
</p>
<div id="fig-structure-plot-k35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/51a34ba74cc7ca9872a65214967f9ca45d71fd6f/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k35-1.png" target="_blank">51a34ba</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/842c681de6f4a38c5b793cc40c341d2589c843a5/docs/figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k35-1.png" target="_blank">842c681</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-40" class="section level3">
<h3><span class="math inline">\(k = 40\)</span></h3>
<pre class="r"><code>k &lt;- 40
fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(sprintf(&quot;%s/fit-Cusanovich2018-scd-ex-k=%d.rds&quot;, fit.dir, k))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- sample(col_74, k)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:k,colors = colors_topics,
                              num_threads = 8,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/compare_different_k_Cusanovich2018.Rmd/structure-plot-k40-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] tools     stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
# [1] RColorBrewer_1.1-2 dplyr_1.0.4        reshape2_1.4.3     Matrix_1.2-18     
# [5] cowplot_1.1.1      ggplot2_3.3.3      fastTopics_0.5-20  workflowr_1.6.2   
# 
# loaded via a namespace (and not attached):
#  [1] httr_1.4.2         tidyr_1.1.2        jsonlite_1.7.2     viridisLite_0.3.0 
#  [5] RcppParallel_5.0.3 mixsqp_0.3-43      yaml_2.2.1         progress_1.2.2    
#  [9] ggrepel_0.9.1      pillar_1.5.0       lattice_0.20-41    quantreg_5.85     
# [13] glue_1.4.2         quadprog_1.5-8     digest_0.6.27      promises_1.2.0.1  
# [17] colorspace_2.0-0   plyr_1.8.6         htmltools_0.5.1.1  httpuv_1.5.4      
# [21] conquer_1.0.2      pkgconfig_2.0.3    invgamma_1.1       SparseM_1.81      
# [25] purrr_0.3.4        scales_1.1.1       whisker_0.4        later_1.1.0.1     
# [29] Rtsne_0.15         MatrixModels_0.4-1 git2r_0.27.1       tibble_3.1.0      
# [33] farver_2.0.3       generics_0.1.0     ellipsis_0.3.1     withr_2.4.1       
# [37] ashr_2.2-47        lazyeval_0.2.2     magrittr_2.0.1     crayon_1.4.1      
# [41] mcmc_0.9-7         evaluate_0.14      fs_1.3.1           fansi_0.4.2       
# [45] MASS_7.3-53        truncnorm_1.0-8    data.table_1.14.0  prettyunits_1.1.1 
# [49] hms_1.0.0          lifecycle_1.0.0    matrixStats_0.58.0 stringr_1.4.0     
# [53] MCMCpack_1.5-0     plotly_4.9.3       munsell_0.5.0      irlba_2.3.3       
# [57] compiler_3.6.1     rlang_0.4.10       debugme_1.1.0      grid_3.6.1        
# [61] htmlwidgets_1.5.3  labeling_0.4.2     rmarkdown_2.6      gtable_0.3.0      
# [65] DBI_1.1.0          R6_2.5.0           knitr_1.30         utf8_1.1.4        
# [69] rprojroot_2.0.2    stringi_1.5.3      SQUAREM_2021.1     Rcpp_1.0.6        
# [73] vctrs_0.3.6        tidyselect_1.1.0   xfun_0.19          coda_0.19-4</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
