<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Topic modeling analysis of Cusanovich et al (2018) kidney cells with k = 10 topics</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Topic modeling analysis of Cusanovich et al
(2018) kidney cells with k = 10 topics</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree992fdf78a48586abe1f6daa7c72ef6914324f544targetblank992fdf7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/scATACseq-topics/tree/992fdf78a48586abe1f6daa7c72ef6914324f544" target="_blank">992fdf7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree992fdf78a48586abe1f6daa7c72ef6914324f544targetblank992fdf7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/scATACseq-topics/tree/992fdf78a48586abe1f6daa7c72ef6914324f544" target="_blank">992fdf7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Buenrostro_2018/
    Ignored:    data/Cusanovich_2018/processed_data/Cusanovich_2018_Kidney.RData
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/de-buenrostro2018-k=10-noshrink.RData
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/fit-Buenrostro2018-binarized-filtered-scd-ex-k=10.rds
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/fit-Buenrostro2018-binarized-filtered-scd-ex-k=8.rds
    Ignored:    output/Cusanovich_2018/.DS_Store
    Ignored:    output/Cusanovich_2018/fit-Cusanovich2018-scd-ex-k=13.rds
    Ignored:    output/Cusanovich_2018/tissues/.DS_Store
    Ignored:    output/Cusanovich_2018/tissues/de-cusanovich2018-kidney-k=10-noshrink.RData
    Ignored:    output/Cusanovich_2018/tissues/fit-Cusanovich2018-Kidney-scd-ex-k=10.rds
    Ignored:    output/Cusanovich_2018/tissues/gene-scores-cusanovich2018-kidney-k=10.RData
    Ignored:    scripts/seq_gene.md.gz

Untracked files:
    Untracked:  analysis/fit-Buenrostro2018-binarized-scd-ex-k=10.rds
    Untracked:  data/Buenrostro_2018_binarized.RData
    Untracked:  output/Buenrostro_2018/binarized/filtered_peaks/Buenrostro_2018_binarized_filtered.RData
    Untracked:  plots/
    Untracked:  scripts/fit-buenrostro-2018-k=8.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/cusanovich2018_kidney_k10.Rmd</code>) and HTML
(<code>docs/cusanovich2018_kidney_k10.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/992fdf78a48586abe1f6daa7c72ef6914324f544/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">992fdf7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-10
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/979a0ab0654930bbc1264aab4ed984c9062b78a9/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">979a0ab</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-05
</td>
<td>
Added step to create eps for two volcano plots in
cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/397cff5e6a0544c82e1c6f618a57946c79e22ed6/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">397cff5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
<td>
Added a couple notes.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/b98cc33f770a652463e9ee2fb957fc85d2bd0211/docs/cusanovich2018_kidney_k10.html" target="_blank">b98cc33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
<td>
Added volcano plots for remaining topics to cusanovich2018_kidney_k10
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/dd40726398ded849d21b6b0fbf75ddb5bc1167a2/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">dd40726</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/7419f5c76dbe89b325b2c7e8295aad14fe8f243b/docs/cusanovich2018_kidney_k10.html" target="_blank">7419f5c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-03
</td>
<td>
Added label counts to cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/dade9f6dd882ff30c82020373fe3a5b6104999c9/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">dade9f6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-03
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1250bce9f094da9f59a36f5336ccba958c410ee9/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">1250bce</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-30
</td>
<td>
Wrote script compile_de_gene_table.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/648fbbc7255adfa4b6080eaab2a00c01f0377542/docs/cusanovich2018_kidney_k10.html" target="_blank">648fbbc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b779c73fecaa7fecf081c77db6619264b7b9253a/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">b779c73</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-30
</td>
<td>
workflowr::wflow_publish("cusanovich2018_kidney_k10.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/3912476a1e5e983079846dfce7c21b751863a0d7/docs/cusanovich2018_kidney_k10.html" target="_blank">3912476</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/91dda41504a25c50696ab146ff7596d78c17c9b9/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">91dda41</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-29
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/36327f7d2d4243fdbda8765e5e877214a0ec9c19/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">36327f7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-29
</td>
<td>
I have a first draft of the function for creating the volcano enrichment
plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d111ca69fa57462245b4484f627fdaa352ee90e8/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">d111ca6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-28
</td>
<td>
Edit to cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/494e06a19c25dfc3794fd970b1080bd4d8b287a5/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">494e06a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-28
</td>
<td>
Working on adding volcano plots to cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b3b73501abb4450302f1856e18b180023bc04da4/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">b3b7350</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-16
</td>
<td>
Made some improvements to exploratory analysis temp.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/822da57191bc554a3c23976a99a29cdc389e23ca/docs/cusanovich2018_kidney_k10.html" target="_blank">822da57</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
Added some accompanying text to the cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/8d165908cf23b0711e0bb2ab347c8d51ca6d89af/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">8d16590</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/0d6c9fd1830cd25f3f990e48777b4c25da827027/docs/cusanovich2018_kidney_k10.html" target="_blank">0d6c9fd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
Added structure plots to cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d9cfa44abbcdd197c0f8dcac52cdfac25667d6a9/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">d9cfa44</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/68fba7a1568278e00ed9fd1ec95c179553fdc407/docs/cusanovich2018_kidney_k10.html" target="_blank">68fba7a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
First build of cusanovich2018_kidney_k10 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/4549e6a9ffb45575d49f23be4af0a65ffc536977/analysis/cusanovich2018_kidney_k10.Rmd" target="_blank">4549e6a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
workflowr::wflow_publish("analysis/cusanovich2018_kidney_k10.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we examine the structure inferred from the topic model fit to
the Mouse sci-ATAC-seq Atlas kidney cells, with <span
class="math inline">\(K = 10\)</span> topics.</p>
<p>Load the packages used in the analysis below.</p>
<pre class="r"><code>library(Matrix)
library(pathways)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source(&quot;code/kidney_genes.R&quot;)
source(&quot;code/volcano_plots.R&quot;)</code></pre>
<p>Load the binarized count data.</p>
<pre class="r"><code>load(&quot;data/Cusanovich_2018/processed_data/Cusanovich_2018_Kidney.RData&quot;)</code></pre>
<p>Load the <span class="math inline">\(K = 10\)</span> topic model
fit.</p>
<pre class="r"><code>fit &lt;- readRDS(file.path(&quot;output/Cusanovich_2018/tissues&quot;,
                         &quot;fit-Cusanovich2018-Kidney-scd-ex-k=10.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)</code></pre>
<p>Load the gene-wise results obtained from the <code>de_analysis</code>
of the peaks.</p>
<pre class="r"><code>load(file.path(&quot;output/Cusanovich_2018/tissues&quot;,
               &quot;de-gene-cusanovich2018-kidney-k=10.RData&quot;))</code></pre>
<p>This next Structure plot shows the correspondence between the topics
and the final cell-type labels from Cusanovich <em>et al</em>
(“cell_label”). Some of more distinct cell-types (e.g., loop of henle,
podocytes) are captured by a single topic, Note that some of the
cell-type labels are combined to improve the Structure plot; in
particular, several of the labels that only appear once or a few times
are combined into the “Other or unknown” grouping.</p>
<pre class="r"><code>set.seed(1)
x &lt;- samples$cell_label
x[x == &quot;Activated B cells&quot; | x == &quot;B cells&quot; | x == &quot;Alveolar macrophages&quot; |
  x == &quot;Monocytes&quot; | x == &quot;NK cells&quot; | x == &quot;T cells&quot; | 
  x == &quot;Regulatory T cells&quot; | x == &quot;Dendritic cells&quot; |
  x == &quot;Macrophages&quot;] &lt;- &quot;Immune cells&quot;
x[x == &quot;Endothelial I (glomerular)&quot; |
  x == &quot;Endothelial I cells&quot;] &lt;- &quot;Endothelial I cells&quot;
x[x == &quot;Collisions&quot; | x == &quot;Enterocytes&quot; | x == &quot;Hepatocytes&quot; |
  x == &quot;Sperm&quot; | x == &quot;Type II pneumocytes&quot; | x == &quot;Endothelial II cells&quot; |
  x == &quot;Hematopoietic progenitors&quot; | x == &quot;Unknown&quot;] &lt;- &quot;Other or unknown&quot;
x &lt;- factor(x,c(&quot;Collecting duct&quot;,&quot;DCT/CD&quot;,&quot;Distal convoluted tubule&quot;,
                &quot;Loop of henle&quot;,&quot;Proximal tubule&quot;,&quot;Proximal tubule S3&quot;,
                &quot;Podocytes&quot;,&quot;Endothelial I cells&quot;,&quot;Immune cells&quot;,
                &quot;Other or unknown&quot;))
samples &lt;- transform(samples,cell_label = x)
topic_colors &lt;- c(&quot;orange&quot;,&quot;darkmagenta&quot;,&quot;magenta&quot;,&quot;gold&quot;,&quot;peru&quot;,
                  &quot;red&quot;,&quot;lightgray&quot;,&quot;limegreen&quot;,&quot;dodgerblue&quot;,&quot;royalblue&quot;)
p1 &lt;- structure_plot(fit,grouping = samples$cell_label,gap = 20,
                     colors = topic_colors,verbose = FALSE)
table(grouping = samples$cell_label)
print(p1)</code></pre>
<p><img src="figure/cusanovich2018_kidney_k10.Rmd/structure-plot-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1-1">
Past versions of structure-plot-1-1.png
</button>
</p>
<div id="fig-structure-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0d6c9fd1830cd25f3f990e48777b4c25da827027/docs/figure/cusanovich2018_kidney_k10.Rmd/structure-plot-1-1.png" target="_blank">0d6c9fd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># grouping
#          Collecting duct                   DCT/CD Distal convoluted tubule 
#                      164                      499                      319 
#            Loop of henle          Proximal tubule       Proximal tubule S3 
#                      814                     2565                      594 
#                Podocytes      Endothelial I cells             Immune cells 
#                      447                      558                      133 
#         Other or unknown 
#                      338</code></pre>
<p>Let’s now visualize the topics without the benefit of the previously
inferred clusters. Instead, we can use PCs computed from the topic
proportions to subdivide the cells into 5 clusters. (Here we are
assigning labels to these clusters only to make it easier to refer to
them.)</p>
<pre class="r"><code>pca &lt;- prcomp(fit$L)$x
pc1 &lt;- pca[,1]
pc2 &lt;- pca[,2]
x &lt;- rep(&quot;other&quot;,nrow(pca))
x[pc1 &lt; 0] &lt;- &quot;proximal tubule&quot;
x[pc1 &gt; 0 &amp; pc2 &gt; 0.2] &lt;- &quot;endothelial I&quot;
rows &lt;- which(x == &quot;other&quot;)
fit2 &lt;- select_loadings(fit,loadings = rows)
pca  &lt;- prcomp(fit2$L)$x
y    &lt;- rep(&quot;DCT + CD&quot;,nrow(pca))
pc1  &lt;- pca[,1]
pc2  &lt;- pca[,2]
y[pc1 &lt; 0 &amp; pc2 &gt; -0.15] &lt;- &quot;loop of henle&quot;
y[pc1 &gt;= 0 &amp; pc2 &gt; -0.15] &lt;- &quot;podocytes + other&quot;
x[rows] &lt;- y
samples$cluster &lt;- factor(x,c(&quot;DCT + CD&quot;,&quot;loop of henle&quot;,&quot;proximal tubule&quot;,
                              &quot;podocytes + other&quot;,&quot;endothelial I&quot;))</code></pre>
<p>This is what the Structure plot looks like without any external
grouping of the cells:</p>
<pre class="r"><code>set.seed(1)
p2 &lt;- structure_plot(fit,grouping = samples$cluster,gap = 20,
                     colors = topic_colors,verbose = FALSE)
table(grouping = samples$cluster)
print(p2)</code></pre>
<p><img src="figure/cusanovich2018_kidney_k10.Rmd/structure-plot-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-2-1">
Past versions of structure-plot-2-1.png
</button>
</p>
<div id="fig-structure-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0d6c9fd1830cd25f3f990e48777b4c25da827027/docs/figure/cusanovich2018_kidney_k10.Rmd/structure-plot-2-1.png" target="_blank">0d6c9fd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># grouping
#          DCT + CD     loop of henle   proximal tubule podocytes + other 
#              1012               843              2878              1071 
#     endothelial I 
#               627</code></pre>
<p>The topics clearly identify four distinctive cell types—(1) loop of
henle, (2) proximal tubule, (3) distal convoluted tubule (DCT) and
collecting duct (CD), and (4) endothelial cells—and within these
distinctive cell types there is sometimes a good amount of continuous
variation in expression that is further captured by the topics, such as
the two topics for DCT and CD, and the heterogeneity among the proximal
tubule cells captured by several topics.</p>
<p>Next, let’s see if the gene ranking generated from the differential
accessibility analysis of the topics supports these interpretations, and
helps us interpret some of the other topics that are less clear such as
topic 9.</p>
<p>In this next code chunk we prepare the <code>gene_info</code> and
<code>de_gene</code> data structures for the volcano plots.</p>
<pre class="r"><code>data(gene_sets_mouse)
gene_info &lt;- gene_sets_mouse$gene_info
gene_info &lt;- subset(gene_info,!is.na(Ensembl))
gene_info &lt;- subset(gene_info,!duplicated(Ensembl))
ids &lt;- gene_info$Ensembl
ids &lt;- intersect(ids,rownames(de_gene$coef))
de_gene$coef  &lt;- de_gene$coef[ids,]
de_gene$logLR &lt;- de_gene$logLR[ids,]
ids       &lt;- rownames(de_gene$coef)
gene_info &lt;- subset(gene_info,is.element(Ensembl,ids))
rownames(gene_info) &lt;- gene_info$Ensembl
gene_info &lt;- gene_info[ids,]
genes     &lt;- gene_info$Symbol</code></pre>
<p>These next few volcano plots show, for each gene tested for
enrichment, the mean LFC on the x-axis against the logBF for enrichment
on the y-axis. Marker genes identified in the <a
href="https://doi.org/10.1126/science.aar2131">Park <em>et al</em>
paper</a> are highlighted in red.</p>
<pre class="r"><code>k &lt;- 2
label_genes &lt;- (is.element(genes,cd_genes) &amp; de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 35 &amp; de_gene$coef[,k] &gt; 0) |
               (de_gene$coef[,k] &gt; 3)
k2_genes &lt;- genes
k2_genes[!label_genes] &lt;- &quot;&quot;
p2 &lt;- volcano_plot_enrich(de_gene,k = 2,labels = k2_genes,
                          highlight = is.element(genes,cd_genes),
                          ymax = 100)
k &lt;- 5
label_genes &lt;- (is.element(genes,dct_genes) &amp; de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 40 &amp; de_gene$coef[,k] &gt; 1) |
               (de_gene$coef[,k] &gt; 3)
k5_genes &lt;- genes
k5_genes[!label_genes] &lt;- &quot;&quot;
p5 &lt;- volcano_plot_enrich(de_gene,k = 5,labels = k5_genes,
                          highlight = is.element(genes,dct_genes),
                          ymax = 100)
k &lt;- 3
label_genes &lt;- (is.element(genes,endo_genes) &amp; de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 85 &amp; de_gene$coef[,k] &gt; 2.25) |
               (de_gene$coef[,k] &gt; 4.5)
k3_genes &lt;- genes
k3_genes[!label_genes] &lt;- &quot;&quot;
p3 &lt;- volcano_plot_enrich(de_gene,k = 3,labels = k3_genes,
                          highlight = is.element(genes,endo_genes),
                          ymax = 125)
k &lt;- 1
label_genes &lt;- (is.element(genes,podo_genes) &amp; de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 50 &amp; de_gene$coef[,k] &gt; 2) |
               (de_gene$logLR[,k] &gt; 20 &amp; de_gene$coef[,k] &gt; 3.75)
k1_genes &lt;- genes
k1_genes[!label_genes] &lt;- &quot;&quot;
p1 &lt;- volcano_plot_enrich(de_gene,k = 1,labels = k1_genes,
                          highlight = is.element(genes,podo_genes),
                          ymax = 80)
k &lt;- 8
label_genes &lt;- (is.element(genes,loh_genes) &amp; de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 60 &amp; de_gene$coef[,k] &gt; 0) |
               (de_gene$coef[,k] &gt; 3)
k8_genes &lt;- genes
k8_genes[!label_genes] &lt;- &quot;&quot;
p8 &lt;- volcano_plot_enrich(de_gene,k = 8,labels = k8_genes,
                          highlight = is.element(genes,loh_genes),
                          ymax = 140)
plot_grid(p2,p5,
          p3,p1,
          p8,
          nrow = 3,ncol = 2)</code></pre>
<p><img src="figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-1-1">
Past versions of volcano-plots-1-1.png
</button>
</p>
<div id="fig-volcano-plots-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b98cc33f770a652463e9ee2fb957fc85d2bd0211/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-1-1.png" target="_blank">b98cc33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7419f5c76dbe89b325b2c7e8295aad14fe8f243b/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-1-1.png" target="_blank">7419f5c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/648fbbc7255adfa4b6080eaab2a00c01f0377542/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-1-1.png" target="_blank">648fbbc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3912476a1e5e983079846dfce7c21b751863a0d7/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-1-1.png" target="_blank">3912476</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These next two volcano plots also highlight, in blue, the marker
genes identified in <a
href="https://doi.org/10.1007/s12079-022-00685-z">Wu <em>et al</em></a>
for the S1 and S3 segments of the proximal tubule.</p>
<pre class="r"><code>k &lt;- 9
label_genes &lt;- (is.element(genes,S1_genes) &amp;
                de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 40 &amp; de_gene$coef[,k] &gt; 2.25) |
               (de_gene$logLR[,k] &gt; 100 &amp; de_gene$coef[,k] &gt; 1) |
               (de_gene$coef[,k] &gt; 3)
k9_genes &lt;- genes
k9_genes[!label_genes] &lt;- &quot;&quot;
k9_highlight &lt;- rep(0,length(genes))
k9_highlight[is.element(genes,pt_genes)] &lt;- 1
k9_highlight[is.element(genes,S1_genes)] &lt;- 2
p9 &lt;- volcano_plot_enrich(de_gene,k = 9,labels = k9_genes,
                          highlight = factor(k9_highlight),ymax = 200)
k &lt;- 10
label_genes &lt;- (is.element(genes,S3_genes) &amp;
                de_gene$logLR[,k] &gt; 4) |
               (de_gene$logLR[,k] &gt; 65 &amp; de_gene$coef[,k] &gt; 1.5) |
               (de_gene$coef[,k] &gt; 3)
k10_genes &lt;- genes
k10_genes[!label_genes] &lt;- &quot;&quot;
k10_highlight &lt;- rep(0,length(genes))
k10_highlight[is.element(genes,pt_genes)] &lt;- 1
k10_highlight[is.element(genes,S3_genes)] &lt;- 2
p10 &lt;- volcano_plot_enrich(de_gene,k = 10,labels = k10_genes,
                           highlight = factor(k10_highlight),ymax = 200)
plot_grid(p9,p10,nrow = 1,ncol = 2)</code></pre>
<p><img src="figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-2-1">
Past versions of volcano-plots-2-1.png
</button>
</p>
<div id="fig-volcano-plots-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b98cc33f770a652463e9ee2fb957fc85d2bd0211/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-2-1.png" target="_blank">b98cc33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7419f5c76dbe89b325b2c7e8295aad14fe8f243b/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-2-1.png" target="_blank">7419f5c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/648fbbc7255adfa4b6080eaab2a00c01f0377542/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-2-1.png" target="_blank">648fbbc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3912476a1e5e983079846dfce7c21b751863a0d7/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-2-1.png" target="_blank">3912476</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-09-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These volcano plots summarize the gene enrichment results for the
remaining topics.</p>
<p>To do:</p>
<ul>
<li>Create volcano plots for paper. In particular, volcano plots for
topics 9 and 10 will go in the main text, and the remaining will go in
the supplement. Also, fill out “selected top genes”.</li>
</ul>
<pre class="r"><code>k &lt;- 4
label_genes &lt;- (de_gene$logLR[,k] &gt; 40 &amp; de_gene$coef[,k] &gt; 1.75) |
               (de_gene$logLR[,k] &gt; 10 &amp; de_gene$coef[,k] &gt; 2)
k4_genes &lt;- genes
k4_genes[!label_genes] &lt;- &quot;&quot;
p4 &lt;- volcano_plot_enrich(de_gene,k = 4,labels = k4_genes)
k &lt;- 6
label_genes &lt;- (de_gene$logLR[,k] &gt; 20) |
               (de_gene$logLR[,k] &gt; 10 &amp; de_gene$coef[,k] &gt; 2)
k6_genes &lt;- genes
k6_genes[!label_genes] &lt;- &quot;&quot;
p6 &lt;- volcano_plot_enrich(de_gene,k = 6,labels = k6_genes,ymax = 40)
k &lt;- 7
label_genes &lt;- (de_gene$logLR[,k] &gt; 75 &amp; de_gene$coef[,k] &gt; 0) |
               (de_gene$logLR[,k] &gt; 10 &amp; de_gene$coef[,k] &gt; 1.75) |
               (grepl(&quot;Zfp&quot;,gene_info$Symbol,fixed = TRUE) &amp;
                de_gene$logLR[,k] &gt; 15)
k7_genes &lt;- genes
k7_genes[!label_genes] &lt;- &quot;&quot;
p7 &lt;- volcano_plot_enrich(de_gene,k = 7,labels = k7_genes,ymax = 200)
plot_grid(p4,p6,p7,nrow = 2,ncol = 2)</code></pre>
<p><img src="figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-3-1">
Past versions of volcano-plots-3-1.png
</button>
</p>
<div id="fig-volcano-plots-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b98cc33f770a652463e9ee2fb957fc85d2bd0211/docs/figure/cusanovich2018_kidney_k10.Rmd/volcano-plots-3-1.png" target="_blank">b98cc33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-10-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Diagram from <a href="https://www.osmosis.org">osmosis.com</a>:</p>
<pre><code># Warning: Removed 2 rows containing missing values (geom_point).
# Warning: Removed 1 rows containing missing values (geom_point).

# Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<div class="figure">
<img src="kidney.jpg" alt="" />
<p class="caption">kidney diagram</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.1.1      ggrepel_0.9.1      ggplot2_3.3.6      fastTopics_0.6-131
# [5] pathways_0.1-20    Matrix_1.4-2       workflowr_1.7.0   
# 
# loaded via a namespace (and not attached):
#  [1] mcmc_0.9-6          fs_1.5.2            progress_1.2.2     
#  [4] httr_1.4.2          rprojroot_1.3-2     tools_3.6.2        
#  [7] backports_1.1.5     bslib_0.3.1         utf8_1.1.4         
# [10] R6_2.4.1            irlba_2.3.3         uwot_0.1.10        
# [13] DBI_1.1.0           lazyeval_0.2.2      colorspace_1.4-1   
# [16] withr_2.5.0         tidyselect_1.1.1    gridExtra_2.3      
# [19] prettyunits_1.1.1   processx_3.5.2      compiler_3.6.2     
# [22] git2r_0.29.0        quantreg_5.54       SparseM_1.78       
# [25] plotly_4.9.2        labeling_0.3        sass_0.4.0         
# [28] scales_1.1.0        SQUAREM_2017.10-1   quadprog_1.5-8     
# [31] callr_3.7.0         pbapply_1.5-1       mixsqp_0.3-46      
# [34] systemfonts_1.0.2   stringr_1.4.0       digest_0.6.23      
# [37] rmarkdown_2.11      MCMCpack_1.4-5      pkgconfig_2.0.3    
# [40] htmltools_0.5.2     highr_0.8           fastmap_1.1.0      
# [43] invgamma_1.1        htmlwidgets_1.5.1   rlang_0.4.11       
# [46] rstudioapi_0.13     farver_2.0.1        jquerylib_0.1.4    
# [49] generics_0.0.2      jsonlite_1.7.2      BiocParallel_1.18.1
# [52] dplyr_1.0.7         magrittr_2.0.1      Rcpp_1.0.8         
# [55] munsell_0.5.0       fansi_0.4.0         lifecycle_1.0.0    
# [58] stringi_1.4.3       whisker_0.4         yaml_2.2.0         
# [61] MASS_7.3-51.4       Rtsne_0.15          grid_3.6.2         
# [64] parallel_3.6.2      promises_1.1.0      crayon_1.4.1       
# [67] lattice_0.20-38     hms_1.1.0           knitr_1.37         
# [70] ps_1.6.0            pillar_1.6.2        fgsea_1.15.1       
# [73] fastmatch_1.1-0     glue_1.4.2          evaluate_0.14      
# [76] getPass_0.2-2       data.table_1.12.8   RcppParallel_5.1.5 
# [79] vctrs_0.3.8         httpuv_1.5.2        MatrixModels_0.4-1 
# [82] gtable_0.3.0        purrr_0.3.4         tidyr_1.1.3        
# [85] assertthat_0.2.1    ashr_2.2-54         xfun_0.29          
# [88] coda_0.19-3         later_1.0.0         ragg_0.3.1         
# [91] viridisLite_0.3.0   truncnorm_1.0-8     tibble_3.1.3       
# [94] ellipsis_0.3.2</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
