<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Gene analysis using topic modeling results for Cusanovich et al (2018) scATAC-seq data</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Gene analysis using topic modeling results for Cusanovich <em>et al</em> (2018) scATAC-seq data</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree2edcd80ae46da8a574639bd407ed0c61367d0851targetblank2edcd80a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/2edcd80ae46da8a574639bd407ed0c61367d0851" target="_blank">2edcd80</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree2edcd80ae46da8a574639bd407ed0c61367d0851targetblank2edcd80a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/2edcd80ae46da8a574639bd407ed0c61367d0851" target="_blank">2edcd80</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/plotly/Buenrostro_2018_Chen2019pipeline/

Untracked files:
    Untracked:  analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/fit_models_Cusanovich2018_tissues.sh

Unstaged changes:
    Modified:   analysis/assess_fits_Cusanovich2018.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   analysis/process_data_Cusanovich2018.Rmd
    Modified:   scripts/fit_poisson_nmf.sbatch
    Modified:   scripts/prefit_poisson_nmf.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gene_analysis_Cusanovich2018.Rmd</code>) and HTML (<code>docs/gene_analysis_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/2edcd80ae46da8a574639bd407ed0c61367d0851/analysis/gene_analysis_Cusanovich2018.Rmd" target="_blank">2edcd80</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-04
</td>
<td>
added marker gene plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/5e832d255d83249a393569a12c556494251896a0/docs/gene_analysis_Cusanovich2018.html" target="_blank">5e832d2</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/8a4475242196bbd642c7f3e66d374a35146124f7/analysis/gene_analysis_Cusanovich2018.Rmd" target="_blank">8a44752</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
<td>
minor update on highlighting the gene sets of GSEA results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/gene_analysis_Cusanovich2018.html" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/8f3685f1574ddddb04d9f98bad4e667a2e0a0e73/analysis/gene_analysis_Cusanovich2018.Rmd" target="_blank">8f3685f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
<td>
update figure directory and links to plotly html files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform gene analysis for the Cusanovich <em>et al</em> (2018) scATAC-seq result inferred from the multinomial topic model with <span class="math inline">\(k = 13\)</span>.</p>
<div id="load-packages-and-some-functions-used-in-this-analysis" class="section level2">
<h2>Load packages and some functions used in this analysis</h2>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape)
source(&quot;code/plots.R&quot;)</code></pre>
</div>
<div id="load-samples-and-topic-model-k-13-results." class="section level2">
<h2>Load samples and topic model (<span class="math inline">\(k = 13\)</span>) results.</h2>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/&quot;
samples &lt;- readRDS(paste0(out.dir, &quot;/samples-clustering-Cusanovich2018.rds&quot;))</code></pre>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
<div id="visualize-by-structure-plot-grouped-by-tissues" class="section level2">
<h2>Visualize by Structure plot grouped by tissues</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
rows &lt;- sample(nrow(fit$L),4000)
samples$tissue &lt;- as.factor(samples$tissue)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = rows),
                              grouping = samples[rows, &quot;tissue&quot;],n = Inf,gap = 40,
                              perplexity = 50,topics = 1:13,colors = colors_topics,
                              num_threads = 4,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-tissue-1">
Past versions of structure-plot-tissue-1.png
</button>
</p>
<div id="fig-structure-plot-tissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/figure/gene_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Load the clustering results</p>
<pre class="r"><code>set.seed(10)
rows &lt;- sample(nrow(fit$L),4000)

p.structure.kmeans &lt;- structure_plot(select(fit_multinom,loadings = rows),
                                     grouping = samples$cluster_kmeans[rows],n = Inf,gap = 40,
                                     perplexity = 50,topics = 1:13,colors = colors_topics,
                                     num_threads = 4,verbose = FALSE)
print(p.structure.kmeans)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/structure-plot-kmeans-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Distribution of tissue labels by cluster.</p>
<pre class="r"><code>freq_table_cluster_tissue &lt;- with(samples,table(tissue,cluster_kmeans))

freq_table_cluster_tissue &lt;- as.data.frame.matrix(freq_table_cluster_tissue)
DT::datatable(freq_table_cluster_tissue, 
              options = list(pageLength = nrow(freq_table_cluster_tissue)), 
              rownames = T, caption = &quot;Number of cells&quot;)

create_celllabel_cluster_heatmap(samples$tissue, samples$cluster_kmeans, normalize_by = &quot;column&quot;)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/freq-table-cluster-tissue-1.png" width="720" style="display: block; margin: auto;" /><div id="htmlwidget-59411a0818dfc507bcdb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-59411a0818dfc507bcdb">{"x":{"filter":"none","caption":"<caption>Number of cells<\/caption>","data":[["BoneMarrow","Cerebellum","Heart","Kidney","LargeIntestine","Liver","Lung","PreFrontalCortex","SmallIntestine","Spleen","Testes","Thymus","WholeBrain"],[2455,0,0,0,0,0,2,0,0,43,240,0,0],[15,51,1374,757,73,268,1093,75,22,3,19,2,116],[0,0,3768,0,1,0,5,0,0,0,0,1,0],[1675,37,194,95,178,136,4002,215,84,2937,3,172,196],[0,525,3,0,4,0,2,1169,0,0,0,0,1942],[33,44,1225,402,665,84,659,48,341,5,37,9,99],[1060,191,803,98,2778,31,1010,118,2112,24,2379,32,1401],[0,0,5,2950,0,0,0,0,0,0,0,0,0],[3063,42,259,280,172,50,465,49,1457,116,38,30,47],[0,0,3,1614,441,1,1521,0,1,0,0,1,0],[102,0,13,19,94,34,773,1,27,892,0,7363,1],[0,15,3,215,2680,4,434,19,18,0,4,7,12],[0,1177,0,0,0,1,2,4,0,0,0,0,2988],[0,196,0,0,0,0,2,4261,0,0,0,0,1964],[0,0,0,1,0,5558,26,0,15,0,3,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>c1<\/th>\n      <th>c2<\/th>\n      <th>c3<\/th>\n      <th>c4<\/th>\n      <th>c5<\/th>\n      <th>c6<\/th>\n      <th>c7<\/th>\n      <th>c8<\/th>\n      <th>c9<\/th>\n      <th>c10<\/th>\n      <th>c11<\/th>\n      <th>c12<\/th>\n      <th>c13<\/th>\n      <th>c14<\/th>\n      <th>c15<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":13,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,13,25,50,100]}},"evals":[],"jsHooks":[]}</script></p>
<p>Distribution of cell labels by cluster.</p>
<pre class="r"><code>freq_table_cluster_celllabel &lt;- with(samples,table(cell_label,cluster_kmeans))

freq_table_cluster_celllabel &lt;- as.data.frame.matrix(freq_table_cluster_celllabel)
DT::datatable(freq_table_cluster_celllabel, 
              options = list(pageLength = nrow(freq_table_cluster_celllabel)), 
              rownames = T, caption = &quot;Number of cells&quot;)

create_celllabel_cluster_heatmap(samples$cell_label, samples$cluster_kmeans, normalize_by = &quot;column&quot;)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/freq-table-cluster-celllabel-1.png" width="720" style="display: block; margin: auto;" /><div id="htmlwidget-a01ef22ab629f6d8e82d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a01ef22ab629f6d8e82d">{"x":{"filter":"none","caption":"<caption>Number of cells<\/caption>","data":[["Activated B cells","Alveolar macrophages","Astrocytes","B cells","Cardiomyocytes","Cerebellar granule cells","Collecting duct","Collisions","DCT/CD","Dendritic cells","Distal convoluted tubule","Endothelial I (glomerular)","Endothelial I cells","Endothelial II cells","Enterocytes","Erythroblasts","Ex. neurons CPN","Ex. neurons CThPN","Ex. neurons SCPN","Hematopoietic progenitors","Hepatocytes","Immature B cells","Inhibitory neurons","Loop of henle","Macrophages","Microglia","Monocytes","NK cells","Oligodendrocytes","Podocytes","Proximal tubule","Proximal tubule S3","Purkinje cells","Regulatory T cells","SOM+ Interneurons","Sperm","T cells","Type I pneumocytes","Type II pneumocytes","Unknown"],[0,0,0,0,0,0,0,6,0,1,0,0,0,0,0,2420,0,0,0,0,0,0,0,0,0,0,69,0,0,0,0,0,0,3,0,240,1,0,0,0],[0,0,0,0,15,0,0,2,0,0,0,386,512,2405,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,369,1,0,0,0,0,0,1,0,8,167],[0,0,0,0,3773,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2],[477,507,0,5086,1,9,0,340,0,840,0,0,0,15,0,37,0,0,0,26,2,353,3,0,464,385,1019,231,1,0,1,0,0,17,0,0,35,1,0,74],[0,0,1574,0,0,0,0,423,0,0,0,0,0,0,1,0,0,0,0,0,0,0,68,0,0,1,0,0,1506,0,0,0,1,0,7,0,1,0,0,63],[0,2,11,3,42,3,10,15,1,2,0,158,438,574,11,0,0,0,0,1,5,0,27,8,14,3,1,1,1,113,16,0,2,1,8,1,7,3,14,2155],[0,16,58,648,169,306,0,166,0,12,0,0,1,7,1585,79,10,91,40,899,11,0,754,2,191,31,50,2,30,0,33,2,50,7,66,1814,104,0,0,4803],[0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,2361,592,0,0,0,0,0,0,0,0],[23,34,20,34,76,19,5,64,6,102,6,8,1,15,54,275,0,0,32,2499,44,218,8,11,42,1,32,16,17,16,132,0,0,4,5,32,50,27,1,2139],[0,0,0,0,0,0,124,0,441,0,301,0,0,0,442,0,0,0,0,0,0,0,0,718,0,0,0,0,0,0,7,0,0,0,0,0,0,1415,66,68],[0,0,0,0,0,0,0,11,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,2,71,0,0,0,0,0,475,0,0,8753,6,0,0],[0,0,0,1,0,0,25,8,57,1,12,0,0,1,2690,0,0,0,0,0,0,0,0,74,0,0,0,0,0,0,19,0,0,0,0,2,2,170,68,281],[0,0,0,0,0,3762,0,97,0,0,0,0,0,1,0,0,0,0,0,0,0,0,3,0,0,0,0,0,0,0,0,0,262,0,1,0,0,0,0,46],[0,0,3,0,0,0,0,86,0,0,0,0,0,0,0,0,1822,1449,1394,0,0,0,965,0,0,0,0,0,3,0,0,0,5,0,466,0,0,0,0,230],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,5602,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>c1<\/th>\n      <th>c2<\/th>\n      <th>c3<\/th>\n      <th>c4<\/th>\n      <th>c5<\/th>\n      <th>c6<\/th>\n      <th>c7<\/th>\n      <th>c8<\/th>\n      <th>c9<\/th>\n      <th>c10<\/th>\n      <th>c11<\/th>\n      <th>c12<\/th>\n      <th>c13<\/th>\n      <th>c14<\/th>\n      <th>c15<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":40,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,40,50,100]}},"evals":[],"jsHooks":[]}</script></p>
<p>Top 5 cell types</p>
<pre class="r"><code>top_celltypes_table &lt;- data.frame(matrix(nrow=5, ncol = ncol(freq_table_cluster_celllabel)))
colnames(top_celltypes_table) &lt;- colnames(freq_table_cluster_celllabel)
for (k in 1:ncol(freq_table_cluster_celllabel)){
  top_celltypes &lt;- rownames(freq_table_cluster_celllabel)[head(order(freq_table_cluster_celllabel[,k], decreasing=TRUE), 5)]
  freq_top_celltypes &lt;- freq_table_cluster_celllabel[top_celltypes, k]
  percent_top_celltypes &lt;- freq_table_cluster_celllabel[top_celltypes, k]/sum(freq_table_cluster_celllabel[,k])
  top_celltypes_table[,k] &lt;- sprintf(&quot;%s (%.1f%%)&quot;, top_celltypes, percent_top_celltypes*100)
}

DT::datatable(top_celltypes_table, rownames = T, caption = &quot;Top 5 cell types in each cluster&quot;)</code></pre>
<div id="htmlwidget-49186b68cba9b93e8d46" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-49186b68cba9b93e8d46">{"x":{"filter":"none","caption":"<caption>Top 5 cell types in each cluster<\/caption>","data":[["1","2","3","4","5"],["Erythroblasts (88.3%)","Sperm (8.8%)","Monocytes (2.5%)","Collisions (0.2%)","Regulatory T cells (0.1%)"],["Endothelial II cells (62.2%)","Endothelial I cells (13.2%)","Endothelial I (glomerular) (10.0%)","Podocytes (9.5%)","Unknown (4.3%)"],["Cardiomyocytes (99.9%)","Unknown (0.1%)","Activated B cells (0.0%)","Alveolar macrophages (0.0%)","Astrocytes (0.0%)"],["B cells (51.2%)","Monocytes (10.3%)","Dendritic cells (8.5%)","Alveolar macrophages (5.1%)","Activated B cells (4.8%)"],["Astrocytes (43.2%)","Oligodendrocytes (41.3%)","Collisions (11.6%)","Inhibitory neurons (1.9%)","Unknown (1.7%)"],["Unknown (59.0%)","Endothelial II cells (15.7%)","Endothelial I cells (12.0%)","Endothelial I (glomerular) (4.3%)","Podocytes (3.1%)"],["Unknown (39.9%)","Sperm (15.1%)","Enterocytes (13.2%)","Hematopoietic progenitors (7.5%)","Inhibitory neurons (6.3%)"],["Proximal tubule (79.9%)","Proximal tubule S3 (20.0%)","DCT/CD (0.0%)","Loop of henle (0.0%)","Activated B cells (0.0%)"],["Hematopoietic progenitors (41.2%)","Unknown (35.3%)","Erythroblasts (4.5%)","Immature B cells (3.6%)","Proximal tubule (2.2%)"],["Type I pneumocytes (39.5%)","Loop of henle (20.0%)","Enterocytes (12.3%)","DCT/CD (12.3%)","Distal convoluted tubule (8.4%)"],["T cells (93.9%)","Regulatory T cells (5.1%)","NK cells (0.8%)","Collisions (0.1%)","Type I pneumocytes (0.1%)"],["Enterocytes (78.9%)","Unknown (8.2%)","Type I pneumocytes (5.0%)","Loop of henle (2.2%)","Type II pneumocytes (2.0%)"],["Cerebellar granule cells (90.2%)","Purkinje cells (6.3%)","Collisions (2.3%)","Unknown (1.1%)","Inhibitory neurons (0.1%)"],["Ex. neurons CPN (28.4%)","Ex. neurons CThPN (22.6%)","Ex. neurons SCPN (21.7%)","Inhibitory neurons (15.0%)","SOM+ Interneurons (7.3%)"],["Hepatocytes (100.0%)","Unknown (0.0%)","Activated B cells (0.0%)","Alveolar macrophages (0.0%)","Astrocytes (0.0%)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>c1<\/th>\n      <th>c2<\/th>\n      <th>c3<\/th>\n      <th>c4<\/th>\n      <th>c5<\/th>\n      <th>c6<\/th>\n      <th>c7<\/th>\n      <th>c8<\/th>\n      <th>c9<\/th>\n      <th>c10<\/th>\n      <th>c11<\/th>\n      <th>c12<\/th>\n      <th>c13<\/th>\n      <th>c14<\/th>\n      <th>c15<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<p>We can see the major cell types in the clusters (topics): * cluster 1 (topic 1): Erythroblasts * cluster 2 (topic 3): Endothelial cells * cluster 3 (topic 7): Cardiomyocytes cells * cluster 4 (topic 6): B cells, Monocytes, Dendritic cells * cluster 5 (topic 9): Astrocytes, Oligodendrocytes * cluster 8 (topic 5): Proximal tubule * cluster 10 (topic 4): a mixture of pneumocytes, Loop of henle, Enterocytes, DCT/CD * cluster 11 (topic 12): T cells * cluster 13 (topic 2): Cerebellar granule cells * cluster 14 (topic 11): Ex. neurons and Inhibitory neurons * cluster 15 (topic 8): Hepatocytes</p>
</div>
<div id="gene-score-analysis" class="section level2">
<h2>Gene score analysis</h2>
<p>Set output directorry</p>
<pre class="r"><code>fig.dir &lt;- &quot;output/plotly/Cusanovich2018&quot;
dir.create(fig.dir, showWarnings = F, recursive = T)</code></pre>
<div id="tss-model" class="section level3">
<h3>TSS model</h3>
<p>Gene scores were computed using TSS based method as in Lareau <em>et al</em> Nature Biotech, 2019 as well as the <code>model 21</code> in <code>archR</code> paper. This model weights chromatin accessibility around gene promoters by using bi-directional exponential decays from the TSS.</p>
<ul>
<li>Top genes</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-TSS-sum&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
genescore_res_tss &lt;- readRDS(file.path(gene.dir, &quot;genescore_result_topics.rds&quot;))
genescore_res &lt;- genescore_res_tss

genes &lt;- genescore_res$genes
gene_mean_acc &lt;- genescore_res$colmeans
gene_scores &lt;- genescore_res$Z
gene_logFC &lt;- genescore_res$beta

topics &lt;- colnames(gene_scores)
top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- topics

for (k in topics){
  top_genes[,k] &lt;- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = &quot;Top 10 genes by abs(gene z-scores)&quot;)</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018//geneanalysis-Cusanovich2018-k=13-TSS-sum</code></pre>
<div id="htmlwidget-aef3d3705df5c6636aed" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-aef3d3705df5c6636aed">{"x":{"filter":"none","caption":"<caption>Top 10 genes by abs(gene z-scores)<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Hbb-bs","Spta1","Gypa","Slc4a1","Suco","H2afx","Clta","Eif2ak1","Ska2","Prr11"],["Dtna","Galnt13","Ptprr","Slc4a10","Tenm1","Nhlh2","Ppfia2","Kctd8","Grid2","Plcxd3"],["Gli3","Mill2","Adamts12","Lox","Col1a2","Slc10a6","Wnt2","Col5a2","Sparc","Rhoj"],["Grhl2","Cldn8","Scnn1a","Rab11fip1","Slc6a14","Tspan8","Krt8","Tacstd2","Elf3","Thbs1"],["Kap","Veph1","Col4a4","Col4a3","Glb1l2","C1qtnf3","Mfsd4b5","Aass","Cyp2j5","Olfr338"],["Rab39","H2-Eb1","H2-Ab1","H2-Aa","Trim30b","Cd53","Adgre5","Cd74","Spata6","Cd83"],["Mir133a-2","Kcnk3","Myh7","Asb2","Ldb3","Trdn","A530016L24Rik","3425401B19Rik","Myh6","Mhrt"],["Slc2a2","Itih1","Aadac","Tm6sf2","Ftcd","Fgg","Cyp2d9","C4b","Slc10a1","Hsd3b5"],["Mdga2","Grid2","Kif1a","Mir99ahg","Grin3a","Pantr1","Gpr37","Dtna","Pou3f3","Ptprz1"],["Mroh2a","Coil","Elobl","2210409E12Rik","Spag17","Trpm8","Sycp1","Gm16381","Gm2016","Gm2022"],["Pnma2","Mir99ahg","Sntg1","Tenm1","Unc79","Grm5","Opcml","March4","Slc7a14","Npy1r"],["Chil6","Cd247","Try5","Cd3d","Satb1","Itk","Cd3g","Tcf7","Spata6","Mei4"],["Ifi211","Mroh2a","Ifi204","Vwa5a","Gm9733","Olfr796","Aldoart1","Ifi205","Pabpc4l","Olfm5"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Volcano plots</li>
</ul>
<p>Explore the volcano plots [interactively][volcano-plotly-tss-k]</p>
<pre class="r"><code>for ( k in 1:ncol(genescore_res$beta) ){
  p.volcano.plotly &lt;- genescore_volcano_plotly(genescore_res,k,
                                             file = sprintf(&quot;%s/volcano_topic_%s_%s.html&quot;, fig.dir, k, &quot;tss-sum&quot;),
                                             labels = genescore_res$genes$SYMBOL)
}</code></pre>
<p><strong>Topic 1</strong></p>
<pre class="r"><code>k &lt;- 1
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gene-volcano-plot-tss-1-1">
Past versions of gene-volcano-plot-tss-1-1.png
</button>
</p>
<div id="fig-gene-volcano-plot-tss-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-1-1.png" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Check some known marker genes for endothelial cells</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;Hbb-b1&quot;, &quot;Hbb-b2&quot;, &quot;Gypa&quot;)

gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-tss-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 3</strong> (Endothelial cells)</p>
<pre class="r"><code>k &lt;- 3
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-3-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for erythrocyte</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;PECAM1&quot;, &quot;CD106&quot;, &quot;CD62E&quot;, &quot;Sele&quot;, &quot;Kdr&quot;, &quot;ENG&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-tss-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 5</strong> (Proximal tubule)</p>
<pre class="r"><code>k &lt;- 5
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-5-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Proximal tubule</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;PALDOB&quot;, &quot;CUBN&quot;, &quot;LRP2&quot;, &quot;SLC34A1&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-tss-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 7</strong> (Cardiomyocytes)</p>
<pre class="r"><code>k &lt;- 7
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-7-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Cardiomyocytes</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;Nppa&quot;, &quot;Myl4&quot;, &quot;SLN&quot;, &quot;PITX2&quot;, &quot;Myl7&quot;, &quot;Gja5&quot;, &quot;Myl2&quot;, &quot;Myl3&quot;, &quot;IRX4&quot;, &quot;HAND1&quot;, &quot;HEY2&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/3),3))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-tss-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 8</strong> (Hepatocytes)</p>
<pre class="r"><code>k &lt;- 8
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-tss-8-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Hepatocytes</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;SERPINA1&quot;, &quot;TTR&quot;, &quot;ALB&quot;,&quot;AFP&quot;,&quot;CYP3A4&quot;,&quot;CYP7A1&quot;,&quot;FABP1&quot;,&quot;ALR&quot;,&quot;Glut1&quot;,&quot;MET&quot;,&quot;FoxA1&quot;,&quot;FoxA2&quot;,&quot;CD29&quot;,&quot;PTP4A2&quot;,&quot;Prox1&quot;, &quot;HNF1B&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/3),3))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-8-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-body-model" class="section level3">
<h3>Gene body model</h3>
<p>Gene scores were computed using the gene score model (model 42) in the <code>archR</code> paper with some modifications. This model uses bi-directional exponential decays from the gene TSS (extended upstream by 5 kb by default) and the gene transcription termination site (TTS). Note: the current version of the function does not account for neighboring gene boundaries.</p>
<ul>
<li>Top genes</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-genebody-sum&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
genescore_res_gb &lt;- readRDS(file.path(gene.dir, &quot;genescore_result_topics.rds&quot;))
genescore_res &lt;- genescore_res_gb

genes &lt;- genescore_res$genes
gene_mean_acc &lt;- genescore_res$colmeans
gene_scores &lt;- genescore_res$Z
gene_logFC &lt;- genescore_res$beta

topics &lt;- colnames(gene_scores)
top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- topics

for (k in topics){
  top_genes[,k] &lt;- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = &quot;Top 10 genes by abs(gene z-scores)&quot;)</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018//geneanalysis-Cusanovich2018-k=13-genebody-sum</code></pre>
<div id="htmlwidget-4e78d746f33f1d03a2f0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4e78d746f33f1d03a2f0">{"x":{"filter":"none","caption":"<caption>Top 10 genes by abs(gene z-scores)<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Hbb-bs","Hbb-y","Olfr65","Olfr64","Olfr631","Olfr66","Zfp600","Hbb-b1","Olfr633","Olfr68"],["Olfr1324","Zic4","Zic1","Ldoc1","C130071C03Rik","Syt4","Pax6","Slitrk2","Mir9-2","Slitrk4"],["Eda2r","Col1a2","Col3a1","Mageb16","Sulf1","Rhoj","Tcf21","Adamts1","Col4a1","C1qtnf7"],["Vmn1r185","Heph","Hoxb9","Mir196a-1","Hoxb8","Hoxb6","Hoxb7","Hoxb5","Hoxb13","Irx2"],["Olfr871","Olfr872","Kap","Zfp97","Rhox7a","Rhox6","Acsm1","Zfp960","Vmn1r7","Acsm2"],["H2-Ab1","Notch4","H2-Eb1","H2-Aa","H2-Eb2","H2-Ob","Btnl2","Psmb8","Tap2","Psmb9"],["Hand2","4930415L06Rik","Rbm20","Irx4","Gata4","Myh6","Ttn","Atp2a2","Myocd","Jph2"],["F11","Mup3","Serpina1a","Klkb1","Cyp2d9","Mug1","Mup20","Sult2a8","Slc2a2","Serpina1b"],["Sox2","Pou3f4","Mir1897","Npas3","Slitrk2","C130071C03Rik","Pou3f2","Sox2ot","Olig2","Pou3f3"],["Gm16381","Gm2016","Gm2022","Smok2b","Smok2a","Gm10512","Krt19","Vmn1r214","Vmn1r213","Krt15"],["Gpr165","Foxg1","3110039M20Rik","Slitrk2","Mir124-2hg","Mir124a-2","Myt1l","Syt4","Ajap1","Rbfox3"],["Zdhhc15","Olfr1361","Hist1h2bm","Hist1h2bl","Rhoh","Hist1h3h","Hist1h2ai","Hist1h4j","Olfr1359","Hist1h4k"],["Zfp600","Gm16381","Gm2022","Gm2016","Sirpb1b","Prl2c2","Zfp982","Zfp985","Zfp979","Casp1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Volcano plots</li>
</ul>
<p>Explore the volcano plot <a href="plotly/Cusanovich2018/volcano_topic_1_genebody-sum.html">interactively</a></p>
<pre class="r"><code>for ( k in 1:ncol(genescore_res$beta) ){
  p.volcano.plotly &lt;- genescore_volcano_plotly(genescore_res,k,
                                             file = sprintf(&quot;%s/volcano_topic_%s_%s.html&quot;, fig.dir, k, &quot;genebody-sum&quot;),
                                             labels = genescore_res$genes$SYMBOL)
}</code></pre>
<p><strong>Topic 1</strong></p>
<pre class="r"><code>k &lt;- 1
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-genebody-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for endothelial cells</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;Hbb-b1&quot;, &quot;Hbb-b2&quot;, &quot;Gypa&quot;)

gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-genebody-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 3</strong> (Endothelial cells)</p>
<pre class="r"><code>k &lt;- 3
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-genebody-3-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for erythrocyte</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;PECAM1&quot;, &quot;CD106&quot;, &quot;CD62E&quot;, &quot;Sele&quot;, &quot;Kdr&quot;, &quot;ENG&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-genebody-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 5</strong> (Proximal tubule)</p>
<pre class="r"><code>k &lt;- 5
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-genebody-5-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Proximal tubule</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;PALDOB&quot;, &quot;CUBN&quot;, &quot;LRP2&quot;, &quot;SLC34A1&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/2),2))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-genebody-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 7</strong> (Cardiomyocytes)</p>
<pre class="r"><code>k &lt;- 7
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-genebody-7-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Cardiomyocytes</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;Nppa&quot;, &quot;Myl4&quot;, &quot;SLN&quot;, &quot;PITX2&quot;, &quot;Myl7&quot;, &quot;Gja5&quot;, &quot;Myl2&quot;, &quot;Myl3&quot;, &quot;IRX4&quot;, &quot;HAND1&quot;, &quot;HEY2&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/3),3))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-genebody-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Topic 8</strong> (Hepatocytes)</p>
<pre class="r"><code>k &lt;- 8
genescore_volcano_plot(genescore_res, k, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gene-volcano-plot-genebody-8-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>Check some known marker genes for Hepatocytes</p>
<pre class="r"><code>marker_genes &lt;- c(&quot;SERPINA1&quot;, &quot;TTR&quot;, &quot;ALB&quot;,&quot;AFP&quot;,&quot;CYP3A4&quot;,&quot;CYP7A1&quot;,&quot;FABP1&quot;,&quot;ALR&quot;,&quot;Glut1&quot;,&quot;MET&quot;,&quot;FoxA1&quot;,&quot;FoxA2&quot;,&quot;CD29&quot;,&quot;PTP4A2&quot;,&quot;Prox1&quot;, &quot;HNF1B&quot;)
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genescore_res$genes$SYMBOL
marker_gene_scores &lt;- gene_scores[grep(paste(sprintf(&quot;^%s$&quot;, marker_genes), collapse = &quot;|&quot;), rownames(gene_scores), ignore.case = T),]

par(mfrow = c(ceiling(nrow(marker_gene_scores)/3),3))
for(i in 1:nrow(marker_gene_scores)){
  barplot(marker_gene_scores[i,], xlab = &quot;topics&quot;, ylab = &quot;gene score&quot;, main = rownames(marker_gene_scores)[i], col = colors_topics)
}</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/marker-gene-scores-genebody-8-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-gene-scores-from-the-gene-body-model-vs.-the-tss-model." class="section level3">
<h3>Compare gene scores from the gene-body model vs. the TSS model.</h3>
<pre class="r"><code>m     &lt;- ncol(genescore_res_gb$Z)
plots &lt;- vector(&quot;list&quot;,m)
names(plots) &lt;- colnames(genescore_res_gb$Z)
for (i in 1:m) {
  dat &lt;- data.frame(genebody = genescore_res_gb$Z[,i], tss = genescore_res_tss$Z[,i])
  plots[[i]] &lt;- 
    ggplot(dat,aes_string(x = &quot;genebody&quot;,y = &quot;tss&quot;)) +
    geom_point(shape = 21, na.rm = TRUE, size = 1, alpha = 1/10) +
    geom_abline(intercept = 0, slope = 1, color=&quot;blue&quot;) + 
    labs(x = &quot;gene body model&quot;,y = &quot;TSS model&quot;, 
         title = paste(&quot;topic&quot;,i)) +
    theme_cowplot(9)
}

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/compare-genebody-tss-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare-genebody-tss-1">
Past versions of compare-genebody-tss-1.png
</button>
</p>
<div id="fig-compare-genebody-tss-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/figure/gene_analysis_Cusanovich2018.Rmd/compare-genebody-tss-1.png" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2>Gene-set enrichment analysis (GSEA)</h2>
<div id="tss-model-1" class="section level3">
<h3>TSS model</h3>
<p>Top gene sets/pathways</p>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-TSS-sum&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_IDs_up &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_up[,k] &lt;- paste0(top_IDs_up$name, &quot;(&quot;, top_IDs_up$id, &quot;)&quot;)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_IDs_down &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_down[,k] &lt;- paste0(top_IDs_down$name, &quot;(&quot;, top_IDs_down$id, &quot;)&quot;)
  
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018//geneanalysis-Cusanovich2018-k=13-TSS-sum</code></pre>
<div id="htmlwidget-b9cc5042315085f2f32a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b9cc5042315085f2f32a">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["HU_FETAL_RETINA_BLOOD(M39263)","REACTOME_M_PHASE(M27662)","IVANOVA_HEMATOPOIESIS_MATURE_CELL(M11205)","ZHONG_PFC_C1_OPC(M39096)","FAN_EMBRYONIC_CTX_NSC_2(M39036)","GO_CHROMOSOME_SEGREGATION(M19628)","HALLMARK_HEME_METABOLISM(M5945)","REACTOME_CELL_CYCLE_CHECKPOINTS(M16647)","FAN_EMBRYONIC_CTX_MICROGLIA_1(M39041)","Systemic lupus erythematosus(83315)"],["MANNO_MIDBRAIN_NEUROTYPES_HNBM(M39063)","GAO_LARGE_INTESTINE_ADULT_CA_ENTEROENDOCRINE_CELLS(M39161)","GO_NEURON_TO_NEURON_SYNAPSE(M26002)","REACTOME_NEURONAL_SYSTEM(M735)","GO_POSTSYNAPTIC_MEMBRANE(M17414)","ZHONG_PFC_C3_ASTROCYTE(M39102)","KRAS.KIDNEY_UP.V1_UP(M2892)","MEISSNER_NPC_HCP_WITH_H3K4ME2_AND_H3K27ME3(M1949)","MURARO_PANCREAS_DELTA_CELL(M39170)","GO_SYNAPSE_ASSEMBLY(M10639)"],["PASINI_SUZ12_TARGETS_DN(M2293)","CUI_DEVELOPING_HEART_C3_FIBROBLAST_LIKE_CELL(M39300)","AIZARANI_LIVER_C9_LSECS_1(M39113)","GO_REGULATION_OF_VASCULATURE_DEVELOPMENT(M16954)","BERENJENO_TRANSFORMED_BY_RHOA_DN(M7514)","AIZARANI_LIVER_C20_LSECS_3(M39121)","GO_CELL_SUBSTRATE_ADHESION(M15640)","MODULE_12(M18929)","GAO_LARGE_INTESTINE_24W_C1_DCLK1POS_PROGENITOR(M39152)","CUI_DEVELOPING_HEART_C4_ENDOTHELIAL_CELL(M39301)"],["HOLLERN_EMT_BREAST_TUMOR_DN(M624)","AIZARANI_LIVER_C7_EPCAM_POS_BILE_DUCT_CELLS_2(M39111)","COLDREN_GEFITINIB_RESISTANCE_DN(M6744)","MCBRYAN_PUBERTAL_BREAST_4_5WK_UP(M19432)","AIZARANI_LIVER_C24_EPCAM_POS_BILE_DUCT_CELLS_3(M39125)","CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP(M12795)","AIZARANI_LIVER_C4_EPCAM_POS_BILE_DUCT_CELLS_1(M39108)","JAEGER_METASTASIS_DN(M10702)","MODULE_342(M16667)","GO_APICAL_PLASMA_MEMBRANE(M1446)"],["GO_APICAL_PLASMA_MEMBRANE(M1446)","GO_ORGANIC_ACID_CATABOLIC_PROCESS(M13128)","MEBARKI_HCC_PROGENITOR_FZD8CRD_DN(M38993)","AIZARANI_LIVER_C14_HEPATOCYTES_2(M39118)","LAKE_ADULT_KIDNEY_C3_PROXIMAL_TUBULE_EPITHELIAL_CELLS_S1_S2(M39222)","HP_ABNORMAL_URINE_PH(M38554)","AIZARANI_LIVER_C11_HEPATOCYTES_1(M39115)","OHGUCHI_LIVER_HNF4A_TARGETS_DN(M2194)","LAKE_ADULT_KIDNEY_C5_PROXIMAL_TUBULE_EPITHELIAL_CELLS_STRESS_INFLAM(M39224)","GO_ORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY(M18663)"],["CUI_DEVELOPING_HEART_C8_MACROPHAGE(M39305)","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA(M39022)","GAO_LARGE_INTESTINE_24W_C11_PANETH_LIKE_CELL(M39160)","GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS(M39166)","GSE33424_CD161_HIGH_VS_NEG_CD8_TCELL_UP(M8541)","AIZARANI_LIVER_C2_KUPFFER_CELLS_1(M39106)","AIZARANI_LIVER_C25_KUPFFER_CELLS_4(M39126)","GSE43863_DAY6_EFF_VS_DAY150_MEM_LY6C_INT_CXCR5POS_CD4_TCELL_DN(M9733)","GO_ADAPTIVE_IMMUNE_RESPONSE(M13847)","GO_B_CELL_ACTIVATION(M10657)"],["GO_CARDIAC_MUSCLE_CONTRACTION(M24087)","GO_MUSCLE_CELL_DEVELOPMENT(M16122)","GO_MUSCLE_ORGAN_DEVELOPMENT(M774)","KAAB_HEART_ATRIUM_VS_VENTRICLE_DN(M2584)","HALLMARK_MYOGENESIS(M5909)","HP_EMG_ABNORMALITY(M36311)","HP_ABNORMALITY_OF_CIRCULATING_ENZYME_LEVEL(M37737)","GO_REGULATION_OF_MUSCLE_SYSTEM_PROCESS(M15722)","GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN(M10033)","CTAWWWATA_RSRFC4_Q2(M16022)"],["GO_LIPID_CATABOLIC_PROCESS(M15533)","GO_CELLULAR_AMINO_ACID_METABOLIC_PROCESS(M9857)","GO_ALPHA_AMINO_ACID_METABOLIC_PROCESS(M10829)","GNF2_CEBPA(M3221)","REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES(M727)","CAR_IGFBP1(M18034)","GO_BLOOD_MICROPARTICLE(M17273)","GO_MONOCARBOXYLIC_ACID_BIOSYNTHETIC_PROCESS(M14020)","Cytochrome P450 - arranged by substrate type(1367856)","WP_NUCLEAR_RECEPTORS_METAPATHWAY(M39428)"],["ZHONG_PFC_MAJOR_TYPES_ASTROCYTES(M39074)","MANNO_MIDBRAIN_NEUROTYPES_HRN(M39065)","MANNO_MIDBRAIN_NEUROTYPES_HNBML1(M39064)","GO_SYNAPTIC_MEMBRANE(M17824)","MANNO_MIDBRAIN_NEUROTYPES_HNBM(M39063)","MANNO_MIDBRAIN_NEUROTYPES_HOMTN(M39073)","GAO_LARGE_INTESTINE_ADULT_CA_ENTEROENDOCRINE_CELLS(M39161)","GO_NEURON_TO_NEURON_SYNAPSE(M26002)","GO_GLUTAMATERGIC_SYNAPSE(M25999)","GO_POSTSYNAPTIC_MEMBRANE(M17414)"],["Keratinization(1457932)","GO_KERATINIZATION(M15449)","REACTOME_KERATINIZATION(M27640)","GO_KERATINOCYTE_DIFFERENTIATION(M29131)","REACTOME_FORMATION_OF_THE_CORNIFIED_ENVELOPE(M27649)","GO_EPIDERMAL_CELL_DIFFERENTIATION(M12194)","REACTOME_CLASS_A_1_RHODOPSIN_LIKE_RECEPTORS_(M18334)","GO_CORNIFICATION(M24369)","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON(M17620)","MIKKELSEN_MEF_HCP_WITH_H3_UNMETHYLATED(M2020)"],["REACTOME_TRANSMISSION_ACROSS_CHEMICAL_SYNAPSES(M15514)","GO_VOLTAGE_GATED_CATION_CHANNEL_ACTIVITY(M17981)","ZHONG_PFC_C3_ASTROCYTE(M39102)","MANNO_MIDBRAIN_NEUROTYPES_HOPC(M39052)","GO_REGULATION_OF_SYNAPSE_STRUCTURE_OR_ACTIVITY(M14201)","GO_REGULATION_OF_POSTSYNAPTIC_MEMBRANE_POTENTIAL(M12831)","GO_POSTSYNAPTIC_SPECIALIZATION_MEMBRANE(M26026)","GO_CELL_CELL_ADHESION_VIA_PLASMA_MEMBRANE_ADHESION_MOLECULES(M16429)","GO_NEUROTRANSMITTER_RECEPTOR_ACTIVITY(M18960)","GO_POTASSIUM_CHANNEL_ACTIVITY(M18977)"],["GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS(M39166)","CUI_DEVELOPING_HEART_C9_B_T_CELL(M39306)","AIZARANI_LIVER_C1_NK_NKT_CELLS_1(M39105)","FAN_EMBRYONIC_CTX_BRAIN_NAIVE_LIKE_T_CELL(M39047)","WALLACE_PROSTATE_CANCER_RACE_UP(M10319)","AIZARANI_LIVER_C3_NK_NKT_CELLS_2(M39107)","HAY_BONE_MARROW_NAIVE_T_CELL(M39202)","GSE2405_S_AUREUS_VS_UNTREATED_NEUTROPHIL_DN(M6205)","MODULE_46(M15983)","Systemic lupus erythematosus(83315)"],["GO_ODORANT_BINDING(M19181)","GPCR ligand binding(1367054)","G alpha (i) signalling events(1367086)","chr11q12(M5761)","Prolactin receptor signaling(1367010)","Growth hormone receptor signaling(1367608)","Class C/3 (Metabotropic glutamate/pheromone receptors)(1367083)","Taste transduction(83285)","chr11q11(M3966)","chr11p15(M9514)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Scatterplot of GSEA results <strong>Topic 1</strong> (highlighting &quot;Erythroblast&quot; cells)</li>
</ul>
<pre class="r"><code>k   &lt;- 1
name_interest &lt;- &quot;erythroblast&quot;
label_geneset_ids &lt;- gene_set_info[gsea_res$pval[,k] &lt; 1e-6 &amp; grepl(name_interest, gene_set_info$name, ignore.case = T), &quot;id&quot;]
cat(sprintf(&quot;highlighting %d %s gene sets \n&quot;, length(label_geneset_ids), name_interest))

create_genescore_gsea_plot(gene_set_info, gsea_res, k, label_geneset_ids, 
                                     title = sprintf(&quot;topic %d GSEA plot&quot;,k))</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-tss-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gsea-plot-tss-1-1">
Past versions of gsea-plot-tss-1-1.png
</button>
</p>
<div id="fig-gsea-plot-tss-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/5e832d255d83249a393569a12c556494251896a0/docs/figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-tss-1-1.png" target="_blank">5e832d2</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-tss-1-1.png" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>if(length(label_geneset_ids) &gt; 0)
   DT::datatable(gene_set_info[gene_set_info$id %in% label_geneset_ids, 
                               c(&quot;name&quot;, &quot;id&quot;, &quot;database&quot;, &quot;category_code&quot;, &quot;description_brief&quot;)],  
                 rownames = F)</code></pre>
<pre><code># highlighting 3 erythroblast gene sets</code></pre>
<div id="htmlwidget-6c63caabe14d1889eb58" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6c63caabe14d1889eb58">{"x":{"filter":"none","data":[["GSE27786_LIN_NEG_VS_ERYTHROBLAST_DN","GSE27786_ERYTHROBLAST_VS_NEUTROPHIL_UP","GSE27786_ERYTHROBLAST_VS_MONO_MAC_UP"],["M4793","M4875","M4877"],["MSigDB","MSigDB","MSigDB"],["C7","C7","C7"],["Genes down-regulated in comparison of lineage negative versus erythroblasts.","Genes up-regulated in comparison of erythroblasts versus neutrophils.","Genes up-regulated in comparison of erythroblasts versus monocyte macrophages."]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>name<\/th>\n      <th>id<\/th>\n      <th>database<\/th>\n      <th>category_code<\/th>\n      <th>description_brief<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Explore the GSEA plot <a href="plotly/Cusanovich2018/gsea_topic_1_tss-sum.html">interactively</a></p>
<pre class="r"><code>for ( k in 1:ncol(gsea_res$pval)){
  p.gsea.plotly &lt;- create_genescore_gsea_plotly(gene_set_info, gsea_res, k)
  saveWidget(p.gsea.plotly,file = sprintf(&quot;%s/gsea_topic_%s_%s.html&quot;, fig.dir, k, &quot;tss-sum&quot;),
             selfcontained = TRUE,title = &quot;GSEA&quot;)
}</code></pre>
</div>
<div id="gene-body-model-1" class="section level3">
<h3>Gene body model</h3>
<ul>
<li>Top gene sets</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-genebody-sum&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_pathways_up[,k] &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
    unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_pathways_down[,k] &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
    unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018//geneanalysis-Cusanovich2018-k=13-genebody-sum</code></pre>
<div id="htmlwidget-49d1d76ae7db2bc1aeb7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-49d1d76ae7db2bc1aeb7">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Systemic lupus erythematosus.83315","DNA methylation.1367468","ERCC6 (CSB) and EHMT2 (G9a) positively regulate rRNA expression.1367464","NoRC negatively regulates rRNA expression.1368132","NoRC negatively regulates rRNA expression.1367467","Negative epigenetic regulation of rRNA expression.1367465","Meiotic synapsis.1367999","Meiotic Synapsis.1368131","Meiotic Recombination.1368130","HU_FETAL_RETINA_BLOOD.M39263"],["MANNO_MIDBRAIN_NEUROTYPES_HRN.M39065","GO_GLUTAMATERGIC_SYNAPSE.M25999","MANNO_MIDBRAIN_NEUROTYPES_HNBML1.M39064","GO_NEURON_TO_NEURON_SYNAPSE.M26002","GAO_LARGE_INTESTINE_ADULT_CA_ENTEROENDOCRINE_CELLS.M39161","MANNO_MIDBRAIN_NEUROTYPES_HNBM.M39063","REACTOME_NEURONAL_SYSTEM.M735","GO_POSTSYNAPTIC_MEMBRANE.M17414","ZHONG_PFC_C3_ASTROCYTE.M39102","GNF2_DNM1.M3373"],["CUI_DEVELOPING_HEART_C3_FIBROBLAST_LIKE_CELL.M39300","AIZARANI_LIVER_C9_LSECS_1.M39113","REACTOME_EXTRACELLULAR_MATRIX_ORGANIZATION.M610","NABA_CORE_MATRISOME.M5884","ONDER_CDH1_TARGETS_2_UP.M13867","GO_EXTRACELLULAR_MATRIX_STRUCTURAL_CONSTITUENT.M12927","Extracellular matrix organization.1367333","MODULE_2.M9982","AIZARANI_LIVER_C20_LSECS_3.M39121","PASINI_SUZ12_TARGETS_DN.M2293"],["AIZARANI_LIVER_C7_EPCAM_POS_BILE_DUCT_CELLS_2.M39111","AIZARANI_LIVER_C24_EPCAM_POS_BILE_DUCT_CELLS_3.M39125","AIZARANI_LIVER_C39_EPCAM_POS_BILE_DUCT_CELLS_4.M39135","MCBRYAN_PUBERTAL_BREAST_4_5WK_UP.M19432","COLDREN_GEFITINIB_RESISTANCE_DN.M6744","CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP.M12795","AIZARANI_LIVER_C4_EPCAM_POS_BILE_DUCT_CELLS_1.M39108","GO_EPIDERMIS_DEVELOPMENT.M14065","GO_SKIN_DEVELOPMENT.M15889","GO_EPIDERMAL_CELL_DIFFERENTIATION.M12194"],["Biological oxidations.1367854","REACTOME_BIOLOGICAL_OXIDATIONS.M10320","LAKE_ADULT_KIDNEY_C3_PROXIMAL_TUBULE_EPITHELIAL_CELLS_S1_S2.M39222","GO_ORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY.M18663","GO_FATTY_ACID_METABOLIC_PROCESS.M17829","MENON_FETAL_KIDNEY_8_CONNECTING_TUBULE_CELLS.M39258","NIKOLSKY_BREAST_CANCER_7P15_AMPLICON.M18241","GO_EMBRYONIC_SKELETAL_SYSTEM_MORPHOGENESIS.M10927","REACTOME_PHASE_II_CONJUGATION_OF_COMPOUNDS.M18788","SHETH_LIVER_CANCER_VS_TXNIP_LOSS_PAM4.M1280"],["AIZARANI_LIVER_C2_KUPFFER_CELLS_1.M39106","MCLACHLAN_DENTAL_CARIES_UP.M10454","AIZARANI_LIVER_C25_KUPFFER_CELLS_4.M39126","MODULE_75.M12401","GSE23568_ID3_KO_VS_WT_CD8_TCELL_DN.M8518","GO_IMMUNE_RESPONSE_REGULATING_SIGNALING_PATHWAY.M22284","FAN_EMBRYONIC_CTX_BIG_GROUPS_BRAIN_IMMUNE.M39021","POOLA_INVASIVE_BREAST_CANCER_UP.M5369","FAN_EMBRYONIC_CTX_BRAIN_B_CELL.M39045","Systemic lupus erythematosus.83315"],["GO_REGULATION_OF_BLOOD_CIRCULATION.M16818","GO_STRIATED_MUSCLE_CONTRACTION.M19924","GO_MUSCLE_TISSUE_DEVELOPMENT.M10365","GO_I_BAND.M16979","HP_ABNORMALITY_OF_CARDIOVASCULAR_SYSTEM_ELECTROPHYSIOLOGY.M38428","GSE32533_WT_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_DN.M8465","GO_CARDIAC_MUSCLE_TISSUE_DEVELOPMENT.M13993","GO_MUSCLE_CELL_DIFFERENTIATION.M17152","GO_CARDIAC_MUSCLE_CONTRACTION.M24087","GO_STRIATED_MUSCLE_CELL_DIFFERENTIATION.M11134"],["GNF2_LCAT.M132","Retinol metabolism.83218","GO_ORGANIC_ACID_CATABOLIC_PROCESS.M13128","MODULE_212.M2027","AIZARANI_LIVER_C30_HEPATOCYTES_4.M39129","DESERT_PERIPORTAL_HEPATOCELLULAR_CARCINOMA_SUBCLASS_UP.M34031","MEBARKI_HCC_PROGENITOR_FZD8CRD_DN.M38993","SHETH_LIVER_CANCER_VS_TXNIP_LOSS_PAM4.M1280","GNF2_GSTM1.M10693","WP_METAPATHWAY_BIOTRANSFORMATION_PHASE_I_AND_II.M39588"],["MANNO_MIDBRAIN_NEUROTYPES_HRGL2C.M39053","MANNO_MIDBRAIN_NEUROTYPES_HNBML1.M39064","GO_SYNAPTIC_MEMBRANE.M17824","FAN_EMBRYONIC_CTX_BIG_GROUPS_GLIAL.M39020","GO_NEGATIVE_REGULATION_OF_NERVOUS_SYSTEM_DEVELOPMENT.M11740","FAN_EMBRYONIC_CTX_ASTROCYTE_2.M39034","GO_GLIOGENESIS.M3208","GO_GLIAL_CELL_DIFFERENTIATION.M15366","MIR4498.M30868","ZHONG_PFC_MAJOR_TYPES_OPC.M39079"],["Keratinization.1457932","REACTOME_KERATINIZATION.M27640","GO_KERATINIZATION.M15449","GO_CORNIFICATION.M24369","REACTOME_FORMATION_OF_THE_CORNIFIED_ENVELOPE.M27649","NIKOLSKY_BREAST_CANCER_20Q12_Q13_AMPLICON.M18973","GO_KERATINOCYTE_DIFFERENTIATION.M29131","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON.M17620","GO_SKIN_DEVELOPMENT.M15889","GO_KERATIN_FILAMENT.M17671"],["GO_CATION_CHANNEL_ACTIVITY.M19194","GAO_LARGE_INTESTINE_ADULT_CA_ENTEROENDOCRINE_CELLS.M39161","GO_CATION_CHANNEL_COMPLEX.M17419","ZHONG_PFC_C3_ASTROCYTE.M39102","GO_REGULATION_OF_POSTSYNAPTIC_MEMBRANE_POTENTIAL.M12831","MEISSNER_NPC_HCP_WITH_H3K4ME2_AND_H3K27ME3.M1949","GO_SYNAPSE_ASSEMBLY.M10639","CAHOY_NEURONAL.M2808","REACTOME_TRANSMISSION_ACROSS_CHEMICAL_SYNAPSES.M15514","GO_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_MEMBRANE.M25996"],["LEE_DIFFERENTIATING_T_LYMPHOCYTE.M2200","FAN_EMBRYONIC_CTX_BRAIN_NAIVE_LIKE_T_CELL.M39047","Mus musculus biological processes.1368125","HAY_BONE_MARROW_NAIVE_T_CELL.M39202","AIZARANI_LIVER_C3_NK_NKT_CELLS_2.M39107","Systemic lupus erythematosus.83315","AIZARANI_LIVER_C1_NK_NKT_CELLS_1.M39105","GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS.M39166","Meiosis.1367997","Meiotic synapsis.1367999"],["GO_ODORANT_BINDING.M19181","chr11q12.M5761","chr11q11.M3966","Growth hormone receptor signaling.1367608","Taste transduction.83285","chr11p15.M9514","GPCR ligand binding.1367054","Prolactin receptor signaling.1367010","G alpha (i) signalling events.1367086","chr4q13.M1196"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Scatterplot of GSEA results <strong>Topic 1</strong> (highlighting &quot;Erythroblast&quot; cells)</li>
</ul>
<pre class="r"><code>k   &lt;- 1
name_interest &lt;- &quot;erythroblast&quot;
label_geneset_ids &lt;- gene_set_info[gsea_res$pval[,k] &lt; 1e-6 &amp; grepl(name_interest, gene_set_info$name, ignore.case = T), &quot;id&quot;]
cat(sprintf(&quot;highlighting %d %s gene sets \n&quot;, length(label_geneset_ids), name_interest))

create_genescore_gsea_plot(gene_set_info, gsea_res, k, label_geneset_ids, 
                                     title = sprintf(&quot;topic %d GSEA plot&quot;,k))</code></pre>
<p><img src="figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-genebody-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gsea-plot-genebody-1-1">
Past versions of gsea-plot-genebody-1-1.png
</button>
</p>
<div id="fig-gsea-plot-genebody-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/5e832d255d83249a393569a12c556494251896a0/docs/figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-genebody-1-1.png" target="_blank">5e832d2</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ca6ea6fe4813bd285095282e60b7b1828844106a/docs/figure/gene_analysis_Cusanovich2018.Rmd/gsea-plot-genebody-1-1.png" target="_blank">ca6ea6f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>if(length(label_geneset_ids) &gt; 0)
   DT::datatable(gene_set_info[gene_set_info$id %in% label_geneset_ids, 
                               c(&quot;name&quot;, &quot;id&quot;, &quot;database&quot;, &quot;category_code&quot;, &quot;description_brief&quot;)],  
                 rownames = F)</code></pre>
<pre><code># highlighting 0 erythroblast gene sets</code></pre>
<p>Explore the GSEA plot <a href="plotly/Cusanovich2018/gsea_topic_1_genebody-sum.html">interactively</a></p>
<pre class="r"><code>for ( k in 1:ncol(gsea_res$pval)){
  p.gsea.plotly &lt;- create_genescore_gsea_plotly(gene_set_info, gsea_res, k)
saveWidget(p.gsea.plotly,file = sprintf(&quot;%s/gsea_topic_%s_%s.html&quot;, fig.dir, k, &quot;genebody-sum&quot;),
           selfcontained = TRUE,title = &quot;GSEA&quot;)
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#  [1] reshape_0.8.8     DT_0.16           htmlwidgets_1.5.3 plotly_4.9.3     
#  [5] cowplot_1.1.1     ggrepel_0.9.1     ggplot2_3.3.3     tidyr_1.1.2      
#  [9] dplyr_1.0.3       fastTopics_0.4-29 Matrix_1.2-18     workflowr_1.6.2  
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.6         lattice_0.20-41    prettyunits_1.1.1  rprojroot_2.0.2   
#  [5] digest_0.6.27      plyr_1.8.6         R6_2.5.0           MatrixModels_0.4-1
#  [9] evaluate_0.14      coda_0.19-4        httr_1.4.2         pillar_1.4.7      
# [13] rlang_0.4.10       progress_1.2.2     lazyeval_0.2.2     data.table_1.13.6 
# [17] irlba_2.3.3        SparseM_1.78       whisker_0.4        rmarkdown_2.6     
# [21] labeling_0.4.2     Rtsne_0.15         stringr_1.4.0      munsell_0.5.0     
# [25] compiler_3.6.1     httpuv_1.5.4       xfun_0.19          pkgconfig_2.0.3   
# [29] mcmc_0.9-7         htmltools_0.5.1.1  tidyselect_1.1.0   tibble_3.0.6      
# [33] quadprog_1.5-8     matrixStats_0.58.0 viridisLite_0.3.0  withr_2.4.1       
# [37] crayon_1.4.0       conquer_1.0.2      later_1.1.0.1      MASS_7.3-53       
# [41] grid_3.6.1         jsonlite_1.7.2     gtable_0.3.0       lifecycle_0.2.0   
# [45] DBI_1.1.0          git2r_0.27.1       magrittr_2.0.1     scales_1.1.1      
# [49] RcppParallel_5.0.2 stringi_1.5.3      farver_2.0.3       fs_1.3.1          
# [53] promises_1.1.1     ellipsis_0.3.1     generics_0.1.0     vctrs_0.3.6       
# [57] tools_3.6.1        glue_1.4.2         purrr_0.3.4        crosstalk_1.1.1   
# [61] hms_1.0.0          yaml_2.2.1         colorspace_2.0-0   knitr_1.30        
# [65] quantreg_5.83      MCMCpack_1.5-0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
