<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Topic modeling analysis of single-cell ATAC-seq data</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Topic modeling analysis of single-cell
ATAC-seq data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree432b2e6405e74d4aab5b048153a64507dc886cbftargetblank432b2e6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/scATACseq-topics/tree/432b2e6405e74d4aab5b048153a64507dc886cbf" target="_blank">432b2e6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree432b2e6405e74d4aab5b048153a64507dc886cbftargetblank432b2e6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/scATACseq-topics/tree/432b2e6405e74d4aab5b048153a64507dc886cbf" target="_blank">432b2e6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Buenrostro_2018/
    Ignored:    data/Cusanovich_2018/processed_data/Cusanovich_2018_Kidney.RData
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/de-buenrostro2018-k=10-noshrink.RData
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/fit-Buenrostro2018-binarized-filtered-scd-ex-k=10.rds
    Ignored:    output/Buenrostro_2018/binarized/filtered_peaks/fit-Buenrostro2018-binarized-filtered-scd-ex-k=8.rds
    Ignored:    output/Cusanovich_2018/

Untracked files:
    Untracked:  analysis/cusanovich2018_kidney_k10.Rmd
    Untracked:  analysis/fit-Buenrostro2018-binarized-scd-ex-k=10.rds
    Untracked:  data/Buenrostro_2018_binarized.RData
    Untracked:  output/Buenrostro_2018/binarized/filtered_peaks/Buenrostro_2018_binarized_filtered.RData
    Untracked:  plots/
    Untracked:  scripts/fit-buenrostro-2018-k=8.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/432b2e6405e74d4aab5b048153a64507dc886cbf/analysis/index.Rmd" target="_blank">432b2e6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-05
</td>
<td>
workflowr::wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/f33b30424200e783afdcf36d508573bdddfcb6bc/docs/index.html" target="_blank">f33b304</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-01
</td>
<td>
Fixed link.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/5f007aeecfb21f566eb15ca820e75dcd0e69b17d/analysis/index.Rmd" target="_blank">5f007ae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-01
</td>
<td>
workflowr::wflow_publish("analysis/index.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/2f96251f30bd1b3d440281df01b8639f3b3b7dda/docs/index.html" target="_blank">2f96251</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-01
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/08d32c286e6e8db9161480f2fa0b3728ee06aa5e/analysis/index.Rmd" target="_blank">08d32c2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-08-01
</td>
<td>
workflowr::wflow_publish("analysis/index.Rmd", verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/c04a77d75067ba71b193b1f32c77bf0da094f1ad/docs/index.html" target="_blank">c04a77d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-05-03
</td>
<td>
Updated link in overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/16eb0d97545c1a56c0940a5b19d5679fbeac0e95/analysis/index.Rmd" target="_blank">16eb0d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-05-03
</td>
<td>
workflowr::wflow_publish("index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/e64430147051405ce0b2f51dbf442acf9cfcdcd7/docs/index.html" target="_blank">e644301</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-05-03
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/bb549947073fdcce0e004e9a9faaa98ac7175bbe/analysis/index.Rmd" target="_blank">bb54994</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-05-03
</td>
<td>
workflowr::wflow_publish("index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/0e152c8106c094c26187d641411339b06ab19c1d/docs/index.html" target="_blank">0e152c8</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/66081146b3a402e4244bb23fe8289a8b4424c4a9/analysis/index.Rmd" target="_blank">6608114</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-29
</td>
<td>
added results using filtered peaks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/a94d1711b3c92d8c3aa86b992b5bd5f930a98943/docs/index.html" target="_blank">a94d171</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7359d789902490075e24c5b3a9f78ccb8a46b24f/analysis/index.Rmd" target="_blank">7359d78</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-11
</td>
<td>
Updated the link for the structure plots and DA analysis for Buenrostro
data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/71126fbda005b4b4e6216e85857972fe7cf8e270/docs/index.html" target="_blank">71126fb</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/e5e3bb2e97c7765cb204ca56d79ff2dc37b0689d/analysis/index.Rmd" target="_blank">e5e3bb2</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-09
</td>
<td>
added gene score analysis for Buenrostro data with k = 10
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/fa0665ea9d8c4f522cedd76d5106c74aaf7ba012/docs/index.html" target="_blank">fa0665e</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/bd51c665ce66d6101684e44f7451eed7f0c1d9f8/analysis/index.Rmd" target="_blank">bd51c66</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-09
</td>
<td>
added motif_analysis_Buenrostro2018_k10_v2.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/01c66379f796cfc3804d51d1aab6f42b57fcdec1/docs/index.html" target="_blank">01c6637</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/64a4ed233a4d66a4db85dd44dc72c275551eb9f6/analysis/index.Rmd" target="_blank">64a4ed2</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
updated with version 2 of DA analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/379795853765625c0d9b774a2c648ca0a4db06ac/docs/index.html" target="_blank">3797958</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/6ffed16040e1cc2235cdbbfb16e18e8152f007a9/analysis/index.Rmd" target="_blank">6ffed16</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-24
</td>
<td>
added compare_different_k_Cusanovich2018.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/72b0497ac27e6854c5e9cdd11c5cc0ce935849de/docs/index.html" target="_blank">72b0497</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/02f35cc8a970e4d6036133c8d8b2bb511061e952/analysis/index.Rmd" target="_blank">02f35cc</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-05
</td>
<td>
clean the page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/a1be8951fdbe9d7550b4d195ad4f9e8f90a9fb38/docs/index.html" target="_blank">a1be895</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14e2172cbcd6be48dd4cc495f7aa6b5941b91484/analysis/index.Rmd" target="_blank">14e2172</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-05
</td>
<td>
reorganized the Buenrostro2018 results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/bce9235271540caff4eb619a31ecfde69897fe7f/docs/index.html" target="_blank">bce9235</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/277a73ca4fcc3c14235bcf017db0510cc61d4b89/analysis/index.Rmd" target="_blank">277a73c</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
fix a link
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/85a1700eb4e7bbcd34c90c5f8ffa1067aa1d973d/docs/index.html" target="_blank">85a1700</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cbb2e61bfef811e67be56b75161e735206363344/analysis/index.Rmd" target="_blank">cbb2e61</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
created separate analysis files of motif and gene analysis for
Cusanovich 2018 data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/d9a11c5ac57557c4716b4fd0aef673ea6c92fc74/docs/index.html" target="_blank">d9a11c5</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d96bcdc2ac8f2aedf814cbe4559e40490ec8d89e/analysis/index.Rmd" target="_blank">d96bcdc</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-05
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/7ba8b9d5ba5543b91be39ebdccad21b2658167b7/docs/index.html" target="_blank">7ba8b9d</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/322233d1a0b40f6281513b17b0319d9afe7b7aa6/analysis/index.Rmd" target="_blank">322233d</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-05
</td>
<td>
add results for chomVAR_scPeaks_filtered data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ecfafe9ffc363d02dc299636f5af7114cd091c83/docs/index.html" target="_blank">ecfafe9</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c4a3ba4a50a94d7b108e3a3e14fc382ad2806249/analysis/index.Rmd" target="_blank">c4a3ba4</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-23
</td>
<td>
add motif and gene analysis results for Cusanovich 2018 and Buenrostro
2018 datasets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/5aa86fc76186537b8b0b5a289500e86c49cdc094/docs/index.html" target="_blank">5aa86fc</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3d1285c1ffc6b4df29f14597edf25c30bf3cb5ba/analysis/index.Rmd" target="_blank">3d1285c</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-02
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/aca6d62d8c896f63c86c15eed7c9cc5cd84ecefc/docs/index.html" target="_blank">aca6d62</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-11-09
</td>
<td>
Added "source" link to navbar.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/e1cc037fe45d77094baf32e46e230dd65beee4dd/docs/index.html" target="_blank">e1cc037</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d2cc33a0682ac15d75b72c0431e2559c06cc42e1/analysis/index.Rmd" target="_blank">d2cc33a</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-06
</td>
<td>
add PCA and k-means clustering of topic proportions for Buenrostro 2018
results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/8047ab1ac1224c129fe8d14035a884230acd178b/docs/index.html" target="_blank">8047ab1</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/310562487ffa072297c7c5670db35142949a25a7/analysis/index.Rmd" target="_blank">3105624</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-21
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/861ab21fbfa6850a5c208906272020819837fcb3/docs/index.html" target="_blank">861ab21</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/4a0657bc0bd980c9b7912a24d7ba69423e82859e/analysis/index.Rmd" target="_blank">4a0657b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-20
</td>
<td>
add clustering with k = 13 topics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ce007aa783cf34a2ff5d68e860f387b812146c13/docs/index.html" target="_blank">ce007aa</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d8d17304fb86ad7e83bd3d6883b221882af1de01/analysis/index.Rmd" target="_blank">d8d1730</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-07
</td>
<td>
add PCA plots of topic proportions for Cusanovich 2018 dataset
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/49e6a0ad06264e4207b20283d4bca7120ecabe64/docs/index.html" target="_blank">49e6a0a</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/e7d466d54afe2f31c16a2f0d0f94dccff8cfadea/analysis/index.Rmd" target="_blank">e7d466d</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-10-07
</td>
<td>
add PCA plots of topic proportions for Cusanovich 2018 dataset
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/181b7693580f368673e27322252fe967b35dc1af/docs/index.html" target="_blank">181b769</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/30928925d30c0c5ab837e5e873c942ea52cdbade/analysis/index.Rmd" target="_blank">3092892</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/bc761e3d2bbbb193e5959d3f42dc341b75bfaeaa/docs/index.html" target="_blank">bc761e3</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f37667a7a9e0cfdaf994bf37cef6404aeb9a190d/analysis/index.Rmd" target="_blank">f37667a</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/936df8b4d9cf28d2dbd0ffc664d593af22c0e004/docs/index.html" target="_blank">936df8b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f48e62ceca24a8433506b157bf6479775deec03c/analysis/index.Rmd" target="_blank">f48e62c</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-24
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/cf1f1ab374680aa373378f008e347db6637c9d78/docs/index.html" target="_blank">cf1f1ab</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/9f397864eef044d439dfb4e4d4ee3540421f83b9/analysis/index.Rmd" target="_blank">9f39786</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-22
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/0cfe3a8426143294b77a5f401801d15b5266ef1e/docs/index.html" target="_blank">0cfe3a8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3c2ef95c03bc7a2c611963b077503206942ca167/analysis/index.Rmd" target="_blank">3c2ef95</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/d94e173e0ace503d30d2364c36f72571efaf366b/docs/index.html" target="_blank">d94e173</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/9bb909ea8c4e08b7ed215afb6b7451063899382c/analysis/index.Rmd" target="_blank">9bb909e</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-29
</td>
<td>
add a link to the literatures for scATAC-seq data resources
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/5e9854eb09f656a8731951987b32133053669541/docs/index.html" target="_blank">5e9854e</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b09eb42f471d345624985bde509dac11c4cdfe6c/analysis/index.Rmd" target="_blank">b09eb42</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-29
</td>
<td>
wflow_publish("analysis/*")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/758af0e88403ffab87b980b42a2d98ff72e0867d/analysis/index.Rmd" target="_blank">758af0e</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-07-29
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Topic modeling analysis of single-cell ATAC-seq data</p>
<ul>
<li>Human hematopoietic differentiation dataset from Buenrostro <em>et
al.</em> 2018
<ul>
<li><a href="process_data_Buenrostro2018.html">Process Buenrostro 2018
dataset</a></li>
<li><a href="assess_fits_Buenrostro2018_Chen2019pipeline.html">Assess
model fits</a></li>
<li><a href="structureplot_DA_Buenrostro2018_k10.html">Structure plots
and DA analysis (K=10)</a></li>
<li><a href="motif_analysis_Buenrostro2018_k10_v2.html">Motif analysis
(K=10)</a></li>
<li><a href="gene_analysis_Buenrostro2018_k10_v2.html">Gene score
analysis (K=10)</a></li>
<li><a
href="structureplot_DA_Buenrostro2018_filteredpeaks_k10.html">Structure
plots and DA analysis (K=10) using filtered peaks</a></li>
<li><a href="motif_analysis_Buenrostro2018_k10_filterpeaks.html">Motif
analysis (K=10 using filtered peaks)</a></li>
<li><a href="buenrostro2018_k10.html">Topic modeling analysis with k =
10 topics</a></li>
</ul></li>
<li>Mouse Atlas dataset from Cusanovich <em>et al.</em> 2018
<ul>
<li><a href="process_data_Cusanovich2018.html">Process Cusanovich 2018
dataset</a></li>
<li><a href="assess_fits_Cusanovich2018.html">Assess model fits</a></li>
<li><a href="compare_different_k_Cusanovich2018.html">Visualize all
tissues with different numbers of topics (k’s)</a></li>
<li><a href="plots_Cusanovich2018.html">Visualize all tissues (k=13) and
then focus on a few tissues</a></li>
<li><a href="clusters_Cusanovich2018_k13.html">Clustering by topic
proportions (K=13)</a></li>
<li><a href="motif_analysis_Cusanovich2018_v2.html">Motif analysis
(K=13, new DA analysis)</a></li>
<li><a href="gene_analysis_Cusanovich2018_v2.html">Gene analysis (K=13,
new DA analysis)</a></li>
<li><a href="cusanovich2018_k13.html">Topic modeling analysis with k =
13 topics</a></li>
<li><a href="cusanovich2018_kidney_k10.html">Topic modeling analysis of
kidney cells with k = 10 topics</a></li>
</ul></li>
</ul>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
