<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Motif analysis using topic modeling results for Buenrostro et al (2018) scATAC-seq result (data processed using chromVAR processed counts on peaks called from Chen 2019, filtering regions with accessible in &gt; 1% samples)</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Motif analysis using topic modeling results for Buenrostro <em>et al</em> (2018) scATAC-seq result (data processed using chromVAR processed counts on peaks called from Chen 2019, filtering regions with accessible in &gt; 1% samples)</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee6bac50b9427c988811cf17f613608c579b0e903targetblanke6bac50a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/e6bac50b9427c988811cf17f613608c579b0e903" target="_blank">e6bac50</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee6bac50b9427c988811cf17f613608c579b0e903targetblanke6bac50a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/e6bac50b9427c988811cf17f613608c579b0e903" target="_blank">e6bac50</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/diff_count_Buenrostro2018_chomVAR_scPeaks.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  analysis/single_cell_rnaseq_demo.Rmd
    Untracked:  output/Cusanovich2018/
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum.html
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum.png
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum_files/
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum.html
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum.png
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum_files/
    Untracked:  output/plotly/
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum.html
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum.png
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum_files/
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum.html
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum.png
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum_files/
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/postfit_Buenrostro2018_Chen2019pipeline.sh
    Untracked:  scripts/postfit_Buenrostro2018_chromVar_scPeaks.sh
    Untracked:  scripts/postfit_Buenrostro2018_chromVar_scPeaks_filtered.sh

Unstaged changes:
    Modified:   analysis/cisTopic_Buenrostro2018_chomVAR_scPeaks.Rmd
    Modified:   analysis/motif_gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/plots_Lareau2019_bonemarrow.Rmd
    Modified:   code/plots.R
    Modified:   scripts/fit_all_models_Buenrostro_2018.sbatch
    Modified:   scripts/fit_cisTopic_Buenrostro_2018_chromVAR_scPeaks.sh
    Modified:   scripts/postfit_Buenrostro2018.sh
    Modified:   scripts/postfit_Cusanovich2018.sh
    Modified:   scripts/postfit_gene_analysis.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd</code>) and HTML (<code>docs/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/e6bac50b9427c988811cf17f613608c579b0e903/analysis/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd" target="_blank">e6bac50</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
<td>
update figure sizes for motif logos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.html" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/658591bc2892ca35bb9cb8f9ed87f6ac311e80f7/analysis/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd" target="_blank">658591b</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
<td>
updated motif analysis for Buenrostro2018 data processed by chomVAR using scPeaks filtered &gt; 1% samples
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform TF motif analysis for the Buenrostro <em>et al</em> (2018) scATAC-seq result inferred from the multinomial topic model with <span class="math inline">\(k = 11\)</span>.</p>
<p>scATAC-seq peaks were called using Chen <em>et al</em> (2019) pipeline, and we counted the reads in peaks using chromVAR, and further filtering peaks requiring peaks to be accessible in &gt; 1% samples.</p>
<div id="load-packages-and-some-functions-used-in-this-analysis" class="section level2">
<h2>Load packages and some functions used in this analysis</h2>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape2)
library(Logolas)
library(grid)
source(&quot;code/motif_analysis.R&quot;)
source(&quot;code/plots.R&quot;)</code></pre>
</div>
<div id="load-data-and-topic-model-results" class="section level2">
<h2>Load data and topic model results</h2>
<p>Load the <strong>binarized</strong> data and the <span class="math inline">\(k = 11\)</span> Poisson NMF fit results from Buenrostro <em>et al</em> 2018 dataset using binarized scPeaks based on chromVAR processed counts with peaks accessible in at least 1 sample.</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/chromVAR/&quot;
load(file.path(data.dir, &quot;Buenrostro_2018_binarized_scPeaks_filtered.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))</code></pre>
<pre><code># 2034 x 104502 counts matrix.</code></pre>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized/&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Buenrostro2018-binarized-scd-ex-k=11.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
<div id="visualize-by-structure-plot-grouped-by-cell-labels." class="section level2">
<h2>Visualize by Structure plot grouped by cell labels.</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
samples$label &lt;- as.factor(samples$label)

p.structure &lt;- structure_plot(fit_multinom,
                     grouping = samples[, &quot;label&quot;],n = Inf,gap = 40,
                     perplexity = 50,topics = 1:11,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/structure-plot-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1">
Past versions of structure-plot-1.png
</button>
</p>
<div id="fig-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/structure-plot-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-accessbility-analysis-of-the-atac-seq-regions-for-the-topics" class="section level2">
<h2>Differential accessbility analysis of the ATAC-seq regions for the topics</h2>
<p>Load results from differential accessbility analysis for the topics</p>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized/&quot;
cat(sprintf(&quot;Load results from %s \n&quot;, out.dir))</code></pre>
<pre><code># Load results from /project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized/</code></pre>
<pre class="r"><code>diffcount_file &lt;- file.path(out.dir, &quot;diffcount-Buenrostro2018-11topics.rds&quot;)
if(file.exists(diffcount_file)){
  cat(&quot;Load precomputed differential accessbility statistics.\n&quot;)
  diff_count_topics &lt;- readRDS(diffcount_file)
}else{
  cat(&quot;Computing differential accessbility statistics from topic model.\n&quot;)
  timing &lt;- system.time(diff_count_topics &lt;- diff_count_analysis(fit,counts))
  cat(sprintf(&quot;Computation took %0.2f seconds.\n&quot;,timing[&quot;elapsed&quot;]))
  cat(&quot;Saving results.\n&quot;)
  saveRDS(diff_count_topics, diffcount_file)
}</code></pre>
<pre><code># Load precomputed differential accessbility statistics.</code></pre>
<p>Distribution of z-scores</p>
<pre class="r"><code>zscore_topics &lt;-  melt(diff_count_topics$Z)
colnames(zscore_topics) &lt;- c(&quot;region&quot;, &quot;topic&quot;, &quot;zscore&quot;)
levels(zscore_topics$topic) &lt;- colnames(diff_count_topics$Z)

z.quantile.99 &lt;- apply(abs(diff_count_topics$Z), 2, quantile, 0.99)
cat(&quot;z-score 99% quantile: \n&quot;)
print(z.quantile.99)

p.hist.zscores &lt;- ggplot(zscore_topics, aes(x=zscore)) + 
  geom_histogram(binwidth=1, color=&quot;black&quot;, fill=&quot;white&quot;) + 
  coord_cartesian(xlim = c(-10, 30)) + theme_cowplot(font_size = 10) +
  facet_wrap(~ topic, ncol=4)

print(p.hist.zscores)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/hist-zscore-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hist-zscore-1">
Past versions of hist-zscore-1.png
</button>
</p>
<div id="fig-hist-zscore-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/hist-zscore-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># z-score 99% quantile: 
#       k1       k2       k3       k4       k5       k6       k7       k8 
# 6.979974 7.017651 5.880457 8.865264 9.698530 5.077929 8.030697 7.756501 
#       k9      k10      k11 
# 6.079376 6.375808 5.822375</code></pre>
<p>Volcano plot of the regions</p>
<p><strong>topic 5 example</strong></p>
<pre class="r"><code>volcano_plot(diff_count_topics,k = 5,label_above_quantile = Inf, 
             subsample_below_quantile = 0.7, subsample_rate = 0.1)</code></pre>
<pre><code># 38667 out of 104502 data points will be included in plot</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/region-volcano-plot-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-region-volcano-plot-1">
Past versions of region-volcano-plot-1.png
</button>
</p>
<div id="fig-region-volcano-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/region-volcano-plot-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="motif-enrichment-analysis-using-homer" class="section level2">
<h2>Motif enrichment analysis using <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">HOMER</a></h2>
<ul>
<li>Details about HOMER motif analysis:
<ul>
<li><a href="http://homer.ucsd.edu/homer/motif/" class="uri">http://homer.ucsd.edu/homer/motif/</a></li>
<li><a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html" class="uri">http://homer.ucsd.edu/homer/ngs/peakMotifs.html</a></li>
</ul></li>
<li>Motif enrichment result using regions with z-score above 99% quantile.</li>
</ul>
<p>Compile Homer results across topics</p>
<pre class="r"><code>homer.dir &lt;- paste0(out.dir, &quot;/motifanalysis-Buenrostro2018-k=11-quantile/HOMER/&quot;)
cat(sprintf(&quot;Directory of motif analysis result: %s \n&quot;, homer.dir))
homer_res_topics &lt;- readRDS(file.path(homer.dir, &quot;/homer_knownResults.rds&quot;))
selected_regions &lt;- readRDS(file.path(homer.dir, &quot;/selected_regions.rds&quot;))

# Compile Homer results (pvalue and ranking) across topics
motif_res &lt;- compile_homer_motif_res(homer_res_topics)
saveRDS(motif_res, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))
cat(&quot;compiled homer motif results are saved in&quot;, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))

motif_table &lt;- data.frame(motif = gsub(&quot;/.*&quot;, &quot;&quot;, rownames(motif_res$mlog10P)),
                          round(motif_res$mlog10P,2))
DT::datatable(motif_table, rownames = F, caption = &quot;Motif enrichment (-log10P)&quot;)</code></pre>
<pre><code># Directory of motif analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized//motifanalysis-Buenrostro2018-k=11-quantile/HOMER/ 
# compiled homer motif results are saved in /project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized//motifanalysis-Buenrostro2018-k=11-quantile/HOMER//homer_motif_enrichment_results.rds</code></pre>
<div id="htmlwidget-e248df790b973720ac44" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e248df790b973720ac44">{"x":{"filter":"none","caption":"<caption>Motif enrichment (-log10P)<\/caption>","data":[["AMYB(HTH)","AP-1(bZIP)","AP-2alpha(AP2)","AP-2gamma(AP2)","Ap4(bHLH)","AR-halfsite(NR)","ARE(NR)","Arnt:Ahr(bHLH)","Ascl1(bHLH)","Ascl2(bHLH)","Atf1(bZIP)","Atf2(bZIP)","Atf3(bZIP)","Atf4(bZIP)","Atf7(bZIP)","Atoh1(bHLH)","Bach1(bZIP)","Bach2(bZIP)","Bapx1(Homeobox)","Barx1(Homeobox)","BATF(bZIP)","Bcl11a(Zf)","Bcl6(Zf)","BHLHA15(bHLH)","bHLHE40(bHLH)","bHLHE41(bHLH)","BMAL1(bHLH)","BMYB(HTH)","BORIS(Zf)","Brachyury(T-box)","Brn1(POU,Homeobox)","Brn2(POU,Homeobox)","bZIP:IRF(bZIP,IRF)","c-Jun-CRE(bZIP)","c-Myc(bHLH)","c-Myc(bHLH)","CArG(MADS)","Cdx2(Homeobox)","CDX4(Homeobox)","CEBP:AP1(bZIP)","CEBP:CEBP(bZIP)","CEBP(bZIP)","Chop(bZIP)","CHR(?)","CLOCK(bHLH)","COUP-TFII(NR)","COUP-TFII(NR)","CRE(bZIP)","CREB5(bZIP)","CRX(Homeobox)","CTCF-SatelliteElement(Zf?)","CTCF(Zf)","CUX1(Homeobox)","Cux2(Homeobox)","DLX1(Homeobox)","DLX2(Homeobox)","Dlx3(Homeobox)","DLX5(Homeobox)","DMRT1(DM)","DMRT6(DM)","DUX(Homeobox)","DUX4(Homeobox)","Duxbl(Homeobox)","E-box(bHLH)","E2A(bHLH),near_PU.1","E2A(bHLH)","E2F(E2F)","E2F1(E2F)","E2F3(E2F)","E2F4(E2F)","E2F6(E2F)","E2F7(E2F)","EAR2(NR)","EBF(EBF)","EBF1(EBF)","EBF2(EBF)","EBNA1(EBV-virus)","Egr1(Zf)","Egr2(Zf)","EHF(ETS)","EKLF(Zf)","ELF1(ETS)","ELF3(ETS)","Elf4(ETS)","ELF5(ETS)","Elk1(ETS)","Elk4(ETS)","En1(Homeobox)","Eomes(T-box)","ERE(NR),IR3","ERG(ETS)","Erra(NR)","ERRg(NR)","Esrrb(NR)","ETS:E-box(ETS,bHLH)","ETS:RUNX(ETS,Runt)","ETS(ETS)","Ets1-distal(ETS)","ETS1(ETS)","ETV1(ETS)","Etv2(ETS)","ETV4(ETS)","EWS:ERG-fusion(ETS)","EWS:FLI1-fusion(ETS)","Fli1(ETS)","Fos(bZIP)","Fosl2(bZIP)","Fox:Ebox(Forkhead,bHLH)","FOXA1:AR(Forkhead,NR)","FOXA1(Forkhead)","FOXA1(Forkhead)","Foxa2(Forkhead)","Foxa3(Forkhead)","FoxD3(forkhead)","Foxf1(Forkhead)","Foxh1(Forkhead)","FOXK1(Forkhead)","FOXK2(Forkhead)","FoxL2(Forkhead)","FOXM1(Forkhead)","Foxo1(Forkhead)","Foxo3(Forkhead)","FOXP1(Forkhead)","Fra1(bZIP)","Fra2(bZIP)","FXR(NR),IR1","GABPA(ETS)","GATA:SCL(Zf,bHLH)","GATA(Zf),IR3","GATA(Zf),IR4","Gata1(Zf)","Gata2(Zf)","GATA3(Zf),DR4","GATA3(Zf),DR8","GATA3(Zf)","Gata4(Zf)","Gata6(Zf)","Gfi1b(Zf)","GFX(?)","GFY-Staf(?,Zf)","GFY(?)","Gli2(Zf)","GLI3(Zf)","GLIS3(Zf)","GRE(NR),IR3","GRE(NR),IR3","GRHL2(CP2)","GSC(Homeobox)","Hand2(bHLH)","HEB(bHLH)","HIC1(Zf)","HIF-1a(bHLH)","HIF-1b(HLH)","HIF2a(bHLH)","HINFP(Zf)","HLF(bZIP)","Hnf1(Homeobox)","HNF1b(Homeobox)","HNF4a(NR),DR1","HNF6(Homeobox)","Hnf6b(Homeobox)","HOXA1(Homeobox)","Hoxa10(Homeobox)","Hoxa11(Homeobox)","Hoxa13(Homeobox)","HOXA2(Homeobox)","Hoxa9(Homeobox)","HOXB13(Homeobox)","Hoxb4(Homeobox)","Hoxc9(Homeobox)","Hoxd10(Homeobox)","Hoxd11(Homeobox)","Hoxd12(Homeobox)","Hoxd13(Homeobox)","HRE(HSF)","HRE(HSF)","IRF:BATF(IRF:bZIP)","IRF1(IRF)","IRF2(IRF)","IRF3(IRF)","IRF4(IRF)","IRF8(IRF)","Isl1(Homeobox)","ISRE(IRF)","Jun-AP1(bZIP)","JunB(bZIP)","JunD(bZIP)","KLF1(Zf)","KLF10(Zf)","KLF14(Zf)","KLF3(Zf)","Klf4(Zf)","KLF5(Zf)","KLF6(Zf)","Klf9(Zf)","LEF1(HMG)","Lhx1(Homeobox)","Lhx2(Homeobox)","Lhx3(Homeobox)","LHX9(Homeobox)","LRF(Zf)","LXRE(NR),DR4","MafA(bZIP)","MafB(bZIP)","MafF(bZIP)","MafK(bZIP)","Max(bHLH)","Maz(Zf)","Mef2a(MADS)","Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Meis1(Homeobox)","MITF(bHLH)","MNT(bHLH)","Mouse_Recombination_Hotspot(Zf)","MYB(HTH)","Myf5(bHLH)","MYNN(Zf)","MyoD(bHLH)","MyoG(bHLH)","n-Myc(bHLH)","Nanog(Homeobox)","NeuroD1(bHLH)","NeuroG2(bHLH)","NF-E2(bZIP)","NF1-halfsite(CTF)","NF1:FOXA1(CTF,Forkhead)","NF1(CTF)","NFAT:AP1(RHD,bZIP)","NFAT(RHD)","NFE2L2(bZIP)","NFIL3(bZIP)","NFkB-p50,p52(RHD)","NFkB-p65-Rel(RHD)","NFkB-p65(RHD)","NFY(CCAAT)","Nkx2.1(Homeobox)","Nkx2.2(Homeobox)","Nkx2.5(Homeobox)","Nkx3.1(Homeobox)","Nkx6.1(Homeobox)","NPAS(bHLH)","NPAS2(bHLH)","Npas4(bHLH)","Nr5a2(NR)","Nr5a2(NR)","NRF(NRF)","NRF1(NRF)","Nrf2(bZIP)","Nur77(NR)","OCT:OCT-short(POU,Homeobox)","OCT:OCT(POU,Homeobox,IR1)","OCT:OCT(POU,Homeobox)","OCT:OCT(POU,Homeobox)","Oct11(POU,Homeobox)","Oct2(POU,Homeobox)","OCT4-SOX2-TCF-NANOG(POU,Homeobox,HMG)","Oct4:Sox17(POU,Homeobox,HMG)","Oct4(POU,Homeobox)","Oct6(POU,Homeobox)","Olig2(bHLH)","Otx2(Homeobox)","p53(p53)","p53(p53)","p53(p53)","p63(p53)","p73(p53)","PAX3:FKHR-fusion(Paired,Homeobox)","PAX5(Paired,Homeobox),condensed","PAX5(Paired,Homeobox)","PAX6(Paired,Homeobox)","Pax7(Paired,Homeobox),long","Pax7(Paired,Homeobox),longest","Pax7(Paired,Homeobox)","Pax8(Paired,Homeobox)","PBX1(Homeobox)","PBX2(Homeobox)","Pbx3(Homeobox)","Pdx1(Homeobox)","PGR(NR)","Phox2a(Homeobox)","Phox2b(Homeobox)","Pit1(Homeobox)","Pit1+1bp(Homeobox)","Pitx1:Ebox(Homeobox,bHLH)","Pitx1(Homeobox)","Pknox1(Homeobox)","PPARa(NR),DR1","PPARE(NR),DR1","PR(NR)","PRDM1(Zf)","PRDM10(Zf)","PRDM14(Zf)","PRDM15(Zf)","PRDM9(Zf)","Prop1(Homeobox)","PSE(SNAPc)","Ptf1a(bHLH)","PU.1-IRF(ETS:IRF)","PU.1:IRF8(ETS:IRF)","PU.1(ETS)","RAR:RXR(NR),DR0","RAR:RXR(NR),DR5","RARa(NR)","RARg(NR)","RBPJ:Ebox(?,bHLH)","Rbpj1(?)","REST-NRSF(Zf)","Reverb(NR),DR2","RFX(HTH)","Rfx1(HTH)","Rfx2(HTH)","Rfx5(HTH)","Rfx6(HTH)","Ronin(THAP)","RORa(NR)","RORg(NR)","RORgt(NR)","RUNX-AML(Runt)","RUNX(Runt)","RUNX1(Runt)","RUNX2(Runt)","RXR(NR),DR1","SCL(bHLH)","SCRT1(Zf)","SF1(NR)","Six1(Homeobox)","Six2(Homeobox)","Six4(Homeobox)","Slug(Zf)","Smad2(MAD)","Smad3(MAD)","Smad4(MAD)","Snail1(Zf)","Sox10(HMG)","Sox15(HMG)","Sox17(HMG)","Sox2(HMG)","Sox21(HMG)","Sox3(HMG)","Sox4(HMG)","Sox6(HMG)","Sox7(HMG)","Sox9(HMG)","Sp1(Zf)","Sp2(Zf)","Sp5(Zf)","SPDEF(ETS)","SpiB(ETS)","Srebp1a(bHLH)","Srebp2(bHLH)","STAT1(Stat)","Stat3(Stat)","Stat3+il21(Stat)","STAT4(Stat)","STAT5(Stat)","STAT6(Stat)","STAT6(Stat)","T1ISRE(IRF)","TATA-Box(TBP)","Tbet(T-box)","Tbox:Smad(T-box,MAD)","Tbr1(T-box)","Tbx20(T-box)","Tbx21(T-box)","Tbx5(T-box)","Tbx6(T-box)","Tcf12(bHLH)","Tcf21(bHLH)","Tcf3(HMG)","TCF4(bHLH)","Tcf7(HMG)","Tcfcp2l1(CP2)","TCFL2(HMG)","TEAD(TEA)","TEAD1(TEAD)","TEAD2(TEA)","TEAD3(TEA)","TEAD4(TEA)","TFE3(bHLH)","Tgif1(Homeobox)","Tgif2(Homeobox)","THRa(NR)","THRb(NR)","THRb(NR)","Tlx?(NR)","TR4(NR),DR1","TRPS1(Zf)","Twist(bHLH)","Twist2(bHLH)","Unknown-ESC-element(?)","Unknown(Homeobox)","USF1(bHLH)","Usf2(bHLH)","VDR(NR),DR3","WT1(Zf)","X-box(HTH)","YY1(Zf)","Zac1(Zf)","ZBTB12(Zf)","ZBTB18(Zf)","ZBTB33(Zf)","ZEB1(Zf)","ZEB2(Zf)","Zfp281(Zf)","ZFP3(Zf)","Zfp57(Zf)","Zfp809(Zf)","ZFX(Zf)","Zic(Zf)","Zic2(Zf)","Zic3(Zf)","ZKSCAN1(Zf)","ZNF136(Zf)","ZNF143|STAF(Zf)","ZNF16(Zf)","ZNF165(Zf)","ZNF189(Zf)","Znf263(Zf)","ZNF264(Zf)","ZNF317(Zf)","ZNF322(Zf)","ZNF341(Zf)","ZNF382(Zf)","ZNF41(Zf)","ZNF415(Zf)","ZNF416(Zf)","ZNF467(Zf)","ZNF519(Zf)","ZNF528(Zf)","ZNF652","ZNF669(Zf)","ZNF675(Zf)","ZNF692(Zf)","ZNF7(Zf)","ZNF711(Zf)","ZNF768(Zf)","ZSCAN22(Zf)"],[15.42,72.92,2.85,3.02,4.52,0.44,4.99,0,3.08,0.15,7.39,1.97,74,96.8,2.13,8.66,6.48,23.83,0,0.39,77.78,0.03,0.47,9.58,0.01,0,0,10.57,0.95,0.56,0.19,0.06,0.73,1.6,0.03,0.05,0.34,0.35,0.37,119.82,7.6,186.14,68.92,0.13,0.02,0,0,0.36,4.96,0,0.1,2.19,0,0,0.44,0.96,0.47,0.83,1.03,0.97,0,0.02,0.01,0.15,0,0,0,0.02,0.12,0.02,0.37,0,0,0.43,0.93,1.15,0.6,0.22,1.97,34.89,0.02,8.49,37.42,43.56,40.49,9.48,8.35,1.57,0,0.39,54.72,0,0.03,0.03,5.54,10.6,8.43,33.56,54.11,51.03,43.6,23.42,33.11,21.59,37.33,82.86,66.88,0.74,0.02,0,0.01,0.08,0.41,0.1,0,0.26,0.01,0.04,0,0.02,1.79,0,0,80.52,78.52,0.01,33.84,0.29,1.6,0.03,0.11,0.15,0.04,0.81,0.09,0.14,0.11,0.01,0.27,0.03,0.33,0.97,0.8,0,2.55,3.47,0.05,0,1.42,0.01,1.03,0.52,0.14,0.27,0,135.93,0.23,0.02,0.12,0,0,0.23,5.89,1.34,1.78,0.63,0.6,0.04,1.22,0.14,1.09,0.83,0.47,0.68,0.16,0.5,1.1,2.77,3.16,1.12,1.78,3.5,1.01,1.29,51.07,77.22,1.22,0.07,0,0.25,0.32,0.1,0.22,0.4,0.35,0.23,0.16,0.71,0.45,0.5,0.17,0,4.07,6.68,0.45,8.28,0.02,0.64,0,0,0,0.65,1,1.92,0,0,14.65,3.1,0.15,4.4,2.57,0.02,0.75,6.59,10.15,9.17,5.08,0.8,1.27,0.96,0.03,5.59,128.03,0.07,0.57,0.74,1.23,0.01,0.01,0.02,0.01,0.78,0.01,0,0.31,0.05,0.02,0,0,6.77,0.76,3.01,0.14,0,0.02,0.35,0.04,0.1,0.95,0.07,0.08,9.89,0.02,0.05,0.01,0.01,0.02,0.19,0.2,0.75,1.58,0.42,0.08,0.58,0.01,0.97,0.09,0.15,0.04,0.53,1.83,0.01,0.01,0.1,0.7,0.16,0,0.03,0.26,0.27,1.02,0.36,0,0.6,0.77,1.44,0.01,2.52,0.01,4.63,3.53,45.47,0.09,0.18,0,0.02,0.89,0.05,0.33,0.66,0.54,2.83,1,2.04,3.57,0,0.23,0.15,0.36,28.08,21.95,24.21,33.82,0.13,5.39,0,0.12,0.84,1.21,0.26,0,1.53,1.6,0.28,0,0.26,0.11,0.44,0.07,0.32,0.58,0.07,0.36,0.18,0.67,0.03,1.31,1.76,17.82,24.25,0.83,0.45,0.63,1.63,3.13,1.91,0.2,0.72,0.64,0.22,0.17,0.03,0.41,0.27,2.25,0,0,0,4.15,7.93,0.25,8.29,0.87,0.21,0.07,0,0,0,0.01,0,0.37,2.74,2.51,0,0.01,0.03,1.22,0.08,0.02,1.28,10.03,0.01,0,0.34,1.03,0.02,0.37,0.6,0,0.12,0.04,3.43,0.17,0,0,0.01,0,0.2,0.17,0.67,0.18,0.01,0.07,0.01,0.01,0.4,0,0.14,0,0.23,0.73,0.34,0.01,0.3,0.77,0.15,0.86,0.01,0.72,2.9,0,0.28,0.05,0.01,0.01,0.06,0.71,0.21,0.01],[8.35,84.34,1.22,1.42,1.37,0.53,1.55,0.05,0.58,0,5.08,1.64,87.99,34.62,0.8,2.53,3.68,17.21,0,0.79,88.64,0.1,0.33,1.73,0.02,0.06,0.71,7.77,1.72,0.84,0.57,0.37,7.22,0.62,0.03,0.1,1.11,0.19,0.44,45.17,4.29,66.62,21.92,0.42,0.27,0.09,0.02,0.03,1.48,0,0.18,0.52,0.01,0,0.11,0.97,0.18,0.46,1.96,1.84,0.56,0.17,0.05,0.3,0,0,0.09,0.04,0.42,0.27,0.29,0,0.09,0.05,0.77,1.06,0.52,1.03,1.13,31.36,3.78,8.24,33.79,46.21,54.94,9.23,9.74,0.66,0.01,0.01,36.32,0.1,0.07,0.02,2.81,7.13,4.85,19.71,46.08,40.74,34.45,25.29,20.13,10.03,31.89,88.68,68.97,0.16,0.1,0.02,0,0.73,0.3,0,0,0.11,0.01,0.06,0.01,0,1.72,0.01,0.03,90.38,85.9,0.29,24.82,0.03,1.32,0.02,0.01,0,0.03,0.19,0,0,0,0.01,0,0,0.51,0.58,1.47,0,1.62,0.83,0.01,0,0.57,0,0.01,0,0,0.01,0.66,49.55,0.71,0.19,0.01,0,0.09,0.33,13,1.61,1.15,0.33,3.73,0.26,1.36,0,4.77,1.59,2.22,0.37,0.06,0.11,6.76,79.48,56.89,120.99,50.16,179.23,0.49,26.37,58.89,89.64,0.53,2.96,1.63,3.65,1.9,2.41,3.4,3.48,3.04,0.16,0.1,0.25,1.01,0.47,0.04,0.01,3.69,2.98,0.29,5.72,0.72,2.53,0.46,2.06,0.1,2.94,0.77,1.52,0.18,0.12,5.55,0.44,0.46,0.79,0.29,0.16,0.03,1.28,1.81,6.96,3.93,0.15,0.66,2.58,0.18,4.95,58.93,0.41,1.31,0.62,0.21,0,0,0.01,0.03,0.36,0.33,0.01,1.06,0.55,0.79,0,0.09,2.71,0.07,4.24,0,0,0.15,0.6,0.05,0.03,0.99,0.36,0.02,1.72,0,0.31,0.04,0.04,0.29,0.01,0.63,1.41,0.01,0.2,0,0.05,0.57,0.02,0.81,0.65,0.63,2.49,1.55,0.03,0.03,0.25,0.61,0.25,0,0.27,0.01,0.52,3.18,20.75,0.02,0.44,0.22,1.18,0.04,0.43,0,106.32,217.19,76.26,0.01,0.89,0,0.19,0.11,0,0,0.27,1.66,2.78,0.97,3.76,1.79,0,1.25,1.68,1.5,7.64,9.99,12.03,11.8,0.04,5.45,0,0.78,0.3,3.4,0.39,0,2.13,1.51,1.77,0,1.03,0.16,0.79,0.6,0.91,1.33,0.65,0.57,0.54,0.33,0.16,2.93,2.55,14.84,75.13,0.54,0.44,0.24,0.24,0.54,0.16,0.1,0.51,0.42,5.97,0.37,0.04,0.03,0.04,0.79,0,0,0,0.49,1.4,0.31,3.17,0.08,0.1,0.05,0,0,0,0,0,0.08,1.47,1.04,0,0.04,0.09,1.25,0.19,0,1.88,2.66,0.07,0.05,0.12,0.27,0.59,0.73,1.83,0.09,0.08,0.08,0.13,0,0,0,1.71,0,0.36,0.07,0.77,0.01,0.11,0.34,0,0,0.67,0,0.61,3.8,0.32,0.17,0.04,0.78,1.39,2.22,0.1,0.53,0,1.95,1.11,0,0.02,0.73,1.1,0.09,0.72,0.39,0.43,0.99],[1.97,0.07,0.19,0.39,0.11,0.43,0.21,1.4,0.02,0,0.71,0.34,0.06,2.36,0.08,0.01,0.36,0.26,0,0.9,0.08,0.49,0.5,0.02,1.9,2.32,0.91,2.11,29.78,0.05,0.97,0.28,0.07,0.06,1.58,0.92,0.81,0.2,0.2,2.96,0.92,0.36,2.58,0.14,2.16,0.02,0.02,2.93,0.3,0.05,2.75,45.04,0.92,0.36,0.88,0.75,0.3,0.59,0.04,0.17,0,0,1.2,1.95,0,0,2.54,1.55,1.34,3.63,1.77,3.05,0,0.01,0.09,0.9,0.12,1.97,7.08,2.67,0.69,9.88,1.55,3.79,3.09,12.83,11.77,6.39,0.04,0.02,1.38,0.01,0.46,0.29,1.57,0.17,5.21,0.17,3.24,5.97,0.97,7.42,0.08,3.75,9.98,0.05,0.09,0.12,0.12,0.01,0.01,0.03,0.05,0.01,0.03,0.06,0.22,0.12,0.03,0.01,0.3,0.11,0.42,0.16,0.05,0.02,7.38,3.73,0.11,0,7.2,5.98,0.13,0.54,3.88,5.87,5.92,0.1,0.97,13.7,13.08,0.05,0.08,0.17,0.28,0.38,0.14,0,0,0,0.06,1.69,2.44,1.3,0.14,1.29,0,0.15,0,0.25,0.87,0.01,0.26,1.07,0.65,0.04,3.31,0.14,0.02,0.48,0.85,1.8,0,0.45,1.9,1.58,0.06,0,0.03,0.39,0.27,0.09,1.83,0.07,0.05,0.13,0.7,22.84,12.87,14.86,29.48,12.73,15.25,15.14,29.36,0.2,0.13,0.16,0.83,2.05,0.01,0.03,0.3,0.64,0.17,0.34,2.56,5.82,0,0,0,0.25,0.27,0.76,1.84,0,2.14,0.08,0.48,0.01,0.03,2.54,4.05,0.08,0.01,0.41,0.32,0.15,0.25,0.28,0.19,0.22,3.11,0.02,0,0.2,28.24,0.13,0.1,0.26,0.01,0.3,1.06,1.35,0.86,0.09,0.12,26.12,27.19,0.59,0.11,0.19,1.21,0.51,0,0.71,1.06,0.04,0.13,1.53,0.6,0.02,0.14,0,0.01,0.01,0.89,0.13,0.54,0.71,0.05,0.35,0,0.31,0.45,0.02,0.11,0.1,0.94,0.01,0.34,0.12,0.38,0.01,0.12,0.15,0.03,1.55,0.31,0.23,0.43,0.18,0.18,0.37,0.51,0.01,0.13,0.44,0,0.2,0.11,0.21,0,0.06,0,0,0.41,0.04,0.34,0.05,2.63,1.85,2.89,1.16,0.17,11.72,0.18,0.25,1.26,1.19,1.61,0.91,1.1,0.43,0.03,0.06,0.02,0.5,0.81,0,0.02,0.26,0.37,0.47,0,0.12,0.01,0.03,0.05,0.23,0.03,0.04,0.25,0.13,0,35.01,17.25,21.17,1.98,0.88,0.02,0.03,0.76,2.06,1.23,2.29,0.59,0.35,0.52,0,2.22,0.03,0.17,0.17,0.38,0.03,0.06,0,0.01,0.07,0.03,0.01,0.54,0.75,0.6,0,0.03,0.01,0.02,0.01,3.02,0.09,0.11,0.16,0.87,0.11,0.07,0.2,3.51,0.42,0.01,0.29,0.84,1.58,1.1,0.06,1.38,1.74,7.2,0.48,0.73,0.07,1.25,0,0,1.04,0,0.79,0.66,1.98,0.48,0.28,0.38,0.4,0.09,5.95,0,0.02,1.02,0.11,0.03,0.17,0.61,1.62,0.38,0,0.13,0,0.01,0.14,0.85,0.06,0.13,0.26,0.3,1.06,4.26,0.19,0.68],[0.85,5.9,1.21,0.8,62.36,5.12,1,0.35,114.05,79,0.24,1.75,8.94,0.19,0.52,74.96,0.81,1.81,0,0.05,9.31,0.03,0,77.35,0.12,0.91,0.76,2.85,1.35,0.45,0.81,0.24,4.79,0.66,0.41,0.18,0.7,0.08,0.23,0.36,0.63,0,0.32,2.02,0.44,0,0,3.27,1.04,0,0.23,0.54,0.08,0.02,0.12,0.11,0.16,0.11,0.25,0.06,1.36,0.37,0.24,0.69,26.43,85.03,0.19,0.29,0.31,0.13,1.48,0.88,0,0.15,0.19,0.16,0.17,1.56,2.04,19.55,0.85,2.89,17.48,30.24,33.95,3,2.79,0.1,0.97,0.12,23.17,0,0.01,0,5.82,14.94,2.93,18.23,25.88,20.52,24.8,11.21,15.29,4.29,21.56,8.36,6.55,1.09,0.05,0.82,0.76,0.82,0.43,0.05,0.37,0.1,0.12,0.57,0.87,0.67,2.62,0.05,0,9.34,5.75,0,12.71,0.17,0.07,0.02,0,0,0.46,0.27,0,0,0,0,0,0.07,1.19,1.25,2.77,0,0.43,2.02,0.47,0,0.21,89.16,1.36,0.8,0.17,0.28,0.59,0.88,1.01,0.36,0,0.01,0.03,0.06,8.35,0.49,0.1,0.29,1.31,0.1,0.18,0.91,2.3,1.45,0.32,0.08,0.09,0.03,7.42,49.94,42.27,76,33.48,114.96,0.29,23.65,6.44,9.65,0.77,0.31,1.29,2.28,0.13,0.35,0.23,0.39,0.89,1.37,0.34,0.38,0.07,0.64,0.02,0,1.08,1.23,0.61,2.36,0.45,3.53,0.57,1.73,0.51,8.13,2.09,0.48,0.57,0.74,1.38,50.73,0.89,46.82,55.89,0.77,0,46.04,58.28,1.07,3.98,0.13,0.21,0.58,0.15,0.2,0.42,0.4,0.55,0.17,0.17,0.05,0,0,0.02,0.1,0.43,0.82,0.01,0.12,0,0.08,0.12,0.76,0.08,0.29,0.63,0.36,0.13,0.93,0.45,0.56,0.01,1.14,0.44,46.08,0,0.02,0.24,0.24,0.1,0.21,1.05,0.11,0.03,0.52,0,0.41,0.31,0.02,0.87,0.55,0.4,1.34,0.84,0.1,0.03,1.7,1.32,3.89,0,0.53,0,0,0.38,15.6,0.01,0.09,0.04,2.18,0.14,2.41,88.64,67.45,127.77,53.72,0.01,0.77,0,0.01,1.64,0.01,0,0.01,2.21,1.27,2.77,1.31,2.66,0.07,0.11,0.3,0.25,52.64,62.67,68.31,64.36,0.01,40.88,0.15,0.05,0.17,2.59,0.46,18.61,6.48,7.44,8.54,20.06,1.51,0.01,0.51,0.33,0.66,0.87,0.34,0.47,0.73,0.32,0.19,0.74,1.6,3.82,80.47,0.02,0.04,0.05,0.02,0.21,0.27,0.05,0.49,0.47,2.51,0.6,0.77,0.72,0.2,1.71,1.29,1.91,0.35,79.48,79.43,0.52,66.66,2.33,0.32,0.03,0,0,0,0,0,0.59,3.44,6.41,0,0,0,0.68,0.12,0.01,5.01,83.69,4.56,0,0.43,0.24,0.05,0.79,2.35,0,0.49,0.41,23.9,0,10.53,3.2,1.26,0,0.21,0.66,1.57,0.6,2.09,3.75,0,0.04,0,0.38,0.42,4.29,0.94,1.61,1.83,0.46,2.48,2.25,0.04,0.3,0,0.99,0.96,0,0.02,0,0.39,0.04,1.34,0.38,0.6,1.94],[11.9,4.32,0.46,0.59,5.23,5.85,0.16,1.19,2.01,0.16,0.22,0.38,5.81,0.21,0.08,7.08,3.44,4.2,0,0.11,6.16,0.12,0.45,6.41,0.06,0.01,1.4,10.46,0.14,0.47,0.33,0.07,0.03,0.11,0.01,0.56,1.96,0.23,1.05,0.36,0.93,0.07,0.07,0,0.5,0,0,0.67,0.62,2.67,0.09,0.03,0.02,0.02,1.52,1.07,0.44,0.21,0.13,0.09,0.64,0.91,1.09,0.13,0,0,0.63,0.05,0.39,0.78,0.27,0.32,0,0.09,1.1,0.25,1.39,3.62,1.91,9.8,4.44,5.39,9.76,9.02,7.57,8.39,5.83,2.98,3.14,0.05,9.49,0,0.11,0.07,2.03,4.37,4.48,9.61,14.22,16.39,8.11,9.12,8.07,7.33,13.73,8.71,8.04,0.08,0,0.05,0.07,0.29,0.12,0.09,2.73,0.37,3.81,1.93,1.06,0.03,5.58,5.1,3.6,6.48,5.49,0,10.42,191.26,0.06,0.01,289.15,303.14,1.8,0.21,327.94,323.51,335.54,0,0,0.22,0.19,0.75,0.21,0,0.08,0.38,1.01,0.16,1.67,0,0.62,0.26,0.13,0.28,0.26,1.23,0.69,0.35,0,0.01,0.05,0.1,0.63,0.15,0.31,0.33,0.04,2.59,0.35,0.05,0.19,0.28,0,0.54,0.11,0.32,0.54,0.1,0.18,0.17,0.23,0.49,0.01,0.37,8.88,6.5,1.54,4.53,1.62,4.15,2.13,4.33,3.43,4.82,2.28,0.53,0.1,0.25,0.69,1.46,0,0,3.06,0.84,0.5,1.32,0.85,3.7,0,0,0,0.49,0.57,2.51,1.32,0.38,16.01,4.59,1.46,8.3,3.98,1.73,0.06,2.22,7.38,5.3,7.3,1.11,4.53,1.89,1.49,5.75,0.5,0.34,0.43,0.3,0.21,0,0,0.01,0,1.35,0.71,0.02,0,0.11,0.23,0.01,0.01,7.5,0.04,0.85,0.17,0,0,0.33,0.63,0.14,0.34,0.15,0.3,3.59,1.96,0.05,0.54,0.54,0.79,0.18,0.6,0.11,0,0.7,0,0.04,0.59,0,0.14,2.06,0.84,1.09,0.04,1.36,0.12,0.01,0,0.21,0.01,0.88,0.24,0.19,1.95,0.01,0.03,0.56,0.22,0.69,0.54,0.33,0.06,1.73,0.46,6.58,0.02,0.17,0,0.05,0.96,0.02,0,0.01,2.24,2.28,2.47,1.13,0.92,0.12,0.09,0.29,0.02,21.83,32.73,30.52,29.37,0.37,11.3,0,0.1,0.02,0.5,0.19,0,4.6,4.78,4.91,0,2.45,0.05,1.03,0.8,1.96,1.04,0.39,1.51,0.05,3.56,0.04,3.89,3.39,1.72,0.57,0.2,0.01,0.92,0.13,0.63,1.03,0.38,0.97,0.3,0,0.52,0.02,0.29,1.23,0.81,0.04,0,0,7.52,5.22,0.42,8.51,0.62,0.76,0.06,0.67,0.45,0,0.11,0.22,0.06,1.17,1.3,0.01,0.01,0.04,2.41,0.03,294.23,0.01,9.11,0.8,0.4,0.1,1.02,0.18,3.58,1.89,0.02,1.84,0.04,2.49,0,0,0,0.5,0,0.06,0.18,5.08,2.04,2.13,1.92,0,0.4,0.33,0.18,2.13,0.65,0.43,0.2,0.23,1.23,0.94,0,0.06,0.01,0,2.07,0,0.39,0.26,0.55,0.51,0.85,1.36,2.58,0.02,0.23],[1.69,312.87,0.35,0.19,4.41,0.02,0.01,5.01,5.2,0.65,3.59,3.5,341.75,3.3,1.85,5.44,37.86,157.39,0,0.04,338.01,0,0.63,4.67,0.06,0.33,0.18,0.31,1.99,0.01,0.54,1.08,0.52,2.59,0.23,0.01,1.74,5.22,10.71,2.39,0.31,0.39,2.74,0.51,0.22,0,0,1.2,2.17,0,0.12,0.57,0.99,0.66,0.16,0.82,0.31,0.17,0.16,0.3,0,0.39,0.17,0.09,0,0.41,0.7,0.02,2.14,0.65,0.98,0.29,0,0.63,0.36,0.27,0.67,4.53,3.83,44.04,0.02,27.29,41.74,48.03,32.84,30.01,25.12,1.12,0.27,0.15,98.98,0,0,0,13.14,21.68,19.1,41.89,82.17,79.26,90.55,60.84,64.36,56.33,83.6,339.49,296.45,0.6,0,0.13,0.18,0.59,0.61,0.44,1.39,0.22,2.19,0.75,1.83,0.33,4.04,2.48,1.51,341.53,327.33,0.13,56.55,3.11,0.39,0.33,0.81,1.37,0.06,0.31,1.08,2.28,1.75,0.01,0,0.34,0.03,0.03,0.27,0.04,0.05,0,0.27,0,1.01,0.44,2.17,0.06,0.37,0.12,0.04,3.34,0.49,0.97,0,0.3,0.11,4.54,0.67,5.2,2.27,1.41,1.9,7.43,4.9,20.01,1.59,4.88,0.01,3.22,0.86,0.25,0.66,0.05,0.14,0.7,0.07,0.06,1.45,0.12,277.6,324.11,2.25,0.34,0.2,3.15,0.06,0.04,0.63,0.83,0.8,0.71,1.03,0.65,0.93,0.73,0.05,0,26.89,17.58,0.02,50.51,0.42,9.82,0,0.09,0.01,1.69,1.99,1.98,0.13,0.9,2.43,3.34,0.28,3.83,2.45,0.11,1.01,4.65,6.6,55.07,5.92,0.13,5.54,14.64,1.31,24.74,6.26,6.31,8.64,13.27,0.11,0,0,0,0.02,1.26,0.08,0.02,0.31,0.03,0,0.05,0.03,46.25,0,3.55,0.87,0,0.02,0.69,0.72,1.39,0.02,0.38,1.35,6.55,0.02,0.68,1.12,1.12,0.63,0.92,1.16,1.28,0.22,1.08,0.24,0.2,0.57,0.06,0.6,22.61,0.09,15.9,0.01,0.82,0.45,0.51,3.68,0.33,0.01,0.15,0,0.02,0.41,0.2,0.98,0.23,0.03,1.7,0.4,2.19,0.87,2.79,1.01,40.71,0.04,0.27,0,0.27,0.3,0.04,0.56,1.12,9.22,5.66,8.3,3.45,2.54,0.32,0.04,0.71,0.73,33.73,31.99,44.08,32.53,0.02,6.76,0.2,0,0.21,0.23,0.01,0.03,5.33,5.29,4.94,0.01,2.08,0.09,0.15,0.86,1.52,1.36,0.77,0.71,0.34,0.44,1.15,0.67,1.98,19.96,11.56,0,0,3.06,0.65,2.81,1.94,1.15,0.38,0.13,0.34,5.61,0.9,0.41,0.08,0.83,0.01,0.1,0.01,5.09,4.07,0.48,6.07,0.1,0.97,0.3,0.01,0,0.06,0.01,0.09,0.24,6.62,8.53,0,0,0,4.59,0,0.48,0.39,5.56,1.97,4.05,0.39,0.09,0.09,7.54,5.64,0.05,0.53,0.06,1.5,0.02,0,0,2.34,0,0.43,0.44,1.04,0.92,1.64,2.79,0.06,0,0.17,0,0.09,0.01,0.07,0.61,0.18,0.11,0.44,0,0.03,0.04,0.01,4.5,0.69,0.47,0.07,0.75,0.02,0.04,0.26,0.58,0.02,0.63],[4.79,88.68,1.34,0.93,0.55,0.95,0.37,0.34,0,0,17.95,9.78,86.73,55.03,10.91,0.44,9.18,34.12,0,0.3,87.6,0.06,0.26,0.41,0.99,0.46,0.91,4.96,1.89,0.05,0.03,0.29,4.24,5.8,1.5,0.8,1.01,0.37,0.14,88.94,3.91,126.42,37.77,0.36,2.75,3.34,1.26,3.03,10.68,0,0.21,1.67,0,0,0.12,0.45,0.21,0.47,0.75,0.49,0,0,0.1,0.2,0,0,0,0.3,0.24,0.11,0.08,0.04,2.37,0.17,0.57,0.36,1.92,0.44,0.96,50.03,4.34,17.11,53.16,72.74,85.03,12.43,11.02,1.18,0,1.3,47.12,0.67,1.69,1.75,0.48,4.67,10.44,27.2,58.54,57.85,53.55,30.33,28.87,8.36,41.4,88.34,77.91,0.64,0.08,0.19,0.03,0.44,0.95,0.28,0.09,0.16,0.8,1.12,0.27,0.02,5.72,1.26,0.48,91.64,87.25,0.86,42.45,0.04,1.89,0.16,0.01,0,0.64,0.17,0,0,0,0.09,0,0.12,0.59,0.14,0.67,0,0.17,0.5,0.11,0,1.95,0,0.01,3.12,0.6,0.3,0.09,79.26,0.05,0.2,1.38,0,0,0.09,5.09,0.75,0.41,0.38,2.75,0.1,0.22,0.07,2.43,0.3,0.22,0.02,0.38,0.37,3.84,24.02,18.04,33.86,10.55,49.16,0.34,6.2,67.75,89.12,2.7,4.29,0.91,3.43,2.22,3.98,6.86,7.69,1.67,0.12,0.01,0.59,0.07,1.01,0,0,6.85,6.33,1.07,15.37,0.98,2.06,0.05,0.16,0.01,1.53,0.51,11.19,0.73,0.39,3.46,0.33,0.64,0.43,0.07,0.41,0.63,0.26,1.16,11.01,1.56,0.08,1.71,3.48,0.52,8.09,85.12,1.46,2.39,4.91,2.45,0.01,0.01,0.03,0.23,0.29,0.68,0.25,0.21,1.52,1.69,0,0.11,8.67,0.63,2.62,0,0,0.01,0.46,0.11,0.5,0.83,0.2,0.01,1.32,0.01,0.29,0,0,0.04,0.01,0.63,0.93,0.77,0.47,0,0.4,0.04,0.29,0.08,0.31,0.07,1.12,0.17,0.01,0.01,0.06,0.43,0.53,0,0.04,1.93,2.08,0.3,4.79,0,0.08,0.17,0.85,0.07,3.12,0,45.95,67.36,91.42,0.47,2.39,0.03,0.7,0.13,0.08,0,1.27,4.57,5.16,3.71,4.25,3.78,0.44,0.18,0.52,0.33,8.23,9.78,10.58,9.41,1.73,2.2,0,0.88,0.96,0.91,0.24,0,0.12,0.23,0.39,0,2.95,0.01,0.33,0.6,1.34,0.68,0.23,0.48,0.23,0.45,0.18,5.62,1.72,11.3,73.48,1.15,0.82,1.6,0.98,0.97,1.68,0.61,0.95,0.83,4.63,0.18,0.01,0.05,0.07,1.52,0,0,0,0.32,0.44,0.08,0.4,0.48,0.63,0.45,0,0,0,0,0,0.4,5.71,4.39,0.01,0.2,1.6,1.39,0.25,0,0.43,0.23,1.02,0.2,2.89,2.39,0.52,0.89,3.09,0.34,0.04,0.01,0.38,0,0,0,0.83,0,2.16,0.93,0.58,0.48,0.08,0.41,0.01,0.06,2.06,0,0.82,1.84,0.06,0.4,0.85,0.22,0.74,1.57,0.1,2.13,0.04,4.02,1.63,0.21,0.03,0.98,0.01,0.2,0.08,0.09,0.64,0.56],[1.57,266.01,0.58,0.65,6.35,0.08,0.54,0.36,1.17,0,2.15,3.93,283.07,0.63,2.55,8.57,35.38,101.06,0,2.24,277.12,0.01,1.14,7.84,0.47,2.6,1.87,1.15,0.14,0.36,0.59,2.89,0.81,2.61,0.54,0.25,0.03,4.35,15.94,0.84,0.21,0,0.87,0.03,1.1,0,0,3.76,3.01,0,0.47,0.18,0.5,0.02,1.3,1.42,0.87,0.74,0,0,0,0.13,0.11,0.75,0,0,1.34,0.58,1.01,0.59,0.64,0.96,0,0.55,0.12,0.24,0.57,0.89,1.25,78.22,0.01,28.83,54.24,80.56,40.58,29.64,35.31,3.3,0.04,0.36,145.36,0,0.01,0,37.93,8.13,23.38,60.37,102.88,109.44,112.96,82.82,94.11,102.58,121.3,276.95,227.18,0.55,0,0.25,0.13,0.59,1.03,1,1.27,0.24,2.55,1.9,1.43,0.52,14.87,2.74,2.29,281.21,267.35,0.25,73.66,34.29,3.7,1.4,35.12,33.41,3.45,4.08,22.74,36.48,36.07,0.01,0.28,0.45,0.07,0.95,0.9,0,0.69,0.59,0.54,0,1.24,0,1.92,1.48,0.43,1.25,0.48,1.01,0,0.09,0,0.02,0,4.41,1.15,5.41,2.35,1.59,1.39,3.71,6.65,40.16,2.61,7.93,0.14,3.34,0.29,0,2.66,0.1,0.13,0.99,1.12,0.5,1.9,0.41,195.65,267.48,3.39,0.21,0.02,0.03,0,0.01,0.35,0.03,0.03,0.74,3.03,3.1,4.88,2.88,0,0.01,27.43,14.91,0.71,41.08,2.39,1.53,0.04,0.01,0.01,1.36,3.18,6.52,1.56,0.69,5.12,0.89,0.48,1.93,1.64,0.88,2.06,9.07,10.22,38.84,12.23,0.04,9.19,13.33,2.06,29.41,0.76,2.43,2.21,1.44,0.13,0,0,0,0,3.35,0.91,0.05,0.72,0,0,0.44,0.06,31.91,0.87,2.56,0.13,0.24,0.07,0.3,0.33,1.06,0.31,0.5,0.44,8.69,0.06,0.08,0,0,0.41,0.01,1.37,0.26,0.89,0.61,0.4,0.26,1.22,1.19,0.65,40.04,0.28,25.01,1.47,1.61,0.16,0.14,3.34,1.06,0,0.24,0.38,0.46,1.26,0.43,0.63,1.01,0.29,0.32,0.68,2.16,0.34,3.66,0.72,30.38,0.01,0.19,0,0.02,0.43,0.29,0,0.49,3.98,2.56,3.37,1.26,2.12,0.35,0.83,1.08,1.75,29.35,25.48,36.71,27.95,0.16,12.81,0,0,1.39,3.23,0.27,0,3.5,5.01,2.35,0,0.31,0,0.12,0.28,0.4,0.08,0.02,0.72,0.02,0.1,0.18,0.27,0.54,42.3,11.97,0.08,0,3.22,1.45,5.35,5.24,2.61,1.61,1.46,0.14,0.38,0.07,0.03,0.52,0.73,0,0,0,1.37,5.42,0.48,8.32,1.09,0.41,0.45,0.01,0,0.01,0,0.17,1.7,7.34,8.03,0,0.01,0,9.41,0,16.99,0.21,7.73,2.4,3.5,4.44,2.01,0.05,2.52,2.58,0.09,0.93,0.36,1.46,1.6,0,0,0.18,0,0.25,1.69,3.31,2.01,1.16,2.51,0,0.01,0,0.21,0.63,0.01,0.04,1.09,0.55,0.03,1.26,0.24,2.02,0,0,1.26,2,0.85,0.01,0.73,0.08,0.1,0.84,2.15,0.05,0.08],[6.2,1.71,0.73,0.81,7.29,0.21,0.77,0.88,8.1,0.12,0.93,1.77,1.65,0.24,0.3,16.27,2.4,1.95,0,1.42,2.35,0.29,3.85,18.52,0.77,0.63,2.55,3.55,0.12,0.86,1.3,0.76,0.87,0.58,1.56,0.63,0.37,2.87,12,0.44,0.05,0.04,0.21,0.09,1.57,0,0,1.99,0.82,0.01,0,0.3,0.48,0.12,0.8,0.57,0.04,0.1,0.46,0.66,0,0.03,0.16,1.18,0,0.01,0.5,0.09,0.29,0.22,0.13,0.57,0,0.97,0.42,0.5,0.58,1.93,0.86,95.11,0.01,29.35,72.44,91.85,57.85,35.92,42.32,0.58,0.77,0.3,168.07,0,0,0,72.53,17.16,15.19,55.46,102.45,138.19,131.59,102.71,102.15,112.87,146.66,2.26,1.25,1.14,0,0.15,0.16,0.55,1.34,1.72,0.58,0.16,1.26,0.49,0.98,0.07,24.17,1.13,0.94,1.78,1.74,0.02,82.56,49.25,2.71,5.72,96.37,94.72,3.85,2.69,78.17,96.85,101.58,0,0.33,0.02,0.92,0.89,0.34,0,0.56,0.11,0.29,0,1.09,0.23,0.87,0.9,0.58,0.55,1.34,1.2,0.6,0.43,0.02,0.08,0.07,4.56,0.39,5.78,2.72,3.36,1.32,7.07,5.65,36.99,2.07,6.52,0.82,2.57,0.01,0.07,0.17,3.52,3.64,5,2.25,1.04,0.03,2.5,1.41,2.38,1.12,0.04,0.12,0.27,0.01,0.02,0.07,0.08,0.15,1.8,0.18,0.13,1.13,0.47,0.01,0,2.41,0.84,0.43,1.79,4.37,0.38,0.01,0.23,0.01,0.32,0.48,2.6,2.68,0.41,6.9,1.34,0.22,2.09,0.85,2.18,0.11,17.06,19.22,0.85,6.34,0.08,10.88,0.87,0.1,4.12,1.89,0.83,1.15,1.2,0.1,0.13,0.04,0.03,0.11,0.98,2.47,0.3,0,0.02,0,0.39,0.02,2.13,0.05,2.64,0,0,0,0.76,0.17,0.81,0.41,1.27,2.27,16.12,0.12,0,0.08,0.08,0.53,1.1,0.63,0.42,0.01,0.22,0.15,0.02,0.61,0.36,0.48,32.36,1.83,13.06,0.49,0.81,0.03,0.13,3.92,1.93,0.01,1.37,0.03,0.09,1.28,3.01,0.29,2.09,0.44,0.63,0.22,0.51,3.66,3.76,0.68,37.91,0.07,0.07,0,0,1.34,0.14,0,0.37,2.01,1.53,2.03,1.57,3.07,0.13,0.02,0.09,0.09,44.04,53.29,64.97,47.38,0.98,11.38,0.04,0,1.3,1.86,1.18,0,0.72,1.36,0.32,0,0.17,0,0,0.04,0.03,0.01,0.12,0.04,0.22,0.27,0.6,0.23,0.57,55.55,18.13,0.07,0.01,2.82,3.21,5.88,4.4,0.75,2.88,1.31,0.17,0.64,0.07,0.09,0.75,0.03,0.03,0,0,2.59,14.55,0.8,18.66,0.26,0.99,0.91,0,0,0,0,0,0.68,1.32,2.25,0,0,0,10.17,0.01,71.01,0.27,19.08,3.58,2.18,4.04,3.62,0.12,3.46,0.94,0.05,1.87,0.44,2.47,1.27,0,0,0.16,0,0.63,0.7,1.11,1.68,2.19,1.79,0,0.68,0.17,0.22,2.09,0.74,0.12,1.12,0.14,0.26,0.27,0.38,0.51,0.3,0,2.29,0.16,1.18,0.03,0.35,0.44,0.03,0.43,0.83,0.01,0.03],[10.6,10.56,0.43,0.31,5.3,1.41,1.62,0.87,0.83,0.12,1.38,1.55,9.38,2.85,0.75,6,6.16,12.82,0,0.85,10.13,0.46,0,6.26,0.02,0.24,2.11,13.09,0.08,0.17,0.3,0.08,0.48,0.32,0.16,1.05,1.63,0.21,0.99,3.47,0.34,1.14,1.38,0,0.5,0,0.01,0.59,1.24,1.09,0,0.09,0.08,0.05,0.88,1.15,0.73,0.4,0.06,0.04,0,0.41,0.15,0.01,0,0,0.34,0.02,0.42,0.37,1.06,0.01,0,0,0.15,0.02,1.31,2.6,1.08,39.39,0.03,15.5,36.92,48.68,42.94,14.56,10.42,2.31,0.81,0.17,52.98,0,0.35,0.01,6.85,13.99,13.08,35.66,64.41,50.6,48.34,29.67,33.23,27.64,46.43,11.04,15.51,0.57,0.02,0.12,0.35,0.08,0.04,0.32,1.75,0.38,1.91,0.37,0.89,0.39,4.19,3.29,0.95,10.63,12.82,0.16,43.15,180.84,3.67,2.18,294.76,305.79,9,1.97,299.62,307.52,318.6,0,0,0.03,0.04,0.25,0.38,0,1.01,1.74,1.11,0,0.33,0,0.29,0.07,0.37,0.88,0.06,1.46,0.35,0.14,0.12,0.01,0,0.79,0.87,0.02,0.02,0.55,0.02,0.17,1.84,2.6,0.27,0.08,0,0.01,0.21,0.15,0.03,0.02,0,0,0.13,0.03,0.01,0,16.38,10.86,1.44,0.17,0,0.77,0.15,0.03,0.41,1.03,0.11,0.91,0.12,0.34,0.74,1.15,0.01,0,3.46,2.03,0.76,2.56,1.37,2.95,0,0.07,0,0.03,0.75,2.74,0.49,0.77,10.85,4.03,0.39,4.39,1.55,0.91,0.37,4.26,8.58,7.27,2.14,0.17,0.88,3.31,0.99,7.09,1.85,0.26,0.32,0.67,0.29,0,0,0.01,0,0.89,1.1,0.03,0.04,0.19,0.25,0.02,0.13,7.24,0.01,0.04,0.1,0,0,0.96,1.05,0.45,0.33,0.67,1.05,7.14,0.18,0,0.6,0.6,0.57,0.61,0,0.17,0.1,0.35,0.54,0.15,0.07,0.01,0.12,4.51,0.48,1.98,1.62,1.1,0.09,0,0.03,1.11,0.01,0.78,0.07,0.02,1.87,0.01,0,0.65,0.05,0.29,0.32,0.9,0.14,1.58,0.11,40.3,0.02,0.03,0,0.02,0.38,0,0,0,0.86,1.05,0.61,0.26,1.06,0,0.02,0.18,0.12,38.18,41.53,41.75,42,0.03,8.27,0,0.08,0,0,0.18,0,6.31,6.51,6.05,0,6.76,2.19,2.72,2.6,4.21,4.48,1.64,2.29,1.64,1.93,0.08,0.33,0.54,12.05,16.58,0.25,0.03,0,0.01,0.13,0.07,0.01,0.42,0.15,0,0.46,0,0.7,1.23,0.98,0,0,0,5.79,5.74,1.04,4.61,0.34,1.53,0.64,0,0,0,0.01,0,0.08,2.48,2.7,0.02,0.03,0,1.2,0.1,256.58,0.75,5.43,1.04,0,0.71,1.2,0.36,1.18,0.52,0.02,2.31,0.02,2.91,0,0,0,0.53,0,0.28,0.56,3.93,0.92,1.58,1.89,0.01,0.06,0,0.22,0.61,0.02,0.1,0.1,0.58,0.2,1.11,0.31,0,0.13,0,2.8,0.14,1.09,0.77,1.35,0.47,0.51,0.75,1.73,0.2,0.55],[9.33,0.17,0.49,1.1,47.95,3.56,0.67,0.04,128.29,101.67,0.14,0.41,0.18,0.09,0.05,74.09,0.06,0.02,0,0.42,0.03,0,0.72,74.87,1.57,2.39,2.12,5.45,1.35,0.53,4.24,1.08,4.59,0.18,2.43,1.08,0.3,1.07,4.66,1.23,0.16,0.05,0.04,1.9,2.63,0,0,0.33,0.7,0,0.09,2.11,0.01,0.02,0.36,0.29,0.22,0.67,0.05,0.19,0,0.24,0.12,0.7,55.5,100.24,0.14,0.91,0.98,0.6,1.04,0.73,0,29.77,16.23,44.08,0.42,2.83,0.97,53.81,0,20.58,33.68,60.28,30.38,26.66,28.71,0.32,0.74,0.24,127.81,0,0,0,128.94,13.69,19.06,85.03,99.8,95.46,126.94,77.91,94.07,82.82,102.23,0.43,0.01,3.17,0,1.78,1.74,0.93,1.31,4.75,2.52,0.11,1.12,2.95,2.44,1.13,27.25,3.09,2.47,0.35,0.08,0,72.05,0.33,0.11,0.16,0.17,0.24,0.12,0.44,0.16,0.37,0.6,0,0,0.07,0.12,0.37,1.64,0.14,1.05,0.92,0.34,0,0.34,103.01,4.77,0.06,0.23,0.09,0.01,2.68,0.09,0.35,0.25,0.03,0,2.18,0.51,4.39,3.06,1.41,3.61,2.92,1.11,8.17,1.14,4.37,0.09,0.99,0.11,0.09,1.36,14.19,12.54,12.18,5.27,17.18,0.66,11.89,0,0.03,0.56,0.14,0.03,1.4,0.01,0,0.06,0.05,0.24,2.95,0.98,1.37,0.8,1.33,0.72,0,0.15,0.01,0.01,0,3.89,2.93,0.52,2.21,1.07,5.88,6.07,1.81,2.34,0.07,5.56,55.5,1.3,41.76,51.68,2.68,0.86,49.77,62.1,0.02,4.66,0.2,2.44,2.35,1.21,0,0.66,0.85,0.28,3.06,0.03,0,0,0,0.05,0.22,1.58,2.2,0.13,0.02,0,0.42,0.18,0.01,0.09,1.92,0.87,2.86,0.06,4.12,3.77,0.22,0.24,3.8,5.58,44.56,0.01,0.02,0.44,0.44,0.05,0.03,0.02,0.44,2.14,1.64,0.69,0.6,0.01,2.88,3.66,5.9,1.7,4.99,1.55,1,0.63,2.99,1.82,10.94,0,1.89,0,0.03,2.22,3.52,0.85,0.46,0.22,6.71,0.53,0.64,102.28,13.92,12.86,41.71,0,0.06,0,0.01,5.6,0.26,0,0.01,4.56,4.23,4.19,4.5,4.93,0.11,0.11,0.06,0,23.76,26.6,34.8,20.35,0.07,39.83,0.9,0,0.02,0.35,0.07,48.16,0.71,1.36,0.89,41.66,1.35,0.12,0.1,0.33,0.61,1.3,0.79,0.85,0.02,1.01,0.01,0.69,0.26,32.2,20.83,0.01,0.03,0.26,0.54,2.27,2.88,0.13,0.98,0.17,1.44,2.01,1.55,1.03,0.37,2,1.4,7.04,1.81,67.14,64.71,1.99,69.4,2.55,0.88,1.34,0,0,0,0,0,0.46,7.26,13.25,0,0,0,3.15,0.03,0.18,0.99,83.34,2.51,0.74,2.56,1.82,0.38,0.57,4.73,0.01,1.22,0.11,15.77,0.09,26.32,5.05,0.47,0,0.68,0.66,0.38,1.39,2.37,4.52,0.01,0,0.3,0,0.2,0.01,2.04,0.19,1.81,0.09,3.28,0.13,0.03,0.35,0,1.3,0.16,0,0.07,0.16,0.85,0.27,1.29,0.12,0.01,0.79]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>motif<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="top-10-motifs-in-each-topic" class="section level3">
<h3>Top 10 motifs in each topic</h3>
<pre class="r"><code>cat(&quot;Number of regions selected for each topic: \n&quot;)
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

colnames_homer &lt;- c(&quot;motif_name&quot;, &quot;consensus&quot;, &quot;P&quot;, &quot;log10P&quot;, &quot;Padj&quot;,  &quot;num_target&quot;, &quot;percent_target&quot;, &quot;num_bg&quot;, &quot;percent_bg&quot;)

top_motifs &lt;- data.frame(matrix(nrow=10, ncol = length(homer_res_topics)))
colnames(top_motifs) &lt;- names(homer_res_topics)
for (k in 1:length(homer_res_topics)){
  homer_res &lt;- homer_res_topics[[k]]
  colnames(homer_res) &lt;- colnames_homer
  homer_res &lt;- homer_res %&gt;% separate(motif_name, c(&quot;motif&quot;, &quot;origin&quot;, &quot;database&quot;), &quot;/&quot;)
  top_motifs[,k] &lt;- head(homer_res$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F, caption = &quot;Top 10 motifs enriched in each topic.&quot;)</code></pre>
<pre><code># Number of regions selected for each topic: 
#   k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11 
# 1046 1046 1046 1046 1046 1046 1046 1046 1046 1046 1046</code></pre>
<div id="htmlwidget-6884c229bd0bdf52b2c9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6884c229bd0bdf52b2c9">{"x":{"filter":"none","caption":"<caption>Top 10 motifs enriched in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["CEBP(bZIP)","HLF(bZIP)","NFIL3(bZIP)","CEBP:AP1(bZIP)","Atf4(bZIP)","Fos(bZIP)","Fra1(bZIP)","Fra2(bZIP)","BATF(bZIP)","JunB(bZIP)"],["PU.1:IRF8(ETS:IRF)","IRF8(IRF)","IRF3(IRF)","PU.1-IRF(ETS:IRF)","Fra1(bZIP)","JunB(bZIP)","Fos(bZIP)","BATF(bZIP)","Atf3(bZIP)","Fra2(bZIP)"],["CTCF(Zf)","Sp1(Zf)","BORIS(Zf)","KLF3(Zf)","Klf9(Zf)","NFY(CCAAT)","NRF1(NRF)","NRF(NRF)","KLF1(Zf)","Sp5(Zf)"],["PU.1:IRF8(ETS:IRF)","IRF8(IRF)","Ascl1(bHLH)","HEB(bHLH)","Ptf1a(bHLH)","E2A(bHLH)","Twist2(bHLH)","SpiB(ETS)","Tcf12(bHLH)","Tcf21(bHLH)"],["Gata6(Zf)","GATA3(Zf)","Gata4(Zf)","Gata2(Zf)","TRPS1(Zf)","Gata1(Zf)","GATA:SCL(Zf,bHLH)","RUNX(Runt)","RUNX1(Runt)","RUNX2(Runt)"],["Atf3(bZIP)","Fra1(bZIP)","Fos(bZIP)","BATF(bZIP)","Fra2(bZIP)","JunB(bZIP)","AP-1(bZIP)","Fosl2(bZIP)","Jun-AP1(bZIP)","Bach2(bZIP)"],["CEBP(bZIP)","Fra1(bZIP)","PU.1(ETS)","JunB(bZIP)","CEBP:AP1(bZIP)","AP-1(bZIP)","Fos(bZIP)","BATF(bZIP)","Fra2(bZIP)","Atf3(bZIP)"],["Atf3(bZIP)","Fra1(bZIP)","BATF(bZIP)","Fos(bZIP)","JunB(bZIP)","Fra2(bZIP)","AP-1(bZIP)","Fosl2(bZIP)","Jun-AP1(bZIP)","ERG(ETS)"],["ERG(ETS)","Fli1(ETS)","ETV1(ETS)","Etv2(ETS)","EWS:FLI1-fusion(ETS)","ETV4(ETS)","ETS1(ETS)","EWS:ERG-fusion(ETS)","Gata6(Zf)","Gata4(Zf)"],["Gata6(Zf)","Gata4(Zf)","Gata2(Zf)","GATA3(Zf)","Gata1(Zf)","TRPS1(Zf)","GATA:SCL(Zf,bHLH)","ETS1(ETS)","ERG(ETS)","ETV1(ETS)"],["ETS:E-box(ETS,bHLH)","Ascl1(bHLH)","ERG(ETS)","Etv2(ETS)","HEB(bHLH)","Ptf1a(bHLH)","Fli1(ETS)","Ascl2(bHLH)","E2A(bHLH)","ETS1(ETS)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-motif-enrichment-across-topics" class="section level3">
<h3>Heatmap of motif enrichment across topics</h3>
<p>Heatmap of motif enrichment -log10(p-value). Order motifs by hierarchical clustering.</p>
<pre class="r"><code>create_motif_enrichment_heatmap(motif_res, enrichment = &quot;-log10(p-value)&quot;, 
                                cluster_motifs = TRUE, cluster_topics = FALSE, motif_filter = 50, 
                                enrichment_range = c(0,100), method_cluster = &quot;average&quot;, 
                                font.size.motifs = 5, font.size.topics = 9)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-heatmap-pvalue-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-heatmap-pvalue-1">
Past versions of motif-heatmap-pvalue-1.png
</button>
</p>
<div id="fig-motif-heatmap-pvalue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-heatmap-pvalue-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># 70 out of 439 motifs included the heatmap</code></pre>
<p>Heatmap of motif enrichment z-score. Order motifs by hierarchical clustering.</p>
<pre class="r"><code>create_motif_enrichment_heatmap(motif_res, enrichment = &quot;z-score&quot;,
                                cluster_motifs = TRUE, cluster_topics = FALSE, motif_filter = 10, 
                                enrichment_range = c(-20,20), method_cluster = &quot;average&quot;, 
                                font.size.motifs = 5, font.size.topics = 9)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-heatmap-zscore-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-heatmap-zscore-1">
Past versions of motif-heatmap-zscore-1.png
</button>
</p>
<div id="fig-motif-heatmap-zscore-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-heatmap-zscore-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># 149 out of 439 motifs included the heatmap</code></pre>
</div>
<div id="scatterplots-of-motif-enrichment" class="section level3">
<h3>Scatterplots of motif enrichment</h3>
<pre class="r"><code># Plot enrichment (-log10 p-value) and ranking of the motifs

plots &lt;- vector(&quot;list&quot;, ncol(motif_res$mlog10P))
names(plots) &lt;- colnames(motif_res$mlog10P)

for( i in 1:length(plots)){
  plots[[i]] &lt;- create_motif_enrichment_ranking_plot(motif_res, k = i, 
                                                     max.overlaps = 20, subsample = FALSE)
}

# do.call(plot_grid,plots)</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) and the ranking</li>
</ul>
<p><strong>Topic 5 example</strong></p>
<pre class="r"><code>print(plots[[5]])</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-enrichment-ranking-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-enrichment-ranking-plot-2-1">
Past versions of motif-enrichment-ranking-plot-2-1.png
</button>
</p>
<div id="fig-motif-enrichment-ranking-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot motif enrichment (-log10 p-value) in each topic vs other topics

plots &lt;- vector(&quot;list&quot;, ncol(motif_res$mlog10P))
names(plots) &lt;- colnames(motif_res$mlog10P)

for( i in 1:length(homer_res_topics)){
  plots[[i]] &lt;- create_motif_enrichment_plot(motif_res, k = i, 
                                             max.overlaps = 20, subsample = TRUE)
}

# do.call(plot_grid,plots)</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) in <strong>topic 5</strong> vs other topics</li>
</ul>
<pre class="r"><code>print(plots[[5]])</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-enrichment-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-enrichment-plot-2-1">
Past versions of motif-enrichment-plot-2-1.png
</button>
</p>
<div id="fig-motif-enrichment-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-enrichment-plot-2-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="motif-enrichment-vs-gene-score" class="section level3">
<h3>Motif enrichment vs gene score</h3>
<p>Load pre-computed gene scores</p>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-TSS-l2&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
genescore_res &lt;- readRDS(file.path(gene.dir, &quot;genescore_result_topics.rds&quot;))

genes &lt;- genescore_res$genes
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genes$SYMBOL
gene_logFC &lt;- genescore_res$beta
rownames(gene_logFC) &lt;- genes$SYMBOL</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized//geneanalysis-Buenrostro2018-k=11-TSS-l2</code></pre>
<p>Get TF genes</p>
<pre class="r"><code>motif_names &lt;- gsub(&quot;\\s*\\(.*&quot;, &quot;&quot;, motif_res$motifs$motif)
gene_names &lt;- genes$SYMBOL

TF_genes &lt;- intersect(toupper(motif_names), toupper(gene_names))
cat(sprintf(&quot;%s TF genes mapped between motif names and gene symbol. \n&quot;, length(TF_genes)))</code></pre>
<pre><code># 266 TF genes mapped between motif names and gene symbol.</code></pre>
<p>Compute correlation between motif enrichment z-score and gene score:</p>
<p><strong>Topic 5 example</strong></p>
<ul>
<li>Compute motif enrichment z-scores from the motif enrichment p-values</li>
<li>Plot motif enrichment (-log10 p-value) and correlation between motif enrichment z-scores and gene scores</li>
<li>Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment z-score and gene scores</li>
</ul>
<pre class="r"><code>motif_gene_mapping &lt;- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 5, cor.motif = &quot;z-score&quot;)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-cor-plot-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-cor-plot-1-1">
Past versions of motif-cor-plot-1-1.png
</button>
</p>
<div id="fig-motif-cor-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-cor-plot-1-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif_gene_mapping &lt;- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_zscore, decreasing = T)),]
rownames(motif_gene_mapping) &lt;- 1:nrow(motif_gene_mapping)

cat(&quot;Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n&quot;)
print(head(motif_gene_mapping[,c(&quot;motif&quot;,&quot;motif_mlog10P&quot;, &quot;gene_score&quot;, &quot;cor_zscore&quot;)], 10))</code></pre>
<pre><code># Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: 
#              motif motif_mlog10P  gene_score cor_zscore
# 1        Gata2(Zf)    303.137548  2.56643713 0.80163329
# 2        Gata1(Zf)    289.153266  9.70484601 0.83293420
# 3        Gata6(Zf)    335.535917  0.91584314 0.51453059
# 4        TRPS1(Zf)    294.234511  0.16910153 0.09456715
# 5        ETV1(ETS)     16.394617  0.75275363 0.70669940
# 6        Fli1(ETS)     13.728049 -0.02383964 0.70401127
# 7      RUNX2(Runt)     29.366993 -0.49937641 0.23541775
# 8      Fosl2(bZIP)      8.043134 -2.78270103 0.78356896
# 9  Foxo1(Forkhead)      5.576341  2.91403281 0.82185685
# 10     Atoh1(bHLH)      7.083343  0.43950708 0.57305420</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores</li>
<li>Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>motif_gene_mapping &lt;- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 5, cor.motif = &quot;-log10(p-value)&quot;)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-cor-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-cor-plot-2-1">
Past versions of motif-cor-plot-2-1.png
</button>
</p>
<div id="fig-motif-cor-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-cor-plot-2-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif_gene_mapping &lt;- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_mlog10P, decreasing = T)),]
rownames(motif_gene_mapping) &lt;- 1:nrow(motif_gene_mapping)

cat(&quot;Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n&quot;)
print(head(motif_gene_mapping[,c(&quot;motif&quot;,&quot;motif_mlog10P&quot;, &quot;gene_score&quot;, &quot;cor_mlog10P&quot;)], 10))</code></pre>
<pre><code># Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: 
#              motif motif_mlog10P  gene_score cor_mlog10P
# 1        Gata1(Zf)    289.153266  9.70484601   0.9008827
# 2        Gata2(Zf)    303.137548  2.56643713   0.6605271
# 3        Gata6(Zf)    335.535917  0.91584314   0.3619495
# 4        ETV1(ETS)     16.394617  0.75275363   0.7930947
# 5        Fli1(ETS)     13.728049 -0.02383964   0.7558193
# 6      Fosl2(bZIP)      8.043134 -2.78270103   0.8442031
# 7      RUNX2(Runt)     29.366993 -0.49937641   0.2179535
# 8  Foxo1(Forkhead)      5.576341  2.91403281   0.8295553
# 9       Atf3(bZIP)      5.806517 -1.93470462   0.5962481
# 10     Atoh1(bHLH)      7.083343  0.43950708   0.4751842</code></pre>
<p><strong>GATA family</strong></p>
<pre class="r"><code>GATA_genes &lt;- grep(&quot;^GATA\\d*$&quot;, TF_genes, ignore.case=T, value=T)</code></pre>
<p>Plot GATA motifs in topic 5</p>
<pre class="r"><code>k = 5
selected_motifs &lt;- rownames(motif_res$motifs)[match(toupper(GATA_genes), toupper(motif_names))]
motif.dir &lt;- paste0(homer.dir, &quot;/homer_result_topic_&quot;, k, &quot;/knownResults/&quot;)

for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = &quot;both&quot;)
}</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-GATA-logos-1-1">
Past versions of plot-GATA-logos-1-1.png
</button>
</p>
<div id="fig-plot-GATA-logos-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-2.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-GATA-logos-1-2">
Past versions of plot-GATA-logos-1-2.png
</button>
</p>
<div id="fig-plot-GATA-logos-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-2.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-3.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-GATA-logos-1-3">
Past versions of plot-GATA-logos-1-3.png
</button>
</p>
<div id="fig-plot-GATA-logos-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-3.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-4.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-GATA-logos-1-4">
Past versions of plot-GATA-logos-1-4.png
</button>
</p>
<div id="fig-plot-GATA-logos-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/plot-GATA-logos-1-4.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># The PWM of the motif (GATA3(Zf),DR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer) was not in HOMER output of enriched motifs.</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = &quot;-log10(p-value)&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-gene-plot-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-1-1">
Past versions of motif-gene-plot-1-1.png
</button>
</p>
<div id="fig-motif-gene-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-gene-plot-1-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Plot motif enrichment (z-score) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = &quot;z-score&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-gene-plot-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-2-1">
Past versions of motif-gene-plot-2-1.png
</button>
</p>
<div id="fig-motif-gene-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/cde66e85bbe9decdf578d8c2cde26a3ef2f60268/docs/figure/motif_analysis_Buenrostro2018_chomVAR_scPeaks_filtered.Rmd/motif-gene-plot-2-1.png" target="_blank">cde66e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] grid      stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
#  [1] Logolas_1.3.1     reshape2_1.4.3    DT_0.16           htmlwidgets_1.5.3
#  [5] plotly_4.9.2.1    cowplot_1.1.0     ggrepel_0.9.0     ggplot2_3.3.3    
#  [9] tidyr_1.1.2       dplyr_1.0.2       fastTopics_0.4-6  Matrix_1.2-18    
# [13] workflowr_1.6.2  
# 
# loaded via a namespace (and not attached):
#  [1] nlme_3.1-140       mcmc_0.9-7         matrixStats_0.57.0 fs_1.3.1          
#  [5] bit64_4.0.5        progress_1.2.2     httr_1.4.2         rprojroot_2.0.2   
#  [9] tools_3.6.1        R6_2.5.0           irlba_2.3.3        lazyeval_0.2.2    
# [13] colorspace_2.0-0   ade4_1.7-16        withr_2.4.0        tidyselect_1.1.0  
# [17] prettyunits_1.1.1  bit_4.0.4          compiler_3.6.1     git2r_0.27.1      
# [21] quantreg_5.75      SparseM_1.78       labeling_0.4.2     scales_1.1.1      
# [25] SQUAREM_2021.1     quadprog_1.5-8     mixsqp_0.3-43      stringr_1.4.0     
# [29] digest_0.6.27      rmarkdown_2.6      MCMCpack_1.4-9     pkgconfig_2.0.3   
# [33] htmltools_0.5.0    invgamma_1.1       rlang_0.4.10       generics_0.1.0    
# [37] farver_2.0.3       jsonlite_1.7.2     crosstalk_1.1.0.1  magrittr_2.0.1    
# [41] Rcpp_1.0.6         munsell_0.5.0      ape_5.4-1          lifecycle_0.2.0   
# [45] CVXR_1.0-9         stringi_1.5.3      whisker_0.4        yaml_2.2.1        
# [49] MASS_7.3-53        Rtsne_0.15         plyr_1.8.6         parallel_3.6.1    
# [53] promises_1.1.1     crayon_1.3.4       lattice_0.20-41    hms_1.0.0         
# [57] knitr_1.30         pillar_1.4.7       seqinr_4.2-5       glue_1.4.2        
# [61] evaluate_0.14      data.table_1.13.6  RcppParallel_5.0.2 vctrs_0.3.6       
# [65] httpuv_1.5.4       MatrixModels_0.4-1 gtable_0.3.0       purrr_0.3.4       
# [69] ashr_2.2-47        xfun_0.19          gridBase_0.4-7     Rmpfr_0.8-2       
# [73] coda_0.19-4        later_1.1.0.1      viridisLite_0.3.0  truncnorm_1.0-8   
# [77] tibble_3.0.5       conquer_1.0.2      gmp_0.6-2          ellipsis_0.3.1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
