<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Motif and gene analysis using topic modeling results for Buenrostro et al (2018) scATAC-seq result (data processed using chromVAR processed counts with scPeaks)</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Motif and gene analysis using topic modeling results for Buenrostro <em>et al</em> (2018) scATAC-seq result (data processed using chromVAR processed counts with scPeaks)</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree0f7f8795fe76d0294e9c50fceaad7b13bbcc1a75targetblank0f7f879a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/0f7f8795fe76d0294e9c50fceaad7b13bbcc1a75" target="_blank">0f7f879</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree0f7f8795fe76d0294e9c50fceaad7b13bbcc1a75targetblank0f7f879a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/0f7f8795fe76d0294e9c50fceaad7b13bbcc1a75" target="_blank">0f7f879</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/clusters_pca_structure_Cusanovich2018.Rmd
    Untracked:  analysis/diff_count_Buenrostro2018_chomVAR_scPeaks.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  analysis/single_cell_rnaseq_demo.Rmd
    Untracked:  buenrostro2018.RData
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch

Unstaged changes:
    Modified:   analysis/cisTopic_Buenrostro2018_chomVAR_scPeaks.Rmd
    Modified:   analysis/diff_count_Cusanovich2018.Rmd
    Modified:   analysis/motif_gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/plots_Lareau2019_bonemarrow.Rmd
    Modified:   analysis/process_data_Buenrostro2018.Rmd
    Modified:   code/gene_annotation.R
    Modified:   code/motif_analysis.R
    Modified:   code/plots.R
    Modified:   scripts/diffcount_motif_analysis.R
    Modified:   scripts/fit_all_models_Buenrostro_2018.sbatch
    Modified:   scripts/fit_cisTopic_Buenrostro_2018_chromVAR_scPeaks.sh
    Modified:   scripts/postfit_Buenrostro2018.sh
    Modified:   scripts/postfit_Cusanovich2018.sh
    Modified:   scripts/postfit_motif_analysis.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/motif_gene_analysis_Buenrostro2018_chomVAR_scPeaks.Rmd</code>) and HTML (<code>docs/motif_gene_analysis_Buenrostro2018_chomVAR_scPeaks.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0f7f8795fe76d0294e9c50fceaad7b13bbcc1a75/analysis/motif_gene_analysis_Buenrostro2018_chomVAR_scPeaks.Rmd" target="_blank">0f7f879</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-12-23
</td>
<td>
motif and gene analysis for Buenrostro result, data processed using chromVAR processed counts with scPeaks
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform TF motif and gene analysis for the Buenrostro <em>et al</em> (2018) scATAC-seq result inferred from the multinomial topic model with <span class="math inline">\(k = 11\)</span>.</p>
<p>We use binarized scPeaks based on chromVAR processed counts with scPeaks (&gt; 1 sample).</p>
<div id="load-packages-and-some-functions-used-in-this-analysis" class="section level2">
<h2>Load packages and some functions used in this analysis</h2>
<pre class="r"><code>library(Matrix)
library(dplyr)
library(ggplot2)
library(cowplot)
library(fastTopics)
library(dplyr)
library(tidyr)
library(DT)
library(reshape)</code></pre>
</div>
<div id="load-data-and-topic-model-results" class="section level2">
<h2>Load data and topic model results</h2>
<p>Load the <strong>binarized</strong> data and the <span class="math inline">\(k = 11\)</span> Poisson NMF fit results from Buenrostro <em>et al</em> 2018 dataset using binarized scPeaks based on chromVAR processed counts with scPeaks (&gt; 1 sample).</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/chromVAR/&quot;
load(file.path(data.dir, &quot;Buenrostro_2018_binarized_scPeaks.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
# 2034 x 228965 counts matrix.</code></pre>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks/binarized/&quot;
fit &lt;- readRDS(file.path(out.dir, &quot;/fit-Buenrostro2018-binarized-scd-ex-k=11.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
<div id="visualize-by-structure-plot-grouped-by-cell-labels." class="section level2">
<h2>Visualize by Structure plot grouped by cell labels.</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
samples$label &lt;- as.factor(samples$label)

p.structure &lt;- structure_plot(fit_multinom,
                     grouping = samples[, &quot;label&quot;],n = Inf,gap = 40,
                     perplexity = 50,topics = 1:11,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)
# Perplexity automatically changed to 24 because original setting of 50 was too large for the number of samples (78)
# Perplexity automatically changed to 44 because original setting of 50 was too large for the number of samples (138)
# Perplexity automatically changed to 20 because original setting of 50 was too large for the number of samples (64)
# Perplexity automatically changed to 46 because original setting of 50 was too large for the number of samples (142)
# Perplexity automatically changed to 45 because original setting of 50 was too large for the number of samples (141)
# Perplexity automatically changed to 18 because original setting of 50 was too large for the number of samples (60)

print(p.structure)</code></pre>
<p><img src="figure/motif_gene_analysis_Buenrostro2018_chomVAR_scPeaks.Rmd/structure-plot-1.png" width="840" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-accessbility-analysis-of-the-atac-seq-regions-for-the-topics" class="section level2">
<h2>Differential accessbility analysis of the ATAC-seq regions for the topics</h2>
<pre class="r"><code>diffcount_file &lt;- file.path(out.dir, &quot;diffcount-Buenrostro2018-11topics.rds&quot;)
if(file.exists(diffcount_file)){
  cat(&quot;Load precomputed differential accessbility statistics.\n&quot;)
  diff_count_topics &lt;- readRDS(diffcount_file)
}else{
  cat(&quot;Computing differential accessbility statistics from topic model.\n&quot;)
  timing &lt;- system.time(diff_count_topics &lt;- diff_count_analysis(fit,counts))
  cat(sprintf(&quot;Computation took %0.2f seconds.\n&quot;,timing[&quot;elapsed&quot;]))
  cat(&quot;Saving results.\n&quot;)
  saveRDS(diff_count_topics, diffcount_file)
}
# Load precomputed differential accessbility statistics.</code></pre>
<p>Distribution of z-scores</p>
<pre class="r"><code>
zscore_topics &lt;-  melt(diff_count_topics$Z)
colnames(zscore_topics) &lt;- c(&quot;region&quot;, &quot;topic&quot;, &quot;zscore&quot;)
levels(zscore_topics$topic) &lt;- colnames(diff_count_topics$Z)

z.quantile.99 &lt;- apply(abs(diff_count_topics$Z), 2, quantile, 0.99)
cat(&quot;z-score 99% quantile: \n&quot;)
print(z.quantile.99)

p.hist.zscores &lt;- ggplot(zscore_topics, aes(x=zscore)) + 
  geom_histogram(binwidth=1, color=&quot;black&quot;, fill=&quot;white&quot;) + 
  coord_cartesian(xlim = c(-10, 20)) + theme_cowplot(font_size = 10) +
  facet_wrap(~ topic, ncol=4)

print(p.hist.zscores)</code></pre>
<p><img src="figure/motif_gene_analysis_Buenrostro2018_chomVAR_scPeaks.Rmd/hist-zscore-1.png" width="840" style="display: block; margin: auto;" /></p>
<pre><code># z-score 99% quantile: 
#       k1       k2       k3       k4       k5       k6       k7       k8 
# 5.429064 7.020322 5.828474 6.308036 6.161995 6.931451 4.862751 7.977127 
#       k9      k10      k11 
# 6.756263 6.858868 9.739615</code></pre>
</div>
<div id="motif-enrichment-analysis-using-homer" class="section level2">
<h2>Motif enrichment analysis using <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">HOMER</a></h2>
<ul>
<li><p>Details about HOMER motif analysis: <a href="http://homer.ucsd.edu/homer/motif/" class="uri">http://homer.ucsd.edu/homer/motif/</a> <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html" class="uri">http://homer.ucsd.edu/homer/ngs/peakMotifs.html</a></p></li>
<li><p>Motif enrichment result using regions with z-score above 99% quantile. RCC: <code>/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks/binarized/motifanalysis-Buenrostro2018-k=11-quantile/</code></p></li>
</ul>
<pre class="r"><code>homer.dir &lt;- paste0(out.dir, &quot;/motifanalysis-Buenrostro2018-k=11-quantile/HOMER/quantile&quot;)
homer_res &lt;- readRDS(file.path(homer.dir, &quot;/homer_knownResults.rds&quot;))
selected_regions &lt;- readRDS(file.path(homer.dir, &quot;/selected_regions.rds&quot;))

cat(&quot;Number of regions selected for each topic: \n&quot;)
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

top_motifs &lt;- data.frame(matrix(nrow=10, ncol = ncol(diff_count_topics$Z)))
colnames(top_motifs) &lt;- colnames(diff_count_topics$Z)
for (k in colnames(top_motifs)){
  homer_motifs &lt;- homer_res[[k]]
  colnames(homer_motifs) &lt;- c(&quot;Motif.name&quot;, &quot;Consensus&quot;, &quot;P-value&quot;, &quot;Log.P-value&quot;, &quot;q-value (Benjamini)&quot;, 
                              &quot;# of Target Sequences with Motif&quot;, &quot;% of Target Sequences with Motif&quot;,
                              &quot;# of Background Sequences with Motif&quot;, &quot;% of Background Sequences with Motif&quot;)
  homer_motifs &lt;- homer_motifs %&gt;% separate(Motif.name, c(&quot;motif&quot;, &quot;experiment&quot;, &quot;database&quot;), &quot;/&quot;)
  top_motifs[,k] &lt;- head(homer_motifs$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F,
              caption = &quot;Top 10 motifs enriched in each topic.&quot;)
# Number of regions selected for each topic: 
#   k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11 
# 2290 2290 2290 2290 2253 2258 2290 2290 2278 2264 2289</code></pre>
<div id="htmlwidget-4ae068db386cace9d91d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4ae068db386cace9d91d">{"x":{"filter":"none","caption":"<caption>Top 10 motifs enriched in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Fosl2(bZIP)","Fra1(bZIP)","Fra2(bZIP)","Fos(bZIP)","Atf4(bZIP)","Jun-AP1(bZIP)","Chop(bZIP)","JunB(bZIP)","AP-1(bZIP)","Atf3(bZIP)"],["BATF(bZIP)","Atf3(bZIP)","Fra1(bZIP)","Fra2(bZIP)","Fos(bZIP)","AP-1(bZIP)","JunB(bZIP)","Fosl2(bZIP)","Jun-AP1(bZIP)","ERG(ETS)"],["CEBP(bZIP)","CEBP:AP1(bZIP)","Fra1(bZIP)","JunB(bZIP)","HLF(bZIP)","Fra2(bZIP)","Fos(bZIP)","NFIL3(bZIP)","Atf3(bZIP)","BATF(bZIP)"],["Gata6(Zf)","Gata4(Zf)","Gata1(Zf)","Gata2(Zf)","GATA3(Zf)","GATA:SCL(Zf,bHLH)","TRPS1(Zf)","KLF6(Zf)","SCL(bHLH)","KLF5(Zf)"],["ERG(ETS)","Fli1(ETS)","ETV1(ETS)","EWS:FLI1-fusion(ETS)","Etv2(ETS)","ETS1(ETS)","ETS:E-box(ETS,bHLH)","ETV4(ETS)","EWS:ERG-fusion(ETS)","GABPA(ETS)"],["ETS:E-box(ETS,bHLH)","Ascl2(bHLH)","Ascl1(bHLH)","Etv2(ETS)","Fli1(ETS)","E2A(bHLH)","ERG(ETS)","EBF2(EBF)","Tcf12(bHLH)","ETV1(ETS)"],["Gata6(Zf)","Gata2(Zf)","Gata1(Zf)","Gata4(Zf)","GATA3(Zf)","TRPS1(Zf)","GATA:SCL(Zf,bHLH)","ETS1(ETS)","ERG(ETS)","Elf4(ETS)"],["Gata6(Zf)","Gata4(Zf)","GATA3(Zf)","Gata2(Zf)","Gata1(Zf)","TRPS1(Zf)","GATA:SCL(Zf,bHLH)","RUNX(Runt)","ETV1(ETS)","RUNX2(Runt)"],["Sp1(Zf)","NRF1(NRF)","NRF(NRF)","NFY(CCAAT)","Klf9(Zf)","KLF3(Zf)","KLF1(Zf)","Sp5(Zf)","GFX(?)","CRE(bZIP)"],["PU.1(ETS)","CEBP(bZIP)","PU.1:IRF8(ETS:IRF)","ELF5(ETS)","SpiB(ETS)","JunB(bZIP)","Fra1(bZIP)","BATF(bZIP)","Fos(bZIP)","AP-1(bZIP)"],["PU.1:IRF8(ETS:IRF)","IRF8(IRF)","IRF3(IRF)","PU.1-IRF(ETS:IRF)","SpiB(ETS)","IRF1(IRF)","Ascl1(bHLH)","HEB(bHLH)","IRF2(IRF)","Ptf1a(bHLH)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="top-genes" class="section level2">
<h2>Top genes</h2>
<div id="gene-body-model" class="section level3">
<h3>Gene body model</h3>
<p>Gene scores were computed using the gene score model (model 42) in the <code>archR</code> paper with some modifications. This model uses bi-directional exponential decays from the gene TSS (extended upstream by 5 kb by default) and the gene transcription termination site (TTS). Note: the current version of the function does not account for neighboring gene boundaries.</p>
<ul>
<li>Gene body model, normalized by the l2 norm of weights, as in Stouffer's z-score method.</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-genebody-l2&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- colnames(gene_scores)

rownames(gene_scores) &lt;- genes[match(rownames(gene_scores), genes$ENSEMBL), &quot;SYMBOL&quot;]

for (k in colnames(top_genes)){
  top_genes[,k] &lt;- rownames(gene_scores)[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F,
              caption = &quot;Top 10 genes in each topic.&quot;)</code></pre>
<div id="htmlwidget-05ecd7f4f064e52a9141" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-05ecd7f4f064e52a9141">{"x":{"filter":"none","caption":"<caption>Top 10 genes in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["IMPG1","ADAMTSL1","MYH8","NLRC5","TEX13A","LOC100505782","DSCR10","MIR548AL","DSCR8","SKA3"],["UGT2B4","TMEM212","MIR4770","MINAR1","FIRRE","HMGA2","RARRES1","INPP4B","ZNF536","ZNF267"],["COLEC11","HGF","NEDD9","TLR4","IRF8","DENND3","COX4I1","XCL1","OR2A5","PRKCD"],["TAOK2","ZFPM1","UBXN10","PLA2G2F","CABP5","MIR133A1","MIR1-2","TLDC2","MIR5189","CHID1"],["SRD5A2","FAM227B","LINC00421","ZNF654","ELMO1","EFNB2","MIR4328","MIR548AE2","P2RY10","GS1-600G8.3"],["TEX36-AS1","LOC283299","MIR4499","ADAMTS9","LINC00221","MAGI1-AS1","MIR548AU","TRPC5","OR4N3P","MTG2"],["UGT1A6","UGT1A7","UGT1A9","LINC00323","SPANXN1","SLC22A9","UGT1A10","MTCL1","UGT1A8","RNF130"],["ABCC4","HBD","OR51V1","HBG1","HBB","OR51B4","CYP3A43","MIR5192","TIMM23","MICAL2"],["DBP","MAMSTR","CA11","FSTL3","GPX4","SNAPC2","TGFBR3L","MIR1909","CTXN1","LOC100288123"],["NEDD9","NAMPT","TGM3","FPR1","P2RX7","FAM49A","NLRP3","LIPN","RBM47","KIR3DL1"],["IL2RA","SC5D","BLK","KTN1-AS1","PLXNA4","MIR4668","TCL1A","IGF2R","AKAP13","HSD17B3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Gene body model, normalized by the total weights (i.e. weighted averge).</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-genebody-sum&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- colnames(gene_scores)

rownames(gene_scores) &lt;- genes[match(rownames(gene_scores), genes$ENSEMBL), &quot;SYMBOL&quot;]

for (k in colnames(top_genes)){
  top_genes[,k] &lt;- rownames(gene_scores)[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F,
              caption = &quot;Top 10 genes in each topic.&quot;)</code></pre>
<div id="htmlwidget-2b5ba16072c1c8b42672" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2b5ba16072c1c8b42672">{"x":{"filter":"none","caption":"<caption>Top 10 genes in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["IMPG1","ADAMTSL1","MYH8","NLRC5","TEX13A","LOC100505782","DSCR10","MIR548AL","DSCR8","SKA3"],["UGT2B4","TMEM212","MIR4770","MINAR1","FIRRE","HMGA2","RARRES1","INPP4B","ZNF536","ZNF267"],["COLEC11","HGF","NEDD9","TLR4","IRF8","DENND3","COX4I1","XCL1","OR2A5","PRKCD"],["TAOK2","ZFPM1","UBXN10","PLA2G2F","CABP5","MIR133A1","MIR1-2","TLDC2","MIR5189","CHID1"],["SRD5A2","FAM227B","LINC00421","ZNF654","ELMO1","EFNB2","MIR4328","MIR548AE2","P2RY10","GS1-600G8.3"],["TEX36-AS1","LOC283299","MIR4499","ADAMTS9","LINC00221","MAGI1-AS1","MIR548AU","TRPC5","OR4N3P","MTG2"],["UGT1A6","UGT1A7","UGT1A9","LINC00323","SPANXN1","SLC22A9","UGT1A10","MTCL1","UGT1A8","RNF130"],["ABCC4","HBD","OR51V1","HBG1","HBB","OR51B4","CYP3A43","MIR5192","TIMM23","MICAL2"],["DBP","MAMSTR","CA11","FSTL3","GPX4","SNAPC2","TGFBR3L","MIR1909","CTXN1","LOC100288123"],["NEDD9","NAMPT","TGM3","FPR1","P2RX7","FAM49A","NLRP3","LIPN","RBM47","KIR3DL1"],["IL2RA","SC5D","BLK","KTN1-AS1","PLXNA4","MIR4668","TCL1A","IGF2R","AKAP13","HSD17B3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="tss-model" class="section level3">
<h3>TSS model</h3>
<p>Gene scores were computed using TSS based method as in Lareau et al. Nature Biotech, 2019 as well as the <code>model 21</code> in <code>archR</code> paper. This model weights chromatin accessibility around gene promoters by using bi-directional exponential decays from the TSS.</p>
<ul>
<li>TSS model, normalized by the l2 norm of weights, as in Stouffer's z-score method.</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-TSS-l2&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- colnames(gene_scores)

rownames(gene_scores) &lt;- genes[match(rownames(gene_scores), genes$ENSEMBL), &quot;SYMBOL&quot;]

for (k in colnames(top_genes)){
  top_genes[,k] &lt;- rownames(gene_scores)[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F,
              caption = &quot;Top 10 genes in each topic.&quot;)</code></pre>
<div id="htmlwidget-c80327190d663c62c9c9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c80327190d663c62c9c9">{"x":{"filter":"none","caption":"<caption>Top 10 genes in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["IMPG1","ADAMTSL1","MYH8","NLRC5","TEX13A","LOC100505782","DSCR10","MIR548AL","DSCR8","SKA3"],["UGT2B4","TMEM212","MIR4770","MINAR1","FIRRE","HMGA2","RARRES1","INPP4B","ZNF536","ZNF267"],["COLEC11","HGF","NEDD9","TLR4","IRF8","DENND3","COX4I1","XCL1","OR2A5","PRKCD"],["TAOK2","ZFPM1","UBXN10","PLA2G2F","CABP5","MIR133A1","MIR1-2","TLDC2","MIR5189","CHID1"],["SRD5A2","FAM227B","LINC00421","ZNF654","ELMO1","EFNB2","MIR4328","MIR548AE2","P2RY10","GS1-600G8.3"],["TEX36-AS1","LOC283299","MIR4499","ADAMTS9","LINC00221","MAGI1-AS1","MIR548AU","TRPC5","OR4N3P","MTG2"],["UGT1A6","UGT1A7","UGT1A9","LINC00323","SPANXN1","SLC22A9","UGT1A10","MTCL1","UGT1A8","RNF130"],["ABCC4","HBD","OR51V1","HBG1","HBB","OR51B4","CYP3A43","MIR5192","TIMM23","MICAL2"],["DBP","MAMSTR","CA11","FSTL3","GPX4","SNAPC2","TGFBR3L","MIR1909","CTXN1","LOC100288123"],["NEDD9","NAMPT","TGM3","FPR1","P2RX7","FAM49A","NLRP3","LIPN","RBM47","KIR3DL1"],["IL2RA","SC5D","BLK","KTN1-AS1","PLXNA4","MIR4668","TCL1A","IGF2R","AKAP13","HSD17B3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>TSS model, normalized by the total weights (i.e. weighted averge).</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-TSS-sum&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_genes &lt;- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) &lt;- colnames(gene_scores)

rownames(gene_scores) &lt;- genes[match(rownames(gene_scores), genes$ENSEMBL), &quot;SYMBOL&quot;]

for (k in colnames(top_genes)){
  top_genes[,k] &lt;- rownames(gene_scores)[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F,
              caption = &quot;Top 10 genes in each topic.&quot;)</code></pre>
<div id="htmlwidget-7db0eac8347c9c29d3d2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7db0eac8347c9c29d3d2">{"x":{"filter":"none","caption":"<caption>Top 10 genes in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["IMPG1","ADAMTSL1","MYH8","NLRC5","TEX13A","LOC100505782","DSCR10","MIR548AL","DSCR8","SKA3"],["UGT2B4","TMEM212","MIR4770","MINAR1","FIRRE","HMGA2","RARRES1","INPP4B","ZNF536","ZNF267"],["COLEC11","HGF","NEDD9","TLR4","IRF8","DENND3","COX4I1","XCL1","OR2A5","PRKCD"],["TAOK2","ZFPM1","UBXN10","PLA2G2F","CABP5","MIR133A1","MIR1-2","TLDC2","MIR5189","CHID1"],["SRD5A2","FAM227B","LINC00421","ZNF654","ELMO1","EFNB2","MIR4328","MIR548AE2","P2RY10","GS1-600G8.3"],["TEX36-AS1","LOC283299","MIR4499","ADAMTS9","LINC00221","MAGI1-AS1","MIR548AU","TRPC5","OR4N3P","MTG2"],["UGT1A6","UGT1A7","UGT1A9","LINC00323","SPANXN1","SLC22A9","UGT1A10","MTCL1","UGT1A8","RNF130"],["ABCC4","HBD","OR51V1","HBG1","HBB","OR51B4","CYP3A43","MIR5192","TIMM23","MICAL2"],["DBP","MAMSTR","CA11","FSTL3","GPX4","SNAPC2","TGFBR3L","MIR1909","CTXN1","LOC100288123"],["NEDD9","NAMPT","TGM3","FPR1","P2RX7","FAM49A","NLRP3","LIPN","RBM47","KIR3DL1"],["IL2RA","SC5D","BLK","KTN1-AS1","PLXNA4","MIR4668","TCL1A","IGF2R","AKAP13","HSD17B3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2>Gene-set enrichment analysis (GSEA)</h2>
<ul>
<li>Gene body model, normalized by the l2 norm of weights, as in Stouffer's z-score method.</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-genebody-l2&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_pathways_up[,k] &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
                     unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_pathways_down[,k] &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
                     unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)</code></pre>
<div id="htmlwidget-53847066bda55c91ee2c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-53847066bda55c91ee2c">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["chr17q25.M8281","chr17p13.M9760","chr17q21.M8783","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON.M15936","chr17q12.M3433","chr16q24.M9818","chr20q11.M1556","Keratinization.http://identifiers.org/reactome/R-HSA-6805567","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON.M17620","NIKOLSKY_BREAST_CANCER_17Q11_Q21_AMPLICON.M3434"],["JAATINEN_HEMATOPOIETIC_STEM_CELL_UP.M15107","HAY_BONE_MARROW_CD34_POS_HSC.M39182","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN.M4235","BLANCO_MELO_BETA_INTERFERON_TREATED_BRONCHIAL_EPITHELIAL_CELLS_UP.M34022","AIZARANI_LIVER_C29_MVECS_2.M39128","MURARO_PANCREAS_ENDOTHELIAL_CELL.M39176","WANG_SMARCE1_TARGETS_UP.M1804","LAKE_ADULT_KIDNEY_C17_COLLECTING_SYSTEM_PCS_STRESSED_DISSOC_SUBSET.M39236","MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN.M2256","EPPERT_HSC_R.M19231"],["CUI_DEVELOPING_HEART_C8_MACROPHAGE.M39305","HU_FETAL_RETINA_MICROGLIA.M39266","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP.M7977","GSE29618_MONOCYTE_VS_PDC_UP.M4945","GAO_LARGE_INTESTINE_24W_C11_PANETH_LIKE_CELL.M39160","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP.M4963","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP.M3120","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA.M39022","GSE29618_MONOCYTE_VS_MDC_UP.M4948","GSE10325_LUPUS_BCELL_VS_LUPUS_MYELOID_DN.M3091"],["chr16p13.M9048","NIKOLSKY_BREAST_CANCER_16P13_AMPLICON.M1788","chr17q21.M8783","GO_DNA_PACKAGING_COMPLEX.M17506","chr6p22.M6858","chr7q22.M3562","KRCTCNNNNMANAGC_UNKNOWN.M9024","REACTOME_RNA_POLYMERASE_I_PROMOTER_ESCAPE.M27691","HDACs deacetylate histones.1270436","REACTOME_HDACS_DEACETYLATE_HISTONES.M27230"],["HAY_BONE_MARROW_CD34_POS_HSC.M39182","JAATINEN_HEMATOPOIETIC_STEM_CELL_UP.M15107","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP.M6899","EPPERT_HSC_R.M19231","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_UP.M6902","TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP.M6670","WANG_SMARCE1_TARGETS_UP.M1804","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN.M4235","chr15q11.M11848","HADDAD_T_LYMPHOCYTE_AND_NK_PROGENITOR_UP.M17082"],["HADDAD_B_LYMPHOCYTE_PROGENITOR.M939","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_DN.M6903","GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_DN.M6908","HAY_BONE_MARROW_CD34_POS_PRE_PC.M39191","chr14q32.M16134","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP.M4970","ZHENG_CORD_BLOOD_C10_MULTILYMPHOID_PROGENITOR.M39219","GSE10325_BCELL_VS_MYELOID_UP.M3081","FAN_EMBRYONIC_CTX_BRAIN_B_CELL.M39045","GSE29618_PDC_VS_MDC_UP.M4951"],["GO_FLAVONOID_GLUCURONIDATION.M24063","GO_XENOBIOTIC_GLUCURONIDATION.M24064","Glucuronidation.http://identifiers.org/reactome/R-HSA-156588","Ascorbate and aldarate metabolism.82932","GO_FLAVONOID_METABOLIC_PROCESS.M15331","REACTOME_GLUCURONIDATION.M17787","KEGG_ASCORBATE_AND_ALDARATE_METABOLISM.M605","GO_CELLULAR_GLUCURONIDATION.M29241","GO_URONIC_ACID_METABOLIC_PROCESS.M14675","Glucuronidation.1270208"],["HU_FETAL_RETINA_BLOOD.M39263","HALLMARK_HEME_METABOLISM.M5945","chr14q32.M16134","chr15q11.M11848","ZHENG_CORD_BLOOD_C3_MEGAKARYOCYTE_ERYTHROID_PROGENITOR.M39212","GNF2_SPTA1.M13711","ROSS_AML_OF_FAB_M7_TYPE.M4621","GNF2_MAP2K3.M1670","EPPERT_PROGENITOR.M19232","GNF2_TAL1.M19762"],["chr17p13.M9760","chr16p13.M9048","chr17q25.M8281","GSE9006_TYPE_1_VS_TYPE_2_DIABETES_PBMC_AT_DX_UP.M5797","AGUIRRE_PANCREATIC_CANCER_COPY_NUMBER_UP.M9565","chr11q13.M11664","GINESTIER_BREAST_CANCER_ZNF217_AMPLIFIED_DN.M17094","chr17q21.M8783","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON.M15936","chr22q13.M17432"],["GSE22886_NAIVE_CD8_TCELL_VS_MONOCYTE_DN.M4496","GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_DN.M3088","AIZARANI_LIVER_C25_KUPFFER_CELLS_4.M39126","GSE29618_MONOCYTE_VS_MDC_UP.M4948","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP.M4963","AIZARANI_LIVER_C2_KUPFFER_CELLS_1.M39106","AIZARANI_LIVER_C6_KUPFFER_CELLS_2.M39110","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA.M39022","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP.M7977","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP.M3120"],["HAY_BONE_MARROW_DENDRITIC_CELL.M39194","GSE29618_PDC_VS_MDC_UP.M4951","GSE29618_BCELL_VS_PDC_DN.M4942","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_DN.M4964","GSE29618_MONOCYTE_VS_PDC_DN.M4946","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP.M4970","GSE29618_BCELL_VS_PDC_DAY7_FLU_VACCINE_DN.M4959","MODULE_75.M12401","GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS.M39166","MODULE_46.M15983"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Gene body model, normalized by the total weights (i.e. weighted averge).</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-genebody-sum&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_pathways_up[,k] &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
                     unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_pathways_down[,k] &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)] %&gt;% 
                     unite(&quot;pathway&quot;, c(&quot;name&quot;, &quot;id&quot;), sep = &quot;.&quot;, remove = TRUE)
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)</code></pre>
<div id="htmlwidget-c6385dc0e77af21e76d8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c6385dc0e77af21e76d8">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["chr17q25.M8281","chr17p13.M9760","chr17q21.M8783","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON.M15936","chr17q12.M3433","chr20q11.M1556","chr16q24.M9818","Keratinization.http://identifiers.org/reactome/R-HSA-6805567","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON.M17620","NIKOLSKY_BREAST_CANCER_17Q11_Q21_AMPLICON.M3434"],["JAATINEN_HEMATOPOIETIC_STEM_CELL_UP.M15107","HAY_BONE_MARROW_CD34_POS_HSC.M39182","BLANCO_MELO_BETA_INTERFERON_TREATED_BRONCHIAL_EPITHELIAL_CELLS_UP.M34022","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN.M4235","AIZARANI_LIVER_C29_MVECS_2.M39128","WANG_SMARCE1_TARGETS_UP.M1804","MURARO_PANCREAS_ENDOTHELIAL_CELL.M39176","LAKE_ADULT_KIDNEY_C17_COLLECTING_SYSTEM_PCS_STRESSED_DISSOC_SUBSET.M39236","MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN.M2256","AIZARANI_LIVER_C20_LSECS_3.M39121"],["CUI_DEVELOPING_HEART_C8_MACROPHAGE.M39305","HU_FETAL_RETINA_MICROGLIA.M39266","GSE29618_MONOCYTE_VS_PDC_UP.M4945","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP.M7977","GAO_LARGE_INTESTINE_24W_C11_PANETH_LIKE_CELL.M39160","GSE29618_MONOCYTE_VS_MDC_UP.M4948","GSE10325_LUPUS_BCELL_VS_LUPUS_MYELOID_DN.M3091","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP.M4963","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA.M39022","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP.M3120"],["chr16p13.M9048","chr17q21.M8783","NIKOLSKY_BREAST_CANCER_16P13_AMPLICON.M1788","GO_DNA_PACKAGING_COMPLEX.M17506","chr6p22.M6858","chr7q22.M3562","KRCTCNNNNMANAGC_UNKNOWN.M9024","RNA Polymerase I Chain Elongation.1269662","HDACs deacetylate histones.1270436","REACTOME_HDACS_DEACETYLATE_HISTONES.M27230"],["HAY_BONE_MARROW_CD34_POS_HSC.M39182","JAATINEN_HEMATOPOIETIC_STEM_CELL_UP.M15107","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP.M6899","EPPERT_HSC_R.M19231","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_UP.M6902","TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP.M6670","WANG_SMARCE1_TARGETS_UP.M1804","chr15q11.M11848","ZHENG_CORD_BLOOD_C6_HSC_MULTIPOTENT_PROGENITOR.M39215","GO_HOMOPHILIC_CELL_ADHESION_VIA_PLASMA_MEMBRANE_ADHESION_MOLECULES.M7823"],["HADDAD_B_LYMPHOCYTE_PROGENITOR.M939","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_DN.M6903","GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_DN.M6908","HAY_BONE_MARROW_CD34_POS_PRE_PC.M39191","chr14q32.M16134","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP.M4970","ZHENG_CORD_BLOOD_C10_MULTILYMPHOID_PROGENITOR.M39219","GSE10325_BCELL_VS_MYELOID_UP.M3081","GSE29618_PDC_VS_MDC_UP.M4951","FAN_EMBRYONIC_CTX_BRAIN_B_CELL.M39045"],["GO_FLAVONOID_GLUCURONIDATION.M24063","GO_XENOBIOTIC_GLUCURONIDATION.M24064","Glucuronidation.http://identifiers.org/reactome/R-HSA-156588","Ascorbate and aldarate metabolism.82932","GO_FLAVONOID_METABOLIC_PROCESS.M15331","GO_CELLULAR_GLUCURONIDATION.M29241","REACTOME_GLUCURONIDATION.M17787","KEGG_ASCORBATE_AND_ALDARATE_METABOLISM.M605","WP_GLUCURONIDATION.M39404","Porphyrin and chlorophyll metabolism.83021"],["HU_FETAL_RETINA_BLOOD.M39263","HALLMARK_HEME_METABOLISM.M5945","chr14q32.M16134","chr15q11.M11848","ZHENG_CORD_BLOOD_C3_MEGAKARYOCYTE_ERYTHROID_PROGENITOR.M39212","ROSS_AML_OF_FAB_M7_TYPE.M4621","GNF2_SPTA1.M13711","GNF2_MAP2K3.M1670","GNF2_TAL1.M19762","EPPERT_PROGENITOR.M19232"],["NIKOLSKY_BREAST_CANCER_16P13_AMPLICON.M1788","chr16p13.M9048","chr17p13.M9760","chr17q25.M8281","GSE9006_TYPE_1_VS_TYPE_2_DIABETES_PBMC_AT_DX_UP.M5797","chr11q13.M11664","AGUIRRE_PANCREATIC_CANCER_COPY_NUMBER_UP.M9565","GINESTIER_BREAST_CANCER_ZNF217_AMPLIFIED_DN.M17094","chr17q21.M8783","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON.M15936"],["GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_DN.M3088","AIZARANI_LIVER_C25_KUPFFER_CELLS_4.M39126","GSE29618_MONOCYTE_VS_MDC_UP.M4948","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP.M4963","GSE22886_NAIVE_CD8_TCELL_VS_MONOCYTE_DN.M4496","AIZARANI_LIVER_C2_KUPFFER_CELLS_1.M39106","AIZARANI_LIVER_C6_KUPFFER_CELLS_2.M39110","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA.M39022","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP.M7977","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP.M3120"],["HAY_BONE_MARROW_DENDRITIC_CELL.M39194","GSE29618_PDC_VS_MDC_UP.M4951","GSE29618_BCELL_VS_PDC_DN.M4942","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_DN.M4964","GSE29618_MONOCYTE_VS_PDC_DN.M4946","GSE29618_BCELL_VS_PDC_DAY7_FLU_VACCINE_DN.M4959","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP.M4970","MODULE_75.M12401","MODULE_46.M15983","GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS.M39166"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>TSS model, normalized by the l2 norm of weights, as in Stouffer's z-score method.</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-TSS-l2&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_IDs_up &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_up[,k] &lt;- paste0(top_IDs_up$name, &quot;(&quot;, top_IDs_up$id, &quot;)&quot;)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_IDs_down &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_down[,k] &lt;- paste0(top_IDs_down$name, &quot;(&quot;, top_IDs_down$id, &quot;)&quot;)
  
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)</code></pre>
<div id="htmlwidget-c475d1395aac45d19559" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c475d1395aac45d19559">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["chr17q25(M8281)","chr17p13(M9760)","chr17q21(M8783)","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON(M15936)","chr17q12(M3433)","chr16q24(M9818)","chr20q11(M1556)","Keratinization(http://identifiers.org/reactome/R-HSA-6805567)","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON(M17620)","NIKOLSKY_BREAST_CANCER_17Q11_Q21_AMPLICON(M3434)"],["JAATINEN_HEMATOPOIETIC_STEM_CELL_UP(M15107)","HAY_BONE_MARROW_CD34_POS_HSC(M39182)","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN(M4235)","BLANCO_MELO_BETA_INTERFERON_TREATED_BRONCHIAL_EPITHELIAL_CELLS_UP(M34022)","WANG_SMARCE1_TARGETS_UP(M1804)","AIZARANI_LIVER_C29_MVECS_2(M39128)","MURARO_PANCREAS_ENDOTHELIAL_CELL(M39176)","LAKE_ADULT_KIDNEY_C17_COLLECTING_SYSTEM_PCS_STRESSED_DISSOC_SUBSET(M39236)","AIZARANI_LIVER_C20_LSECS_3(M39121)","MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN(M2256)"],["CUI_DEVELOPING_HEART_C8_MACROPHAGE(M39305)","GSE29618_MONOCYTE_VS_PDC_UP(M4945)","HU_FETAL_RETINA_MICROGLIA(M39266)","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP(M7977)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP(M4963)","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA(M39022)","GAO_LARGE_INTESTINE_24W_C11_PANETH_LIKE_CELL(M39160)","AIZARANI_LIVER_C23_KUPFFER_CELLS_3(M39124)","GSE29618_MONOCYTE_VS_MDC_UP(M4948)","GSE10325_LUPUS_BCELL_VS_LUPUS_MYELOID_DN(M3091)"],["chr16p13(M9048)","NIKOLSKY_BREAST_CANCER_16P13_AMPLICON(M1788)","chr17q21(M8783)","GO_DNA_PACKAGING_COMPLEX(M17506)","chr6p22(M6858)","chr7q22(M3562)","KRCTCNNNNMANAGC_UNKNOWN(M9024)","HDACs deacetylate histones(1270436)","REACTOME_HDACS_DEACETYLATE_HISTONES(M27230)","RNA Polymerase I Chain Elongation(1269662)"],["JAATINEN_HEMATOPOIETIC_STEM_CELL_UP(M15107)","HAY_BONE_MARROW_CD34_POS_HSC(M39182)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP(M6899)","EPPERT_HSC_R(M19231)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_UP(M6902)","TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP(M6670)","WANG_SMARCE1_TARGETS_UP(M1804)","HADDAD_T_LYMPHOCYTE_AND_NK_PROGENITOR_UP(M17082)","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN(M4235)","chr15q11(M11848)"],["HADDAD_B_LYMPHOCYTE_PROGENITOR(M939)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_DN(M6903)","HAY_BONE_MARROW_CD34_POS_PRE_PC(M39191)","GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_DN(M6908)","chr14q32(M16134)","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP(M4970)","ZHENG_CORD_BLOOD_C10_MULTILYMPHOID_PROGENITOR(M39219)","GSE10325_BCELL_VS_MYELOID_UP(M3081)","FAN_EMBRYONIC_CTX_BRAIN_B_CELL(M39045)","GSE29618_PDC_VS_MDC_UP(M4951)"],["GO_FLAVONOID_GLUCURONIDATION(M24063)","GO_XENOBIOTIC_GLUCURONIDATION(M24064)","Glucuronidation(http://identifiers.org/reactome/R-HSA-156588)","Ascorbate and aldarate metabolism(82932)","GO_FLAVONOID_METABOLIC_PROCESS(M15331)","GO_CELLULAR_GLUCURONIDATION(M29241)","REACTOME_GLUCURONIDATION(M17787)","GO_URONIC_ACID_METABOLIC_PROCESS(M14675)","KEGG_ASCORBATE_AND_ALDARATE_METABOLISM(M605)","WP_GLUCURONIDATION(M39404)"],["HU_FETAL_RETINA_BLOOD(M39263)","HALLMARK_HEME_METABOLISM(M5945)","chr14q32(M16134)","chr15q11(M11848)","ZHENG_CORD_BLOOD_C3_MEGAKARYOCYTE_ERYTHROID_PROGENITOR(M39212)","ROSS_AML_OF_FAB_M7_TYPE(M4621)","GNF2_SPTA1(M13711)","GNF2_MAP2K3(M1670)","GNF2_TAL1(M19762)","ZHENG_CORD_BLOOD_C4_PUTATIVE_EARLY_ERYTHROID_COMMITMENT(M39213)"],["chr16p13(M9048)","chr17p13(M9760)","chr17q25(M8281)","GSE9006_TYPE_1_VS_TYPE_2_DIABETES_PBMC_AT_DX_UP(M5797)","AGUIRRE_PANCREATIC_CANCER_COPY_NUMBER_UP(M9565)","chr11q13(M11664)","GINESTIER_BREAST_CANCER_ZNF217_AMPLIFIED_DN(M17094)","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON(M15936)","chr17q21(M8783)","chr22q13(M17432)"],["AIZARANI_LIVER_C25_KUPFFER_CELLS_4(M39126)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP(M4963)","GSE29618_MONOCYTE_VS_MDC_UP(M4948)","GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_DN(M3088)","GSE22886_NAIVE_CD8_TCELL_VS_MONOCYTE_DN(M4496)","AIZARANI_LIVER_C6_KUPFFER_CELLS_2(M39110)","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA(M39022)","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP(M7977)","AIZARANI_LIVER_C2_KUPFFER_CELLS_1(M39106)","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP(M3120)"],["HAY_BONE_MARROW_DENDRITIC_CELL(M39194)","GSE29618_PDC_VS_MDC_UP(M4951)","GSE29618_BCELL_VS_PDC_DN(M4942)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_DN(M4964)","GSE29618_MONOCYTE_VS_PDC_DN(M4946)","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP(M4970)","GSE29618_BCELL_VS_PDC_DAY7_FLU_VACCINE_DN(M4959)","MODULE_75(M12401)","MODULE_46(M15983)","GAO_LARGE_INTESTINE_ADULT_CI_MESENCHYMAL_CELLS(M39166)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>TSS model, normalized by the total weights (i.e. weighted averge).</li>
</ul>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Buenrostro2018-k=11-TSS-sum&quot;)
load(file.path(gene.dir, &quot;genescores_gsea.Rdata&quot;))

top_pathways_up &lt;- top_pathways_down &lt;- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) &lt;- colnames(top_pathways_down) &lt;- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic &lt;- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up &lt;- gsea_topic[gsea_topic$ES &gt; 0,]
  top_IDs_up &lt;- as.character(gsea_up[head(order(gsea_up$pval), 10), &quot;pathway&quot;])
  top_IDs_up &lt;- gene_set_info[match(top_IDs_up, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_up[,k] &lt;- paste0(top_IDs_up$name, &quot;(&quot;, top_IDs_up$id, &quot;)&quot;)
  
  gsea_down &lt;- gsea_topic[gsea_topic$ES &lt; 0,]
  top_IDs_down &lt;- as.character(gsea_down[head(order(gsea_down$pval), 10), &quot;pathway&quot;])
  top_IDs_down &lt;- gene_set_info[match(top_IDs_down, gene_set_info$id),c(&quot;name&quot;, &quot;id&quot;)]
  top_pathways_down[,k] &lt;- paste0(top_IDs_down$name, &quot;(&quot;, top_IDs_down$id, &quot;)&quot;)
  
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = &quot;Top 10 pathways enriched at the top of the gene rank list.&quot;)
</code></pre>
<div id="htmlwidget-cbe6b4412fbdb9a1f341" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cbe6b4412fbdb9a1f341">{"x":{"filter":"none","caption":"<caption>Top 10 pathways enriched at the top of the gene rank list.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["chr17q25(M8281)","chr17q21(M8783)","chr17p13(M9760)","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON(M15936)","chr17q12(M3433)","chr16q24(M9818)","chr20q11(M1556)","Keratinization(http://identifiers.org/reactome/R-HSA-6805567)","GO_INTERMEDIATE_FILAMENT_CYTOSKELETON(M17620)","NIKOLSKY_BREAST_CANCER_17Q11_Q21_AMPLICON(M3434)"],["JAATINEN_HEMATOPOIETIC_STEM_CELL_UP(M15107)","HAY_BONE_MARROW_CD34_POS_HSC(M39182)","BLANCO_MELO_BETA_INTERFERON_TREATED_BRONCHIAL_EPITHELIAL_CELLS_UP(M34022)","GRAHAM_CML_DIVIDING_VS_NORMAL_QUIESCENT_DN(M4235)","MURARO_PANCREAS_ENDOTHELIAL_CELL(M39176)","WANG_SMARCE1_TARGETS_UP(M1804)","AIZARANI_LIVER_C29_MVECS_2(M39128)","LAKE_ADULT_KIDNEY_C17_COLLECTING_SYSTEM_PCS_STRESSED_DISSOC_SUBSET(M39236)","EPPERT_HSC_R(M19231)","AIZARANI_LIVER_C20_LSECS_3(M39121)"],["CUI_DEVELOPING_HEART_C8_MACROPHAGE(M39305)","HU_FETAL_RETINA_MICROGLIA(M39266)","GSE29618_MONOCYTE_VS_PDC_UP(M4945)","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP(M7977)","GAO_LARGE_INTESTINE_24W_C11_PANETH_LIKE_CELL(M39160)","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA(M39022)","GSE29618_MONOCYTE_VS_MDC_UP(M4948)","GSE10325_LUPUS_BCELL_VS_LUPUS_MYELOID_DN(M3091)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP(M4963)","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP(M3120)"],["chr16p13(M9048)","chr17q21(M8783)","NIKOLSKY_BREAST_CANCER_16P13_AMPLICON(M1788)","chr6p22(M6858)","GO_DNA_PACKAGING_COMPLEX(M17506)","chr7q22(M3562)","REACTOME_RNA_POLYMERASE_I_PROMOTER_ESCAPE(M27691)","HDACs deacetylate histones(1270436)","REACTOME_HDACS_DEACETYLATE_HISTONES(M27230)","KRCTCNNNNMANAGC_UNKNOWN(M9024)"],["HAY_BONE_MARROW_CD34_POS_HSC(M39182)","JAATINEN_HEMATOPOIETIC_STEM_CELL_UP(M15107)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP(M6899)","EPPERT_HSC_R(M19231)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_UP(M6902)","TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP(M6670)","WANG_SMARCE1_TARGETS_UP(M1804)","chr15q11(M11848)","HADDAD_T_LYMPHOCYTE_AND_NK_PROGENITOR_UP(M17082)","ZHENG_CORD_BLOOD_C6_HSC_MULTIPOTENT_PROGENITOR(M39215)"],["HADDAD_B_LYMPHOCYTE_PROGENITOR(M939)","GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_30MIN_DN(M6903)","HAY_BONE_MARROW_CD34_POS_PRE_PC(M39191)","GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_DN(M6908)","chr14q32(M16134)","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP(M4970)","ZHENG_CORD_BLOOD_C10_MULTILYMPHOID_PROGENITOR(M39219)","GSE10325_BCELL_VS_MYELOID_UP(M3081)","FAN_EMBRYONIC_CTX_BRAIN_B_CELL(M39045)","GSE29618_PDC_VS_MDC_UP(M4951)"],["GO_FLAVONOID_GLUCURONIDATION(M24063)","GO_XENOBIOTIC_GLUCURONIDATION(M24064)","Glucuronidation(http://identifiers.org/reactome/R-HSA-156588)","Ascorbate and aldarate metabolism(82932)","GO_FLAVONOID_METABOLIC_PROCESS(M15331)","GO_CELLULAR_GLUCURONIDATION(M29241)","REACTOME_GLUCURONIDATION(M17787)","KEGG_ASCORBATE_AND_ALDARATE_METABOLISM(M605)","WP_GLUCURONIDATION(M39404)","GO_URONIC_ACID_METABOLIC_PROCESS(M14675)"],["HU_FETAL_RETINA_BLOOD(M39263)","HALLMARK_HEME_METABOLISM(M5945)","chr14q32(M16134)","chr15q11(M11848)","ZHENG_CORD_BLOOD_C3_MEGAKARYOCYTE_ERYTHROID_PROGENITOR(M39212)","GNF2_SPTA1(M13711)","ROSS_AML_OF_FAB_M7_TYPE(M4621)","GNF2_MAP2K3(M1670)","ZHENG_CORD_BLOOD_C4_PUTATIVE_EARLY_ERYTHROID_COMMITMENT(M39213)","GNF2_TAL1(M19762)"],["NIKOLSKY_BREAST_CANCER_16P13_AMPLICON(M1788)","chr16p13(M9048)","chr17p13(M9760)","chr17q25(M8281)","AGUIRRE_PANCREATIC_CANCER_COPY_NUMBER_UP(M9565)","GSE9006_TYPE_1_VS_TYPE_2_DIABETES_PBMC_AT_DX_UP(M5797)","chr11q13(M11664)","GINESTIER_BREAST_CANCER_ZNF217_AMPLIFIED_DN(M17094)","NIKOLSKY_BREAST_CANCER_17Q21_Q25_AMPLICON(M15936)","chr17q21(M8783)"],["AIZARANI_LIVER_C25_KUPFFER_CELLS_4(M39126)","GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_DN(M3088)","GSE29618_MONOCYTE_VS_MDC_UP(M4948)","GSE22886_NAIVE_CD8_TCELL_VS_MONOCYTE_DN(M4496)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_UP(M4963)","AIZARANI_LIVER_C6_KUPFFER_CELLS_2(M39110)","FAN_EMBRYONIC_CTX_BIG_GROUPS_MICROGLIA(M39022)","AIZARANI_LIVER_C2_KUPFFER_CELLS_1(M39106)","GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_PPARG_KO_MACROPHAGE_UP(M7977)","GSE11057_PBMC_VS_MEM_CD4_TCELL_UP(M3120)"],["HAY_BONE_MARROW_DENDRITIC_CELL(M39194)","GSE29618_PDC_VS_MDC_UP(M4951)","GSE29618_BCELL_VS_PDC_DN(M4942)","GSE29618_MONOCYTE_VS_PDC_DAY7_FLU_VACCINE_DN(M4964)","GSE29618_MONOCYTE_VS_PDC_DN(M4946)","GSE29618_PDC_VS_MDC_DAY7_FLU_VACCINE_UP(M4970)","GSE29618_BCELL_VS_PDC_DAY7_FLU_VACCINE_DN(M4959)","MODULE_46(M15983)","MODULE_75(M12401)","RYAN_MANTLE_CELL_LYMPHOMA_NOTCH_DIRECT_UP(M39016)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] reshape_0.8.8    DT_0.16          tidyr_1.1.2      fastTopics_0.4-6
# [5] cowplot_1.1.0    ggplot2_3.3.2    dplyr_1.0.2      Matrix_1.2-18   
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.0      Rcpp_1.0.5         lattice_0.20-41    prettyunits_1.1.1 
#  [5] rprojroot_2.0.2    digest_0.6.27      plyr_1.8.6         R6_2.5.0          
#  [9] MatrixModels_0.4-1 evaluate_0.14      coda_0.19-4        httr_1.4.2        
# [13] pillar_1.4.7       rlang_0.4.9        progress_1.2.2     lazyeval_0.2.2    
# [17] data.table_1.13.4  irlba_2.3.3        SparseM_1.78       whisker_0.4       
# [21] rmarkdown_2.6      labeling_0.4.2     Rtsne_0.15         stringr_1.4.0     
# [25] htmlwidgets_1.5.3  munsell_0.5.0      compiler_3.6.1     httpuv_1.5.4      
# [29] xfun_0.19          pkgconfig_2.0.3    mcmc_0.9-7         htmltools_0.5.0   
# [33] tidyselect_1.1.0   tibble_3.0.4       workflowr_1.6.2    quadprog_1.5-8    
# [37] matrixStats_0.57.0 viridisLite_0.3.0  crayon_1.3.4       conquer_1.0.2     
# [41] withr_2.3.0        later_1.1.0.1      MASS_7.3-53        grid_3.6.1        
# [45] jsonlite_1.7.2     gtable_0.3.0       lifecycle_0.2.0    git2r_0.27.1      
# [49] magrittr_2.0.1     scales_1.1.1       RcppParallel_5.0.2 stringi_1.5.3     
# [53] farver_2.0.3       fs_1.3.1           promises_1.1.1     ellipsis_0.3.1    
# [57] generics_0.1.0     vctrs_0.3.6        tools_3.6.1        glue_1.4.2        
# [61] purrr_0.3.4        crosstalk_1.1.0.1  hms_0.5.3          yaml_2.2.1        
# [65] colorspace_2.0-0   plotly_4.9.2.1     knitr_1.30         quantreg_5.75     
# [69] MCMCpack_1.4-9</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
