<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Motif analysis using topic modeling results for Cusanovich et al (2018) scATAC-seq data</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Motif analysis using topic modeling results for Cusanovich <em>et al</em> (2018) scATAC-seq data</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree8e3f64121d94f1d3d85aa6063bfe04ce5617e0e2targetblank8e3f641a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/8e3f64121d94f1d3d85aa6063bfe04ce5617e0e2" target="_blank">8e3f641</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree8e3f64121d94f1d3d85aa6063bfe04ce5617e0e2targetblank8e3f641a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/8e3f64121d94f1d3d85aa6063bfe04ce5617e0e2" target="_blank">8e3f641</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/plotly/Buenrostro_2018_Chen2019pipeline/

Untracked files:
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_10_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_11_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_12_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_13_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_1_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_2_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_3_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_4_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_5_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_6_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_7_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_8_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/gsea_topic_9_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_10_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_11_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_12_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_13_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_1_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_2_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_3_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_4_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_5_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_6_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_7_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_8_tss-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum_files/plotly-main-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/crosstalk-1.1.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/jquery-3.5.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-binding-4.9.3/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-htmlwidgets-css-1.57.1/
    Untracked:  output/plotly/Cusanovich2018/volcano_topic_9_tss-sum_files/plotly-main-1.57.1/
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/fit_models_Cusanovich2018_tissues.sh

Unstaged changes:
    Modified:   analysis/assess_fits_Cusanovich2018.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   analysis/process_data_Cusanovich2018.Rmd
    Modified:   output/plotly/Cusanovich2018/gsea_topic_10_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_10_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_11_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_11_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_12_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_12_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_13_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_13_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_1_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_1_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_2_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_2_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_3_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_3_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_4_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_4_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_5_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_5_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_6_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_6_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_7_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_7_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_8_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_8_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_9_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/gsea_topic_9_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_10_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_10_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_11_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_11_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_12_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_12_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_13_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_13_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_1_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_1_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_2_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_2_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_3_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_3_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_4_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_4_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_5_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_5_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_6_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_6_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_7_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_7_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_8_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_8_tss-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_9_genebody-sum.html
    Modified:   output/plotly/Cusanovich2018/volcano_topic_9_tss-sum.html
    Modified:   scripts/fit_all_models_Cusanovich2018.sh
    Modified:   scripts/fit_poisson_nmf.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/motif_analysis_Cusanovich2018.Rmd</code>) and HTML (<code>docs/motif_analysis_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/8e3f64121d94f1d3d85aa6063bfe04ce5617e0e2/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">8e3f641</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-05
</td>
<td>
cleaned results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/40ea1f1d37845883771d1e29a09de326467028e0/docs/motif_analysis_Cusanovich2018.html" target="_blank">40ea1f1</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/780af9ec2b7df605022a5864024ef8ae0e369ad5/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">780af9e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-21
</td>
<td>
added motif logo plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/motif_analysis_Cusanovich2018.html" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c7428cdf6599187564a1d4e4896daa384e155a6d/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">c7428cd</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
minor adjustment on the figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/motif_analysis_Cusanovich2018.html" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/4458ed5bf26503cb8cc53fb412abe62d96fd56e5/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">4458ed5</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
added KLF/SP genes and added an option to compute correlation using -log10P
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/motif_analysis_Cusanovich2018.html" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c9d674286d289488c47518606b422b0d901dde39/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">c9d6742</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
updated the correlations between motif enrichment (using z-score) and gene z-score
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/motif_analysis_Cusanovich2018.html" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0134bdfe702344f01e4ae492b2fdd5c3f0a7d8ac/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">0134bdf</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
fix colors in scatter plots of create_motif_gene_scatterplot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/motif_analysis_Cusanovich2018.html" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f51b56e493e3713f1329a64876152c270b9c4f08/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">f51b56e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
added colors to scatterplots for motif enrichment and gene scores
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/motif_analysis_Cusanovich2018.html" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/5f1841157267142b5306364c1a22a516679ac82e/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">5f18411</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Plot motif enrichment and correlate with gene scores
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform TF motif analysis for the Cusanovich <em>et al</em> (2018) scATAC-seq result inferred from the multinomial topic model with <span class="math inline">\(k = 13\)</span>.</p>
<div id="load-packages-and-some-functions-used-in-this-analysis" class="section level2">
<h2>Load packages and some functions used in this analysis</h2>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape2)
library(Logolas)
library(grid)
source(&quot;code/motif_analysis.R&quot;)
source(&quot;code/plots.R&quot;)</code></pre>
</div>
<div id="load-data-and-topic-model-results" class="section level2">
<h2>Load data and topic model results</h2>
<p>Load the data and the <span class="math inline">\(k = 13\)</span> Poisson NMF fit results.</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
rm(counts)</code></pre>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
<div id="visualize-by-structure-plot-grouped-by-tissues" class="section level2">
<h2>Visualize by Structure plot grouped by tissues</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
rows &lt;- sample(nrow(fit$L),4000)
samples$tissue &lt;- as.factor(samples$tissue)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = rows),
                              grouping = samples[rows, &quot;tissue&quot;],n = Inf,gap = 40,
                              perplexity = 50,topics = 1:13,colors = colors_topics,
                              num_threads = 4,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-tissue-1">
Past versions of structure-plot-tissue-1.png
</button>
</p>
<div id="fig-structure-plot-tissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-accessbility-analysis-of-the-atac-seq-regions-for-the-topics" class="section level2">
<h2>Differential accessbility analysis of the ATAC-seq regions for the topics</h2>
<p>Load results from differential accessbility analysis for the topics</p>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
cat(sprintf(&quot;Load results from %s \n&quot;, out.dir))</code></pre>
<pre><code># Load results from /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018</code></pre>
<pre class="r"><code>diff_count_topics &lt;- readRDS(file.path(out.dir, &quot;/diffcount-Cusanovich2018-13topics.rds&quot;))</code></pre>
<p>Distribution of z-scores</p>
<pre class="r"><code>zscore_topics &lt;-  melt(diff_count_topics$Z)
colnames(zscore_topics) &lt;- c(&quot;region&quot;, &quot;topic&quot;, &quot;zscore&quot;)
levels(zscore_topics$topic) &lt;- colnames(diff_count_topics$Z)

z.quantile.99 &lt;- apply(abs(diff_count_topics$Z), 2, quantile, 0.99)
cat(&quot;z-score 99% quantile: \n&quot;)
print(z.quantile.99)

p.hist.zscores &lt;- ggplot(zscore_topics, aes(x=zscore)) + 
  geom_histogram(binwidth=1, color=&quot;black&quot;, fill=&quot;white&quot;) + 
  coord_cartesian(xlim = c(-10, 30)) + theme_cowplot(font_size = 10) +
  facet_wrap(~ topic, ncol=4)

print(p.hist.zscores)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/hist-zscore-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hist-zscore-1">
Past versions of hist-zscore-1.png
</button>
</p>
<div id="fig-hist-zscore-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/hist-zscore-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># z-score 99% quantile: 
#       k1       k2       k3       k4       k5       k6       k7       k8 
# 21.42992 31.48751 25.46082 25.97670 34.64418 37.09098 32.07250 39.65746 
#       k9      k10      k11      k12      k13 
# 25.93102 15.88394 34.29782 39.80147 20.71928</code></pre>
</div>
<div id="motif-enrichment-analysis-using-homer" class="section level2">
<h2>Motif enrichment analysis using <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">HOMER</a></h2>
<ul>
<li>Details about HOMER motif analysis:</li>
<li><a href="http://homer.ucsd.edu/homer/motif/" class="uri">http://homer.ucsd.edu/homer/motif/</a></li>
<li><p><a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html" class="uri">http://homer.ucsd.edu/homer/ngs/peakMotifs.html</a></p></li>
<li><p>Motif enrichment result using regions with z-score above 99% quantile.</p></li>
</ul>
<p>Compile Homer results across topics</p>
<pre class="r"><code>homer.dir &lt;- paste0(out.dir, &quot;/motifanalysis-Cusanovich2018-k=13-quantile/HOMER&quot;)
cat(sprintf(&quot;Directory of motif analysis result: %s \n&quot;, homer.dir))
homer_res_topics &lt;- readRDS(file.path(homer.dir, &quot;/homer_knownResults.rds&quot;))
selected_regions &lt;- readRDS(file.path(homer.dir, &quot;/selected_regions.rds&quot;))

# Compile Homer results (pvalue and ranking) across topics
motif_res &lt;- compile_homer_motif_res(homer_res_topics)
saveRDS(motif_res, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))
cat(&quot;compiled homer motif results are saved in&quot;, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))

motif_table &lt;- data.frame(motif = gsub(&quot;/.*&quot;, &quot;&quot;, rownames(motif_res$mlog10P)),
                          round(motif_res$mlog10P,2))
DT::datatable(motif_table, rownames = F, caption = &quot;Motif enrichment (-log10P)&quot;)</code></pre>
<pre><code># Directory of motif analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/motifanalysis-Cusanovich2018-k=13-quantile/HOMER 
# compiled homer motif results are saved in /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/motifanalysis-Cusanovich2018-k=13-quantile/HOMER/homer_motif_enrichment_results.rds</code></pre>
<div id="htmlwidget-a6d95f92de83d3e66151" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6d95f92de83d3e66151">{"x":{"filter":"none","caption":"<caption>Motif enrichment (-log10P)<\/caption>","data":[["AMYB(HTH)","AP-1(bZIP)","AP-2alpha(AP2)","AP-2gamma(AP2)","Ap4(bHLH)","AR-halfsite(NR)","ARE(NR)","Arnt:Ahr(bHLH)","Ascl1(bHLH)","Ascl2(bHLH)","Atf1(bZIP)","Atf2(bZIP)","Atf3(bZIP)","Atf4(bZIP)","Atf7(bZIP)","Atoh1(bHLH)","Bach1(bZIP)","Bach2(bZIP)","Bapx1(Homeobox)","Barx1(Homeobox)","BATF(bZIP)","Bcl11a(Zf)","Bcl6(Zf)","BHLHA15(bHLH)","bHLHE40(bHLH)","bHLHE41(bHLH)","BMAL1(bHLH)","BMYB(HTH)","BORIS(Zf)","Brachyury(T-box)","Brn1(POU,Homeobox)","Brn2(POU,Homeobox)","bZIP:IRF(bZIP,IRF)","c-Jun-CRE(bZIP)","c-Myc(bHLH)","c-Myc(bHLH)","CArG(MADS)","Cdx2(Homeobox)","CDX4(Homeobox)","CEBP:AP1(bZIP)","CEBP:CEBP(bZIP)","CEBP(bZIP)","Chop(bZIP)","CHR(?)","CLOCK(bHLH)","COUP-TFII(NR)","COUP-TFII(NR)","CRE(bZIP)","CREB5(bZIP)","CRX(Homeobox)","CTCF-SatelliteElement(Zf?)","CTCF(Zf)","CUX1(Homeobox)","Cux2(Homeobox)","DLX1(Homeobox)","DLX2(Homeobox)","Dlx3(Homeobox)","DLX5(Homeobox)","DMRT1(DM)","DMRT6(DM)","DUX(Homeobox)","DUX4(Homeobox)","Duxbl(Homeobox)","E-box(bHLH)","E2A(bHLH),near_PU.1","E2A(bHLH)","E2F(E2F)","E2F1(E2F)","E2F3(E2F)","E2F4(E2F)","E2F6(E2F)","E2F7(E2F)","EAR2(NR)","EBF(EBF)","EBF1(EBF)","EBF2(EBF)","EBNA1(EBV-virus)","Egr1(Zf)","Egr2(Zf)","EHF(ETS)","EKLF(Zf)","ELF1(ETS)","ELF3(ETS)","Elf4(ETS)","ELF5(ETS)","Elk1(ETS)","Elk4(ETS)","En1(Homeobox)","Eomes(T-box)","ERE(NR),IR3","ERG(ETS)","Erra(NR)","ERRg(NR)","Esrrb(NR)","ETS:E-box(ETS,bHLH)","ETS:RUNX(ETS,Runt)","ETS(ETS)","Ets1-distal(ETS)","ETS1(ETS)","ETV1(ETS)","Etv2(ETS)","ETV4(ETS)","EWS:ERG-fusion(ETS)","EWS:FLI1-fusion(ETS)","Fli1(ETS)","Fos(bZIP)","Fosl2(bZIP)","Fox:Ebox(Forkhead,bHLH)","FOXA1:AR(Forkhead,NR)","FOXA1(Forkhead)","FOXA1(Forkhead)","Foxa2(Forkhead)","Foxa3(Forkhead)","FoxD3(forkhead)","Foxf1(Forkhead)","Foxh1(Forkhead)","FOXK1(Forkhead)","FOXK2(Forkhead)","FoxL2(Forkhead)","FOXM1(Forkhead)","Foxo1(Forkhead)","Foxo3(Forkhead)","FOXP1(Forkhead)","Fra1(bZIP)","Fra2(bZIP)","FXR(NR),IR1","GABPA(ETS)","GATA:SCL(Zf,bHLH)","GATA(Zf),IR3","GATA(Zf),IR4","Gata1(Zf)","Gata2(Zf)","GATA3(Zf),DR4","GATA3(Zf),DR8","GATA3(Zf)","Gata4(Zf)","Gata6(Zf)","Gfi1b(Zf)","GFX(?)","GFY-Staf(?,Zf)","GFY(?)","Gli2(Zf)","GLI3(Zf)","GLIS3(Zf)","GRE(NR),IR3","GRE(NR),IR3","GRHL2(CP2)","GSC(Homeobox)","Hand2(bHLH)","HEB(bHLH)","HIC1(Zf)","HIF-1a(bHLH)","HIF-1b(HLH)","HIF2a(bHLH)","HINFP(Zf)","HLF(bZIP)","Hnf1(Homeobox)","HNF1b(Homeobox)","HNF4a(NR),DR1","HNF6(Homeobox)","Hnf6b(Homeobox)","HOXA1(Homeobox)","Hoxa10(Homeobox)","Hoxa11(Homeobox)","Hoxa13(Homeobox)","HOXA2(Homeobox)","Hoxa9(Homeobox)","HOXB13(Homeobox)","Hoxb4(Homeobox)","Hoxc9(Homeobox)","Hoxd10(Homeobox)","Hoxd11(Homeobox)","Hoxd12(Homeobox)","Hoxd13(Homeobox)","HRE(HSF)","HRE(HSF)","IRF:BATF(IRF:bZIP)","IRF1(IRF)","IRF2(IRF)","IRF3(IRF)","IRF4(IRF)","IRF8(IRF)","Isl1(Homeobox)","ISRE(IRF)","Jun-AP1(bZIP)","JunB(bZIP)","JunD(bZIP)","KLF1(Zf)","KLF10(Zf)","KLF14(Zf)","KLF3(Zf)","Klf4(Zf)","KLF5(Zf)","KLF6(Zf)","Klf9(Zf)","LEF1(HMG)","Lhx1(Homeobox)","Lhx2(Homeobox)","Lhx3(Homeobox)","LHX9(Homeobox)","LRF(Zf)","LXRE(NR),DR4","MafA(bZIP)","MafB(bZIP)","MafF(bZIP)","MafK(bZIP)","Max(bHLH)","Maz(Zf)","Mef2a(MADS)","Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Meis1(Homeobox)","MITF(bHLH)","MNT(bHLH)","Mouse_Recombination_Hotspot(Zf)","MYB(HTH)","Myf5(bHLH)","MYNN(Zf)","MyoD(bHLH)","MyoG(bHLH)","n-Myc(bHLH)","Nanog(Homeobox)","NeuroD1(bHLH)","NeuroG2(bHLH)","NF-E2(bZIP)","NF1-halfsite(CTF)","NF1:FOXA1(CTF,Forkhead)","NF1(CTF)","NFAT:AP1(RHD,bZIP)","NFAT(RHD)","NFE2L2(bZIP)","NFIL3(bZIP)","NFkB-p50,p52(RHD)","NFkB-p65-Rel(RHD)","NFkB-p65(RHD)","NFY(CCAAT)","Nkx2.1(Homeobox)","Nkx2.2(Homeobox)","Nkx2.5(Homeobox)","Nkx3.1(Homeobox)","Nkx6.1(Homeobox)","NPAS(bHLH)","NPAS2(bHLH)","Npas4(bHLH)","Nr5a2(NR)","Nr5a2(NR)","NRF(NRF)","NRF1(NRF)","Nrf2(bZIP)","Nur77(NR)","OCT:OCT-short(POU,Homeobox)","OCT:OCT(POU,Homeobox,IR1)","OCT:OCT(POU,Homeobox)","OCT:OCT(POU,Homeobox)","Oct11(POU,Homeobox)","Oct2(POU,Homeobox)","OCT4-SOX2-TCF-NANOG(POU,Homeobox,HMG)","Oct4:Sox17(POU,Homeobox,HMG)","Oct4(POU,Homeobox)","Oct6(POU,Homeobox)","Olig2(bHLH)","Otx2(Homeobox)","p53(p53)","p53(p53)","p53(p53)","p63(p53)","p73(p53)","PAX3:FKHR-fusion(Paired,Homeobox)","PAX5(Paired,Homeobox),condensed","PAX5(Paired,Homeobox)","PAX6(Paired,Homeobox)","Pax7(Paired,Homeobox),long","Pax7(Paired,Homeobox),longest","Pax7(Paired,Homeobox)","Pax8(Paired,Homeobox)","PBX1(Homeobox)","PBX2(Homeobox)","Pbx3(Homeobox)","Pdx1(Homeobox)","PGR(NR)","Phox2a(Homeobox)","Phox2b(Homeobox)","Pit1(Homeobox)","Pit1+1bp(Homeobox)","Pitx1:Ebox(Homeobox,bHLH)","Pitx1(Homeobox)","Pknox1(Homeobox)","PPARa(NR),DR1","PPARE(NR),DR1","PR(NR)","PRDM1(Zf)","PRDM10(Zf)","PRDM14(Zf)","PRDM15(Zf)","PRDM9(Zf)","Prop1(Homeobox)","PSE(SNAPc)","Ptf1a(bHLH)","PU.1-IRF(ETS:IRF)","PU.1:IRF8(ETS:IRF)","PU.1(ETS)","RAR:RXR(NR),DR0","RAR:RXR(NR),DR5","RARa(NR)","RARg(NR)","RBPJ:Ebox(?,bHLH)","Rbpj1(?)","REST-NRSF(Zf)","Reverb(NR),DR2","RFX(HTH)","Rfx1(HTH)","Rfx2(HTH)","Rfx5(HTH)","Rfx6(HTH)","Ronin(THAP)","RORa(NR)","RORg(NR)","RORgt(NR)","RUNX-AML(Runt)","RUNX(Runt)","RUNX1(Runt)","RUNX2(Runt)","RXR(NR),DR1","SCL(bHLH)","SCRT1(Zf)","SF1(NR)","Six1(Homeobox)","Six2(Homeobox)","Six4(Homeobox)","Slug(Zf)","Smad2(MAD)","Smad3(MAD)","Smad4(MAD)","Snail1(Zf)","Sox10(HMG)","Sox15(HMG)","Sox17(HMG)","Sox2(HMG)","Sox21(HMG)","Sox3(HMG)","Sox4(HMG)","Sox6(HMG)","Sox7(HMG)","Sox9(HMG)","Sp1(Zf)","Sp2(Zf)","Sp5(Zf)","SPDEF(ETS)","SpiB(ETS)","Srebp1a(bHLH)","Srebp2(bHLH)","STAT1(Stat)","Stat3(Stat)","Stat3+il21(Stat)","STAT4(Stat)","STAT5(Stat)","STAT6(Stat)","STAT6(Stat)","T1ISRE(IRF)","TATA-Box(TBP)","Tbet(T-box)","Tbox:Smad(T-box,MAD)","Tbr1(T-box)","Tbx20(T-box)","Tbx21(T-box)","Tbx5(T-box)","Tbx6(T-box)","Tcf12(bHLH)","Tcf21(bHLH)","Tcf3(HMG)","TCF4(bHLH)","Tcf7(HMG)","Tcfcp2l1(CP2)","TCFL2(HMG)","TEAD(TEA)","TEAD1(TEAD)","TEAD2(TEA)","TEAD3(TEA)","TEAD4(TEA)","TFE3(bHLH)","Tgif1(Homeobox)","Tgif2(Homeobox)","THRa(NR)","THRb(NR)","THRb(NR)","Tlx?(NR)","TR4(NR),DR1","TRPS1(Zf)","Twist(bHLH)","Twist2(bHLH)","Unknown-ESC-element(?)","Unknown(Homeobox)","USF1(bHLH)","Usf2(bHLH)","VDR(NR),DR3","WT1(Zf)","X-box(HTH)","YY1(Zf)","Zac1(Zf)","ZBTB12(Zf)","ZBTB18(Zf)","ZBTB33(Zf)","ZEB1(Zf)","ZEB2(Zf)","Zfp281(Zf)","ZFP3(Zf)","Zfp57(Zf)","Zfp809(Zf)","ZFX(Zf)","Zic(Zf)","Zic2(Zf)","Zic3(Zf)","ZKSCAN1(Zf)","ZNF136(Zf)","ZNF143|STAF(Zf)","ZNF16(Zf)","ZNF165(Zf)","ZNF189(Zf)","Znf263(Zf)","ZNF264(Zf)","ZNF317(Zf)","ZNF322(Zf)","ZNF341(Zf)","ZNF382(Zf)","ZNF41(Zf)","ZNF415(Zf)","ZNF416(Zf)","ZNF467(Zf)","ZNF519(Zf)","ZNF528(Zf)","ZNF652","ZNF669(Zf)","ZNF675(Zf)","ZNF692(Zf)","ZNF7(Zf)","ZNF711(Zf)","ZNF768(Zf)","ZSCAN22(Zf)"],[12.81,7.26,0.03,0.08,4.85,0.31,0.92,1.11,1.35,0.01,5.73,3.1,6.79,4.21,4.35,6.89,21.3,16.38,0.02,0.06,7.25,0.2,0,8.66,0.37,0.17,0.22,15.64,0,0.89,0.66,0.05,0.57,2.08,0.2,0,2.44,0,0.01,7.52,0.09,6.54,3.86,0.4,0.94,0.06,0.01,2.73,3.85,3.37,0.25,0.21,0.05,0,0.67,1.68,0.34,0.6,0.14,0.04,0,1.05,1.19,1.07,0,0,1.09,1.02,0.95,3.43,0.51,1.7,0.01,0.23,0.43,0.18,0.53,0.38,0.32,6.8,41.87,3.91,5.66,5.63,6.71,4.42,4.1,7.76,9.24,0.06,8.59,0,0.13,0,1.9,3.15,4.71,4.74,7.99,7.4,8.82,4.29,4.38,7.15,8.63,9.67,10.02,3.59,0.09,1.64,1.25,4.76,4.54,1.42,3.79,0.52,6.49,9.5,3.05,1.14,8.67,9.45,8.82,8.82,8.28,0.23,6.39,257.49,0,0.25,387.65,395.9,1.8,0.37,381.27,393.69,393.6,0,5.72,0.6,1.23,0.06,0.12,1.53,2.58,1.05,1.71,0.09,0.98,0.04,0.86,0.04,0.04,0.28,0.1,4.91,0.21,0.28,0.03,0,0,0.07,1.56,0.02,0.01,0.27,1.84,0.03,0.84,0,0.45,0.14,0.25,0.01,0.52,0.56,1.14,1.42,1.72,0.85,1.12,1.56,0.04,1.12,11.01,9.09,3.46,57.85,13.45,26.75,51.12,51.33,53.37,48.81,19.88,0.24,0.47,0.22,2.45,1.54,0,0.15,4.24,2.21,0.23,5.7,0.14,8.69,0.98,0.13,0.8,0.9,5.52,11.31,0.03,0.05,15.11,3.35,0.3,1.46,3.54,0.02,3.14,3.5,8.47,18.21,7.84,1.16,10.19,1.67,0.12,12.7,7.89,0.08,0.83,0.07,18.61,0.04,0.35,0.74,0.49,1.2,0.17,0.17,0.76,0,0,1.14,1.78,22.04,0.63,0,0.1,0.45,0,0.32,0.85,0.14,0.17,0.11,0.85,3.62,0,0.39,0.69,0.69,0.18,0.36,0.18,0.82,0.64,0.21,0,0,0.01,0.31,2.01,0.46,2.05,0.44,0.5,0.1,0.2,0,0.03,0.56,0,1.89,0.08,0.59,0.06,0.05,0.02,0.43,0.21,0.11,0.05,0.01,0.23,1.48,0.66,7.88,0,0.86,0,0,0.01,0,0.01,2.07,4.55,1.89,3.3,0.87,1.57,3.25,0.02,0.25,0.37,7.44,5.43,4.73,4.11,0.21,3.56,0.01,0,0.18,0.47,0.08,0,1.83,3.6,1.7,0,0.01,0.04,0.01,0.02,0.01,0,0.01,0,0.11,0.11,13.97,53.77,39.95,1.11,5.91,4.84,2.89,0.91,0.46,1.42,2.42,2.17,0.02,0.05,0.18,0.02,3.4,1.65,1.92,2.68,1.91,1.17,1.25,1.76,7.23,0.02,6.3,0.26,0.54,0.01,0.04,0.01,0.03,0.04,0.02,2.94,6.11,8.71,0.86,0.22,0,4.2,0.03,320.68,0.3,4.01,2.37,0.13,1.08,3.71,0.58,0.81,3.76,1.23,0.91,0.01,3.23,4.28,0,0,1.04,0.37,0.63,0.83,2.17,1.06,1.64,1,0.12,0.07,0.15,0.27,0.21,0.01,1.49,0.23,0.39,0.06,0.15,0.42,0.02,0.08,0,1.88,0.29,0,0.41,0.7,0.16,1.75,0.21,2.69,0.52,0.47],[1.97,4.11,0.99,0.98,1.23,7.88,0.04,1.63,81.65,1.54,4.73,3.76,4.68,1.84,3.64,198.69,0.74,1.99,0,0.42,4.27,3.41,0.16,169.85,0.07,0.06,0,1.79,1.05,0.83,0.19,0.93,0.06,4.52,0,0,1.75,0.01,0.01,1.33,0.66,0.16,1.41,0.9,0.07,0,0,1.89,5.79,0,0.11,1.29,0,0,2,3.93,2.31,2.74,0.03,0.03,0.25,0.02,0,0.19,4.23,2.57,0.19,0.24,0.89,1.14,0.33,0.04,0,0.1,0.24,0.05,0.26,0.29,0.22,1.21,0,0.02,0.93,0.23,0.13,0.07,0.14,4.73,1.55,2.07,5.65,0,0.24,0.08,4.44,0.01,0,1.38,1.21,1.13,2.5,0.85,3.91,1.3,1.87,5.49,3.99,1.26,0.02,1.11,0.83,1.73,1.17,0.31,1.19,0.84,2.49,1.68,1.32,0.39,8.99,2.71,1.39,5.62,4.3,0.01,0.68,0.16,0.85,0.34,0.03,0.06,0.01,0.36,0.12,0.07,0.05,0.49,0.05,0,0.36,0.28,0.51,1.61,0.15,1.16,0.05,0,6.58,40.01,45.04,0.42,0.09,0.5,0.01,6.49,0.16,0.06,0,0,0.02,0.06,2.4,0.14,0.23,0.01,0.12,0.12,0.81,0.09,0.5,0.14,0.31,0.23,0.27,0.13,0.84,0.44,0.18,0.07,0.3,0.87,0.96,0.15,3.03,4.22,3.59,0,0,0,0,0,0,0,0,0.07,3.79,4.39,2.34,4.73,0,0.04,3.61,0.08,0.14,0.32,0,0.11,22.91,23.98,21.47,18.41,6.07,0.72,0,0,1.12,0.38,0.58,0.06,2.15,0,8.74,259.27,193.39,0.79,98.8,0.73,73.18,0.7,2.48,0.11,7.77,0.36,0.24,0.14,0,0,0.03,0,0,2.92,0,0.27,0.83,0,0,0,0.01,0.96,1.22,0.75,0.47,1.47,0,0.33,0.07,0,0.21,0.24,0.84,192.57,0,0.24,0.04,0.04,0.01,0.06,0.09,0.04,1.2,0.73,0.06,0.01,0.02,3.54,1.77,0.05,1.33,1.97,0.85,0.74,0.26,0.01,0.42,2.22,0.01,0.58,0.04,0.13,0.49,0,0.92,0.07,0.65,0.87,0.32,0.06,52.2,1.37,0.49,0.06,0,0.1,0,0,6.75,1.03,1.85,0.01,4.2,4.29,3.85,0.7,1.18,0.14,16.11,17.91,14.13,0.2,0.07,0.03,0.01,0.46,55.98,0.22,0.01,0.28,0.17,0.02,7.21,0.45,1.04,0.8,3.48,0.21,0.3,0.54,0.26,0,0.11,0.24,0.36,0.32,0.23,0,0,0,1.53,0.27,0.1,0.31,0.02,2.12,2.87,3.41,0.02,0.41,0.37,0.06,0.59,1.05,0.89,2.39,3.58,1.4,4.78,4.65,0.67,13.02,0.03,154.91,0.03,0.54,0.04,0.14,0.51,0.16,0.41,0.24,0.13,14.33,33.29,0.01,0.31,0,44.17,0,0.11,4.67,134.67,20.3,0.07,0.27,0.4,0.11,0.97,6.43,0,0.17,0.31,0,0.05,2.33,0.6,0.61,0,0.41,0.17,0.04,21.12,18.08,21.48,0.35,0.16,0,0.86,1.68,1.07,1.25,0.88,0.9,0.99,0.09,1.8,0.17,0.11,1.48,0.83,0.17,0.07,0.51,0.49,2.64,0,2.37,0,1.03,2.55],[2.42,44.91,0.41,0.74,92.94,2.26,2.4,0.71,41.43,14.31,22.22,28.5,45.12,13.05,32.48,76.13,5.98,21.6,0.04,1.9,41.07,10.12,6.39,99.02,0.05,0,0.41,5.01,0,0.39,0.02,0.3,0.46,24.79,0.04,0.01,3.97,1.02,4.69,11.13,1.81,33.87,7.3,0.5,1.29,12.86,15.79,3.52,30.25,0,0.57,0.03,0,0,1.26,2.48,4.55,4.16,0.03,0.03,0.22,0.13,0.24,2.95,0,16.48,0.33,0.01,0.52,1.04,0.16,0.03,21.58,10.24,8.39,24.49,0.06,1.42,0.09,9.85,6.64,6.76,10.89,11.07,4.12,10.17,12.49,3.41,0.11,0.32,40.81,1.86,0.82,1.13,3.43,7.12,6.37,29.4,40.98,29.53,40.55,22.62,35.84,24.13,33.85,45.04,39.39,8.2,0.03,7.58,5.84,9.07,8.63,7.61,12.84,3.92,16.99,12.82,9.29,6.02,22.42,17.15,16.26,48.29,44.99,0.94,24.26,2.31,0.89,1.68,5.02,6.55,0.51,0,5.34,9.04,5.22,0.52,0.05,0,0,0.05,0,1.67,8.9,12.59,0,0,3.15,18.77,0.01,0.24,0.03,1,0.02,18.39,0.24,0.15,0.2,0.01,0,0.2,2.32,1.75,3.69,0.31,0.16,3.56,0.46,4.2,1.8,2.69,0.25,4.98,6.37,8.2,3.32,1.11,0.95,1.01,0.09,1.55,2.87,0.17,32.65,43.09,8.47,3.72,0.85,3.39,4.37,4.27,2.91,2.69,0.52,0.04,1.47,2.94,2.73,3.4,0,1.93,4.74,0.67,2.38,1.19,0.08,4.36,12.71,15.67,12.06,15.98,0.37,8.02,0.58,0.44,2.6,65.8,0.81,49.21,61.41,0.03,0.02,49.38,74.96,7.08,20.78,16.95,30.59,5.42,11.33,0.79,24.94,0,12.29,10.44,0,0,0.23,0.02,0.04,2.74,0.52,0,0.59,0.09,0.21,0.01,0.05,5.25,2.98,0.93,0,0.17,0.05,0.15,0.23,0.03,0.08,0,0,52.85,0,0.34,0.2,0.2,0.96,0.56,0.17,0.72,0.44,0.01,1.17,0.03,0.28,0.7,0.41,1.97,0,3.5,9.73,0.5,0.22,0,0.04,27.83,0.3,0,1.62,1.18,2.04,0.69,0.76,0.02,0.29,0,0.56,0,23.22,1.67,1.71,9.92,0,5.72,4.88,0,3.81,16.88,0.01,2.49,0.65,0.28,0.88,0,5.1,0.01,4.55,7.2,12.88,0.68,0.56,0.16,0.27,3.24,34.67,0,0.3,0.53,1.05,1.15,0,0.8,4.3,0.46,0,0.34,0.37,0.38,0.53,0,0.27,0.38,0.9,0.45,0.31,0.17,6.78,6.32,19.15,1.86,1.75,1.39,6.73,9.1,12.49,11.6,6.86,4.15,4.35,1.37,0.12,0.08,0.08,0.8,1.96,0.04,0,0,71.09,109.49,0,83.99,0,2.18,0.2,32.77,35.52,19.37,34.78,29.51,4.62,3.51,2.65,5.38,5.99,2.06,9.14,1.02,3.98,2.99,80.21,3.87,1.33,2.39,5.49,0.34,3.67,1.24,0,0.18,0.04,62.41,0.01,0,0,0.39,0.66,0.08,3.29,3.49,0.42,0.47,6.08,0.17,0.17,1.86,0.87,0.75,2.79,1.67,0.02,0.87,1.62,4.4,0.16,0.15,0.44,1.56,2.15,2.17,0,0.19,0.08,0.04,0.73,2.57,4,1.14,0.72],[1.44,28.34,6.29,6.29,1.24,2.64,0.1,0.64,0.41,1.76,3.97,4.79,35.49,0.73,5.81,0.01,1.56,6.25,5.29,1.11,27.8,2.33,1.67,0,4.15,5.59,0.46,1.5,0,0.67,2.72,3.38,0.3,6.07,1.71,0,0.6,20.24,27.3,1.3,0.29,8.67,0.08,0.32,4.7,9.88,12.16,1.13,7.32,0,0.02,0,0.03,0.07,0.97,0.76,2.42,2.16,0.24,0.23,0,1.13,0.45,3.31,8.39,2.64,0.04,0.28,0.61,0.57,0.65,0.3,9.91,0.55,4.7,4.74,0.16,0.91,0.09,10.62,18.22,2.83,14.24,4.83,10.4,0.17,0.12,3.81,4.68,2.34,10.02,4.15,22.72,17.38,0.13,0.24,0.92,2.61,3.55,4.11,11.01,2.47,8.41,3.07,4.3,31.3,30.02,34.42,0.11,23.4,25.61,29.37,26.86,8.48,13.35,1.71,19.02,14.72,13.7,22.75,9.82,22.9,16.53,35.13,32.85,4.28,5.09,0.42,2.59,5.2,10.49,10.97,1.77,0.17,6.36,9.39,8.73,0.25,0.33,0,0.1,0,0,2.12,2.43,2.29,26.1,0,1.8,1.76,6.88,0.02,0.64,0.28,0.03,5.32,38.01,53.68,4.15,0.27,0.3,0.64,2.73,25.72,18.11,1.15,18.84,24.13,2.57,5.87,10.76,23.56,8.27,28.61,5.88,3.75,2.46,0.4,0.38,0.16,1.26,0.14,2.35,0.37,25.18,34.87,3.62,25.05,2.99,14.57,15.08,18.84,24.38,24.63,4.21,0.02,1.82,1.62,3.76,1.86,0,1.74,1.25,0.04,0.4,0.29,0.04,8.3,1.54,1.06,1.79,4.58,0.01,5.33,0.01,0.05,1.21,2.94,1.17,0.38,3.58,0.14,4.07,0.05,0,1.43,25.75,7.92,36.12,1.27,3.48,1.13,7.85,0.07,3.34,3.07,0.09,4.19,6.67,4.33,2.97,4.33,0.07,1.82,1.06,8.48,8.2,0,0.01,1.89,1.34,0.82,2.24,1.27,0.01,2.48,1.18,0.56,0.77,2.15,0.77,0.02,0,0.42,0.82,0.82,3.91,1.7,0.94,2.23,2.99,1.54,0.13,0.35,0.29,1.48,0.39,4.45,0.5,3.4,3.21,0.49,0.01,0.35,2.42,0.57,0.36,1.08,5.32,4.31,0.07,0.98,0.27,0.2,0.21,0,0.36,0.03,1.73,0.02,0.1,3.67,0,1.18,2.92,0,0.15,3.77,0,1.65,1.4,1.58,1.8,1,4.48,0.03,0.68,1.3,1.11,0.37,0.24,0.03,0.05,4.87,0.35,1.72,2,1.09,4.49,0.98,0.81,1.84,3.01,1.02,3.19,0.08,0.42,0.18,0.21,0.01,0.02,0.92,0,0.05,0.23,2.63,20.07,12.82,4.13,0.47,0.31,0.01,5.34,7.17,4.35,3.69,5.99,1.69,3.96,0.29,1.47,3.47,0.28,5.03,0.11,4.49,6.46,5.35,1.19,0.3,0,0,0.15,8.45,0,89.25,97.2,69.92,89.9,93.63,4.85,0.03,0.75,2.75,3.61,2.7,13.1,0.99,7.99,0.15,0,1.85,14.77,7.66,7.77,0.45,5.35,0.73,0,0.05,0.01,0.15,0.02,15,14.17,2.89,0.43,0.44,3.75,0.86,0.71,1.05,0.59,0.11,0.46,0.23,0.41,0.56,0.59,1.16,0.34,0.02,4.22,0.11,0.62,0.46,0.04,5.29,2.09,0.63,0.3,0.98,0.18,1.04,1.51,0.12,0.24,0.12,0.75],[3.19,5.71,2.73,2.74,2.7,3.25,6.71,0.22,0.16,0.02,0.08,0.06,12.23,0.09,0.1,0.65,2.58,3.02,0.01,1.27,11.7,10.53,0.39,0.48,5.82,6.57,0.87,2.49,0.09,0.24,3,0.44,1.16,0.11,3.2,0,0.14,0.1,0.51,0.09,0.07,1.73,0.02,0.01,4.68,27.24,53.9,0.84,0.11,0.29,0.03,0.06,0.55,0.02,7.31,8.56,13.27,6.42,2.58,1.78,0,0.93,0.33,5,0,0.07,0.01,0.03,0.37,0.33,1.12,0.18,29.88,2.82,1.53,3.69,0.6,0.01,0.01,0.07,0.21,0.06,0.2,0.09,0.3,0.05,0.08,13.62,1.56,2.55,0.14,57.11,26.88,17.77,0.51,0,0.06,0.4,0.02,0.02,0.17,0.02,0.19,0.01,0.06,13.99,2.96,4.51,0,1.23,1.56,0.95,1.87,0.39,3.14,0.08,3.77,2.08,1.16,1.83,1.58,4.03,2.35,14.64,5.91,7.63,0.25,0.02,0.03,0.31,0,0,0.99,0.02,0,0,0,0.18,0.86,0,0.05,0,0.08,6,12.45,13.15,1.05,0.64,0.85,0.03,2.44,1.61,0.62,0.84,0.02,2.08,161.12,206.72,205.38,0.17,1.51,1.39,0.43,0.68,1.13,1.01,2.07,0.48,3.42,1.97,0.89,1.78,0.14,4.21,2.36,1.92,1.56,0,0.06,0.03,0.41,0.01,2.74,0.14,2.08,10.42,0.05,0.2,0.14,0.51,0.31,0.36,0.01,0.07,0.79,4.29,14.97,11.72,14.43,17.56,0,2.24,5.13,3.56,3.94,2.53,0.27,0.71,1.03,0.19,0.6,2.54,0.01,8.31,0.14,0.27,2.25,0.54,1.05,1.16,1.12,0.36,6.67,0.22,0.28,4.74,25.64,0.99,28.7,0.82,0.4,1.87,3.23,0.04,0.94,0.69,0.02,0.05,0.46,0.07,0.3,17.38,0.4,0.92,2.02,0.32,1.12,0.01,0.16,2.13,21.47,0.82,1.13,0.32,0.01,1.09,0.71,1.11,0.05,0.54,0.39,0.46,0.29,0.04,0.3,0.3,0.75,0.61,1.32,0.15,18.42,14.46,2.61,0.07,0.81,6.06,1.46,8.9,1.82,9.92,7.74,0.97,1.89,0.98,5.68,0.13,0.05,1.9,119.39,71.66,3.47,0.16,0.04,0.01,0.63,0.15,1.28,0.02,0.01,0,0.08,0.01,0,0.94,49.16,0,1.28,0.46,0,1.19,0.07,0.56,0.18,0,0.72,0,1.79,3.6,4.45,1.47,0.28,0.08,0.35,69.36,3.11,0.13,0.21,0.7,0.34,0.58,0,0.5,1.6,0.47,0,2.55,2.07,0.47,0.79,0.03,2.02,2.86,0.95,0.25,2.36,0.91,0.44,2.46,0.06,0.02,0.25,0.48,1.93,1.98,3.05,3.66,3.4,1.81,2.63,0.01,0.01,0.08,0.31,4.2,0.16,0.1,0,0.43,1.62,0.95,2.31,0.15,0.6,1.59,0.97,8.65,8.11,8.42,11.76,9.92,8.9,0.13,0.68,5.47,3.04,21.44,6.99,30.44,0,0.12,0.05,3.42,1.08,11.3,10.56,1.98,1.85,0.38,0,0.64,0.02,0.27,0.16,0,0.2,0.39,0,0.07,0.58,0.53,1.75,1.9,1.99,0.07,0.01,0.03,0,0.06,0.02,3.26,0.22,0.31,1.19,0,0.02,0.03,1.51,2.95,0.09,0.18,1.14,10.34,0.09,0.19,0.03,0.6,0.3,0.34,0.61],[3.75,5.66,2.01,3.44,8.57,0.21,0.03,0.27,6.32,4.19,8.66,7.01,8.79,2.99,9.25,5.91,2.17,2.79,0.01,0.15,5.62,1.62,0,5.41,4.05,3.51,1.23,3.48,0.48,0.9,16.16,0.17,3.15,3.84,3.29,0.13,2.64,0.07,0.86,8.43,0.13,4.69,1.74,2.3,4.96,0.03,0.11,4.51,7.49,0,0.02,1.42,0.01,0.02,0.19,0.26,0.76,0.53,0.12,0.18,0.77,0.27,0.52,6.04,0.02,3.63,0.05,0.28,0.21,0.1,0.22,0.2,0.11,7.87,6.05,12.94,0.12,2.87,0.7,149.31,4.42,94.55,166.38,215.54,208.64,52.25,43.73,0.4,3.82,0.03,164.34,0,0.89,0.12,5.94,34.97,87.73,143.71,201.56,167.72,182.06,124.21,126.47,61.71,171.03,8,4.68,1.39,0.04,0.21,0.15,1.46,0.24,0.17,2.1,0.05,2.09,3.17,0.72,0.05,13.2,2.56,2.31,8.82,5.65,0.18,173.98,0.24,1.03,0.27,1.8,1.31,0.33,0.06,0.44,0.64,0.95,0.05,0.16,0.13,0.19,0.46,1.83,2.51,0.24,0.19,0.18,0,2.75,2.98,0.24,0.71,0.1,1.25,0,3.22,0.01,0.28,0.09,0.02,0.01,0.14,3.51,0.65,0.95,0.32,0.13,0.69,0.1,0.38,0.79,0.68,0,0.7,1.88,0.83,4.92,50.16,42.1,72.48,21.99,145.58,0.04,16.24,3.34,6.38,3.71,15.61,1.18,7.02,11.18,6.35,11.09,10.29,2.49,0.15,1.06,0.37,0.53,0.81,0,2.14,0.89,0.01,1.48,0.06,0.4,7.06,14.65,19.7,24.31,19.06,2.25,18.44,0.17,1.81,1.84,5.29,0.58,2.75,5.22,0.83,0.61,4.67,5.76,1.28,0.66,0.14,0.83,1.46,6,0.47,8.65,8.38,17.41,25.37,0.86,0,0.12,0.01,0.95,0.06,1.34,1.17,0.59,0,0,0.47,0.26,2.08,2.19,0.36,1.39,0.25,0.02,27.42,23.95,1.3,0.3,16.46,12.33,1.42,0,1.07,0.36,0.36,0.21,0.05,0.7,0.29,5.59,2.16,0.61,0.74,0.02,3,0.8,0.08,2.13,0.25,0.99,0.03,0.77,0.3,0.46,0.07,0,2.09,0.09,0.13,0.49,11.13,0,0.09,0,0.14,0.34,0.31,3.14,87.55,147.83,291.76,0,0.47,0,0,0.81,0,0.03,1.17,2.84,0.96,1.66,0.25,0.27,0.38,0.24,0.53,0.17,37.96,28.78,30.2,27.09,0.75,5.87,0.14,0.01,1.41,1.46,0.61,0.98,0.58,0.52,1.85,0.77,1.65,0.19,0.21,0.15,0.16,1.34,0.41,0.84,0.48,0.76,5.22,12.13,13.5,40.46,291.76,1.04,0.83,0.25,0.06,0.11,0.06,0.13,0.16,0.24,6.51,2.43,7.56,0.44,1.77,0.2,3.15,3.85,2,6.1,6.03,0.02,4.25,0.03,1.51,0.23,0,0,0,0,0,8.26,6.14,8,0.34,1.32,0.02,1.03,0.21,1.24,0.97,4.4,3.09,0.13,7.6,9.29,0.71,3.47,2.23,0.06,1.25,0,1.15,0.39,0,0,1.73,0,0.34,0.41,0.7,1.11,1.77,3.26,0.05,0,0.02,0.15,1.14,0.71,0.01,0.03,2.68,0.55,0.28,1.09,0.09,0.08,0,3.32,0.1,1.25,0.29,0.18,0.94,2.17,0.4,2.16,0.47,0.15],[3.77,3.6,1,2.34,12.24,8.2,1.87,1.12,2.32,1.1,5.68,7.81,2.9,0.39,8.52,2.61,4.28,4.27,4.96,0.98,3.58,1.49,0.2,1.81,1.44,12.01,10.31,2.26,0,0.78,0.12,0.24,1.36,5.2,4.12,1.52,8.93,0.03,1.37,0.42,0.03,1.65,0.34,1.47,11.56,15.14,26.29,0.96,10.01,1.22,0.54,0.01,0.01,0,1.36,1.79,2.87,2.11,1.01,0.71,0,0.05,0.39,2.83,0,0.6,0.02,0.01,0.04,0.01,0.03,0,25.19,1.53,2.9,3.64,0.02,0.04,0.01,0,0,0.01,0,0,0,0,0,2.25,8.66,3.37,0.01,7.26,42.5,41.37,0.03,0.03,0.03,0.04,0,0,0.01,0,0.01,0,0,3.09,4.12,0.91,0.04,0.4,0.25,2.07,0.46,0.1,3.5,0.76,4.02,2.8,5.67,0.02,2.43,5.86,2.11,3.61,4.11,4.17,0,1.51,0.39,0.33,41.98,41.85,0.72,0.01,45.95,52.55,48.25,0.03,0,0,0.04,0,0.05,0.52,5.71,5.76,0.29,0.01,8.3,0.67,24.42,1.65,2.41,3.34,0.08,0.95,0.3,0.15,0.88,0,0,0.01,0.84,1.84,1.17,0.01,1.51,2.53,0.22,2.41,0.44,3.18,0.44,2.28,0.74,0.66,0.56,0.07,0.02,0.07,0.32,0.01,0.72,0.16,3.18,4.51,0.98,0,0,0,0,0,0,0,0,1.18,1.64,3.03,1.37,5.22,0,6.41,2.28,0.17,0.29,0.11,3.05,0,205.68,245.94,235.26,215.76,16.48,32.88,3.74,0.02,2.68,5.41,0.43,3.38,8.51,2.48,0.77,1.46,1.52,2.64,64.93,2.23,58.02,2.01,2.66,0.56,1.65,0.12,2.6,0.43,0.01,2.55,6.06,6.48,1.42,0.7,8.86,6.61,1.67,6.99,8.63,0.12,0,3.05,11.38,0.03,0.08,0.34,0.01,0.23,0.15,0.04,0.58,0.13,0.01,0.42,0,0.29,0.92,0.92,1.29,0.7,1.28,1.46,1.33,0.3,0.27,0,0.19,2.33,1.34,0.69,1.27,1.48,4.42,0.17,0.08,0.08,0.04,0.74,0,0.35,7.04,4.95,0.4,0.1,0.14,0.34,0.11,0.01,0.16,0.21,1.28,0.03,0.02,0.01,0.02,2.48,8.19,0,1.67,1.81,0.34,1.35,0.62,0.51,0.7,0.29,4.07,0.03,4.53,4.59,3.28,0.73,0.17,0.08,0.1,7.56,4.38,0.11,5.51,9.03,10.74,2.75,0,0.42,4.37,1.87,0,1.82,0.95,1.47,1.16,0.13,1.28,1.49,0.65,1.22,0.2,0.01,0,0,0,0.09,0.63,0.23,3.74,8.51,10.77,7.19,4.39,1.45,1.5,0.01,1.74,8.02,1.17,4.77,7.78,5.22,10.64,8.04,5.16,5.67,0.15,1.62,1.24,0.92,0.43,10.34,17.12,10.81,20.02,15.54,2.67,23.52,31.76,16.3,14.8,13.12,31.73,1.35,39.66,2.37,0.52,1.87,1.62,14.18,15.96,1.85,0.09,1.68,0,1.27,0.01,0.29,0.25,0,0,0.15,0,0.13,1.46,0.56,0.15,1.23,1.21,0.31,0.23,0.08,0.56,0.28,0.01,0.48,0.35,1.64,2.06,0.22,0.07,0.05,0.04,2.48,0.46,0.65,0.03,0.47,0.5,1.73,0.5,2.55,0.38,0.19,0.22],[1.84,1.7,3.56,3.67,0.14,1.67,0.39,0.14,0.01,0.05,5.06,2.23,1.84,11.89,3.22,0.13,2.02,1.91,0.04,0.08,0.87,4.64,0,0.06,0.4,0.87,0.05,1.19,0,0.08,0.07,0.91,0.39,1.33,0.8,0,0.45,0.03,0.06,24.32,1.03,59.41,8.7,0.33,1.09,26.29,43.99,0.35,1.51,0.55,0.22,0,33.58,52.72,1.06,1.18,1.44,0.96,0.22,0.25,0.22,0.15,1.07,1.38,0,0.05,0.05,0.13,0.59,0.4,0.71,0.47,35.56,0.45,1.48,2.84,1.05,0.41,0.13,0.58,0.01,0.2,0.22,0.12,0.19,0.48,0.49,2.56,2.49,0.5,0.4,43.01,19.36,16.67,1.72,0.05,0.11,0.34,0.03,0.7,0.11,0.42,0.14,0.17,0.36,3.41,0.87,39.27,0.17,36.55,33.61,41.92,28.69,8.49,14.33,2.98,25.38,18.87,11.48,34.31,11.3,21.47,14.56,1.87,1.71,17.53,0.36,0.73,0.31,0.95,8.99,11.31,1.27,0,9.68,10.99,10.2,0.43,0.1,0,0.16,0.01,0.4,1.06,1.11,2.42,0.3,0.03,1.4,0.06,5.07,0.69,0.05,0.84,0.06,22.65,38.23,57.63,120.69,59.28,46.34,1,2.94,0.45,0.51,0.93,1.16,0.46,1.67,0.72,0.96,0.58,0.1,0.59,2.15,1.43,0.8,1.25,1.71,1.47,0.98,1.1,0.73,1.81,0.55,1.57,0.72,0,0.06,0.21,0.04,0.01,0,0,0.05,4.38,1.02,1.53,2.24,2.68,0,5.07,0.4,1.13,0.46,1.57,0,0.18,1.3,0.94,1.55,1.16,0.1,5.19,0.02,1.18,1.21,0.19,0.82,0.47,0.1,0,0.67,0.01,0.04,2.93,19.7,13.53,15.41,1.93,1.62,0.83,23.78,0.06,0.53,0.07,2.67,0,0.02,0,0.03,2.41,0.01,0.73,0.09,1.43,1.88,0.04,0.04,2.39,16.74,0.37,0.37,0.73,0.01,0.15,0.32,0.06,0.05,0.01,0.06,0.14,0.05,0.55,0.96,0.96,0.01,1.55,1.26,0.49,0.15,0.48,0.21,0.22,2.05,0.04,0.18,2.75,0.29,4.65,2.04,0.02,0.17,0.15,1.05,0.54,0.19,0.36,80.95,57.5,0.41,0.59,0.06,1.7,0,0.02,0.02,0.05,0.01,1.07,1.4,0,0.59,0.62,36.32,0.01,0.5,3.13,0.15,2.32,0.11,0.47,0.2,0,0.94,0.03,12.75,11.22,7.63,0.08,0,0,0,55.63,0.3,0.11,0.31,0.42,0.73,0.06,0,0.06,0.53,0.14,0.03,0.45,0.3,0.07,0.29,0,0.73,0.64,0.15,0.04,0.51,0.09,0.48,1.8,0.06,0.07,2.67,1.54,0.38,0.93,1.39,1.29,0.53,0.01,0.08,0.44,0.01,0.61,0.16,0.68,0.62,0.91,2.09,2.67,0.31,0.49,3.74,0.01,2.97,1.17,2.63,0.62,0.19,0.5,0.43,1.43,1.07,1.18,4.32,5.07,2.91,20.96,3.54,26.56,11.22,0.43,0,0.53,0.01,2.9,5.11,4.08,0.96,0.35,0.01,0.18,0,0.54,0.04,0.28,0.09,3.19,0.62,0.04,0.59,0.06,0.69,0.68,0.29,0.1,0.16,0,0.82,0.88,0.18,1.43,1.1,0.06,0.91,0.13,0.06,0.23,0.46,1.54,0.42,0.71,0.25,3.74,0.09,1.27,0.12,0.43,0.25,0.81,1.04],[1.52,1.98,0.29,0.16,20.91,5.8,2.78,0.46,5.28,1.13,6.95,12.47,1.97,1.02,8.31,14.64,1.44,1.23,0.05,3.4,1.52,2.57,0.79,14.71,0.3,0,0,3.95,0.05,0.18,7.14,0.9,0.42,10.7,0.03,0,0.63,0.11,0.03,0.74,0.26,2.81,2.03,1.58,0.16,0,0,3.64,12.87,0,0.32,0.73,1.35,0.64,12.08,11.85,11.81,13.84,0.03,0.02,0.32,0.62,0.1,0.19,0,0.48,0.13,0.06,0.14,0.07,0.7,0.15,0,2.83,4.11,0.66,0.17,0.68,0,0,0,0,0.03,0,0,0,0,16.19,0.23,0.52,0,0,0.2,0,0.15,0.17,0,0.13,0,0,0,0,0.01,0,0,2.39,1.75,3.89,0.33,1.77,1.83,4.07,1.16,1.25,4.91,3.39,6.71,8.4,2.93,1.29,4.49,6.67,7.17,2,1.51,0.36,0,0.9,0.36,0.02,0.01,0.03,0.02,0,0,0,0,0.43,0.25,0,0.41,0.21,0.14,0.72,4.27,7.74,0.06,0,0.78,0.74,24.26,0.02,0,0.1,0.02,1.89,0.42,0.09,0.36,0.71,0.4,0.5,4.03,0.41,0.84,0.4,2.27,0.25,0.99,0.05,2.81,1.8,3.53,1.76,0.64,0.66,1.16,0.81,0.44,0.63,0.08,0.19,4.84,0.27,1.77,2.46,6.15,0,0,1.03,0,0,0,0,0,8.99,18.03,22.63,15.23,15.9,0.03,0.95,2.24,0.03,0.24,0.24,0,2.31,3.29,1.99,3.31,2.29,0.95,1.32,0,0,2.37,12.11,0.39,10.97,13.22,0,7.98,6.87,12.43,0.92,64.32,3.63,29.61,1.5,3.12,0.31,5.15,0.04,0.01,0.03,0.4,0,0.08,0.08,0.09,15.99,0,0,1.24,0.01,0,0.03,0.23,1.49,0.4,2.52,0.42,3.22,0,4.35,1.59,6.79,9.43,5.88,11.99,3.82,0,0.07,0.01,0.01,0.09,0.7,0.01,0.35,1.15,0.76,0.8,0.18,0.18,0.87,0.87,0.11,0.14,1.04,4.73,0.53,0.84,0.97,3.25,0.12,0.16,0.62,0.01,0.23,1.49,0.34,0.29,0.62,0.54,0.07,0.56,0.12,0.31,0.02,0.81,0.04,0,0.07,0,0,0.75,2.76,0.12,1.13,15.35,16.42,13.88,5.77,14.24,0.09,8.98,7.13,5.54,0.24,0.01,0,0.02,0.18,14.86,0,0,0.03,0.01,0.05,0,0.18,5.42,0.16,0,138.11,107.4,87.55,121.78,117.26,129.2,106.92,102.36,54.5,118.35,0,0.31,0.19,0,0.03,1.55,1.28,0.21,0.8,1.38,2.37,1.31,2.19,2.82,0.03,1.01,0,0.84,1.58,2.8,0,0,0,14.71,18.9,0.19,10.06,5.92,0.83,0.12,0.34,0.1,0.14,0.3,0.05,0.73,3.6,3.2,1.69,0.44,0,10.68,0,0,0.02,12,0.6,0.79,1.25,1.73,0,1.75,11.9,0,1.51,0,2.5,0.27,0,0,8.79,0,0.43,1.38,5.83,0.47,0.26,1.66,0.44,2.3,1.25,0.14,0.1,0.98,0.89,0.5,0.24,1.63,0.93,0.01,0.72,0.01,4.15,5.22,0.62,0.05,1.33,0.09,2.36,0.42,2.52,4.89,0.85,0.56],[3.44,40.53,0.91,1.84,1.05,1.26,0.02,0.31,3.49,2.36,6.58,9.36,47.42,1.02,10.71,1.2,4.66,18.39,0.03,0.54,41.56,0.49,1.58,1.21,0.38,0.37,0,1.92,0,0.73,0.04,0.36,0.81,8.58,0.88,0,0.41,103.49,141.36,0.68,1.37,1.8,1.08,0.65,1.29,4.12,14.91,2.38,10.48,0.14,0,0,0,0.07,0.93,0.61,1.48,2.34,1.3,0.46,0.22,1.18,0.26,2.17,3.53,4.88,1.28,0.44,0.04,0.85,0.87,0.25,6.87,1.51,1.08,1.61,0.24,0.27,0.45,42.84,73.7,20.3,45.34,29.86,41.84,7.99,6.3,0.83,5.3,0.02,32.45,12.9,7.1,5.61,1.83,0.92,10.87,6.67,24.89,22.26,29.88,22.17,20.56,14.54,20.71,46.51,48.34,18.33,0.09,19.3,16.34,17.37,14.28,6.51,6.35,0.81,12.9,6.99,8.3,15.55,11.95,13.58,9.26,49.64,44.82,3.1,23.45,0.41,0.66,0.32,20.66,21.25,0.3,0.02,14.15,19.47,24.56,0.24,0.3,0,0.02,0.01,0.06,0.37,0.43,2.03,1.67,0,0.13,4.33,2.53,0.02,0.01,0.44,0,3.51,7.21,7.05,57.15,0.66,0.86,0.09,3.19,102.06,86.69,0.25,51.99,110.4,0.69,4.95,38.43,85.56,16.14,94.63,0.58,0.66,3.41,0.58,0.58,1.15,0.21,0.54,0.59,0.14,38.69,48.21,3.22,78.65,16.95,23.83,62.45,82.78,108.75,81.69,19.14,3.41,0.75,1.06,0.78,0.78,0,1.53,1.38,0.39,0.15,1.25,0,3.02,0.76,0.38,0.15,0.68,2.54,1.55,0,0.17,3.74,3.76,1.48,1.48,4.44,0.03,4.72,1.09,0.6,4.71,12.73,2.8,6.58,2.13,0.19,1.77,5.39,0.02,1.99,1.09,1.05,0,0.07,0.07,0.03,1.03,0,0.79,1.76,1.31,1.2,0.01,0.4,3.88,8.27,0.01,0.39,0.27,0,0.27,0.22,0.01,2.53,0.07,0.01,0.25,0,0.27,0.36,0.36,0.32,0.42,0.26,1.19,1.02,0.05,0.1,0.05,0.15,1.08,1.76,2.31,2.37,0.53,1.27,0.37,0.08,0,0,1.06,0.67,1.86,26.01,23.77,1.04,0.27,0,0.01,0.47,0.11,0.06,0.26,2.67,0.78,0.93,4.76,0,1,10.91,0,0.23,1.77,0.02,1.97,4.13,4.96,4.74,5.55,3.04,0,2.8,2.27,0.99,0.12,0.11,0.16,0.04,25.2,1.24,1.82,1.54,0.07,0.92,0.04,0.84,0.44,0.26,0.55,2.23,6.91,4.25,5.39,5.27,2.8,3.91,4.44,5.76,1.13,1.54,3.32,46.99,19.29,21.51,0.47,1.06,0.2,2.46,5.29,4.26,5.87,2.34,0.18,0.8,0.01,11.78,3.69,0.36,5.06,0.02,5.49,12.33,9.91,3.21,1.52,1.17,0.54,1.97,0.15,0.17,1,1.46,0.19,0.68,1.61,1.68,4.19,4.54,0.62,1.31,2.3,0.91,13.75,12.56,0.28,0.14,0.87,69.18,1.74,1.27,1.6,1.65,5.22,0.16,0.1,0.15,0.15,0.26,5.4,5.25,0.03,0,0.64,1.2,1.28,1.08,2.07,1.19,0.04,0.05,0.07,0.17,0.33,0.22,4.25,0.3,0.11,1.71,0.82,0.59,0.07,1.27,1,1.37,0.34,1.15,0.9,0.59,0.07,0.96,0.11,0.67,0.95,0.36],[2.68,19.28,0.05,0.06,4.01,4.15,0.23,1.78,14.45,2.03,1.74,0.97,24.55,3.77,0.87,42.12,3.05,13.39,0,0.2,20.59,0.01,0.22,39.88,0.81,0.15,0.06,2.33,0,0.5,2.51,3.86,0.25,1.6,0.63,0.08,1.99,0,0.01,1.74,0.34,2.75,0.93,1.47,1.84,0,0,0.06,1.55,0,0.35,0,0,0.04,0.09,0.16,1.73,1.1,0.13,0.16,0,0.37,0.04,0.43,0,2.02,0.04,0.18,1.33,0.63,0.49,0.03,0,0.52,0.59,1.22,0.68,38.7,12.2,0,0.02,0,0,0,0,0,0.01,0.02,4.88,0.01,0,0,0,0,0.44,0.18,0,0.25,0.05,0,0,0,0.03,0,0,29.19,21.66,0.7,0,0.01,0.01,0.05,0.08,0.12,0.04,0.54,0.06,0.66,0.08,0,1.94,0.63,0.37,25.66,24.83,0,0,0.11,0.33,0.24,0.01,0.03,0.22,0.08,0.24,0.04,0.07,1.09,0.03,0,0.24,0.61,1.2,3.15,1.41,3.6,0.06,0,1.43,5.76,6.18,0.75,0.07,0.56,0,18.24,0.02,0.01,0,0.03,0,0.04,0.81,0.04,0.06,0.01,0,0.01,0,0.87,0.01,0.37,0,0,0.69,0.54,0.45,0.58,1.08,0.42,0.3,0.28,0,0.58,24.02,19.43,0.31,0,1.29,9.61,0,0.01,0,0.03,0.04,0.11,0.29,0.54,0,0.4,0.43,0.04,0.79,0.06,0.08,0.78,0.06,2.09,51.38,84.56,66.1,48.25,1.79,0.41,0.05,0.02,2.98,5.27,0.59,4.45,4.78,0.04,0,36.97,43.52,5.73,21.72,1.19,4.86,0.41,1.04,0.74,19.66,0.59,0.46,1.12,0,0,0.92,0.06,0,0.02,0.08,0.1,1.67,0,0,0,0,2.93,0,1.42,3.21,0.18,0,2.02,1,0.06,0.72,2.46,4.01,41.27,0,0.03,0.02,0.02,0.56,0,0.29,0.36,0.17,0.03,0.08,0.09,0.27,0.23,0.2,0.31,0.76,0.51,3.96,0.06,0.02,0.29,1.36,1.07,0.35,0.19,0,0,0.04,0.02,0.05,0.01,0.09,1.78,0.14,0,8.75,0.75,0.22,0.01,0,0,0,0,4.16,6.27,2.49,0.21,37.21,25.48,40.79,7.2,6.9,0.04,1.49,1.07,1,0.4,0,0.02,0.06,0,21.86,0.01,0,0.32,0.16,0.16,0.02,1.87,2.03,4.69,0,0,0,0,0,0,0,0,0,0,0.13,0,0.05,1.32,0.04,1.18,0.49,0.89,0.31,0.75,1.29,1.2,0.98,2.79,2.65,0.01,2.39,4.32,2.87,7.87,1.15,4.83,2.99,3.95,4.08,10.83,0.06,34.65,0.04,0.73,0,0.09,0.13,0.25,0.34,0.11,1.09,4.03,8.93,0,0,0,5.19,0,0.07,0.23,32.7,0.04,0.01,1.82,0.99,0,29.37,30.73,0,0,0.01,0.43,0,0,0,3.9,0.65,1.04,2.33,0.01,0.07,0.01,0.1,0.36,0.01,0.03,0.03,1.67,0.82,5.45,0,0.46,0.19,1.61,0.55,0.27,0.21,0.47,1.36,0.09,1.06,0.37,0.07,0.6,0.05,1.21,0,0.56,0.13],[10.23,1.27,0.18,0.17,29.49,0.02,0.01,0.02,41.98,59.15,8.92,8.2,0.79,2.88,9.19,24.7,1.22,1.84,0.07,0.03,1.27,0.06,0,34.1,4.52,1.24,0.33,10.38,0,0.96,1.22,0.02,1.77,6.34,2.41,0.46,0.13,0.01,0.9,5.49,0.03,0.81,1.36,1.72,3.22,0,0,5.77,6.37,0,0.21,0.02,0.02,0.01,0,0,0.16,0.07,1.01,0.72,0,0.01,0.06,2.71,23.34,64.54,0.02,0.16,0.83,0.68,0.41,0.23,0,0.37,0.38,0.1,0.26,0.47,0.01,81.65,0.7,75.35,73.44,109.62,46.3,94.2,93.59,0,8.84,0,167.07,0,0.04,0,20.09,82.73,69.96,145.88,168.51,168.77,174.02,167.33,142.93,133.42,181.14,1.03,0.93,2.9,0.03,1.3,0.89,4.86,2.55,2.37,4.19,0.01,5.29,7.66,3.97,0.69,16.54,6.55,5.38,0.99,1.27,0.01,139.84,2.43,4.11,2.35,4.19,4.26,0.35,0.39,8.24,6.12,8.58,0,0.02,0.03,0.54,0.15,0.22,1.73,0.36,0.6,0.01,0,0.97,59.19,1.67,0.74,0.08,0.42,0.01,2.18,0.31,0.03,0.36,0,0,0.14,0.75,0.69,0.23,0.16,0.04,0.83,0.01,0.1,0.44,0.23,0.03,0.07,0.41,0.64,3.57,10.33,10.14,9.78,7.12,14.54,0,9.12,1.22,1.21,3.79,5.63,0.02,3.93,2.98,0.77,2.89,1.7,0.54,29.06,0,0,0.01,0,0.02,0.24,0.19,0.21,0.32,0.21,0.57,10.94,2.63,1.78,3.22,4.95,6.66,6.02,0.03,1.07,7.51,32.51,1.82,22.34,33.13,0.31,0,11.49,18.81,0.94,3.17,0.06,0.36,1.7,1.2,0.25,6.06,1.26,2.75,2.05,0.12,0,0.2,0,0.62,0,0.09,1.23,0.86,0,0,0,0,0.77,0.19,0.21,0.71,0.06,0.05,0.5,0.98,0.66,0.65,0.91,1,10.32,0,0.83,0,0,0.01,0.02,0.05,0.22,0.04,0.01,0.1,0.1,0.12,0.02,0.27,0.12,1.13,0.06,0.73,0.05,0.12,0.01,0.02,1.2,0,1.88,0.06,0.2,0.05,2.07,0,0.02,0.01,1.02,0.04,0.08,49.73,8.82,7.78,86.25,0,0.01,0,0,0.97,0.16,0.01,0.39,2.03,1.07,2.37,1.58,4.33,0.51,12.71,16.84,18.59,71.4,57.41,76.7,61.89,0.04,10.52,1.09,0,1.33,0.91,1.6,24.22,0.49,2.07,1.02,23.93,9.92,2.15,0.88,1.78,3.6,7.85,4.68,1.66,0.79,1.98,2.3,6.92,8.79,45.47,27.79,0.61,0.2,1.57,2.13,3.81,3.82,2.93,0.89,0.98,0.85,0.05,7.01,0.29,3.33,1.43,5.75,14.48,7.58,43.36,36.77,18.3,23.46,20.19,0.99,8.3,0,0,0,0,0,1.65,10.26,10.87,0.11,0.11,0,0.35,0.11,6.45,1.16,26.17,4.56,0.02,5.87,4.66,0.01,4.23,1.65,0.02,1.2,0,20.33,0.04,9.23,7.43,7.9,0,0.14,1.95,0.31,1.61,3.29,4.78,0.18,0,0.06,0.23,0.78,0,0.73,0.06,0.82,0.22,0.42,0.3,0.28,0.01,0,7.17,0.07,0.12,1.93,0.19,0.55,3.63,2.42,0.02,0.47,0.27],[0.28,0.35,2.28,3.75,1.26,2.78,0,0.04,1.76,0,0.9,0.01,0.55,1.59,0,0.18,1.73,1.07,0.14,2.39,0.7,2.5,0,0.51,0,0,0,0.05,0,1.41,0,0.01,0.06,0,0.01,0.29,3.85,3.24,0.23,4.89,0.25,5.96,1.09,0,0,5.02,5.25,0,0.01,0,0,0,0.91,0.11,0.21,0.4,0.03,0.13,0,0,1.1,0.5,1.35,0.01,1.09,0,0.73,0.11,0,0.1,0,0.29,2.87,0.15,6.88,2.22,0.05,0.07,0.01,0.41,0,1.11,0.86,0.66,2.73,0.87,2.74,0.08,1.89,0,4.16,0.5,0.03,0.01,0.29,0.36,1.92,1.09,3.31,3.94,0.74,4.2,3.43,0.23,1.03,0.38,0.85,0,0.04,0.41,0.13,0.52,0.94,0,0.11,0.29,0.28,0.65,0.54,0.02,0.78,0.7,0.26,0.52,1.06,1.23,5.72,1.84,0,0.01,2.78,2.51,0,0,0.76,2.56,1.13,0.01,0.93,0,0.06,1.97,3.09,8.74,0.22,0.2,0.03,0,0,0.02,0,0.06,0,0.46,0.24,1.73,0.05,0.32,0,0.1,0.01,0,0.14,0.06,0,0.01,2.55,0,0.03,0,0.05,0.77,0.01,0.12,0.02,0,0.2,1.57,0.73,1.18,0.28,1.69,0,0.13,0.81,0.72,0,0.01,0,0,0,0,0,0.01,0,0,0.24,0.32,0.03,0.42,0.01,0.01,11.96,0,0.02,0.05,0,0,0.14,2.68,0.31,0.37,4.45,0,0.03,0.01,0.48,5.17,0.18,0.86,2.32,0.21,0.66,0,2.15,0.78,2.3,0.92,0,1.55,0.03,0.29,0.43,0.07,0.04,0,11.5,1.53,0.09,2.32,0.04,0.01,0,0,0.05,0.3,0.3,0.03,0.11,0.43,0.05,0.03,0.07,0,0,0.05,0.01,0.58,0.19,0.06,0.02,0.97,0,0.07,0,0,0.02,0.03,0.17,0.03,2.61,0.16,0.1,0,0.03,0.11,0.14,0.02,0.04,0,0.13,0.07,0.3,0,0.03,0.12,0,0.61,1.69,8.84,0.08,0.28,0.32,0.04,0.07,1.62,0.1,0,2.93,0.42,1.04,0.99,0,0.1,0,0,0.28,0.01,0.05,0,0,0.95,0.01,0.08,0.02,0.01,2.5,0.16,0.21,1.89,2.47,1.89,0.7,2.03,4.04,1.78,0.02,0.38,0,0.29,0,10.15,7.9,7.43,0.01,0.2,1.71,0.76,4.73,0.76,3.07,1.84,3.89,4.67,0.01,0.01,0.05,0,1.79,3.94,0.18,0.05,0,0,0.15,0,0,0,0,0.23,0.94,0.01,0.03,0.21,6.52,0.92,0.01,0.13,0.38,0.27,0,1,0,0.15,0,0,0,0.02,0,0,0,3.05,2.41,0.05,0.14,0.01,0,0,0.47,0,2.12,0,0.01,0.97,0,0.04,1.82,0.36,0.07,4.47,0,1.89,0.9,0.33,2.04,0,0,0.81,0,5.97,0,0.01,0.08,0,0.03,0,0.05,0.08,0.01,3.28,1.44,0.01,2.6,0.79,0,0.34,0.25,0,0.1,0.01,0.04,0.08,0.32,0.15,0.01,0,0.34,0.15,0.04]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>motif<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="top-10-motifs-in-each-topic" class="section level3">
<h3>Top 10 motifs in each topic</h3>
<pre class="r"><code>cat(&quot;Number of regions selected for each topic: \n&quot;)
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

colnames_homer &lt;- c(&quot;motif_name&quot;, &quot;consensus&quot;, &quot;P&quot;, &quot;log10P&quot;, &quot;Padj&quot;,  &quot;num_target&quot;, &quot;percent_target&quot;, &quot;num_bg&quot;, &quot;percent_bg&quot;)

top_motifs &lt;- data.frame(matrix(nrow=10, ncol = length(homer_res_topics)))
colnames(top_motifs) &lt;- names(homer_res_topics)
for (k in 1:length(homer_res_topics)){
  homer_res &lt;- homer_res_topics[[k]]
  colnames(homer_res) &lt;- colnames_homer
  homer_res &lt;- homer_res %&gt;% separate(motif_name, c(&quot;motif&quot;, &quot;origin&quot;, &quot;database&quot;), &quot;/&quot;)
  top_motifs[,k] &lt;- head(homer_res$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F, caption = &quot;Top 10 motifs enriched in each topic.&quot;)</code></pre>
<pre><code># Number of regions selected for each topic: 
#   k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11  k12  k13 
# 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363</code></pre>
<div id="htmlwidget-d57c2a72e51faef143c1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d57c2a72e51faef143c1">{"x":{"filter":"none","caption":"<caption>Top 10 motifs enriched in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Gata2(Zf)","Gata4(Zf)","Gata6(Zf)","Gata1(Zf)","GATA3(Zf)","TRPS1(Zf)","GATA:SCL(Zf,bHLH)","KLF1(Zf)","Sp2(Zf)","KLF5(Zf)"],["NeuroD1(bHLH)","Atoh1(bHLH)","NeuroG2(bHLH)","Olig2(bHLH)","BHLHA15(bHLH)","TCF4(bHLH)","Twist2(bHLH)","NF1-halfsite(CTF)","Ascl1(bHLH)","NF1(CTF)"],["Tcf21(bHLH)","BHLHA15(bHLH)","Ap4(bHLH)","TCF4(bHLH)","Twist2(bHLH)","Atoh1(bHLH)","NeuroG2(bHLH)","Tcf12(bHLH)","Myf5(bHLH)","ZBTB18(Zf)"],["TEAD1(TEAD)","TEAD4(TEA)","TEAD3(TEA)","TEAD(TEA)","TEAD2(TEA)","HNF1b(Homeobox)","Hnf1(Homeobox)","NF1(CTF)","Atf3(bZIP)","Fra1(bZIP)"],["HNF1b(Homeobox)","HNF4a(NR),DR1","Hnf1(Homeobox)","PPARa(NR),DR1","PPARE(NR),DR1","RXR(NR),DR1","Erra(NR)","COUP-TFII(NR)","RARa(NR)","TR4(NR),DR1"],["SpiB(ETS)","PU.1(ETS)","Elf4(ETS)","ELF5(ETS)","ETS1(ETS)","Etv2(ETS)","GABPA(ETS)","Fli1(ETS)","ETV1(ETS)","ELF3(ETS)"],["Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Mef2a(MADS)","NF1-halfsite(CTF)","NF1(CTF)","Gata4(Zf)","Gata6(Zf)","GATA3(Zf)","ERRg(NR)"],["HNF4a(NR),DR1","PPARa(NR),DR1","CEBP(bZIP)","HNF6(Homeobox)","HNF1b(Homeobox)","PPARE(NR),DR1","RXR(NR),DR1","Cux2(Homeobox)","Hnf6b(Homeobox)","COUP-TFII(NR)"],["Sox10(HMG)","Sox3(HMG)","Sox2(HMG)","Sox9(HMG)","Sox21(HMG)","Sox15(HMG)","Sox4(HMG)","Sox6(HMG)","Sox17(HMG)","NF1-halfsite(CTF)"],["CDX4(Homeobox)","HOXB13(Homeobox)","KLF5(Zf)","Cdx2(Homeobox)","Hoxa11(Homeobox)","Hoxd13(Homeobox)","Hoxa13(Homeobox)","Hoxd11(Homeobox)","Klf4(Zf)","KLF6(Zf)"],["Mef2b(MADS)","Mef2c(MADS)","Mef2a(MADS)","Mef2d(MADS)","NeuroG2(bHLH)","Atoh1(bHLH)","Olig2(bHLH)","Rfx2(HTH)","BHLHA15(bHLH)","Egr1(Zf)"],["Fli1(ETS)","Etv2(ETS)","ETV1(ETS)","ETS1(ETS)","ETV4(ETS)","ERG(ETS)","Ets1-distal(ETS)","EWS:ERG-fusion(ETS)","GABPA(ETS)","EWS:FLI1-fusion(ETS)"],["MafA(bZIP)","NFY(CCAAT)","Smad2(MAD)","PPARE(NR),DR1","GLIS3(Zf)","Smad3(MAD)","Smad4(MAD)","EBF1(EBF)","Tbx20(T-box)","ZFX(Zf)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-motif-enrichment-across-topics" class="section level3">
<h3>Heatmap of motif enrichment across topics</h3>
<p>Heatmap of motif enrichment -log10(p-value).</p>
<pre class="r"><code>create_motif_enrichment_heatmap(motif_res, enrichment = &quot;-log10(p-value)&quot;, 
                                cluster_motifs = TRUE, cluster_topics = TRUE, motif_filter = 10, horizontal = FALSE,
                                enrichment_range = c(0,100), method_cluster = &quot;average&quot;, font.size.motifs = 4, font.size.topics = 9)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-pvalue-1.png" width="840" style="display: block; margin: auto;" /></p>
<pre><code># 243 out of 439 motifs included the heatmap</code></pre>
</div>
<div id="scatterplots-of-motif-enrichment" class="section level3">
<h3>Scatterplots of motif enrichment</h3>
<p>Plot motif enrichment (-log10 p-value) and the ranking</p>
<pre class="r"><code># Plot enrichment (-log10 p-value) and ranking of the motifs

plots &lt;- vector(&quot;list&quot;, ncol(motif_res$mlog10P))
names(plots) &lt;- colnames(motif_res$mlog10P)

for( i in 1:length(plots)){
  plots[[i]] &lt;- create_motif_enrichment_ranking_plot(motif_res, k = i, 
                                                     max.overlaps = 20, subsample = FALSE)
}

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-1-1.png" width="1200" style="display: block; margin: auto;" /></p>
<p><strong>Topic 1 example</strong></p>
<pre class="r"><code>print(plots[[1]])</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-enrichment-ranking-plot-2-1">
Past versions of motif-enrichment-ranking-plot-2-1.png
</button>
</p>
<div id="fig-motif-enrichment-ranking-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/40ea1f1d37845883771d1e29a09de326467028e0/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">40ea1f1</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="motif-enrichment-vs-gene-score" class="section level3">
<h3>Motif enrichment vs gene score</h3>
<p>Load pre-computed gene scores</p>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-TSS-l2&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
genescore_res &lt;- readRDS(file.path(gene.dir, &quot;genescore_result_topics.rds&quot;))

genes &lt;- genescore_res$genes
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genes$SYMBOL
gene_logFC &lt;- genescore_res$beta
rownames(gene_logFC) &lt;- genes$SYMBOL</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/geneanalysis-Cusanovich2018-k=13-TSS-l2</code></pre>
<p>Get TF genes</p>
<pre class="r"><code>motif_names &lt;- motif_res$motifs$motif
gene_names &lt;- genescore_res$genes$SYMBOL
common_genes &lt;- intersect(toupper(motif_names), toupper(gene_names))
cat(sprintf(&quot;%s TF genes mapped between motif names and gene symbol. \n&quot;, length(common_genes)))

motif_gene_table &lt;- data.frame(motif = motif_names[match(common_genes, toupper(motif_names))], 
                                      gene = gene_names[match(common_genes, toupper(gene_names))])</code></pre>
<pre><code># 247 TF genes mapped between motif names and gene symbol.</code></pre>
<p>Compute correlation between motif enrichment z-score and gene score:</p>
<p><strong>Topic 1 example</strong></p>
<ul>
<li>Compute motif enrichment z-scores from the motif enrichment p-values</li>
<li>Plot motif enrichment (-log10 p-value) and correlation between motif enrichment z-scores and gene scores</li>
<li>Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment z-score and gene scores</li>
</ul>
<pre class="r"><code>motif_gene_mapping &lt;- create_motif_gene_cor_scatterplot(motif_res, genescore_res, motif_gene_table, 
                                                        k = 4, cor.motif = &quot;z-score&quot;)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-cor-plot-1-1">
Past versions of motif-cor-plot-1-1.png
</button>
</p>
<div id="fig-motif-cor-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/40ea1f1d37845883771d1e29a09de326467028e0/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" target="_blank">40ea1f1</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif_gene_mapping &lt;- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_zscore, decreasing = T)),]
rownames(motif_gene_mapping) &lt;- 1:nrow(motif_gene_mapping)

cat(&quot;Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n&quot;)
print(head(motif_gene_mapping[,c(&quot;motif&quot;,&quot;motif_mlog10P&quot;, &quot;gene_score&quot;, &quot;cor_zscore&quot;)], 10))</code></pre>
<pre><code># Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: 
#               motif motif_mlog10P gene_score cor_zscore
# 1        TEAD3(TEA)      89.89896   33.96213  0.8305369
# 2       TEAD1(TEAD)      97.19511   37.87814  0.6162972
# 3        TEAD4(TEA)      93.63389   43.41048  0.6033918
# 4   HNF1b(Homeobox)      53.67880   55.95734  0.9691212
# 5   Foxa2(Forkhead)      29.36699   46.10279  0.8435864
# 6        GRHL2(CP2)      26.10110   88.26210  0.9316662
# 7   FOXA1(Forkhead)      25.61469   67.55695  0.8544556
# 8  HOXB13(Homeobox)      24.12940   35.16238  0.8467272
# 9  Hoxd13(Homeobox)      28.61132   11.97401  0.6732085
# 10   Cdx2(Homeobox)      20.24247   22.57429  0.9163834</code></pre>
<p><strong>GATA family</strong></p>
<pre class="r"><code>motif_names &lt;- motif_res$motifs$motif
gene_names &lt;- genescore_res$genes$SYMBOL
TF_motifs &lt;- sort(unique(grep(&quot;^GATA\\d*$&quot;, motif_names, ignore.case=T, value=T)))
TF_genes &lt;- sort(unique(grep(&quot;^GATA\\d*$&quot;, gene_names, ignore.case=T, value=T)))
common_genes &lt;- intersect(toupper(TF_motifs), toupper(TF_genes))

motif_gene_table &lt;- data.frame(motif = TF_motifs[match(common_genes, toupper(TF_motifs))], 
                                      gene = TF_genes[match(common_genes, toupper(TF_genes))])
print(motif_gene_table)</code></pre>
<pre><code>#   motif  gene
# 1 Gata1 Gata1
# 2 Gata2 Gata2
# 3 GATA3 Gata3
# 4 Gata4 Gata4
# 5 Gata6 Gata6</code></pre>
<p>Plot GATA motifs in topic 1</p>
<pre class="r"><code># Plot GATA motifs in topic 1
k = 1
motif_order &lt;- order(motif_res$mlog10P[,k], decreasing = T)
motifs &lt;- rownames(motif_res$motifs[motif_order,])
motif_names &lt;- motif_res$motifs[motif_order, &quot;motif&quot;]
selected_motifs &lt;- unique(motifs[match(toupper(motif_gene_table$motif), toupper(motif_names))])
motif.dir &lt;- paste0(homer.dir, &quot;/homer_result_topic_&quot;, k, &quot;/knownResults/&quot;)

for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = &quot;both&quot;)
}</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-2.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-3.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-4.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-5.png" width="600" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, genescore_res, 
                                       motif_gene_table,
                                       k = 1, 
                                       y = &quot;-log10(p-value)&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-GATA-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot motif enrichment (zscore) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, genescore_res, 
                                       motif_gene_table,
                                       k = 1, 
                                       y = &quot;z-score&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-GATA-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] grid      stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
#  [1] Logolas_1.3.1     reshape2_1.4.3    DT_0.16           htmlwidgets_1.5.3
#  [5] plotly_4.9.3      cowplot_1.1.1     ggrepel_0.9.1     ggplot2_3.3.3    
#  [9] tidyr_1.1.2       dplyr_1.0.3       fastTopics_0.4-29 Matrix_1.2-18    
# [13] workflowr_1.6.2  
# 
# loaded via a namespace (and not attached):
#  [1] nlme_3.1-140       mcmc_0.9-7         matrixStats_0.58.0 fs_1.3.1          
#  [5] bit64_4.0.5        progress_1.2.2     httr_1.4.2         rprojroot_2.0.2   
#  [9] tools_3.6.1        R6_2.5.0           irlba_2.3.3        DBI_1.1.0         
# [13] lazyeval_0.2.2     colorspace_2.0-0   ade4_1.7-16        withr_2.4.1       
# [17] tidyselect_1.1.0   prettyunits_1.1.1  bit_4.0.4          compiler_3.6.1    
# [21] git2r_0.27.1       quantreg_5.83      SparseM_1.78       labeling_0.4.2    
# [25] scales_1.1.1       SQUAREM_2021.1     quadprog_1.5-8     mixsqp_0.3-43     
# [29] stringr_1.4.0      digest_0.6.27      rmarkdown_2.6      MCMCpack_1.5-0    
# [33] pkgconfig_2.0.3    htmltools_0.5.1.1  invgamma_1.1       rlang_0.4.10      
# [37] farver_2.0.3       generics_0.1.0     jsonlite_1.7.2     crosstalk_1.1.1   
# [41] magrittr_2.0.1     Rcpp_1.0.6         munsell_0.5.0      ape_5.4-1         
# [45] lifecycle_0.2.0    CVXR_1.0-9         stringi_1.5.3      whisker_0.4       
# [49] yaml_2.2.1         MASS_7.3-53        Rtsne_0.15         plyr_1.8.6        
# [53] parallel_3.6.1     promises_1.1.1     crayon_1.4.0       lattice_0.20-41   
# [57] hms_1.0.0          knitr_1.30         pillar_1.4.7       seqinr_4.2-5      
# [61] glue_1.4.2         evaluate_0.14      data.table_1.13.6  RcppParallel_5.0.2
# [65] vctrs_0.3.6        httpuv_1.5.4       MatrixModels_0.4-1 gtable_0.3.0      
# [69] purrr_0.3.4        ashr_2.2-47        xfun_0.19          gridBase_0.4-7    
# [73] Rmpfr_0.8-2        coda_0.19-4        later_1.1.0.1      viridisLite_0.3.0 
# [77] truncnorm_1.0-8    tibble_3.0.6       conquer_1.0.2      gmp_0.6-2         
# [81] ellipsis_0.3.1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
