<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Motif analysis using topic modeling results for Cusanovich et al (2018) scATAC-seq data</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Motif analysis using topic modeling results for Cusanovich <em>et al</em> (2018) scATAC-seq data</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree780af9ec2b7df605022a5864024ef8ae0e369ad5targetblank780af9ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/780af9ec2b7df605022a5864024ef8ae0e369ad5" target="_blank">780af9e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree780af9ec2b7df605022a5864024ef8ae0e369ad5targetblank780af9ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/780af9ec2b7df605022a5864024ef8ae0e369ad5" target="_blank">780af9e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/diff_count_Buenrostro2018_chomVAR_scPeaks.Rmd
    Untracked:  analysis/gene_analysis_Cusanovich2018.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  analysis/single_cell_rnaseq_demo.Rmd
    Untracked:  output/Cusanovich2018/
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum.html
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum.png
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-TSS-sum_files/
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum.html
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum.png
    Untracked:  output/gsea_topic_1_Cusanovich2018-k=13-genebody-sum_files/
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum.html
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum.png
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-TSS-sum_files/
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum.html
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum.png
    Untracked:  output/volcano_topic_1_Cusanovich2018-k=13-genebody-sum_files/
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch

Unstaged changes:
    Modified:   analysis/cisTopic_Buenrostro2018_chomVAR_scPeaks.Rmd
    Modified:   analysis/motif_gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/plots_Lareau2019_bonemarrow.Rmd
    Modified:   code/plots.R
    Modified:   scripts/fit_all_models_Buenrostro_2018.sbatch
    Modified:   scripts/fit_cisTopic_Buenrostro_2018_chromVAR_scPeaks.sh
    Modified:   scripts/postfit_Buenrostro2018.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/motif_analysis_Cusanovich2018.Rmd</code>) and HTML (<code>docs/motif_analysis_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/780af9ec2b7df605022a5864024ef8ae0e369ad5/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">780af9e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-21
</td>
<td>
added motif logo plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/motif_analysis_Cusanovich2018.html" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c7428cdf6599187564a1d4e4896daa384e155a6d/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">c7428cd</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
minor adjustment on the figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/motif_analysis_Cusanovich2018.html" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/4458ed5bf26503cb8cc53fb412abe62d96fd56e5/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">4458ed5</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
added KLF/SP genes and added an option to compute correlation using -log10P
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/motif_analysis_Cusanovich2018.html" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c9d674286d289488c47518606b422b0d901dde39/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">c9d6742</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
<td>
updated the correlations between motif enrichment (using z-score) and gene z-score
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/motif_analysis_Cusanovich2018.html" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0134bdfe702344f01e4ae492b2fdd5c3f0a7d8ac/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">0134bdf</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
fix colors in scatter plots of create_motif_gene_scatterplot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/motif_analysis_Cusanovich2018.html" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/f51b56e493e3713f1329a64876152c270b9c4f08/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">f51b56e</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
added colors to scatterplots for motif enrichment and gene scores
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/motif_analysis_Cusanovich2018.html" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/5f1841157267142b5306364c1a22a516679ac82e/analysis/motif_analysis_Cusanovich2018.Rmd" target="_blank">5f18411</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
<td>
Plot motif enrichment and correlate with gene scores
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform TF motif analysis for the Cusanovich <em>et al</em> (2018) scATAC-seq result inferred from the multinomial topic model with <span class="math inline">\(k = 13\)</span>.</p>
<div id="load-packages-and-some-functions-used-in-this-analysis" class="section level2">
<h2>Load packages and some functions used in this analysis</h2>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape2)
library(Logolas)
library(grid)
source(&quot;code/motif_analysis.R&quot;)
source(&quot;code/plots.R&quot;)</code></pre>
</div>
<div id="load-data-and-topic-model-results" class="section level2">
<h2>Load data and topic model results</h2>
<p>Load the data and the <span class="math inline">\(k = 13\)</span> Poisson NMF fit results.</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
rm(counts)</code></pre>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
<div id="visualize-by-structure-plot-grouped-by-tissues" class="section level2">
<h2>Visualize by Structure plot grouped by tissues</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
rows &lt;- sample(nrow(fit$L),4000)
samples$tissue &lt;- as.factor(samples$tissue)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = rows),
                              grouping = samples[rows, &quot;tissue&quot;],n = Inf,gap = 40,
                              perplexity = 50,topics = 1:13,colors = colors_topics,
                              num_threads = 4,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-tissue-1">
Past versions of structure-plot-tissue-1.png
</button>
</p>
<div id="fig-structure-plot-tissue-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/structure-plot-tissue-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-accessbility-analysis-of-the-atac-seq-regions-for-the-topics" class="section level2">
<h2>Differential accessbility analysis of the ATAC-seq regions for the topics</h2>
<p>Load results from differential accessbility analysis for the topics</p>
<pre class="r"><code>out.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
cat(sprintf(&quot;Load results from %s \n&quot;, out.dir))</code></pre>
<pre><code># Load results from /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018</code></pre>
<pre class="r"><code>diff_count_topics &lt;- readRDS(file.path(out.dir, &quot;/diffcount-Cusanovich2018-13topics.rds&quot;))</code></pre>
<p>Distribution of z-scores</p>
<pre class="r"><code>zscore_topics &lt;-  melt(diff_count_topics$Z)
colnames(zscore_topics) &lt;- c(&quot;region&quot;, &quot;topic&quot;, &quot;zscore&quot;)
levels(zscore_topics$topic) &lt;- colnames(diff_count_topics$Z)

z.quantile.99 &lt;- apply(abs(diff_count_topics$Z), 2, quantile, 0.99)
cat(&quot;z-score 99% quantile: \n&quot;)
print(z.quantile.99)

p.hist.zscores &lt;- ggplot(zscore_topics, aes(x=zscore)) + 
  geom_histogram(binwidth=1, color=&quot;black&quot;, fill=&quot;white&quot;) + 
  coord_cartesian(xlim = c(-10, 30)) + theme_cowplot(font_size = 10) +
  facet_wrap(~ topic, ncol=4)

print(p.hist.zscores)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/hist-zscore-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-hist-zscore-1">
Past versions of hist-zscore-1.png
</button>
</p>
<div id="fig-hist-zscore-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/hist-zscore-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># z-score 99% quantile: 
#       k1       k2       k3       k4       k5       k6       k7       k8 
# 21.42992 31.48751 25.46082 25.97670 34.64418 37.09098 32.07250 39.65746 
#       k9      k10      k11      k12      k13 
# 25.93102 15.88394 34.29782 39.80147 20.71928</code></pre>
<p>Volcano plot of the regions</p>
<p><strong>topic 1 example</strong></p>
<pre class="r"><code>volcano_plot(diff_count_topics,k = 1,label_above_quantile = Inf, 
             subsample_below_quantile = 0.7, subsample_rate = 0.1)</code></pre>
<pre><code># 161397 out of 436206 data points will be included in plot</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/region-volcano-plot-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-region-volcano-plot-1">
Past versions of region-volcano-plot-1.png
</button>
</p>
<div id="fig-region-volcano-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/region-volcano-plot-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/region-volcano-plot-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="motif-enrichment-analysis-using-homer" class="section level2">
<h2>Motif enrichment analysis using <a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html">HOMER</a></h2>
<ul>
<li>Details about HOMER motif analysis:</li>
<li><a href="http://homer.ucsd.edu/homer/motif/" class="uri">http://homer.ucsd.edu/homer/motif/</a></li>
<li><p><a href="http://homer.ucsd.edu/homer/ngs/peakMotifs.html" class="uri">http://homer.ucsd.edu/homer/ngs/peakMotifs.html</a></p></li>
<li><p>Motif enrichment result using regions with z-score above 99% quantile.</p></li>
</ul>
<p>Compile Homer results across topics</p>
<pre class="r"><code>homer.dir &lt;- paste0(out.dir, &quot;/motifanalysis-Cusanovich2018-k=13-quantile/HOMER&quot;)
cat(sprintf(&quot;Directory of motif analysis result: %s \n&quot;, homer.dir))
homer_res_topics &lt;- readRDS(file.path(homer.dir, &quot;/homer_knownResults.rds&quot;))
selected_regions &lt;- readRDS(file.path(homer.dir, &quot;/selected_regions.rds&quot;))

# Compile Homer results (pvalue and ranking) across topics
motif_res &lt;- compile_homer_motif_res(homer_res_topics)
saveRDS(motif_res, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))
cat(&quot;compiled homer motif results are saved in&quot;, paste0(homer.dir, &quot;/homer_motif_enrichment_results.rds&quot;))

motif_table &lt;- data.frame(motif = gsub(&quot;/.*&quot;, &quot;&quot;, rownames(motif_res$mlog10P)),
                          round(motif_res$mlog10P,2))
DT::datatable(motif_table, rownames = F, caption = &quot;Motif enrichment (-log10P)&quot;)</code></pre>
<pre><code># Directory of motif analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/motifanalysis-Cusanovich2018-k=13-quantile/HOMER 
# compiled homer motif results are saved in /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/motifanalysis-Cusanovich2018-k=13-quantile/HOMER/homer_motif_enrichment_results.rds</code></pre>
<div id="htmlwidget-9217405887d9db89f01d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9217405887d9db89f01d">{"x":{"filter":"none","caption":"<caption>Motif enrichment (-log10P)<\/caption>","data":[["AMYB(HTH)","AP-1(bZIP)","AP-2alpha(AP2)","AP-2gamma(AP2)","Ap4(bHLH)","AR-halfsite(NR)","ARE(NR)","Arnt:Ahr(bHLH)","Ascl1(bHLH)","Ascl2(bHLH)","Atf1(bZIP)","Atf2(bZIP)","Atf3(bZIP)","Atf4(bZIP)","Atf7(bZIP)","Atoh1(bHLH)","Bach1(bZIP)","Bach2(bZIP)","Bapx1(Homeobox)","Barx1(Homeobox)","BATF(bZIP)","Bcl11a(Zf)","Bcl6(Zf)","BHLHA15(bHLH)","bHLHE40(bHLH)","bHLHE41(bHLH)","BMAL1(bHLH)","BMYB(HTH)","BORIS(Zf)","Brachyury(T-box)","Brn1(POU,Homeobox)","Brn2(POU,Homeobox)","bZIP:IRF(bZIP,IRF)","c-Jun-CRE(bZIP)","c-Myc(bHLH)","c-Myc(bHLH)","CArG(MADS)","Cdx2(Homeobox)","CDX4(Homeobox)","CEBP:AP1(bZIP)","CEBP:CEBP(bZIP)","CEBP(bZIP)","Chop(bZIP)","CHR(?)","CLOCK(bHLH)","COUP-TFII(NR)","COUP-TFII(NR)","CRE(bZIP)","CREB5(bZIP)","CRX(Homeobox)","CTCF-SatelliteElement(Zf?)","CTCF(Zf)","CUX1(Homeobox)","Cux2(Homeobox)","DLX1(Homeobox)","DLX2(Homeobox)","Dlx3(Homeobox)","DLX5(Homeobox)","DMRT1(DM)","DMRT6(DM)","DUX(Homeobox)","DUX4(Homeobox)","Duxbl(Homeobox)","E-box(bHLH)","E2A(bHLH),near_PU.1","E2A(bHLH)","E2F(E2F)","E2F1(E2F)","E2F3(E2F)","E2F4(E2F)","E2F6(E2F)","E2F7(E2F)","EAR2(NR)","EBF(EBF)","EBF1(EBF)","EBF2(EBF)","EBNA1(EBV-virus)","Egr1(Zf)","Egr2(Zf)","EHF(ETS)","EKLF(Zf)","ELF1(ETS)","ELF3(ETS)","Elf4(ETS)","ELF5(ETS)","Elk1(ETS)","Elk4(ETS)","En1(Homeobox)","Eomes(T-box)","ERE(NR),IR3","ERG(ETS)","Erra(NR)","ERRg(NR)","Esrrb(NR)","ETS:E-box(ETS,bHLH)","ETS:RUNX(ETS,Runt)","ETS(ETS)","Ets1-distal(ETS)","ETS1(ETS)","ETV1(ETS)","Etv2(ETS)","ETV4(ETS)","EWS:ERG-fusion(ETS)","EWS:FLI1-fusion(ETS)","Fli1(ETS)","Fos(bZIP)","Fosl2(bZIP)","Fox:Ebox(Forkhead,bHLH)","FOXA1:AR(Forkhead,NR)","FOXA1(Forkhead)","FOXA1(Forkhead)","Foxa2(Forkhead)","Foxa3(Forkhead)","FoxD3(forkhead)","Foxf1(Forkhead)","Foxh1(Forkhead)","FOXK1(Forkhead)","FOXK2(Forkhead)","FoxL2(Forkhead)","FOXM1(Forkhead)","Foxo1(Forkhead)","Foxo3(Forkhead)","FOXP1(Forkhead)","Fra1(bZIP)","Fra2(bZIP)","FXR(NR),IR1","GABPA(ETS)","GATA:SCL(Zf,bHLH)","GATA(Zf),IR3","GATA(Zf),IR4","Gata1(Zf)","Gata2(Zf)","GATA3(Zf),DR4","GATA3(Zf),DR8","GATA3(Zf)","Gata4(Zf)","Gata6(Zf)","Gfi1b(Zf)","GFX(?)","GFY-Staf(?,Zf)","GFY(?)","Gli2(Zf)","GLI3(Zf)","GLIS3(Zf)","GRE(NR),IR3","GRE(NR),IR3","GRHL2(CP2)","GSC(Homeobox)","Hand2(bHLH)","HEB(bHLH)","HIC1(Zf)","HIF-1a(bHLH)","HIF-1b(HLH)","HIF2a(bHLH)","HINFP(Zf)","HLF(bZIP)","Hnf1(Homeobox)","HNF1b(Homeobox)","HNF4a(NR),DR1","HNF6(Homeobox)","Hnf6b(Homeobox)","HOXA1(Homeobox)","Hoxa10(Homeobox)","Hoxa11(Homeobox)","Hoxa13(Homeobox)","HOXA2(Homeobox)","Hoxa9(Homeobox)","HOXB13(Homeobox)","Hoxb4(Homeobox)","Hoxc9(Homeobox)","Hoxd10(Homeobox)","Hoxd11(Homeobox)","Hoxd12(Homeobox)","Hoxd13(Homeobox)","HRE(HSF)","HRE(HSF)","IRF:BATF(IRF:bZIP)","IRF1(IRF)","IRF2(IRF)","IRF3(IRF)","IRF4(IRF)","IRF8(IRF)","Isl1(Homeobox)","ISRE(IRF)","Jun-AP1(bZIP)","JunB(bZIP)","JunD(bZIP)","KLF1(Zf)","KLF10(Zf)","KLF14(Zf)","KLF3(Zf)","Klf4(Zf)","KLF5(Zf)","KLF6(Zf)","Klf9(Zf)","LEF1(HMG)","Lhx1(Homeobox)","Lhx2(Homeobox)","Lhx3(Homeobox)","LHX9(Homeobox)","LRF(Zf)","LXRE(NR),DR4","MafA(bZIP)","MafB(bZIP)","MafF(bZIP)","MafK(bZIP)","Max(bHLH)","Maz(Zf)","Mef2a(MADS)","Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Meis1(Homeobox)","MITF(bHLH)","MNT(bHLH)","Mouse_Recombination_Hotspot(Zf)","MYB(HTH)","Myf5(bHLH)","MYNN(Zf)","MyoD(bHLH)","MyoG(bHLH)","n-Myc(bHLH)","Nanog(Homeobox)","NeuroD1(bHLH)","NeuroG2(bHLH)","NF-E2(bZIP)","NF1-halfsite(CTF)","NF1:FOXA1(CTF,Forkhead)","NF1(CTF)","NFAT:AP1(RHD,bZIP)","NFAT(RHD)","NFE2L2(bZIP)","NFIL3(bZIP)","NFkB-p50,p52(RHD)","NFkB-p65-Rel(RHD)","NFkB-p65(RHD)","NFY(CCAAT)","Nkx2.1(Homeobox)","Nkx2.2(Homeobox)","Nkx2.5(Homeobox)","Nkx3.1(Homeobox)","Nkx6.1(Homeobox)","NPAS(bHLH)","NPAS2(bHLH)","Npas4(bHLH)","Nr5a2(NR)","Nr5a2(NR)","NRF(NRF)","NRF1(NRF)","Nrf2(bZIP)","Nur77(NR)","OCT:OCT-short(POU,Homeobox)","OCT:OCT(POU,Homeobox,IR1)","OCT:OCT(POU,Homeobox)","OCT:OCT(POU,Homeobox)","Oct11(POU,Homeobox)","Oct2(POU,Homeobox)","OCT4-SOX2-TCF-NANOG(POU,Homeobox,HMG)","Oct4:Sox17(POU,Homeobox,HMG)","Oct4(POU,Homeobox)","Oct6(POU,Homeobox)","Olig2(bHLH)","Otx2(Homeobox)","p53(p53)","p53(p53)","p53(p53)","p63(p53)","p73(p53)","PAX3:FKHR-fusion(Paired,Homeobox)","PAX5(Paired,Homeobox),condensed","PAX5(Paired,Homeobox)","PAX6(Paired,Homeobox)","Pax7(Paired,Homeobox),long","Pax7(Paired,Homeobox),longest","Pax7(Paired,Homeobox)","Pax8(Paired,Homeobox)","PBX1(Homeobox)","PBX2(Homeobox)","Pbx3(Homeobox)","Pdx1(Homeobox)","PGR(NR)","Phox2a(Homeobox)","Phox2b(Homeobox)","Pit1(Homeobox)","Pit1+1bp(Homeobox)","Pitx1:Ebox(Homeobox,bHLH)","Pitx1(Homeobox)","Pknox1(Homeobox)","PPARa(NR),DR1","PPARE(NR),DR1","PR(NR)","PRDM1(Zf)","PRDM10(Zf)","PRDM14(Zf)","PRDM15(Zf)","PRDM9(Zf)","Prop1(Homeobox)","PSE(SNAPc)","Ptf1a(bHLH)","PU.1-IRF(ETS:IRF)","PU.1:IRF8(ETS:IRF)","PU.1(ETS)","RAR:RXR(NR),DR0","RAR:RXR(NR),DR5","RARa(NR)","RARg(NR)","RBPJ:Ebox(?,bHLH)","Rbpj1(?)","REST-NRSF(Zf)","Reverb(NR),DR2","RFX(HTH)","Rfx1(HTH)","Rfx2(HTH)","Rfx5(HTH)","Rfx6(HTH)","Ronin(THAP)","RORa(NR)","RORg(NR)","RORgt(NR)","RUNX-AML(Runt)","RUNX(Runt)","RUNX1(Runt)","RUNX2(Runt)","RXR(NR),DR1","SCL(bHLH)","SCRT1(Zf)","SF1(NR)","Six1(Homeobox)","Six2(Homeobox)","Six4(Homeobox)","Slug(Zf)","Smad2(MAD)","Smad3(MAD)","Smad4(MAD)","Snail1(Zf)","Sox10(HMG)","Sox15(HMG)","Sox17(HMG)","Sox2(HMG)","Sox21(HMG)","Sox3(HMG)","Sox4(HMG)","Sox6(HMG)","Sox7(HMG)","Sox9(HMG)","Sp1(Zf)","Sp2(Zf)","Sp5(Zf)","SPDEF(ETS)","SpiB(ETS)","Srebp1a(bHLH)","Srebp2(bHLH)","STAT1(Stat)","Stat3(Stat)","Stat3+il21(Stat)","STAT4(Stat)","STAT5(Stat)","STAT6(Stat)","STAT6(Stat)","T1ISRE(IRF)","TATA-Box(TBP)","Tbet(T-box)","Tbox:Smad(T-box,MAD)","Tbr1(T-box)","Tbx20(T-box)","Tbx21(T-box)","Tbx5(T-box)","Tbx6(T-box)","Tcf12(bHLH)","Tcf21(bHLH)","Tcf3(HMG)","TCF4(bHLH)","Tcf7(HMG)","Tcfcp2l1(CP2)","TCFL2(HMG)","TEAD(TEA)","TEAD1(TEAD)","TEAD2(TEA)","TEAD3(TEA)","TEAD4(TEA)","TFE3(bHLH)","Tgif1(Homeobox)","Tgif2(Homeobox)","THRa(NR)","THRb(NR)","THRb(NR)","Tlx?(NR)","TR4(NR),DR1","TRPS1(Zf)","Twist(bHLH)","Twist2(bHLH)","Unknown-ESC-element(?)","Unknown(Homeobox)","USF1(bHLH)","Usf2(bHLH)","VDR(NR),DR3","WT1(Zf)","X-box(HTH)","YY1(Zf)","Zac1(Zf)","ZBTB12(Zf)","ZBTB18(Zf)","ZBTB33(Zf)","ZEB1(Zf)","ZEB2(Zf)","Zfp281(Zf)","ZFP3(Zf)","Zfp57(Zf)","Zfp809(Zf)","ZFX(Zf)","Zic(Zf)","Zic2(Zf)","Zic3(Zf)","ZKSCAN1(Zf)","ZNF136(Zf)","ZNF143|STAF(Zf)","ZNF16(Zf)","ZNF165(Zf)","ZNF189(Zf)","Znf263(Zf)","ZNF264(Zf)","ZNF317(Zf)","ZNF322(Zf)","ZNF341(Zf)","ZNF382(Zf)","ZNF41(Zf)","ZNF415(Zf)","ZNF416(Zf)","ZNF467(Zf)","ZNF519(Zf)","ZNF528(Zf)","ZNF652","ZNF669(Zf)","ZNF675(Zf)","ZNF692(Zf)","ZNF7(Zf)","ZNF711(Zf)","ZNF768(Zf)","ZSCAN22(Zf)"],[18.23,7.37,0,0.01,5.05,0.38,0.74,1.18,1.61,0.02,6.19,3.36,6.06,4.5,5.13,8.29,23.76,17.75,0.03,0.12,7.36,0.18,0,9.35,0.31,0.16,0.09,18.25,0.01,0.61,0.26,0.02,0.66,1.92,0.07,0,2.72,0,0,6.84,0.08,6.15,3.82,0.58,0.48,0.13,0.05,3,3.95,3.64,0.3,0.18,0.06,0,0.39,0.97,0.09,0.27,0.2,0.05,0,0.8,1.09,1.16,0,0.01,1.1,1.11,0.74,4.05,0.51,2.19,0.01,0.41,0.2,0.2,0.49,0.71,0.69,6.44,48.08,4.24,4.83,5.06,6.85,3.94,4.83,8.7,11.14,0.02,6.51,0,0.16,0,2.05,3.28,3.99,4.55,7.22,6.24,7.3,3.03,3.42,4.73,6.77,9.76,10.54,3.85,0.07,2.49,1.93,4.22,4.59,1.47,4.92,0.48,8.06,10.35,3.51,2.04,10.97,11.67,10.13,9.95,8.53,0.12,5.53,337.23,0,0.19,447.76,460.35,2.35,0.18,443.85,452.97,465.56,0,6.52,0.8,1.14,0.05,0.15,1.7,2.72,1.5,1.27,0.06,1.16,0.11,0.98,0.02,0.04,0.15,0.13,4.87,0.35,0.31,0.05,0.01,0.01,0.03,1.26,0.02,0.01,0.09,1.19,0.04,0.71,0,0.28,0.08,0.29,0,0.86,0.83,1.94,0.97,1.35,1.09,1.39,1.22,0.02,0.79,11.23,8.75,4.01,66.14,15.39,31.54,58.76,57.07,63.19,56.02,21.78,0.15,0.28,0.13,1.96,1.48,0,0.25,4.81,2.75,0.6,4.12,0.09,9.98,0.97,0.08,0.57,0.52,5.78,8.26,0.02,0.02,19.86,3.54,0.49,1.96,4.37,0.01,4.02,4.23,9.58,20.24,8.35,1.54,12.58,0.86,0.24,12.27,6.66,0.09,1.02,0.1,25.26,0.03,0.44,0.56,0.41,0.9,0.11,0.08,1.18,0,0,1.65,2.91,27.33,0.52,0,0.08,0.29,0,0.21,0.7,0.05,0.11,0.05,0.53,3.83,0,0.42,0.34,0.34,0.21,0.4,0.26,0.52,0.59,0.23,0,0.02,0.02,0.28,2.02,0.55,2.2,0.37,0.61,0.08,0.1,0,0,1.01,0,1.55,0.16,0.46,0.03,0.12,0.01,0.28,0.28,0.09,0.05,0.09,0.31,1.33,0.64,7.94,0,0.84,0,0,0.01,0,0.01,2.28,3.68,1.22,2.21,1.31,2.07,3.65,0.03,0.21,0.31,9.01,5.57,4.02,5.12,0.24,3.38,0,0,0.16,0.28,0.08,0,1.94,3.28,1.51,0,0.02,0.01,0.02,0,0,0,0.01,0,0.16,0.19,16.14,59.8,45.82,0.76,7.45,4.98,2.75,0.21,0.17,0.67,1.43,1.15,0.12,0.18,0.23,0.07,4.49,1,2.47,2.81,2.45,1.57,1.34,2.74,8.23,0.01,6.92,0.23,0.34,0,0.05,0.03,0.04,0.04,0.03,2.44,6.02,11.34,0.92,0.58,0,5.85,0.06,369.76,0.32,4.26,2.05,0.12,0.66,2.58,0.3,1.43,2.99,1.29,0.76,0.02,3.01,5.16,0,0,0.81,0.42,0.69,0.82,2.14,1.17,1.36,0.81,0.15,0.06,0.22,0.52,0.09,0,1.71,0.15,0.5,0.03,0.18,0.43,0.04,0.14,0,2.14,0.4,0,0.39,0.47,0.09,1.32,0.21,2.66,0.59,0.61],[2.37,4.11,0.51,0.77,1.98,7.04,0.04,1.67,91.46,2.12,6.3,4.88,4.66,1.27,5.34,222.58,1.01,2.24,0,0.41,4.41,3.87,0.05,195.82,0.16,0.17,0,1.73,1,0.83,0.09,0.81,0.08,6.26,0.01,0,2.24,0,0,1.23,0.48,0.19,0.9,0.6,0.19,0,0,2.41,8.54,0,0.01,1.6,0,0,2.86,5.04,2.98,3.59,0.03,0.03,0.16,0.02,0,0.08,4.69,3.14,0.1,0.14,0.47,0.6,0.09,0.01,0,0.03,0.14,0.02,0.45,0.31,0.24,1.02,0,0.01,0.81,0.2,0.07,0.08,0.11,6.93,1.33,2.37,5.3,0,0.2,0.05,4.62,0.01,0,2.2,1.62,0.78,2.56,1,5.28,1.33,2.09,6.1,4.07,2.67,0.01,2.33,1.72,2.13,1.6,0.46,1.39,1.17,3.12,2.49,2.05,0.9,9.32,2.46,1.07,5.6,4.05,0,0.4,0.16,1.32,0.66,0.03,0.07,0.01,0.25,0.08,0.09,0.04,0.62,0.05,0,0.28,0.17,0.53,0.96,0.25,1.86,0.32,0,7.78,45.69,48.25,0.53,0.1,0.48,0.03,7.25,0.14,0.03,0,0,0.02,0.03,1.84,0.27,0.21,0,0.09,0.13,0.72,0.02,0.48,0.22,0.26,0.32,0.32,0.13,0.84,0.76,0.35,0.18,0.47,1.13,1.62,0.17,3.51,4.47,3.99,0,0,0,0,0,0,0,0,0.01,4.23,5.39,3.66,5.69,0,0.06,4.39,0.05,0.25,0.27,0,0.13,24.9,27.11,27.47,20.16,6.1,0.65,0,0,1.03,0.6,0.47,0.2,2.86,0,8.5,290.02,217.71,0.95,109.27,0.58,78.04,0.59,3,0.03,10.27,0.25,0.19,0.1,0,0,0.06,0.01,0,4.18,0,0.38,0.46,0,0,0,0.01,0.88,1.76,0.92,0.56,1.83,0.01,0.06,0,0,0.1,0.15,0.69,221.88,0,0.17,0.01,0.01,0,0.04,0.1,0.11,1.78,0.74,0.01,0.02,0.01,3.9,0.65,0.02,0.79,1.15,0.95,0.4,0.11,0,0.44,1.96,0,0.18,0.02,0.2,0.29,0,0.92,0.18,0.84,0.69,0.25,0.12,59.85,1.56,0.56,0.11,0,0.11,0,0,5.44,1.24,1.4,0,3.7,4.18,3.02,0.61,0.86,0.08,17.22,19.57,15.37,0.48,0.16,0.04,0.03,0.69,63.62,0.6,0.01,0.19,0.12,0.01,7.47,0.25,1.04,0.52,3.95,0.21,0.33,0.63,0.25,0,0.14,0.42,0.52,0.4,0.23,0,0,0,1.06,0.23,0.06,0.29,0,0.9,1.94,2.4,0,0.22,0.14,0.04,0.48,0.65,0.91,2.08,4.21,0.91,3.93,3.96,1.13,16.85,0,172.33,0.01,0.16,0.01,0.13,0.54,0.16,0.31,0.08,0.23,14.23,34.58,0.02,0.55,0,41.81,0,0.06,4.78,151.79,22.44,0.14,0.45,0.52,0.08,0.86,7.01,0,0.14,0.17,0,0.08,1.92,0.65,0.82,0,0.16,0.36,0.07,23.83,19.14,24.54,0.48,0.17,0,0.56,1.9,1.01,1,1.69,1.15,1.35,0.1,2.04,0.1,0.04,1.79,0.65,0.06,0.04,0.24,0.51,3.27,0,2.71,0,1.16,1.77],[2.23,50.99,0.58,1.36,100.63,1.86,3.44,0.55,43.78,15.87,23.13,27.86,49.64,16.17,32.65,84.56,6.11,24.18,0.01,1.41,46.34,10.44,5.2,110.96,0.08,0,0.25,4.08,0,0.28,0.06,0.48,0.18,26.5,0.04,0.02,3.99,0.86,4.44,10.09,3.1,38.41,9.53,0.76,1.38,12.29,13.84,3.14,33.85,0,0.56,0.17,0,0,1.35,2.45,4.2,3.73,0.01,0.03,0.27,0.11,0.31,2.24,0,17.24,0.14,0.03,0.52,1.2,0.38,0.02,21,9.84,10.83,24.67,0.04,1.56,0.19,13.52,6.47,9.4,12.59,12.06,5.2,13.66,15.96,3.12,0.21,0.29,44.95,0.84,1.1,1.61,5.05,6.22,8.37,33.72,45.04,29.96,42.94,27.4,41.26,26.57,36.85,48.25,45.34,7.45,0.02,9.85,7.07,9.43,9.15,8.99,14.83,5.16,19.76,14.91,12.23,8.07,25.24,20.07,19.07,53.42,52.42,0.95,26.24,3.23,0.66,1.02,5.29,7.23,0.81,0,4.83,10.28,5.69,0.4,0.05,0,0,0.04,0,1.85,12.1,16.09,0.01,0,4.4,18.38,0,0.49,0.02,0.99,0.04,22.08,0.08,0.08,0.26,0.01,0.01,0.09,2.93,1.22,4.14,0.13,0.12,3.63,0.35,4.21,1.17,2.39,0.26,4.66,7.43,8.09,2.38,1.47,0.91,0.96,0.05,1.93,3.34,0.17,38.94,48.99,9.21,3.65,1.11,4.36,5,4.77,2.69,2.95,0.24,0.01,1.52,2.8,2.89,3.57,0,2.54,5.06,1.04,2.48,2.13,0.11,3.26,14.75,16.2,14.48,21.23,0.62,8.68,0.76,0.47,1.81,70.92,0.89,55.76,67.01,0.03,0.01,59.06,84.95,9.21,22,17.63,27.13,5.12,14.33,0.97,28.85,0,10.64,10.05,0,0,0.3,0.01,0.03,2.95,0.3,0,0.64,0.06,0.22,0.06,0.13,6.66,3.4,0.75,0,0.09,0.03,0.1,0.33,0.04,0.16,0,0,54.85,0,0.2,0.24,0.24,1.3,0.45,0.04,0.28,0.65,0,1.13,0.02,0.11,0.57,0.39,1.96,0,3.25,11.29,0.62,0.35,0.01,0,31.84,0.21,0,0.94,0.54,1.47,0.52,0.71,0.01,0.4,0.06,0.62,0,23.58,1.21,1.22,8.7,0,5.63,3.96,0,6.44,23.42,0.01,2.49,0.88,0.31,1.27,0,6.22,0.01,5.16,8.96,13.12,0.69,0.38,0.09,0.23,2.44,35.02,0,0.23,1.18,2.03,2.23,0,1.65,6.26,0.81,0,0.48,0.31,0.51,0.37,0,0.29,0.54,1.03,0.41,0.21,0.18,8.27,7.77,19.15,1.82,2.12,1.79,6.35,9.37,11.68,11.2,6.91,4.85,4.93,1.67,0.17,0.02,0.09,0.59,2.23,0.05,0,0,72.74,117.82,0,93.76,0,1.86,0.25,41.67,44.17,23.34,39,35.82,3.57,4.17,3.11,4.28,5.51,1.56,8.72,0.66,3.57,3.74,91.11,4.01,0.85,2.3,5.19,0.45,2.79,1.12,0,0.38,0.05,70.62,0.01,0,0,0.65,0.49,0.04,4.11,4.85,0.34,0.73,6.3,0.32,0.17,1.21,0.5,1.15,2.82,2.64,0.04,1.77,1.46,4.89,0.05,0.26,0.32,2.12,1.31,2.51,0,0.16,0.05,0.07,0.49,3.36,4.78,1.26,1.58],[1.02,31.25,6.45,6.31,2.02,1.95,0.17,1.21,0.62,1.88,4.18,4.9,38.9,0.79,5.52,0.01,1.13,7.59,4.42,1.52,31.04,1.83,1.96,0,3.54,5.65,0.61,1.34,0,0.7,3.88,2.76,0.34,6.56,2.29,0.01,0.89,19.56,26.86,1.11,0.3,8.64,0.06,0.44,5.4,9.72,12.95,1.6,8.64,0,0.02,0,0.02,0.02,0.91,1.18,2.89,2.48,0.22,0.17,0,0.52,0.22,4.2,8.2,3.19,0.11,0.2,0.27,0.55,0.37,0.14,10.01,0.63,5.36,5.69,0.21,0.74,0.09,9.51,24.28,2.95,12.7,4.6,10.53,0.24,0.07,4.86,5.57,2.15,7.98,5.35,29.36,21.84,0.12,0.23,1.12,3.17,2.95,3.87,9.66,1.88,8.06,2.92,4.02,37.87,32.32,37.33,0.05,29.68,32.3,31.7,27.77,9.53,15.99,2.3,19.19,17.35,18.36,27.99,12.09,25.12,17.41,40.36,33.9,3.73,4.11,0.51,3,4.25,13.2,13.74,1.35,0.24,8.35,11.1,11.3,0.29,0.55,0,0.27,0,0,2.02,1.75,1.95,26.91,0,1.68,2.15,6.77,0.01,0.81,0.26,0,6.13,49.81,62.84,5,0.19,0.31,0.55,1.5,22.09,18.62,1.18,17.28,24.36,2.58,6.11,8.06,22.39,7.03,27.43,5.82,5.23,1.98,0.64,0.76,0.19,1.4,0.31,2.82,0.89,28.47,40.05,3.87,30.59,4.58,18.02,19.37,26.24,27.72,26.37,5.61,0.08,2.21,2.45,4.91,2.7,0,1.02,2.11,0.09,0.47,0.26,0.08,6.84,2.09,1.23,1.8,4.31,0,6.14,0.02,0.08,0.75,4.26,0.94,0.76,4.38,0.31,5.3,0.05,0,1.32,27.76,7.17,42.21,1.85,4.88,1.79,8.93,0.06,2.31,2.57,0.38,3.6,6.91,4.34,3.37,5.94,0.15,2.3,1.04,9.13,10.03,0,0.01,1.56,1.89,0.5,2.2,1.59,0.01,3.55,1.48,0.99,0.81,3.09,1.69,0.01,0,0.98,0.87,0.87,3.2,0.69,0.82,2.2,4.37,2.33,0.07,0.22,0.35,1.7,0.66,4.86,0.64,3.71,2.59,0.91,0.01,0.1,2.86,0.78,0.16,1.32,7.7,4.52,0.03,0.98,0.15,0.14,0.17,0,0.77,0,2.31,0.01,0.25,2.72,0,0.86,3.51,0,0.15,4.53,0,2.66,1.74,1.57,1.94,1.06,4.1,0.03,1.18,2.1,1.4,0.67,0.54,0.03,0.12,5.83,0.44,1.26,2.58,1.6,6.18,0.94,0.56,1.09,3.09,0.53,2.6,0.29,0.5,0.24,0.29,0.02,0.05,1.4,0.01,0.08,0.36,3.04,25.43,14.34,3.93,0.3,0.26,0.02,4.79,6.03,3.28,3.61,5.69,2.02,4.32,0.27,1.25,5.26,0.31,6.08,0.13,4.62,6.91,7.6,1.88,0.48,0.02,0,0.24,6.3,0.04,107.01,119.39,84.43,108.27,115.13,4.9,0.07,1.45,1.9,2.46,2.61,14.36,1.05,8.53,0.05,0,2.22,15.32,8.78,9.15,0.14,5.77,0.76,0,0.04,0,0.24,0.04,15.57,14.38,2.67,0.27,0.5,3.94,1.06,1.23,1.72,0.86,0.1,0.51,0.44,0.51,0.44,0.71,1.2,0.34,0.02,2.88,0.18,0.33,0.41,0.12,6.55,2.3,1.34,0.75,0.89,0.13,1.11,0.95,0.06,0.35,0.11,0.74],[3.07,6.39,2.69,2.32,2.58,3.05,7.96,0.14,0.33,0.04,0.11,0.17,12.23,0.04,0.32,1,3.07,2.75,0.01,2.02,10.87,12.4,0.36,0.42,6.39,7.28,1.18,2.23,0.12,0.32,2.15,0.43,1.37,0.28,3.52,0,0.24,0.02,0.21,0.01,0.06,2.35,0,0.02,4.92,25.75,58.02,1.02,0.11,0.29,0.04,0.08,0.81,0.05,6.91,9.95,14.24,6.46,3.32,2.97,0,1.38,0.24,5.27,0,0.08,0,0.03,0.26,0.52,1.08,0.29,29.88,4.08,1.71,4.18,0.8,0.06,0,0.03,0.61,0.07,0.08,0.16,0.19,0.07,0.13,16.98,1.23,2.04,0.13,64.97,29.05,18.91,0.56,0,0.06,0.63,0.02,0.06,0.16,0.01,0.08,0,0.05,15.04,3.31,6.47,0,2.1,2.48,1.33,3.1,0.3,3.81,0.05,4.91,2.19,1.34,2.69,1.63,4.72,2.28,14.23,6.42,8.27,0.5,0.01,0.1,0.26,0,0,1.23,0.04,0,0,0,0.07,0.88,0,0.08,0.03,0.08,6.38,16.39,14.98,0.71,0.59,1.08,0.03,4.28,2.16,1.01,0.71,0.04,2.19,191.96,252.5,251.59,0.37,1.27,0.74,0.39,0.77,1.34,0.67,1.49,0.34,2.95,1.18,0.94,2.1,0.22,4.33,2.04,1.35,1.11,0.01,0.13,0.04,1.07,0.01,2.55,0.16,2.27,10.98,0.06,0.74,0.21,1.18,1.27,0.83,0.04,0.21,1.08,6.13,16.13,13.32,17.16,21.35,0,3.34,6.26,5.05,3.18,4.1,0.23,1.48,1.31,0.32,0.66,2.47,0.01,7.9,0.1,0.71,2.51,0.62,0.77,1.34,0.88,0.26,7.78,0.34,0.37,5.62,28.85,1.38,30.42,0.98,0.59,2.26,3.01,0.02,1.69,0.65,0.03,0.06,0.31,0.03,0.08,18.53,0.56,1.2,2.15,0.12,0.9,0.01,0.25,2.41,21.12,0.78,1.38,0.11,0,0.93,0.77,0.79,0.02,0.2,0.18,0.51,0.15,0.06,0.33,0.33,0.38,0.88,1.64,0.21,18.06,12.8,2.66,0.11,1.53,5.65,0.9,7.44,1.64,9.01,7.96,1.13,2.68,0.71,5.83,0.12,0.03,2.23,129.72,71.57,2.96,0.17,0.11,0,0.47,0.1,1.4,0.01,0.02,0,0.12,0.01,0,1.39,52.12,0,0.92,0.47,0,1.05,0.18,0.78,0.52,0,1.09,0,1.11,2.68,3.16,1.11,0.2,0.13,0.48,72.44,4.19,0.19,0.09,1.01,0.16,0.6,0,0.49,1.34,0.49,0,3.2,1.53,0.75,0.99,0.06,2.16,2.42,1,0.47,2.77,1.25,1.15,3.97,0.08,0.01,0.16,0.46,1.79,1.68,3.1,3.46,3.64,2.34,3.41,0.02,0.01,0.03,0.37,3.68,0.05,0.04,0,0.31,1.68,0.73,3.82,0.18,0.97,1.56,1.46,10.08,9.32,9.56,12.51,10.68,9.13,0.2,0.78,5.93,3.03,21.05,6.76,33.03,0,0.07,0.03,3.07,0.64,12.38,11.05,1.8,3.16,0.93,0,0.41,0,0.33,0.18,0,0.24,0.4,0,0.03,0.61,0.31,1.67,3.29,2.21,0.16,0.01,0.02,0,0.11,0.12,3.67,0.28,0.38,1.82,0,0.02,0.02,1.45,1.84,0.17,0.24,1.27,10.23,0.3,0.06,0.11,0.5,0.32,0.44,0.43],[5.12,8.38,1.57,3.09,10.01,0.06,0.01,0.27,7.68,6.33,8.39,8.28,10.67,3.2,10.14,7.83,2.17,2.56,0.03,0.2,7.12,1.72,0,7.04,4.69,3.27,1.62,5,0.28,1.08,19.31,0.1,2.28,4.76,4.2,0.25,1.88,0.09,0.62,10.12,0.19,4.81,1.88,2.01,5.95,0,0.03,5.53,6.65,0,0.02,1.34,0.02,0.04,0.1,0.11,0.81,0.24,0.24,0.28,0.49,0.25,0.4,7.14,0.05,4.91,0.09,0.42,0.38,0.08,0.42,0.22,0.05,11.67,5.91,14.34,0.07,2.84,0.87,178.41,4.65,113.26,207.94,254.06,248.68,60.37,50.03,0.28,3.21,0.02,199.12,0,0.65,0.07,7.06,40.97,109.49,165.34,244.42,197.6,223.01,146.57,149.61,81.17,208.03,10.27,5.69,0.91,0.19,0.07,0.02,0.83,0.34,0.16,1.4,0.01,1.39,2.55,0.51,0,12.43,2.6,1.91,11.08,7.65,0.1,206.2,0.58,1.06,0.3,1.48,0.98,0.29,0.02,0.37,0.35,0.6,0.02,0.14,0.16,0.15,0.86,1.48,3.52,0.18,0.28,0.22,0,2.1,4.73,0.17,0.35,0.04,0.81,0,2.23,0.02,0.09,0.13,0.03,0.03,0.26,4.02,1.06,1.21,0.53,0.29,0.72,0.4,0.41,1.18,1.15,0,1.01,3.63,1.46,5.84,66.14,53.64,86.77,24.44,177.5,0.11,20,3.8,8.51,3.87,13.88,0.94,7.68,11.04,6.74,10.02,9.8,2.34,0.12,1.39,0.47,0.73,0.98,0,1.35,0.79,0,1.43,0.01,0.52,8.64,16.53,21.51,25.53,18.48,2.34,19.51,0.36,2.35,3.3,6.75,0.47,3.6,6.51,1.2,0.93,5.4,7.26,1.12,0.34,0.04,0.78,1.72,6.61,0.42,6,11.44,17.13,26.21,0.75,0,0.09,0,0.68,0.08,1.3,1.12,0.41,0,0,0.27,0.31,1.44,2.51,0.16,1.75,0.2,0.02,27.77,25.75,2.67,0.39,17.79,16.05,1.71,0,0.68,0.29,0.29,0.26,0.01,0.54,0.41,4.59,2.96,0.56,0.52,0.02,2.91,0.66,0.09,1.96,0.55,1.57,0.01,1.02,0.28,0.24,0.04,0,2.54,0.32,0.38,0.49,11.86,0.01,0.16,0,0.08,0.17,0.96,4.48,103.54,196.21,341.36,0,0.42,0,0,0.91,0.01,0.03,1.2,2.77,1.49,1.64,0.41,0.77,0.75,0.36,0.58,0.23,44.6,31.89,33.05,29.71,1.15,5.13,0.39,0,1.01,1.29,0.6,1.34,0.2,0.32,1.29,0.88,1.65,0.2,0.33,0.35,0.29,1.48,0.63,1.75,0.59,1.09,4.94,12,13.63,49.86,377.92,0.97,0.42,0.51,0.07,0.27,0.11,0.3,0.37,0.54,8.8,2.74,9.36,0.84,1.59,0.27,3.12,4.61,2.2,8.24,7.46,0.02,5.1,0.08,1.85,0.12,0,0,0,0,0,10.59,5.46,6.36,0.27,1.17,0,1.31,0.28,1.04,0.98,5.1,1.93,0.17,10.11,11.79,0.34,4.17,2.47,0.04,1.47,0,1.71,0.27,0,0,2.04,0,0.38,0.35,0.84,0.9,1.41,2.03,0.16,0,0.03,0.16,1.15,0.65,0.01,0.01,2.66,0.32,0.27,1.96,0.05,0.11,0,4.15,0.13,0.61,0.17,0.1,0.66,1.94,0.31,1.51,0.48,0.39],[4.18,4.11,0.66,2.09,14.02,11.49,2.07,1.26,3.37,2.31,5.13,9.16,3.66,0.28,9.43,3.71,3.99,3.87,4.36,1.16,4.98,1.59,0.16,2.52,0.88,12.17,9.65,2.78,0,0.81,0.12,0.13,1.21,7.63,2.94,0.61,10.15,0.04,1.89,0.08,0.02,1.26,0.25,1.73,9.29,16,27.92,0.53,11.14,0.8,0.42,0.01,0.06,0,0.91,1.88,3.25,2.67,0.88,0.61,0,0.04,0.59,2.31,0,1.31,0.07,0.03,0.01,0.01,0.02,0,28.01,1.76,3.07,3.54,0.01,0.04,0,0,0,0,0,0,0,0,0,2.09,10.26,2.63,0.01,9.32,43.26,41.97,0.04,0.02,0.03,0.09,0,0,0,0,0.04,0,0,4.37,4.47,1.35,0.13,0.41,0.21,2.77,0.84,0.19,4.53,0.91,6.15,3.53,7.19,0.02,2.12,6.47,2.64,4.38,4.81,4.58,0,1.38,0.64,0.22,47.99,46.99,0.76,0,49.94,55.89,52.16,0.02,0,0,0.03,0,0.02,0.69,5.29,5.37,0.62,0.01,7.37,1.09,24.3,1.93,1.56,3.69,0.02,0.71,0.35,0.16,1.21,0,0,0.04,0.28,1.13,1.3,0.03,1.48,3.06,0.17,2.4,0.24,2.1,0.2,2.37,0.9,0.95,0.43,0.13,0.04,0.08,0.51,0,0.6,0.26,3.82,5.42,1.12,0,0,0,0,0,0,0,0,0.89,1.74,3.34,1.07,5.83,0,7.42,1.95,0.24,0.56,0.15,1.97,0.01,235.13,274.73,268.26,246.68,16.05,35.6,2.93,0.03,3.38,6.47,0.61,5.48,11.17,1.23,0.72,1.42,1.47,2.35,75.48,2.78,64.28,2.82,3.01,0.44,1.78,0.28,2.74,0.32,0.01,1.99,6.05,5.9,1.25,0.35,6.25,5.92,2.02,7.07,9.3,0.13,0.01,2.52,10.81,0.01,0.11,1.15,0,0.25,0.08,0.03,0.12,0.07,0.01,0.34,0,0.19,0.39,0.39,1.11,0.57,0.85,0.97,2.11,0.24,0.3,0.01,0.16,2.91,3.09,0.94,1.76,1.36,4.43,0.14,0.04,0.05,0.05,0.64,0,0.49,8.35,7.02,0.37,0.04,0.09,0.34,0.11,0,0.13,0.25,1.3,0.03,0.01,0.01,0.01,1.94,8.94,0,1.31,1.79,0.32,1.73,0.8,0.56,1.01,0.46,3.92,0.06,5.35,6.17,3.88,0.63,0.09,0.06,0.09,10.02,3.74,0.26,7.05,7.3,8.06,3.06,0,0.35,5.73,2.08,0,1.23,0.77,1.01,1.21,0.18,1.67,1.22,0.57,0.5,0.08,0.01,0,0.01,0,0.09,0.8,0.48,4.6,9.96,11.4,8.45,5.39,2.1,2.01,0,2.46,8.91,1.17,5.48,6.85,6.41,9.92,7.77,7.21,7.93,0.07,1.6,0.83,0.61,0.51,12.51,19.76,12.13,22.24,18.53,2.95,22.97,33.12,18.66,15.4,13.58,30.63,1.5,42.96,2.24,0.48,1.05,1.52,13.5,15.91,1.99,0.06,1.66,0,1.48,0.01,0.22,0.32,0,0,0.27,0,0.08,2.14,0.54,0.02,0.79,0.54,0.19,0.18,0.1,0.78,0.32,0.02,0.51,0.18,1.96,2.41,0.46,0.3,0.07,0.03,2.4,0.34,0.75,0.04,0.43,0.41,1.76,0.04,2.96,0.38,0.04,0.25],[2.41,1.48,3.27,3.34,0.28,1.22,0.65,0.24,0.01,0.02,4.59,2.73,1.6,14.49,3.49,0.18,3.86,2.03,0,0.07,0.64,4.56,0,0.11,0.22,0.66,0.09,1.66,0,0.06,0.05,1.26,0.36,1.57,0.63,0,0.38,0.1,0.21,24.23,0.72,62.93,9.81,0.48,1.23,31.56,48.9,0.32,1.52,0.29,0.34,0,35.02,56.72,1.44,1.44,1.76,1.19,0.42,0.3,0.21,0.45,2.1,1.26,0.01,0.05,0.06,0.14,0.57,0.5,0.69,0.33,38.07,0.45,0.46,2.52,1.06,0.36,0.14,0.54,0,0.37,0.18,0.11,0.21,0.65,0.86,3.14,1.46,0.57,0.44,46.17,20.79,17.08,1.53,0.03,0.2,0.32,0.1,0.79,0.21,0.58,0.11,0.24,0.55,3.39,0.6,44.25,0.05,37.78,34.57,47.12,32.26,9.29,14.02,2.29,24.29,16.41,10.36,35.78,11.9,21.19,14.12,1.89,1.24,17.65,0.55,1.67,0.26,0.61,9.33,11.96,1.9,0.01,10.86,11.12,11.33,0.93,0.17,0,0.15,0.01,0.23,1.51,1.76,3.27,0.22,0.01,1.74,0.05,4.24,0.74,0.02,0.91,0.06,22.58,53.85,76.35,137.28,63.23,48.42,1.94,2.89,0.48,0.63,1.3,1.89,0.53,1.72,1.21,1.08,0.68,0.27,0.67,1.82,1.63,0.78,1.17,1.13,1.63,1.23,1.04,0.84,1.52,0.5,1.34,1.15,0,0.02,0.14,0.01,0,0,0,0.03,4.47,1.38,2.13,3.99,3.18,0,5.44,0.87,1.73,0.53,2.14,0,0.13,0.99,1.06,2.01,1.38,0.14,7.1,0.04,1.34,1.15,0.3,1.19,0.49,0.1,0,0.89,0.03,0.05,5.09,20.03,13.59,14.38,2.49,1.92,1.58,27.09,0.05,0.3,0.1,2.68,0,0.01,0,0.01,3.81,0.01,0.8,0.23,1.65,2.73,0.05,0.05,3.49,18.12,0.35,0.36,0.52,0.01,0.1,0.09,0.04,0.14,0.01,0.03,0.31,0.01,0.58,1.02,1.02,0.05,1.65,1.37,0.81,0.21,0.47,0.13,0.31,2.37,0.06,0.14,4.29,0.23,4.82,3.18,0.01,0.05,0.12,1.08,0.93,0.06,0.2,86.9,56.68,0.88,0.43,0.1,1.13,0.01,0.1,0.02,0.03,0,0.77,1.48,0,0.63,0.45,41.51,0.01,1.39,4.49,0.16,1.94,0.05,0.75,0.22,0,1.19,0.02,13.24,12.09,7.5,0.13,0,0,0,57.15,0.15,0.12,0.28,0.92,1.64,0.15,0,0.08,0.9,0.15,0.01,0.66,0.57,0.12,0.38,0,0.65,1.23,0.29,0.07,0.47,0.1,0.28,1.64,0.08,0.02,2.38,1.56,0.48,0.9,0.98,1.09,0.3,0.06,0.22,0.18,0,0.47,0.21,0.92,0.72,0.82,1.05,1.95,0.31,0.79,3.47,0.01,2.24,2.42,2.62,0.39,0.17,0.33,0.34,1,1.08,1.15,3.82,5.01,2.78,22.55,3.01,30.34,10.61,0.32,0,0.59,0.01,2.6,6.11,3.89,1.19,0.44,0.01,0.24,0,0.45,0.07,0.33,0.13,3.4,0.21,0.1,0.4,0.1,0.54,0.66,0.56,0.04,0.16,0,0.55,0.56,0.34,0.81,1.3,0.09,1.02,0.25,0.1,0.08,0.39,2,0.25,0.52,0.12,3.19,0.12,1.77,0.08,0.38,0.2,0.73,0.85],[1.85,2.19,0.22,0.13,23.61,4.16,2.52,0.78,4.82,1.07,10.88,18,2.78,1.05,11.81,14.76,1.29,1.09,0.05,3.73,1.69,2.73,0.65,16.27,0.2,0,0,4.71,0.05,0.1,6.09,1.54,0.92,13.72,0.02,0,1.34,0.08,0.08,0.45,0.89,3.48,1.96,0.94,0.09,0,0,5.02,13.98,0,0.38,0.37,1.61,0.86,12.01,12.57,15.03,17.43,0.01,0.02,0.3,0.39,0.1,0.15,0,0.51,0.2,0.05,0.17,0.06,0.79,0.1,0,3.52,3.66,1.02,0.03,0.92,0,0,0,0,0,0,0,0,0,18.31,0.21,0.67,0,0,0.12,0,0.1,0.23,0,0.01,0,0,0,0,0,0,0,3.37,1.45,4.04,0.35,2.08,2.52,6.2,1.84,1.01,5.25,2.61,8.06,8.67,3.55,1.87,4.73,7.28,7.26,2.7,2.45,0.28,0,0.91,0.54,0,0.03,0.09,0.01,0,0,0.01,0.01,0.27,0.47,0,0.34,0.57,0.48,1.14,3.45,6.48,0.05,0,0.66,0.61,32.09,0.01,0,0.07,0.01,1.98,0.56,0.16,0.21,1.05,0.76,1.03,3.73,0.63,0.64,0.42,2.26,0.19,1.5,0.09,2.88,2.6,3.39,1.74,1.05,0.45,1.5,0.65,0.21,0.41,0.12,0.15,5.46,0.22,1.53,2.71,9.84,0,0,0.91,0,0,0,0,0,8.63,21.97,27.6,17.81,17.97,0.05,0.84,2.61,0.03,0.18,0.19,0,2.39,3.25,2.3,4.72,3.2,0.94,2.08,0,0,2.97,14.26,0.17,11.32,13.75,0,7.45,7.93,13,0.71,82.69,3.12,26.97,1.1,4.18,0.19,5.81,0.01,0.02,0.04,0.38,0,0.2,0.27,0.02,17.96,0,0,1.23,0,0,0.02,0.11,1.18,0.25,3.57,0.75,2.92,0,3.63,1.05,6.45,6.64,4.57,10.44,4.1,0,0.14,0,0,0.08,0.56,0.03,0.33,1.97,1.34,0.66,0.1,0.17,1,0.94,0.27,0.38,2.01,5.75,0.47,1.07,0.73,3.12,0.12,0.05,0.97,0,0.24,1.02,0.29,0.38,0.47,0.62,0.15,0.59,0.07,0.28,0.03,0.58,0.01,0,0.08,0,0,1.15,2.25,0.04,1.19,20.25,23.33,17.89,7.9,15.8,0.12,10.18,8.73,5.96,0.25,0.01,0.03,0.02,0.35,15.75,0,0,0.04,0.01,0.04,0,0.31,7.4,0.37,0,150.05,119.39,99.63,135.59,117.74,142.1,120.95,109.57,56.68,143.66,0,0.27,0.21,0,0.01,1.52,1.31,0.1,0.6,0.8,1.8,0.76,2.73,3.44,0.14,1.26,0,0.77,1.76,3.91,0,0,0,16.34,21.02,0.1,10.74,5.16,0.48,0.16,0.61,0.28,0.2,0.39,0.08,0.9,4.14,4.53,1.56,0.61,0,13.46,0,0,0.03,12.29,0.62,0.91,0.73,1.97,0,1.72,16.12,0,2.59,0,2.06,0.21,0,0,8.19,0,0.28,0.59,7.03,0.58,0.63,1.81,0.5,2.93,1.47,0.4,0.51,1.4,1.19,0.65,0.18,1.68,0.85,0.11,0.53,0.01,5.97,4.55,0.64,0.03,1.18,0.14,2.9,0.32,3.29,4.8,0.64,0.5],[3.92,47.08,0.73,1.73,1.39,1.12,0.02,0.47,2.96,2.36,7.77,8.56,53.16,1.72,9.92,2.04,4.46,18.18,0.01,0.95,46.9,1.08,0.96,1.53,0.3,0.28,0,1.95,0,0.52,0.02,0.44,1.07,8.49,0.76,0,0.69,113.74,163.73,1.13,1.12,1.62,1.65,0.54,0.88,4.61,14.97,3.56,11.16,0.08,0,0,0,0.08,1.24,0.72,2.22,3.69,1.13,0.49,0.19,1.31,0.25,1.29,2.85,4.48,0.88,0.23,0.01,0.53,0.6,0.14,6.41,1.36,0.86,0.99,0.21,0.14,0.65,47.42,83.56,21.85,48.16,33.37,44.78,7.72,6.65,1.13,5.25,0.03,34.04,13.37,6.93,5.63,2.71,0.8,10.13,6.05,25.2,25.09,30.11,21.56,20.2,14.17,19.98,52.77,50.55,17.88,0.04,18.9,17.38,20.46,15.25,6.27,8.45,0.58,15.31,10.64,11.15,14.75,13.07,15.78,12.2,56.15,51.16,3.21,26.01,0.45,1.14,0.2,21.55,20.92,0.35,0.02,15.43,20.38,24.65,0.09,0.29,0,0.06,0,0.03,0.34,0.39,1.6,2.24,0,0.13,3.99,1.99,0.02,0.01,0.2,0.01,3.85,7.05,8.14,58.5,0.66,0.63,0.13,3.04,120.26,100.02,0.48,66.23,129.81,0.75,7.32,43.29,98.02,19.74,108.4,0.33,0.47,4.1,0.24,0.47,0.81,0.15,0.39,0.42,0.04,42.6,54.76,4.07,91.38,20.44,28.67,69.1,91.11,122.99,93.72,19.36,3.24,0.57,0.97,0.54,0.93,0,1.32,1.85,0.33,0.23,0.64,0,3.7,0.6,0.22,0.16,0.49,1.75,1.24,0,0.25,4.31,5.3,1.21,1.86,4.32,0.03,5.35,1.68,0.78,5.92,13.42,2.51,5.65,3.14,0.22,1.53,6.36,0,2.02,1.06,1.13,0,0.06,0.07,0.01,1.36,0,0.47,2.54,0.96,0.99,0.03,0.75,5.01,7.57,0.01,0.55,0.21,0,0.14,0.28,0.01,3.48,0.06,0.01,0.19,0,0.38,0.24,0.24,0.17,0.59,0.18,1.35,0.92,0.02,0.04,0.05,0.24,1.09,1.64,3.43,2.74,0.76,1.34,0.21,0.03,0,0,0.66,0.78,1.98,25.37,22.3,0.49,0.21,0,0.01,0.54,0.1,0.03,0.31,2.17,1.26,0.92,5.93,0,0.88,11.68,0,0.43,1.21,0.02,2.28,4.69,5.96,5.37,5.54,2.23,0,2.39,2.24,0.48,0.26,0.26,0.3,0.06,24.75,1.62,1.77,2.28,0.05,1.11,0.07,0.41,0.52,0.24,0.67,1.65,8.72,5.11,6.93,6.19,4.21,5.95,5.29,8.81,1.98,1.76,3.9,53.07,21.93,23.55,0.49,2.03,0.41,2.46,6.18,4.08,6.05,1.55,0.11,0.51,0.01,13.98,4.21,0.35,5.11,0,5.55,11.43,8.88,4.08,2.17,1.03,0.66,1.21,0.24,0.08,0.87,1.68,0.32,0.77,1.91,1.24,3.35,3.84,0.34,0.54,1.7,0.82,13.83,13.47,0.26,0.25,1.12,73.83,1.46,0.89,1.88,1.49,6.64,0.13,0.13,0.16,0.2,0.21,5.22,5.82,0.05,0,1.31,1.35,1.05,1.81,2.2,1.6,0.02,0.03,0.06,0.34,0.25,0.22,3.89,0.26,0.43,1.5,0.96,0.73,0.07,0.95,1.01,2.15,0.3,0.87,0.55,1.66,0.1,0.74,0.09,0.68,0.64,0.24],[2.96,16.4,0.06,0.07,5.18,4.04,0.13,1.84,14.77,1.83,1.75,1.09,22.13,3.66,1,47.12,3.35,13.37,0,0.32,18.68,0.03,0.49,42.79,0.29,0.11,0.03,2.58,0,0.37,1.97,3.49,0.11,1.82,0.2,0.01,2.04,0,0,2.13,0.54,3.34,0.87,1.25,1.01,0,0,0.13,1.6,0,0.28,0,0,0.04,0.08,0.06,1.8,1.09,0.02,0.02,0,0.45,0.07,0.36,0,2.55,0.05,0.17,1.35,0.41,0.29,0,0,1.03,0.56,0.87,0.58,43.69,14.51,0,0.01,0,0.01,0,0,0,0.01,0.01,6.97,0.02,0,0,0,0,0.18,0.15,0,0.47,0.04,0,0,0,0.05,0,0,26.12,19.22,0.88,0,0.01,0,0.07,0.03,0.11,0.01,0.4,0.02,0.66,0.06,0,1.74,0.26,0.08,22.67,21.72,0,0,0.12,0.52,0.24,0.02,0.04,0.17,0.02,0.27,0.05,0.08,1.1,0.03,0,0.18,0.66,1.24,2.71,1.24,2.76,0.08,0,0.56,6.68,7.55,0.27,0.04,0.35,0,20.39,0.01,0,0,0.02,0,0.06,0.69,0.01,0.01,0.01,0,0,0,0.31,0,0.05,0,0,1.27,0.37,0.21,0.57,1.34,0.46,0.12,0.67,0,0.87,20.88,16.9,0.4,0,1.25,10.76,0,0,0,0.03,0.05,0.11,0.16,0.55,0,0.31,0.23,0.05,1.03,0.02,0.06,0.47,0.01,3.4,60.28,96.8,80.13,60.54,2.12,0.24,0.01,0.02,3.86,4.91,0.68,5.49,5.55,0,0,41.74,53.51,4.91,26.15,0.87,4.48,0.38,0.89,0.85,22.71,0.24,0.56,1.06,0,0,0.88,0.03,0,0,0.05,0.1,2.53,0,0,0,0,2.23,0,1.49,2.56,0.09,0,1.28,0.37,0.13,0.49,1.88,3.46,44.56,0,0,0,0,0.74,0.01,0.2,0.27,0.06,0.02,0.22,0.02,0.25,0.07,0.07,0.07,0.39,0.34,3.79,0.16,0.04,0.55,1.3,1.08,0.4,0.06,0,0,0.01,0.06,0.16,0.01,0.08,1.51,0.17,0,10.01,1.05,0.65,0.03,0,0,0,0,4.22,6.83,2.38,0.21,44.91,31.78,46.64,7.5,6.16,0.01,2.02,1.99,1.5,0.28,0,0.02,0.03,0,22.45,0.01,0,0.33,0.38,0.42,0.01,2.28,2.5,4.57,0,0,0,0,0,0,0,0,0,0,0.05,0,0.06,1.87,0.05,1.21,0.73,0.5,0.58,0.63,1.95,2.02,1.44,2.73,3.06,0,1.61,6.42,2.79,10.51,0.79,6.57,3.72,5.06,5.52,11.37,0.03,39.59,0.06,0.91,0,0.13,0.2,0.32,0.35,0.12,0.55,3.84,8.18,0,0,0,4.87,0,0.09,0.28,37.31,0.04,0,0.82,0.8,0,35.13,34.52,0,0,0,0.22,0.01,0,0,4.82,0.53,1.42,2.08,0,0.08,0.02,0.11,0.35,0.01,0.03,0.02,1.43,0.73,5.89,0.01,0.49,0.23,1.87,0.36,0.33,0.39,0.52,1.87,0.09,2.01,0.89,0.11,1.04,0.03,0.8,0,1.19,0.21],[10.2,1.97,0.12,0.12,33.11,0.03,0.01,0.04,46.69,61.63,10.2,10.51,1.41,3.61,11.29,24.93,2.29,2.62,0.22,0.02,1.8,0.06,0,37.94,4.21,1.18,0.44,8.69,0,1.52,1.66,0.01,1.4,8.03,3.19,0.45,0.22,0,0.41,5.78,0.02,0.8,2.44,1.72,3.74,0,0.01,6.05,8.31,0,0.23,0.01,0.02,0.01,0,0,0.13,0.1,0.62,0.26,0,0.02,0.13,3.45,24.49,62.84,0.08,0.09,1.07,0.47,0.45,0.15,0.03,0.67,0.87,0.34,0.39,0.47,0,91.94,0.46,86.25,83.82,127.55,59.19,103.54,108.31,0,9.23,0.06,190.26,0,0.06,0.01,21.23,95.85,81.6,176.06,189.48,187.09,195.43,185.83,167.85,158.21,209.42,1.5,1.45,3.31,0.03,1.7,1.08,5.92,3.54,2.39,5.46,0,6.82,9.79,5.05,0.9,19.69,8.72,6.74,1.35,1.45,0.03,155.78,1.69,4.08,2.57,5.86,5.85,0.63,0.41,9.21,7.63,10.58,0,0.03,0.01,0.38,0.15,0.72,1.5,0.68,0.99,0.01,0,1.12,60.5,1.45,0.66,0.06,0.32,0,1.36,0.14,0.02,0.31,0,0,0.31,0.67,0.5,0.06,0.25,0.03,0.52,0.02,0.04,0.31,0.15,0.01,0,0.26,0.6,5.69,11.4,13.86,11.45,8.32,17.64,0,10.48,1.96,1.5,5.08,5.29,0,3.82,2.33,0.24,2.51,1.84,0.23,34.24,0,0,0,0,0.03,0.56,0.05,0.32,0.41,0.14,0.37,10.86,1.53,1.47,3.71,5.62,6.96,6.95,0.02,1.3,7.27,33.71,1.36,24.26,33.81,0.28,0,12.69,19.92,1.51,2.12,0.05,0.15,2.95,1.46,0.21,6.17,1.18,3.03,3.11,0.09,0,0.23,0,1.35,0,0.08,1.62,1.58,0,0,0.01,0,1.49,0.25,0.21,0.47,0.07,0.01,0.37,1.13,0.57,0.79,0.81,1.29,9.89,0,1.77,0,0,0,0.02,0.04,0.35,0.09,0.01,0.02,0.15,0.31,0.04,0.24,0.08,1.44,0.1,1.15,0.02,0.02,0,0.01,1,0,1.8,0.17,0.25,0.05,2.38,0,0.04,0.01,0.92,0.03,0.1,51.2,8.57,9.53,98.45,0,0.03,0,0,0.73,0.1,0.04,0.29,1.73,0.75,1.88,1.69,4.42,0.65,11.07,20.49,23.17,82.08,68.31,90.25,71.92,0.05,12.79,1.31,0,1.37,0.54,1.2,24.86,0.66,2.06,1.28,25.22,10.6,2.71,0.99,2.22,3.07,7.91,6.22,1.71,0.52,2.22,1.95,6.29,9.36,49.68,32.18,0.91,0.35,2.27,1.98,4.43,4.6,2.75,1.24,1.35,0.81,0.04,8.44,0.25,2.78,1.32,7.74,17.23,9.19,45.21,40,20.97,26.92,25.24,0.96,9.11,0,0,0,0,0,1.66,10,11.6,0.12,0.14,0,0.26,0.21,8.85,1.59,28.51,6.1,0,6.49,5.75,0.03,3.79,2.05,0.03,1.27,0,23.93,0.03,8.02,8,8.81,0,0.08,1.83,0.42,2.17,5.05,6.7,0.21,0,0.05,0.41,1.42,0,0.71,0.01,1.22,0.12,0.48,0.42,0.24,0.01,0,8.9,0.05,0.2,2.45,0.19,0.37,2.82,2.37,0.04,0.7,0.35],[0.19,0.33,3.26,4.81,1.59,2.7,0,0.01,1.24,0,1.24,0,0.67,1.6,0,0.16,1.34,0.87,0.14,2.92,0.94,2.77,0,0.44,0,0,0,0.01,0,0.95,0,0.03,0.04,0,0.01,0.23,2.93,3.18,0.26,4.52,0.48,6.39,0.88,0,0,5.89,6.16,0,0.01,0,0,0,0.99,0.05,0.2,0.49,0.01,0.04,0,0,1.05,0.22,1.22,0,0.76,0,0.5,0.14,0,0.06,0,0.14,2.93,0.23,6.84,2.01,0.04,0.09,0,0.5,0,0.88,1.02,0.41,3.39,0.63,2.92,0.06,3.46,0,2.69,0.81,0.04,0.01,0.6,0.05,2.19,1.15,2.26,3.25,0.42,3.5,3.19,0.32,0.56,0.6,0.82,0,0.04,0.86,0.36,1.07,1.91,0.01,0.3,0.31,0.66,0.97,0.9,0.13,1.83,1.14,0.48,0.7,1.07,2.07,5.48,1.71,0.01,0.02,2.81,2.23,0.01,0,1.39,3.53,1.28,0.02,0.75,0,0.02,2.43,2.7,6.82,0.14,0.2,0.1,0,0,0,0,0.04,0,0.45,0.29,2.23,0.13,0.31,0,0.05,0,0,0.05,0,0,0.01,2.01,0,0.08,0,0,0.24,0.01,0.11,0,0,0.13,1.27,0.47,0.81,0.12,1.17,0,0.08,0.72,0.9,0,0,0,0,0,0,0,0.01,0,0,0.22,0.22,0.01,0.45,0.02,0,11.82,0,0.03,0.01,0,0,0.1,3.12,0.43,0.75,3.73,0,0.07,0,0.4,5.69,0.04,0.82,2.36,0.2,0.36,0,2.32,0.8,2.53,0.81,0,1.17,0.1,0.31,0.53,0.2,0.11,0,15.12,1.79,0.11,3.48,0.03,0,0,0,0.03,0.52,0.52,0.06,0.11,0.5,0.05,0.01,0.08,0,0,0.24,0.05,0.64,0.11,0.08,0.04,0.83,0,0.11,0,0,0.01,0.01,0.22,0.08,2.86,0.12,0.16,0,0.01,0.04,0.08,0.02,0.05,0.01,0.13,0.04,0.38,0,0.03,0.03,0,0.4,1.72,8.58,0.07,0.25,0.39,0.07,0.01,0.91,0.04,0,2.58,0.58,0.57,0.44,0,0.18,0,0,0.36,0.02,0.06,0,0,0.5,0.02,0.21,0.06,0,1.91,0.16,0.31,2.14,2.85,2.36,1.41,2.17,2.76,1.62,0.06,0.25,0,0.26,0,9.88,8.27,8.12,0,0.35,2.04,0.6,4.63,1.01,3.42,2.12,4.54,4.07,0.01,0.04,0.02,0,1.54,3.93,0.44,0.18,0,0,0.15,0,0,0,0,0.49,0.61,0.04,0.02,0.3,7.88,0.94,0.01,0.12,0.25,0.2,0,1.02,0,0.12,0,0,0,0,0,0,0,3.77,1.87,0.2,0.36,0.03,0,0,0.86,0.01,2.44,0,0,1.21,0,0.01,1.8,0.25,0.2,5.39,0,1.95,1.03,0.16,2.08,0,0,1.15,0,7.95,0,0.01,0.01,0,0.03,0,0.01,0.1,0.01,3.28,1.87,0.08,3.86,0.76,0,1.19,0.31,0,0.05,0.01,0.06,0.09,0.29,0.05,0.03,0,0.43,0.37,0.05]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>motif<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="top-10-motifs-in-each-topic" class="section level3">
<h3>Top 10 motifs in each topic</h3>
<pre class="r"><code>cat(&quot;Number of regions selected for each topic: \n&quot;)
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

colnames_homer &lt;- c(&quot;motif_name&quot;, &quot;consensus&quot;, &quot;P&quot;, &quot;log10P&quot;, &quot;Padj&quot;,  &quot;num_target&quot;, &quot;percent_target&quot;, &quot;num_bg&quot;, &quot;percent_bg&quot;)

top_motifs &lt;- data.frame(matrix(nrow=10, ncol = length(homer_res_topics)))
colnames(top_motifs) &lt;- names(homer_res_topics)
for (k in 1:length(homer_res_topics)){
  homer_res &lt;- homer_res_topics[[k]]
  colnames(homer_res) &lt;- colnames_homer
  homer_res &lt;- homer_res %&gt;% separate(motif_name, c(&quot;motif&quot;, &quot;origin&quot;, &quot;database&quot;), &quot;/&quot;)
  top_motifs[,k] &lt;- head(homer_res$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F, caption = &quot;Top 10 motifs enriched in each topic.&quot;)</code></pre>
<pre><code># Number of regions selected for each topic: 
#   k1   k2   k3   k4   k5   k6   k7   k8   k9  k10  k11  k12  k13 
# 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363 4363</code></pre>
<div id="htmlwidget-fadc6839d9bee756ac78" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fadc6839d9bee756ac78">{"x":{"filter":"none","caption":"<caption>Top 10 motifs enriched in each topic.<\/caption>","data":[[1,2,3,4,5,6,7,8,9,10],["Gata6(Zf)","Gata2(Zf)","Gata4(Zf)","Gata1(Zf)","GATA3(Zf)","TRPS1(Zf)","GATA:SCL(Zf,bHLH)","KLF1(Zf)","KLF5(Zf)","Sp2(Zf)"],["NeuroD1(bHLH)","Atoh1(bHLH)","Olig2(bHLH)","NeuroG2(bHLH)","BHLHA15(bHLH)","TCF4(bHLH)","Twist2(bHLH)","NF1-halfsite(CTF)","Ascl1(bHLH)","NF1(CTF)"],["Tcf21(bHLH)","BHLHA15(bHLH)","Ap4(bHLH)","TCF4(bHLH)","Twist2(bHLH)","NeuroG2(bHLH)","Atoh1(bHLH)","Tcf12(bHLH)","Myf5(bHLH)","ZBTB18(Zf)"],["TEAD1(TEAD)","TEAD4(TEA)","TEAD3(TEA)","TEAD(TEA)","TEAD2(TEA)","HNF1b(Homeobox)","Hnf1(Homeobox)","NF1(CTF)","Fra1(bZIP)","JunB(bZIP)"],["HNF1b(Homeobox)","HNF4a(NR),DR1","Hnf1(Homeobox)","PPARa(NR),DR1","RXR(NR),DR1","PPARE(NR),DR1","Erra(NR)","COUP-TFII(NR)","RARa(NR)","TR4(NR),DR1"],["SpiB(ETS)","PU.1(ETS)","Elf4(ETS)","ELF5(ETS)","ETS1(ETS)","Etv2(ETS)","Fli1(ETS)","ELF3(ETS)","GABPA(ETS)","ERG(ETS)"],["Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Mef2a(MADS)","NF1-halfsite(CTF)","NF1(CTF)","Gata4(Zf)","Gata6(Zf)","GATA3(Zf)","Gata1(Zf)"],["HNF4a(NR),DR1","PPARa(NR),DR1","HNF1b(Homeobox)","HNF6(Homeobox)","CEBP(bZIP)","RXR(NR),DR1","Cux2(Homeobox)","PPARE(NR),DR1","Hnf1(Homeobox)","COUP-TFII(NR)"],["Sox10(HMG)","Sox9(HMG)","Sox3(HMG)","Sox2(HMG)","Sox4(HMG)","Sox15(HMG)","Sox21(HMG)","Sox6(HMG)","Sox17(HMG)","NF1-halfsite(CTF)"],["CDX4(Homeobox)","HOXB13(Homeobox)","KLF5(Zf)","Hoxa11(Homeobox)","Cdx2(Homeobox)","Hoxd13(Homeobox)","Hoxa13(Homeobox)","Hoxd11(Homeobox)","KLF6(Zf)","KLF1(Zf)"],["Mef2b(MADS)","Mef2c(MADS)","Mef2d(MADS)","Mef2a(MADS)","NeuroG2(bHLH)","Atoh1(bHLH)","Rfx2(HTH)","RFX(HTH)","Olig2(bHLH)","Egr1(Zf)"],["Fli1(ETS)","Etv2(ETS)","ERG(ETS)","ETS1(ETS)","ETV1(ETS)","ETV4(ETS)","Ets1-distal(ETS)","EWS:ERG-fusion(ETS)","EWS:FLI1-fusion(ETS)","GABPA(ETS)"],["NFY(CCAAT)","MafA(bZIP)","Smad2(MAD)","PPARE(NR),DR1","Smad3(MAD)","Smad4(MAD)","ZFX(Zf)","Tbx20(T-box)","EBF1(EBF)","GLIS3(Zf)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>rank<\/th>\n      <th>k1<\/th>\n      <th>k2<\/th>\n      <th>k3<\/th>\n      <th>k4<\/th>\n      <th>k5<\/th>\n      <th>k6<\/th>\n      <th>k7<\/th>\n      <th>k8<\/th>\n      <th>k9<\/th>\n      <th>k10<\/th>\n      <th>k11<\/th>\n      <th>k12<\/th>\n      <th>k13<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-motif-enrichment-across-topics" class="section level3">
<h3>Heatmap of motif enrichment across topics</h3>
<p>Clustering motifs by hierarchical clustering (motifs with similar enrichment across topics are plotted together)</p>
<pre class="r"><code>create_motif_enrichment_heatmap(motif_res, cluster_motifs = TRUE, cluster_topics = FALSE, filter_motifs = TRUE, min_enrichment = 50, 
                                max_enrichment = 100, method_cluster = &quot;average&quot;, font.size.motifs = 4, font.size.topics = 9)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-1-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-heatmap-1-1">
Past versions of motif-heatmap-1-1.png
</button>
</p>
<div id="fig-motif-heatmap-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-1-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-1-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-1-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-1-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># 114 out of 439 motifs included the heatmap</code></pre>
<p>Cluster both motifs and topics by hierarchical clustering</p>
<pre class="r"><code>create_motif_enrichment_heatmap(motif_res, cluster_motifs = TRUE, cluster_topics = TRUE, filter_motifs = TRUE, min_enrichment = 50, 
                                max_enrichment = 100, method_cluster = &quot;average&quot;, font.size.motifs = 4, font.size.topics = 9)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-2-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-heatmap-2-1">
Past versions of motif-heatmap-2-1.png
</button>
</p>
<div id="fig-motif-heatmap-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-2-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-2-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-2-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-heatmap-2-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code># 114 out of 439 motifs included the heatmap</code></pre>
</div>
<div id="scatterplots-of-motif-enrichment" class="section level3">
<h3>Scatterplots of motif enrichment</h3>
<pre class="r"><code># Plot enrichment (-log10 p-value) and ranking of the motifs

plots &lt;- vector(&quot;list&quot;, ncol(motif_res$mlog10P))
names(plots) &lt;- colnames(motif_res$mlog10P)

for( i in 1:length(plots)){
  plots[[i]] &lt;- create_motif_enrichment_ranking_plot(motif_res, k = i, 
                                                     max.overlaps = 20, subsample = FALSE)
}

# do.call(plot_grid,plots)</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) and the ranking</li>
</ul>
<p><strong>Topic 1 example</strong></p>
<pre class="r"><code>print(plots[[1]])</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-enrichment-ranking-plot-2-1">
Past versions of motif-enrichment-ranking-plot-2-1.png
</button>
</p>
<div id="fig-motif-enrichment-ranking-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-ranking-plot-2-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot motif enrichment (-log10 p-value) in each topic vs other topics

plots &lt;- vector(&quot;list&quot;, ncol(motif_res$mlog10P))
names(plots) &lt;- colnames(motif_res$mlog10P)

for( i in 1:length(homer_res_topics)){
  plots[[i]] &lt;- create_motif_enrichment_plot(motif_res, k = i, 
                                             max.overlaps = 20, subsample = TRUE)
}

# do.call(plot_grid,plots)</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) in <strong>topic 1</strong> vs other topics</li>
</ul>
<pre class="r"><code>print(plots[[1]])</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-enrichment-plot-2-1">
Past versions of motif-enrichment-plot-2-1.png
</button>
</p>
<div id="fig-motif-enrichment-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-enrichment-plot-2-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="motif-enrichment-vs-gene-score" class="section level3">
<h3>Motif enrichment vs gene score</h3>
<p>Load pre-computed gene scores</p>
<pre class="r"><code>gene.dir &lt;- paste0(out.dir, &quot;/geneanalysis-Cusanovich2018-k=13-TSS-l2&quot;)
cat(sprintf(&quot;Directory of gene analysis result: %s \n&quot;, gene.dir))
genescore_res &lt;- readRDS(file.path(gene.dir, &quot;genescore_result_topics.rds&quot;))

genes &lt;- genescore_res$genes
gene_scores &lt;- genescore_res$Z
rownames(gene_scores) &lt;- genes$SYMBOL
gene_logFC &lt;- genescore_res$beta
rownames(gene_logFC) &lt;- genes$SYMBOL</code></pre>
<pre><code># Directory of gene analysis result: /project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/geneanalysis-Cusanovich2018-k=13-TSS-l2</code></pre>
<p>Get TF genes</p>
<pre class="r"><code>motif_names &lt;- gsub(&quot;\\s*\\(.*&quot;, &quot;&quot;, motif_res$motifs$motif)
gene_names &lt;- genes$SYMBOL

TF_genes &lt;- intersect(toupper(motif_names), toupper(gene_names))
cat(sprintf(&quot;%s TF genes mapped between motif names and gene symbol. \n&quot;, length(TF_genes)))</code></pre>
<pre><code># 250 TF genes mapped between motif names and gene symbol.</code></pre>
<p>Compute correlation between motif enrichment z-score and gene score:</p>
<p><strong>Topic 1 example</strong></p>
<ul>
<li>Compute motif enrichment z-scores from the motif enrichment p-values</li>
<li>Plot motif enrichment (-log10 p-value) and correlation between motif enrichment z-scores and gene scores</li>
<li>Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment z-score and gene scores</li>
</ul>
<pre class="r"><code>motif_gene_mapping &lt;- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 1, cor.motif = &quot;z-score&quot;)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-cor-plot-1-1">
Past versions of motif-cor-plot-1-1.png
</button>
</p>
<div id="fig-motif-cor-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-1-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif_gene_mapping &lt;- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_zscore, decreasing = T)),]
rownames(motif_gene_mapping) &lt;- 1:nrow(motif_gene_mapping)

cat(&quot;Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n&quot;)
print(head(motif_gene_mapping[,c(&quot;motif&quot;,&quot;motif_mlog10P&quot;, &quot;gene_score&quot;, &quot;cor_zscore&quot;)], 10))</code></pre>
<pre><code># Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: 
#          motif motif_mlog10P gene_score cor_zscore
# 1    Gata2(Zf)     460.35215 27.8830209  0.5920248
# 2    Gata6(Zf)     465.56368 -0.6075940  0.2660333
# 3    Gata4(Zf)     452.96914 -0.9032233  0.2450204
# 4     KLF5(Zf)      63.18985 20.2932849  0.4208190
# 5     KLF1(Zf)      66.14305 51.7778237  0.3657063
# 6     KLF3(Zf)      58.76004 23.3799000  0.3629908
# 7     KLF6(Zf)      56.02399 14.6687082  0.3115715
# 8     Klf4(Zf)      57.06629 -3.7795042  0.2882401
# 9      Sp2(Zf)      59.80235 28.4662015  0.2650298
# 10 Bach1(bZIP)      23.76459 26.3022816  0.3642947</code></pre>
<ul>
<li>Plot motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores</li>
<li>Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>motif_gene_mapping &lt;- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 1, cor.motif = &quot;-log10(p-value)&quot;)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-2-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-cor-plot-2-1">
Past versions of motif-cor-plot-2-1.png
</button>
</p>
<div id="fig-motif-cor-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-2-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-cor-plot-2-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif_gene_mapping &lt;- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_mlog10P, decreasing = T)),]
rownames(motif_gene_mapping) &lt;- 1:nrow(motif_gene_mapping)

cat(&quot;Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n&quot;)
print(head(motif_gene_mapping[,c(&quot;motif&quot;,&quot;motif_mlog10P&quot;, &quot;gene_score&quot;, &quot;cor_mlog10P&quot;)], 10))</code></pre>
<pre><code># Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: 
#          motif motif_mlog10P gene_score cor_mlog10P
# 1    Gata2(Zf)    460.352151 27.8830209  0.62996339
# 2    Gata1(Zf)    447.757611 26.7126627  0.56449139
# 3     KLF1(Zf)     66.143050 51.7778237  0.56294117
# 4     KLF5(Zf)     63.189847 20.2932849  0.38552159
# 5      Sp2(Zf)     59.802350 28.4662015  0.39882226
# 6    Gata4(Zf)    452.969145 -0.9032233  0.05110547
# 7  Bach1(bZIP)     23.764594 26.3022816  0.37361028
# 8    KLF14(Zf)     31.538465  0.4537105  0.25921375
# 9    SpiB(ETS)      7.452493 -3.0070640  0.99145723
# 10 Tcf21(bHLH)      8.234223 -0.5385452  0.89029474</code></pre>
<p><strong>GATA family</strong></p>
<pre class="r"><code>GATA_genes &lt;- grep(&quot;^GATA\\d*$&quot;, TF_genes, ignore.case=T, value=T)</code></pre>
<p>Plot GATA motifs in topic 1</p>
<pre class="r"><code>k = 1
selected_motifs &lt;- rownames(motif_res$motifs)[match(toupper(GATA_genes), toupper(motif_names))]
motif.dir &lt;- paste0(homer.dir, &quot;/homer_result_topic_&quot;, k, &quot;/knownResults/&quot;)

for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = &quot;both&quot;)
}</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1-1.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1-2.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1-3.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1-4.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-GATA-logos-1-5.png" width="600" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = &quot;-log10(p-value)&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-1-1">
Past versions of motif-gene-plot-1-1.png
</button>
</p>
<div id="fig-motif-gene-plot-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1ab3d725d51a0c1546d74aacd5547e80d46059cb/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">1ab3d72</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/a29293a8900803a7f7e071410ba855ed9f1a5f7f/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">a29293a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/14eac34a602ea6bb464f6d9939c56ab2d584ffc8/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-1-1.png" target="_blank">14eac34</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Plot motif enrichment (zscore) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = &quot;z-score&quot;, 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-2-1">
Past versions of motif-gene-plot-2-1.png
</button>
</p>
<div id="fig-motif-gene-plot-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-2-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-2-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1fb78db56c388834bd95c571a4a3c8d6f631579e/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-2-1.png" target="_blank">1fb78db</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>KLF/SP family</strong></p>
<pre class="r"><code>KLF_genes &lt;- grep(&quot;^KLF\\d*$&quot;, TF_genes, ignore.case=T, value=T)
SP_genes &lt;- grep(&quot;^SP\\d*$&quot;, TF_genes, ignore.case=T, value=T)</code></pre>
<p>Plot KLF motif logos in topic 1</p>
<pre class="r"><code>k = 1
selected_motifs &lt;- rownames(motif_res$motifs)[match(toupper(KLF_genes), toupper(motif_names))]
motif.dir &lt;- paste0(homer.dir, &quot;/homer_result_topic_&quot;, k, &quot;/knownResults/&quot;)
for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = &quot;both&quot;)
}</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-1.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-2.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-3.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-4.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-5.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-6.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-7.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-KLF-logos-1-8.png" width="600" style="display: block; margin: auto;" /></p>
<p>Plot SP motif logos in topic 1</p>
<pre class="r"><code>k = 1
selected_motifs &lt;- rownames(motif_res$motifs)[match(toupper(SP_genes), toupper(motif_names))]
motif.dir &lt;- paste0(homer.dir, &quot;/homer_result_topic_&quot;, k, &quot;/knownResults/&quot;)
for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = &quot;both&quot;)
}</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-SP-logos-1-1.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-SP-logos-1-2.png" width="600" style="display: block; margin: auto;" /><img src="figure/motif_analysis_Cusanovich2018.Rmd/plot-SP-logos-1-3.png" width="600" style="display: block; margin: auto;" /></p>
<ul>
<li>Plot motif enrichment (-log10 p-value) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores,
                                       motif_names, gene_names,
                                       selected_genes = c(KLF_genes, SP_genes),
                                       y = &quot;-log10(p-value)&quot;,
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid, plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-3-1">
Past versions of motif-gene-plot-3-1.png
</button>
</p>
<div id="fig-motif-gene-plot-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-3-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-3-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Plot motif enrichment (zscore) and gene scores</li>
</ul>
<pre class="r"><code>plots &lt;- create_motif_gene_scatterplot(motif_res, gene_scores,
                                       motif_names, gene_names,
                                       selected_genes = c(KLF_genes, SP_genes),
                                       y = &quot;z-score&quot;,
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)</code></pre>
<p><img src="figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gene-plot-4-1">
Past versions of motif-gene-plot-4-1.png
</button>
</p>
<div id="fig-motif-gene-plot-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1307c0fadfb5d2d48fba933f3cfd74f8129e86a2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-4-1.png" target="_blank">1307c0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/84b236a37e91e1edbb89a04517b10ce1e23056c2/docs/figure/motif_analysis_Cusanovich2018.Rmd/motif-gene-plot-4-1.png" target="_blank">84b236a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] grid      stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
#  [1] Logolas_1.3.1     reshape2_1.4.3    DT_0.16           htmlwidgets_1.5.3
#  [5] plotly_4.9.2.1    cowplot_1.1.0     ggrepel_0.9.0     ggplot2_3.3.3    
#  [9] tidyr_1.1.2       dplyr_1.0.2       fastTopics_0.4-6  Matrix_1.2-18    
# [13] workflowr_1.6.2  
# 
# loaded via a namespace (and not attached):
#  [1] nlme_3.1-140       mcmc_0.9-7         matrixStats_0.57.0 fs_1.3.1          
#  [5] bit64_4.0.5        progress_1.2.2     httr_1.4.2         rprojroot_2.0.2   
#  [9] tools_3.6.1        R6_2.5.0           irlba_2.3.3        lazyeval_0.2.2    
# [13] colorspace_2.0-0   ade4_1.7-16        withr_2.4.0        tidyselect_1.1.0  
# [17] prettyunits_1.1.1  bit_4.0.4          compiler_3.6.1     git2r_0.27.1      
# [21] quantreg_5.75      SparseM_1.78       labeling_0.4.2     scales_1.1.1      
# [25] SQUAREM_2021.1     quadprog_1.5-8     mixsqp_0.3-43      stringr_1.4.0     
# [29] digest_0.6.27      rmarkdown_2.6      MCMCpack_1.4-9     pkgconfig_2.0.3   
# [33] htmltools_0.5.0    invgamma_1.1       rlang_0.4.10       generics_0.1.0    
# [37] farver_2.0.3       jsonlite_1.7.2     crosstalk_1.1.0.1  magrittr_2.0.1    
# [41] Rcpp_1.0.6         munsell_0.5.0      ape_5.4-1          lifecycle_0.2.0   
# [45] CVXR_1.0-9         stringi_1.5.3      whisker_0.4        yaml_2.2.1        
# [49] MASS_7.3-53        Rtsne_0.15         plyr_1.8.6         parallel_3.6.1    
# [53] promises_1.1.1     crayon_1.3.4       lattice_0.20-41    hms_1.0.0         
# [57] knitr_1.30         pillar_1.4.7       seqinr_4.2-5       glue_1.4.2        
# [61] evaluate_0.14      data.table_1.13.4  RcppParallel_5.0.2 vctrs_0.3.6       
# [65] httpuv_1.5.4       MatrixModels_0.4-1 gtable_0.3.0       purrr_0.3.4       
# [69] ashr_2.2-47        xfun_0.19          gridBase_0.4-7     Rmpfr_0.8-2       
# [73] coda_0.19-4        later_1.1.0.1      viridisLite_0.3.0  truncnorm_1.0-8   
# [77] tibble_3.0.5       conquer_1.0.2      gmp_0.6-2          ellipsis_0.3.1</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
