<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Visualize topics in Cusanovich2018 data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Visualize topics in Cusanovich2018 data</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree0d189ac153653fe96d37cb293698e6239ba20c1ctargetblank0d189aca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/0d189ac153653fe96d37cb293698e6239ba20c1c" target="_blank">0d189ac</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree0d189ac153653fe96d37cb293698e6239ba20c1ctargetblank0d189aca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/0d189ac153653fe96d37cb293698e6239ba20c1c" target="_blank">0d189ac</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/plotly/Buenrostro_2018_Chen2019pipeline/

Untracked files:
    Untracked:  analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  scripts/fit_all_models_Buenrostro_2018_chromVar_scPeaks_filtered.sbatch
    Untracked:  scripts/fit_models_Cusanovich2018_tissues.sh

Unstaged changes:
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   scripts/fit_poisson_nmf.sbatch
    Modified:   scripts/prefit_poisson_nmf.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plots_Cusanovich2018.Rmd</code>) and HTML (<code>docs/plots_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0d189ac153653fe96d37cb293698e6239ba20c1c/analysis/plots_Cusanovich2018.Rmd" target="_blank">0d189ac</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-02
</td>
<td>
added bone marrow plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ad13ef48a8a7e8502e6120a8dd9c573586afab8d/docs/plots_Cusanovich2018.html" target="_blank">ad13ef4</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/6d64de69248fc8bf434009bd289812c0c298095a/analysis/plots_Cusanovich2018.Rmd" target="_blank">6d64de6</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-11
</td>
<td>
add t-SNE and membership pie charts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/7c0b14fc8b62f93f7ccf47c567ff831e5b1b111e/docs/plots_Cusanovich2018.html" target="_blank">7c0b14f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/dc70e46c43134e630fc732877740776370ee24b1/analysis/plots_Cusanovich2018.Rmd" target="_blank">dc70e46</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-11
</td>
<td>
add t-SNE and membership pie charts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/5a8479b47f79b5f1091e745359cd0d807e94dcec/docs/plots_Cusanovich2018.html" target="_blank">5a8479b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b9993e24fe14a5e41c2e051e9a66761f3c5d3971/analysis/plots_Cusanovich2018.Rmd" target="_blank">b9993e2</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
initial structure plots
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we examine and compare the topic modeling results for the scATAC-seq dataset from the mouse single-cell atlas paper <a href="scATACseq_data_Cusanovich2018.html">Cusanovich <em>et al</em> (2018)</a></p>
<p>Load the packages used in the analysis below, as well as additional functions that will be used to generate some of the plots.</p>
<pre class="r"><code>library(fastTopics)
library(Matrix)
library(tools)
library(dplyr)
library(ggplot2)
library(cowplot)
library(mapplots)
source(&quot;code/plots.R&quot;)</code></pre>
<div id="cusanovich-2018-mouse-scatac-seq-data-in-all-13-tissues" class="section level2">
<h2>Cusanovich 2018 mouse scATAC-seq data in all 13 tissues</h2>
<div id="load-the-data-and-poisson-nmf-model-fit" class="section level3">
<h3>Load the data and Poisson NMF model fit</h3>
<p>Load the data</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
rm(counts)
# 81173 x 436206 counts matrix.</code></pre>
<p>About the samples: The study measured single cell chromatin accessibility for 17 samples spanning 13 different tissues in 8-week old mice.</p>
<pre class="r"><code>cat(nrow(samples), &quot;samples (cells). \n&quot;)
# 81173 samples (cells).</code></pre>
<p>Tissues:</p>
<pre class="r"><code>samples$tissue &lt;- as.factor(samples$tissue)
cat(length(levels(samples$tissue)), &quot;tissues. \n&quot;)
table(samples$tissue)
# 13 tissues. 
# 
#       BoneMarrow       Cerebellum            Heart           Kidney 
#             8403             2278             7650             6431 
#   LargeIntestine            Liver             Lung PreFrontalCortex 
#             7086             6167             9996             5959 
#   SmallIntestine           Spleen           Testes           Thymus 
#             4077             4020             2723             7617 
#       WholeBrain 
#             8766</code></pre>
<p>Cell types labels:</p>
<pre class="r"><code>samples$cell_label &lt;- as.factor(samples$cell_label)
cat(length(levels(samples$cell_label)), &quot;cell types \n&quot;)
table(samples$cell_label)
# 40 cell types 
# 
#          Activated B cells       Alveolar macrophages 
#                        500                        559 
#                 Astrocytes                    B cells 
#                       1666                       5772 
#             Cardiomyocytes   Cerebellar granule cells 
#                       4076                       4099 
#            Collecting duct                 Collisions 
#                        164                       1218 
#                     DCT/CD            Dendritic cells 
#                        506                        958 
#   Distal convoluted tubule Endothelial I (glomerular) 
#                        319                        552 
#        Endothelial I cells       Endothelial II cells 
#                        952                       3019 
#                Enterocytes              Erythroblasts 
#                       4783                       2811 
#            Ex. neurons CPN          Ex. neurons CThPN 
#                       1832                       1540 
#           Ex. neurons SCPN  Hematopoietic progenitors 
#                       1466                       3425 
#                Hepatocytes           Immature B cells 
#                       5664                        571 
#         Inhibitory neurons              Loop of henle 
#                       1828                        815 
#                Macrophages                  Microglia 
#                        711                        422 
#                  Monocytes                   NK cells 
#                       1173                        321 
#           Oligodendrocytes                  Podocytes 
#                       1558                        498 
#            Proximal tubule         Proximal tubule S3 
#                       2570                        594 
#             Purkinje cells         Regulatory T cells 
#                        320                        507 
#          SOM+ Interneurons                      Sperm 
#                        553                       2089 
#                    T cells         Type I pneumocytes 
#                       8954                       1622 
#        Type II pneumocytes                    Unknown 
#                        157                      10029</code></pre>
</div>
<div id="structure-plots" class="section level3">
<h3>Structure plots</h3>
<p>The structure plots below summarize the topic proportions in the samples grouped by different tissues.</p>
<p><span class="math inline">\(k = 13\)</span>:</p>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=13.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)</code></pre>
</div>
</div>
<div id="visualize-by-structure-plot-grouped-by-tissues" class="section level2">
<h2>Visualize by Structure plot grouped by tissues</h2>
<pre class="r"><code>set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)
samples$tissue &lt;- as.factor(samples$tissue)

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;tissue&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:13,colors = colors_topics,
                              num_threads = 4,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/plots_Cusanovich2018.Rmd/structure-plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="t-sne-plot" class="section level3">
<h3>t-SNE plot</h3>
<p>Plot samples by tissues on the t-SNE coordinates provided by the original paper:</p>
<pre class="r"><code>colors_tissues &lt;- c(&quot;BoneMarrow&quot; = &quot;darkblue&quot;,
                    &quot;Lung&quot; = &quot;lightblue&quot;,
                    &quot;Thymus&quot; = &quot;orange&quot;,
                    &quot;Spleen&quot; = &quot;pink&quot;,
                    &quot;Liver&quot; = &quot;purple&quot;,
                    &quot;Testes&quot; = &quot;yellow&quot;,
                    &quot;SmallIntestine&quot; = &quot;darkgreen&quot;,
                    &quot;LargeIntestine&quot; = &quot;brown&quot;,
                    &quot;Kidney&quot; = &quot;green&quot;,
                    &quot;Heart&quot; = &quot;red&quot;,
                    &quot;WholeBrain&quot; = &quot;gray&quot;,
                    &quot;PreFrontalCortex&quot; = &quot;black&quot;,
                    &quot;Cerebellum&quot; = &quot;darkgray&quot;)

colors_tissues &lt;- colors_tissues[order(names(colors_tissues))]
dat &lt;- samples[, c(&quot;tsne_1&quot;, &quot;tsne_2&quot;, &quot;tissue&quot;)]

ggplot(dat, aes(x = tsne_1, y = tsne_2, colour=tissue)) + 
      geom_point(size = 0.5, na.rm = TRUE) + 
      scale_colour_manual(values=colors_tissues) +
      labs(x = &#39;t-SNE 1&#39;, y = &#39;t-SNE 2&#39;, title = &#39;&#39;) + 
      theme_cowplot()</code></pre>
<p><img src="figure/plots_Cusanovich2018.Rmd/tSNE-1.png" width="840" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-topic-membership-using-pie-charts-on-t-sne-coordinates" class="section level3">
<h3>Visualize topic membership using pie-charts on t-SNE coordinates</h3>
<p>Visualize topic membership using pie-charts for cells on t-SNE coordinates provided by the original paper</p>
<p>Plot the subset of 2000 samples:</p>
<p><span class="math inline">\(k = 13\)</span>:</p>
<pre class="r"><code>idx_selected_samples &lt;- sample(nrow(fit$L),2000)

plot(NA,NA, main = paste(&quot;K = 13&quot;), 
     xlab = &#39;t-SNE 1&#39;, ylab = &#39;t-SNE 2&#39;, 
     xlim=c(min(samples$tsne_1),max(samples$tsne_1)),
     ylim=c(min(samples$tsne_2),max(samples$tsne_2)))

res &lt;- lapply(idx_selected_samples, function(r)
  add.pie(z=as.integer(100*fit_multinom$L[r,]),
          x=samples[r,&quot;tsne_1&quot;],
          y=samples[r,&quot;tsne_2&quot;], labels=c(&quot;&quot;,&quot;&quot;,&quot;&quot;),
          radius = 1, col = colors_tissues))</code></pre>
<p><img src="figure/plots_Cusanovich2018.Rmd/topic-membership-pies-tSNE-1-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-topic-membership-pies-tSNE-1-1">
Past versions of topic-membership-pies-tSNE-1-1.png
</button>
</p>
<div id="fig-topic-membership-pies-tSNE-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7c0b14fc8b62f93f7ccf47c567ff831e5b1b111e/docs/figure/plots_Cusanovich2018.Rmd/topic-membership-pies-tSNE-1-1.png" target="_blank">7c0b14f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cusanovich-2018-mouse-scatac-seq-data-in-bone-marrow" class="section level2">
<h2>Cusanovich 2018 mouse scATAC-seq data in bone marrow</h2>
<div id="load-the-data-and-poisson-nmf-model-fit-1" class="section level3">
<h3>Load the data and Poisson NMF model fit</h3>
<p>Load the data</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018_BoneMarrow.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
rm(counts)
# 8403 x 436206 counts matrix.</code></pre>
<p>About the samples: The study measured single cell chromatin accessibility for 17 samples spanning 13 different tissues in 8-week old mice.</p>
<pre class="r"><code>cat(nrow(samples), &quot;samples (cells). \n&quot;)
# 8403 samples (cells).</code></pre>
<p>Cell types labels:</p>
<pre class="r"><code>samples$cell_label &lt;- as.factor(samples$cell_label)
cat(length(levels(samples$cell_label)), &quot;cell types \n&quot;)
table(samples$cell_label)
# 18 cell types 
# 
#          Activated B cells       Alveolar macrophages 
#                         17                          1 
#                    B cells                 Collisions 
#                        300                        210 
#            Dendritic cells Endothelial I (glomerular) 
#                        112                          4 
#       Endothelial II cells                Enterocytes 
#                          3                          1 
#              Erythroblasts  Hematopoietic progenitors 
#                       2758                       3047 
#           Immature B cells                Macrophages 
#                        560                        124 
#                  Monocytes                   NK cells 
#                        827                         43 
#                  Podocytes         Regulatory T cells 
#                          3                         51 
#                    T cells                    Unknown 
#                         72                        270</code></pre>
</div>
<div id="structure-plots-1" class="section level3">
<h3>Structure plots</h3>
<p>The structure plots below summarize the topic proportions in the samples grouped by different cell labels.</p>
<p><span class="math inline">\(k = 7\)</span></p>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018_BoneMarrow&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=7.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)

set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)

samples$cell_group &lt;- as.character(samples$cell_label)
samples$cell_group[samples$cell_group %in% names(which(table(samples$cell_group) &lt; 10))] &lt;- &quot;others&quot;
table(samples$cell_group)

samples$cell_group &lt;- factor(samples$cell_group, levels = c(&quot;Hematopoietic progenitors&quot;, 
                                                            &quot;Erythroblasts&quot;,
                                                            &quot;B cells&quot;, &quot;Immature B cells&quot;,&quot;Activated B cells&quot;, 
                                                            &quot;T cells&quot;, &quot;Regulatory T cells&quot;, &quot;NK cells&quot;,&quot;Dendritic cells&quot;, 
                                                            &quot;Monocytes&quot;,&quot;Macrophages&quot;))

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;cell_group&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:7,colors = colors_topics,
                              num_threads = 6,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/plots_Cusanovich2018.Rmd/BoneMarrow-structure-plot-k7-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code># 
#         Activated B cells                   B cells                Collisions 
#                        17                       300                       210 
#           Dendritic cells             Erythroblasts Hematopoietic progenitors 
#                       112                      2758                      3047 
#          Immature B cells               Macrophages                 Monocytes 
#                       560                       124                       827 
#                  NK cells                    others        Regulatory T cells 
#                        43                        12                        51 
#                   T cells                   Unknown 
#                        72                       270</code></pre>
<p><span class="math inline">\(k = 8\)</span></p>
<pre class="r"><code>fit.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018_BoneMarrow&quot;
fit &lt;- readRDS(file.path(fit.dir, &quot;/fit-Cusanovich2018-scd-ex-k=8.rds&quot;))$fit
fit_multinom &lt;- poisson2multinom(fit)

set.seed(10)
colors_topics &lt;- c(&quot;#a6cee3&quot;,&quot;#1f78b4&quot;,&quot;#b2df8a&quot;,&quot;#33a02c&quot;,&quot;#fb9a99&quot;,&quot;#e31a1c&quot;,
                   &quot;#fdbf6f&quot;,&quot;#ff7f00&quot;,&quot;#cab2d6&quot;,&quot;#6a3d9a&quot;,&quot;#ffff99&quot;,&quot;#b15928&quot;,
                   &quot;gray&quot;)
idx_selected_samples &lt;- sample(nrow(fit$L),4000)

samples$cell_group &lt;- as.character(samples$cell_label)
samples$cell_group[samples$cell_group %in% names(which(table(samples$cell_group) &lt; 10))] &lt;- &quot;others&quot;
table(samples$cell_group)

samples$cell_group &lt;- factor(samples$cell_group, levels = c(&quot;Hematopoietic progenitors&quot;, 
                                                            &quot;Erythroblasts&quot;,
                                                            &quot;B cells&quot;, &quot;Immature B cells&quot;,&quot;Activated B cells&quot;, 
                                                            &quot;T cells&quot;, &quot;Regulatory T cells&quot;, &quot;NK cells&quot;,&quot;Dendritic cells&quot;, 
                                                            &quot;Monocytes&quot;,&quot;Macrophages&quot;))

p.structure &lt;- structure_plot(select(fit_multinom,loadings = idx_selected_samples),
                              grouping = samples[idx_selected_samples, &quot;cell_group&quot;],
                              n = Inf,gap = 40, perplexity = 50,
                              topics = 1:8,colors = colors_topics,
                              num_threads = 6,verbose = FALSE)

print(p.structure)</code></pre>
<p><img src="figure/plots_Cusanovich2018.Rmd/BoneMarrow-structure-plot-k8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code># 
#         Activated B cells                   B cells                Collisions 
#                        17                       300                       210 
#           Dendritic cells             Erythroblasts Hematopoietic progenitors 
#                       112                      2758                      3047 
#          Immature B cells               Macrophages                 Monocytes 
#                       560                       124                       827 
#                  NK cells                    others        Regulatory T cells 
#                        43                        12                        51 
#                   T cells                   Unknown 
#                        72                       270</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.1 (2019-07-05)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] tools     stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
# [1] mapplots_1.5.1    cowplot_1.1.1     ggplot2_3.3.3     dplyr_1.0.3      
# [5] Matrix_1.2-18     fastTopics_0.4-29 workflowr_1.6.2  
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.1      Rcpp_1.0.6         lattice_0.20-41    tidyr_1.1.2       
#  [5] prettyunits_1.1.1  rprojroot_2.0.2    digest_0.6.27      R6_2.5.0          
#  [9] MatrixModels_0.4-1 evaluate_0.14      coda_0.19-4        httr_1.4.2        
# [13] pillar_1.4.7       rlang_0.4.10       progress_1.2.2     lazyeval_0.2.2    
# [17] data.table_1.13.6  irlba_2.3.3        SparseM_1.78       whisker_0.4       
# [21] rmarkdown_2.6      labeling_0.4.2     Rtsne_0.15         stringr_1.4.0     
# [25] htmlwidgets_1.5.3  munsell_0.5.0      compiler_3.6.1     httpuv_1.5.4      
# [29] xfun_0.19          pkgconfig_2.0.3    mcmc_0.9-7         htmltools_0.5.1.1 
# [33] tidyselect_1.1.0   tibble_3.0.6       quadprog_1.5-8     matrixStats_0.58.0
# [37] viridisLite_0.3.0  withr_2.4.1        crayon_1.4.0       conquer_1.0.2     
# [41] later_1.1.0.1      MASS_7.3-53        grid_3.6.1         jsonlite_1.7.2    
# [45] gtable_0.3.0       lifecycle_0.2.0    DBI_1.1.0          git2r_0.27.1      
# [49] magrittr_2.0.1     scales_1.1.1       RcppParallel_5.0.2 stringi_1.5.3     
# [53] farver_2.0.3       fs_1.3.1           promises_1.1.1     ellipsis_0.3.1    
# [57] generics_0.1.0     vctrs_0.3.6        glue_1.4.2         purrr_0.3.4       
# [61] hms_1.0.0          yaml_2.2.1         colorspace_2.0-0   plotly_4.9.3      
# [65] knitr_1.30         quantreg_5.83      MCMCpack_1.5-0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
