<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Process data for Buenrostro et al (2018) scATAC-seq dataset</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Process data for Buenrostro <em>et al</em> (2018) scATAC-seq dataset</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee1e23296afa0b5fb02de1915320d345443564234targetblanke1e2329a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/e1e23296afa0b5fb02de1915320d345443564234" target="_blank">e1e2329</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee1e23296afa0b5fb02de1915320d345443564234targetblanke1e2329a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/e1e23296afa0b5fb02de1915320d345443564234" target="_blank">e1e2329</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/motif_analysis_Buenrostro2018_v2.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  paper/
    Untracked:  scripts/postfit_Buenrostro2018_v2.sbatch

Unstaged changes:
    Modified:   analysis/analysis_Buenrostro2018_k10.Rmd
    Modified:   analysis/assess_fits_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/clusters_Cusanovich2018_k13.Rmd
    Modified:   analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/motif_analysis_Cusanovich2018.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   analysis/process_data_Buenrostro2018_different_options.Rmd
    Modified:   scripts/postfit_Buenrostro2018_v2.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/process_data_Buenrostro2018.Rmd</code>) and HTML (<code>docs/process_data_Buenrostro2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/e1e23296afa0b5fb02de1915320d345443564234/analysis/process_data_Buenrostro2018.Rmd" target="_blank">e1e2329</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-11
</td>
<td>
Only kept the data processing option 1 (from the original paper)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/bd24884333b8cdaebf663bf0c93d5f34b736b971/analysis/process_data_Buenrostro2018.Rmd" target="_blank">bd24884</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-11
</td>
<td>
wflow_rename(&quot;analysis/process_data_Buenrostro2018.Rmd&quot;, &quot;analysis/process_data_Buenrostro2018_different_options.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/e1b1e0f7b5b34c67dc26a993cdfc47e8a1efce9d/docs/process_data_Buenrostro2018.html" target="_blank">e1b1e0f</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0be72fdff7a8eaa210f6b58a227b82679f90808b/analysis/process_data_Buenrostro2018.Rmd" target="_blank">0be72fd</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
output data processing messages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/95a1145e6c2e1280c1b8199b3258128241dc8073/docs/process_data_Buenrostro2018.html" target="_blank">95a1145</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/478d9226daca9ad22dc18febe2d547002aa15ed2/docs/process_data_Buenrostro2018.html" target="_blank">478d922</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/3890d389a01fdc0fc92080e5aedf40d3a9dca006/analysis/process_data_Buenrostro2018.Rmd" target="_blank">3890d38</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
wflow_publish(&quot;analysis/process_data_Buenrostro2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/17fc08a917f8d5e94ad1f01eb759474862f7f74f/docs/process_data_Buenrostro2018.html" target="_blank">17fc08a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0c32e6ac7fa0f7e0ddd0e5a9c3eb78254b5f2f72/analysis/process_data_Buenrostro2018.Rmd" target="_blank">0c32e6a</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
wflow_publish(&quot;analysis/process_data_Buenrostro2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/04cb7673cf037042835f1261e1ebcb15b4c4e736/docs/process_data_Buenrostro2018.html" target="_blank">04cb767</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/97c421dcd6a971b87fad64ea010e9032c892af76/analysis/process_data_Buenrostro2018.Rmd" target="_blank">97c421d</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-06
</td>
<td>
put all four data processing options together
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/694fd3b6a6d4cd50b0b97af5f6f99b3252cb155f/docs/process_data_Buenrostro2018.html" target="_blank">694fd3b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/4b7d5faa1e1b0bd86ed0a6df22240335b52aa845/analysis/process_data_Buenrostro2018.Rmd" target="_blank">4b7d5fa</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-19
</td>
<td>
process data with peaks in less than 1% samples filtered out
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/d5b411d2d0713a8d7ce10e76cde7d22316c16eb9/docs/process_data_Buenrostro2018.html" target="_blank">d5b411d</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ee608fdb7bdc1775f74bcc81a9b28eea1aa899f1/analysis/process_data_Buenrostro2018.Rmd" target="_blank">ee608fd</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-18
</td>
<td>
process counts for aggregated single-cell peaks using chromVAR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/594fd86c6639a42167492e74bc21802c3fe1eb06/docs/process_data_Buenrostro2018.html" target="_blank">594fd86</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/614814b2f797674f4bc404f3e986aba9adccc387/analysis/process_data_Buenrostro2018.Rmd" target="_blank">614814b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
process Buenrostro2018 data using Downloaded scATAC-seq processed data from GEO
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c113844e38826d4646b6c8d07a00050215a4d021/analysis/process_data_Buenrostro2018.Rmd" target="_blank">c113844</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
wflow_rename(&quot;analysis/process_data_Buenrostro2018.Rmd&quot;, &quot;analysis/process_data_Buenrostro2018_Chen2019pipeline.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/c113844e38826d4646b6c8d07a00050215a4d021/docs/process_data_Buenrostro2018.html" target="_blank">c113844</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
wflow_rename(&quot;analysis/process_data_Buenrostro2018.Rmd&quot;, &quot;analysis/process_data_Buenrostro2018_Chen2019pipeline.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1905c587884762a4f9e1856e4faa118cf7ad2f34/docs/process_data_Buenrostro2018.html" target="_blank">1905c58</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1da32f8c37df71b84c5000ac6d70eca03c272c18/analysis/process_data_Buenrostro2018.Rmd" target="_blank">1da32f8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-05
</td>
<td>
set binarized counts as sparse matrix
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/89b45be3922b3a5de08b197053ae0474ffdc6698/docs/process_data_Buenrostro2018.html" target="_blank">89b45be</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/857e1e8164281ddb806d57c6e7a604d6e6716a36/analysis/process_data_Buenrostro2018.Rmd" target="_blank">857e1e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
process data from Buenrostro 2018 paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/907fa65bbbb795c9ef5f297ea803e8147dcc74f6/docs/process_data_Buenrostro2018.html" target="_blank">907fa65</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/12bf4b36e8a7bd25cf947db2ab65005d2560c2b4/analysis/process_data_Buenrostro2018.Rmd" target="_blank">12bf4b3</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
process data from Buenrostro 2018 paper
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="about-buenrostro-2018-dataset" class="section level2">
<h2>About Buenrostro 2018 dataset</h2>
<p>Reference: Buenrostro, J. D. et al. Integrated Single-Cell Analysis Maps the Continuous Regulatory Landscape of Human Hematopoietic Differentiation. Cell 173, 1535–1548.e16 (2018).</p>
<p>Data were downloaded from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96772">GSE96772</a></p>
<p>More details about the samples: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2540299" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2540299</a></p>
<p>RCC directory: <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/</code></p>
</div>
<div id="prepare-data-for-topic-modeling" class="section level2">
<h2>Prepare data for topic modeling</h2>
<p>We used the processed counts and peaks from Buenrostro <em>et al</em> (2018) paper.</p>
<p>They called ATAC-seq peaks from the bulk ATAC-seq samples.</p>
<blockquote>
<p>Using the previously described approach (Corces et al., 2016), we defined a peak list using all bulk hematopoietic data analyzed here, resulting in 491,437 500bp non-overlapping peaks which we use for the remainder of this study.</p>
</blockquote>
<p>Downloaded scATAC-seq processed data from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96769">GSE96769</a></p>
<ul>
<li>Peaks called from bulk ATAC-seq samples <code>GSE96769_PeakFile_20160207.bed.gz</code></li>
<li>scATAC-seq sparse count matrix <code>GSE96769_scATACseq_counts.txt.gz</code></li>
</ul>
<pre class="r"><code>library(Matrix)
library(tools)
library(readr)
library(data.table)</code></pre>
<p>Load the fragment counts as a 2,953 x 491,437 sparse matrix.</p>
<pre class="r"><code># The first row has the sample names
file_counts &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/GEO_data/GSE96769_scATACseq_counts.txt.gz&quot;
sample_names &lt;- readLines(file_counts,n = 1)
sample_names &lt;- unlist(strsplit(sample_names,&quot;\t&quot;,fixed = TRUE))
sample_names &lt;- unlist(strsplit(sample_names,&quot;;&quot;,fixed = TRUE))
sample_names &lt;- sample_names[-1]

# Load the fragment counts as sparse matrix.
dat &lt;- fread(file_counts,sep = &quot;\t&quot;,skip = 1)
class(dat) &lt;- &quot;data.frame&quot;
names(dat) &lt;- c(&quot;i&quot;,&quot;j&quot;,&quot;x&quot;)
counts &lt;- sparseMatrix(i = dat$i,j = dat$j,x = dat$x)
counts &lt;- t(counts)
rownames(counts) &lt;- sample_names
dim(counts)</code></pre>
<pre><code># [1]   2953 491437</code></pre>
<p>Peaks (from bulk ATAC-seq)</p>
<pre class="r"><code>peaks &lt;- fread(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/GEO_data/GSE96769_PeakFile_20160207.bed.gz&quot;)
peak_names &lt;- paste(peaks$V1,peaks$V2,peaks$V3,sep = &quot;_&quot;)
rownames(peaks) &lt;- peak_names
cat(sprintf(&quot;Number of peaks: %d\n&quot;,nrow(peaks)))

colnames(counts) &lt;- peak_names</code></pre>
<pre><code># Number of peaks: 491437</code></pre>
<p>The supplementary Table S1 provides more details about these samples. <a href="https://ars.els-cdn.com/content/image/1-s2.0-S009286741830446X-mmc1.xlsx" class="uri">https://ars.els-cdn.com/content/image/1-s2.0-S009286741830446X-mmc1.xlsx</a></p>
<p>Select the 2,034 samples (cells) passing quality filtering</p>
<blockquote>
<p>Single-cell profiles were of consistent high-quality with 2,034 cells passing stringent quality filtering, yielding a median of 8,268 fragments per cell with 76% of those fragments mapping to peaks, resulting in a median of 6,442 fragments in peaks per cell (figure 1E).</p>
</blockquote>
<blockquote>
<p>Single-cells were filtered for quality requiring at least 60% of fragments in peaks and requiring greater than 1,000 fragments passing quality filters, quality filters are previously described (Corces et al., 2016) which includes removal of mitochondrial reads and low alignment quality (Q30).</p>
</blockquote>
<p>The 2,034 samples (pass filter) were included in the <code>metadata.tsv</code> file on the scATAC-benchmarking website from Chen <em>et al</em>.</p>
<pre class="r"><code>samples_filtered &lt;- read.table(&#39;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/data/input_Chen_2019/metadata.tsv&#39;, header = TRUE, stringsAsFactors=FALSE, quote=&quot;&quot;,row.names=1)
samples_filtered$name &lt;- rownames(samples_filtered)

idx_samples_filtered &lt;- match(samples_filtered$name, sample_names)
sample_names_filtered &lt;- sample_names[idx_samples_filtered]
counts &lt;- counts[idx_samples_filtered,]
samples &lt;- samples_filtered[,c(&quot;name&quot;, &quot;label&quot;)]
dim(counts)</code></pre>
<pre><code># [1]   2034 491437</code></pre>
<p>Remove peaks not exist in any of the cells</p>
<pre class="r"><code>j &lt;- which(colSums(counts &gt; 0) &gt;= 1)
peaks &lt;- peaks[j,]
counts &lt;- counts[,j]
cat(length(j), &quot;peaks left after filtering.  \n&quot;)

dim(counts)
sum(colSums(counts) == 0)
sum(rowSums(counts) == 0)</code></pre>
<pre><code># 465536 peaks left after filtering.  
# [1]   2034 465536
# [1] 0
# [1] 0</code></pre>
<p>Binarize counts</p>
<pre class="r"><code>binarized_counts &lt;- as.matrix((counts &gt; 0) + 0)
binarized_counts &lt;- Matrix(binarized_counts, sparse = TRUE) 

cat(sprintf(&quot;Matrix dimension after filtering: %d x %d.\n&quot;,nrow(binarized_counts),ncol(binarized_counts)))</code></pre>
<pre><code># Matrix dimension after filtering: 2034 x 465536.</code></pre>
<p>Save processed data: <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/</code></p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/&quot;
dir.create(data.dir, showWarnings = FALSE, recursive = TRUE)

save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018.RData&quot;))

counts &lt;- binarized_counts
save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018_binarized.RData&quot;))</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Buenrostro_2018_binarized.RData&quot;))
cat(sprintf(&quot;Loaded %d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))

cat(sprintf(&quot;Number of samples (cells): %d\n&quot;,nrow(counts)))
cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))
cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))</code></pre>
<pre><code># Loaded 2034 x 465536 counts matrix.
# Number of samples (cells): 2034
# Number of peaks: 465536
# Proportion of counts that are non-zero: 1.5%.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 4.0.4 (2021-02-15)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] tools     stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
# [1] data.table_1.14.2 readr_2.1.1       Matrix_1.4-0      workflowr_1.7.0  
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.8        highr_0.9         bslib_0.3.1       compiler_4.0.4   
#  [5] pillar_1.7.0      later_1.3.0       git2r_0.29.0      jquerylib_0.1.4  
#  [9] R.methodsS3_1.8.1 R.utils_2.11.0    getPass_0.2-2     digest_0.6.29    
# [13] lattice_0.20-45   jsonlite_1.7.3    evaluate_0.14     tibble_3.1.6     
# [17] lifecycle_1.0.1   pkgconfig_2.0.3   rlang_1.0.1       cli_3.2.0        
# [21] rstudioapi_0.13   yaml_2.2.2        xfun_0.29         fastmap_1.1.0    
# [25] httr_1.4.2        stringr_1.4.0     knitr_1.37        hms_1.1.1        
# [29] sass_0.4.0        fs_1.5.2          vctrs_0.3.8       grid_4.0.4       
# [33] rprojroot_2.0.2   glue_1.6.2        R6_2.5.1          processx_3.5.2   
# [37] fansi_1.0.2       rmarkdown_2.11    tzdb_0.2.0        callr_3.7.0      
# [41] magrittr_2.0.2    whisker_0.4       ps_1.6.0          promises_1.2.0.1 
# [45] htmltools_0.5.2   ellipsis_0.3.2    httpuv_1.6.5      utf8_1.2.2       
# [49] stringi_1.7.6     crayon_1.5.0      R.oo_1.24.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
