<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Process data for Buenrostro 2018 scATAC-seq dataset</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Process data for Buenrostro 2018 scATAC-seq dataset</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreeee608fdb7bdc1775f74bcc81a9b28eea1aa899f1targetblankee608fda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/ee608fdb7bdc1775f74bcc81a9b28eea1aa899f1" target="_blank">ee608fd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreeee608fdb7bdc1775f74bcc81a9b28eea1aa899f1targetblankee608fda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/ee608fdb7bdc1775f74bcc81a9b28eea1aa899f1" target="_blank">ee608fd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/clusters_pca_structure_Cusanovich2018.Rmd
    Untracked:  analysis/process_data_Buenrostro2018_Chen2019.Rmd
    Untracked:  buenrostro2018.RData

Unstaged changes:
    Modified:   analysis/plots_Lareau2019_bonemarrow.Rmd
    Modified:   scripts/fit_all_models_Buenrostro_2018.sbatch

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/process_data_Buenrostro2018.Rmd</code>) and HTML (<code>docs/process_data_Buenrostro2018.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/ee608fdb7bdc1775f74bcc81a9b28eea1aa899f1/analysis/process_data_Buenrostro2018.Rmd" target="_blank">ee608fd</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-18
</td>
<td>
process counts for aggregated single-cell peaks using chromVAR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/594fd86c6639a42167492e74bc21802c3fe1eb06/docs/process_data_Buenrostro2018.html" target="_blank">594fd86</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/614814b2f797674f4bc404f3e986aba9adccc387/analysis/process_data_Buenrostro2018.Rmd" target="_blank">614814b</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
process Buenrostro2018 data using Downloaded scATAC-seq processed data from GEO
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c113844e38826d4646b6c8d07a00050215a4d021/analysis/process_data_Buenrostro2018.Rmd" target="_blank">c113844</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
wflow_rename(“analysis/process_data_Buenrostro2018.Rmd”, “analysis/process_data_Buenrostro2018_Chen2019pipeline.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/c113844e38826d4646b6c8d07a00050215a4d021/docs/process_data_Buenrostro2018.html" target="_blank">c113844</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
<td>
wflow_rename(“analysis/process_data_Buenrostro2018.Rmd”, “analysis/process_data_Buenrostro2018_Chen2019pipeline.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/1905c587884762a4f9e1856e4faa118cf7ad2f34/docs/process_data_Buenrostro2018.html" target="_blank">1905c58</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/1da32f8c37df71b84c5000ac6d70eca03c272c18/analysis/process_data_Buenrostro2018.Rmd" target="_blank">1da32f8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-05
</td>
<td>
set binarized counts as sparse matrix
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/89b45be3922b3a5de08b197053ae0474ffdc6698/docs/process_data_Buenrostro2018.html" target="_blank">89b45be</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/857e1e8164281ddb806d57c6e7a604d6e6716a36/analysis/process_data_Buenrostro2018.Rmd" target="_blank">857e1e8</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
process data from Buenrostro 2018 paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/907fa65bbbb795c9ef5f297ea803e8147dcc74f6/docs/process_data_Buenrostro2018.html" target="_blank">907fa65</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/12bf4b36e8a7bd25cf947db2ab65005d2560c2b4/analysis/process_data_Buenrostro2018.Rmd" target="_blank">12bf4b3</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-04
</td>
<td>
process data from Buenrostro 2018 paper
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="about-buenrostro-2018-dataset" class="section level2">
<h2>About Buenrostro 2018 dataset</h2>
<p>Reference: Buenrostro, J. D. et al. Integrated Single-Cell Analysis Maps the Continuous Regulatory Landscape of Human Hematopoietic Differentiation. Cell 173, 1535–1548.e16 (2018).</p>
<p>Data were downloaded from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96772">GSE96772</a></p>
<p>RCC directory: <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/</code></p>
</div>
<div id="prepare-data-for-topic-modeling" class="section level2">
<h2>Prepare data for topic modeling</h2>
<div id="using-peaks-called-from-bulk-atac-seq-samples" class="section level3">
<h3>Using peaks called from bulk ATAC-seq samples</h3>
<p>The Buenrostro et al paper called ATAC-seq peaks from the bulk ATAC-seq samples. Downloaded scATAC-seq processed data from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96769">GSE96769</a></p>
<ul>
<li>Peaks called from bulk ATAC-seq samples <code>GSE96769_PeakFile_20160207.bed.gz</code></li>
<li>scATAC-seq sparse count matrix <code>GSE96769_scATACseq_counts.txt.gz</code></li>
</ul>
<pre class="r"><code>library(Matrix)
library(tools)
library(readr)
library(data.table)</code></pre>
<p>Load the fragment counts as a 2,953 x 491,437 sparse matrix.</p>
<pre class="r"><code># The first row has the sample names
file_counts &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/GEO_data/GSE96769_scATACseq_counts.txt.gz&quot;
sample_names &lt;- readLines(file_counts,n = 1)
sample_names &lt;- unlist(strsplit(sample_names,&quot;\t&quot;,fixed = TRUE))
sample_names &lt;- unlist(strsplit(sample_names,&quot;;&quot;,fixed = TRUE))
sample_names &lt;- sample_names[-1]

# Load the fragment counts as sparse matrix.
dat &lt;- fread(file_counts,sep = &quot;\t&quot;,skip = 1)
class(dat) &lt;- &quot;data.frame&quot;
names(dat) &lt;- c(&quot;i&quot;,&quot;j&quot;,&quot;x&quot;)
counts &lt;- sparseMatrix(i = dat$i,j = dat$j,x = dat$x)
counts &lt;- t(counts)
rownames(counts) &lt;- sample_names
dim(counts)</code></pre>
<pre><code>[1]   2953 491437</code></pre>
<p>peaks</p>
<pre class="r"><code>peaks &lt;- fread(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/GEO_data/GSE96769_PeakFile_20160207.bed.gz&quot;)
peak_names &lt;- paste(peaks$V1,peaks$V2,peaks$V3,sep = &quot;_&quot;)
rownames(peaks) &lt;- peak_names
cat(sprintf(&quot;Number of peaks: %d\n&quot;,nrow(peaks)))</code></pre>
<pre><code>Number of peaks: 491437</code></pre>
<pre class="r"><code>colnames(counts) &lt;- peak_names</code></pre>
<p>Plot the distribution of fragment counts.</p>
<pre class="r"><code>y &lt;- table(summary(counts)$x)
x &lt;- names(y)
y &lt;- as.numeric(y)
plot(x,y,pch = 20,log = &quot;y&quot;,cex = 0.65,
     xlab = &quot;number of fragments mapping to peak&quot;,
     ylab = &quot;number of peaks&quot;)
lines(x,y)</code></pre>
<p><img src="figure/process_data_Buenrostro2018.Rmd/plot-distribution-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-distribution-1-1">
Past versions of plot-distribution-1-1.png
</button>
</p>
<div id="fig-plot-distribution-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/594fd86c6639a42167492e74bc21802c3fe1eb06/docs/figure/process_data_Buenrostro2018.Rmd/plot-distribution-1-1.png" target="_blank">594fd86</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The supplementary Table S1 provides more details about these samples. <a href="https://ars.els-cdn.com/content/image/1-s2.0-S009286741830446X-mmc1.xlsx" class="uri">https://ars.els-cdn.com/content/image/1-s2.0-S009286741830446X-mmc1.xlsx</a></p>
<p>Use the metadata.tsv from Chen et al. Genome Biology 2019 paper <a href="https://github.com/pinellolab/scATAC-benchmarking/tree/master/Real_Data/Buenrostro_2018/input/metadata.tsv" class="uri">https://github.com/pinellolab/scATAC-benchmarking/tree/master/Real_Data/Buenrostro_2018/input/metadata.tsv</a>)</p>
<pre class="r"><code>samples_filtered &lt;- read.table(&#39;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/data/input_Chen_2019/metadata.tsv&#39;, header = TRUE, stringsAsFactors=FALSE, quote=&quot;&quot;,row.names=1)
samples_filtered$name &lt;- rownames(samples_filtered)

idx_samples_filtered &lt;- match(samples_filtered$name, sample_names)
sample_names_filtered &lt;- sample_names[idx_samples_filtered]
counts &lt;- counts[idx_samples_filtered,]
samples &lt;- samples_filtered[,c(&quot;name&quot;, &quot;label&quot;)]
dim(counts)</code></pre>
<pre><code>[1]   2034 491437</code></pre>
<p>Remove peaks not exist in any of the cells</p>
<pre class="r"><code>j &lt;- which(colSums(counts &gt; 0) &gt;= 1)
peaks &lt;- peaks[j,]
counts &lt;- counts[,j]
cat(length(j), &quot;peaks after filtering.  \n&quot;)</code></pre>
<pre><code>465536 peaks after filtering.  </code></pre>
<p>Plot the distribution of filtered fragment counts.</p>
<pre class="r"><code>y &lt;- table(summary(counts)$x)
x &lt;- names(y)
y &lt;- as.numeric(y)
plot(x,y,pch = 20,log = &quot;y&quot;,cex = 0.65,
     xlab = &quot;number of fragments mapping to peak&quot;,
     ylab = &quot;number of peaks&quot;)
lines(x,y)</code></pre>
<p><img src="figure/process_data_Buenrostro2018.Rmd/plot-distribution-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-distribution-2-1">
Past versions of plot-distribution-2-1.png
</button>
</p>
<div id="fig-plot-distribution-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/594fd86c6639a42167492e74bc21802c3fe1eb06/docs/figure/process_data_Buenrostro2018.Rmd/plot-distribution-2-1.png" target="_blank">594fd86</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Binarize counts</p>
<pre class="r"><code>binarized_counts &lt;- as.matrix((counts &gt; 0) + 0)
binarized_counts &lt;- Matrix(binarized_counts, sparse = TRUE) 

dim(binarized_counts)</code></pre>
<pre><code>[1]   2034 465536</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/&quot;
dir.create(data.dir, showWarnings = FALSE, recursive = TRUE)

saveRDS(counts, file.path(data.dir, &quot;counts_Buenrostro_2018.rds&quot;))
saveRDS(binarized_counts, file.path(data.dir, &quot;binarized_counts_Buenrostro_2018.rds&quot;))

save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018.RData&quot;))

counts &lt;- binarized_counts
save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018_binarized.RData&quot;))</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Buenrostro_2018.RData&quot;))
cat(sprintf(&quot;Loaded %d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))</code></pre>
<pre><code>Loaded 2034 x 465536 counts matrix.</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of samples (cells): %d\n&quot;,nrow(counts)))</code></pre>
<pre><code>Number of samples (cells): 2034</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))</code></pre>
<pre><code>Number of peaks: 465536</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))</code></pre>
<pre><code>Proportion of counts that are non-zero: 1.5%.</code></pre>
</div>
<div id="using-peaks-called-from-aggregated-single-cell-atac-seq-data-processed-by-chen-et-al.-genome-biology-2019" class="section level3">
<h3>Using peaks called from aggregated single-cell ATAC-seq data processed by Chen et al. Genome Biology 2019</h3>
<ul>
<li>Downloaded processed peaks and BAM files from the scATAC-benchmarking website (Chen et al. Genome Biology 2019) (<a href="https://github.com/pinellolab/scATAC-benchmarking/" class="uri">https://github.com/pinellolab/scATAC-benchmarking/</a>)</li>
<li>Count paired-end scATAC-seq reads in peaks using <code>chromVAR</code>, following the pipeline at <a href="https://nbviewer.jupyter.org/github/pinellolab/scATAC-benchmarking/blob/master/Real_Data/Buenrostro_2018/run_methods/chromVAR/chromVAR_buenrostro2018_motifs.ipynb?flush_cache=true" class="uri">https://nbviewer.jupyter.org/github/pinellolab/scATAC-benchmarking/blob/master/Real_Data/Buenrostro_2018/run_methods/chromVAR/chromVAR_buenrostro2018_motifs.ipynb?flush_cache=true</a></li>
</ul>
<p>Load chromVAR and related packages</p>
<pre class="r"><code># if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
#     install.packages(&quot;BiocManager&quot;)
# BiocManager::install(&quot;chromVAR&quot;)
# BiocManager::install(&quot;motifmatchr&quot;)
# BiocManager::install(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)
# BiocManager::install(&quot;JASPAR2016&quot;)</code></pre>
<pre class="r"><code>library(chromVAR)</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(motifmatchr)
library(Matrix)
library(SummarizedExperiment)</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following object is masked from &#39;package:Matrix&#39;:

    which</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following object is masked from &#39;package:Matrix&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: DelayedArray</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>Loading required package: BiocParallel</code></pre>
<pre><code>
Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    aperm, apply, rowsum</code></pre>
<pre class="r"><code>library(BiocParallel)
library(&#39;JASPAR2016&#39;)
library(BSgenome.Hsapiens.UCSC.hg19)</code></pre>
<pre><code>Loading required package: BSgenome</code></pre>
<pre><code>Loading required package: Biostrings</code></pre>
<pre><code>Loading required package: XVector</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:DelayedArray&#39;:

    type</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre><code>Loading required package: rtracklayer</code></pre>
<pre class="r"><code>register(MulticoreParam(10))

data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/chromVAR/&quot;
dir.create(data.dir, showWarnings = FALSE, recursive = TRUE)</code></pre>
<pre class="r"><code>samples &lt;- read.table(&#39;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/data/input_Chen_2019/metadata.tsv&#39;, header = TRUE, stringsAsFactors=FALSE, quote=&quot;&quot;,row.names=1)</code></pre>
<pre class="r"><code>peakfile &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/data/input_Chen_2019/combined.sorted.merged.bed&quot;
peaks &lt;- getPeaks(peakfile, sort_peaks = TRUE)</code></pre>
<pre><code>Warning in getPeaks(peakfile, sort_peaks = TRUE): Peaks are not equal width!Use
resize(peaks, width = x, fix = &quot;center&quot;) to make peaks equal in size, where x is
the desired size of the peaks)</code></pre>
<pre><code>Peaks sorted</code></pre>
<pre class="r"><code>peaks &lt;- resize(peaks, width = 500, fix = &quot;center&quot;)

cat(length(peaks), &quot;peaks \n&quot;)</code></pre>
<pre><code>237450 peaks </code></pre>
<pre class="r"><code>dir_bamfiles &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/data/input_Chen_2019/sc-bams_nodup/&quot;
bamfiles &lt;- list.files(path = dir_bamfiles, pattern = &quot;\\.bam$&quot;)
cellnames &lt;- sapply(strsplit(bamfiles,&#39;.&#39;,fixed = TRUE), &quot;[[&quot;, 1)
sum(cellnames == rownames(samples))</code></pre>
<pre><code>[1] 2034</code></pre>
<p>count scATAC-seq paired-end fragments in the peaks</p>
<pre class="r"><code>fragment_counts &lt;- getCounts(file.path(dir_bamfiles,bamfile), 
                             peaks, 
                             paired = TRUE, 
                             by_rg = TRUE, 
                             format = &quot;bam&quot;, 
                             colData = data.frame(celltype = cellnames))

saveRDS(fragment_counts, file.path(data.dir, &quot;fragment_counts_scPeaks_chromVAR_Buenrostro_2018.rds&quot;))</code></pre>
<pre class="r"><code>fragment_counts &lt;- readRDS(file.path(data.dir, &quot;fragment_counts_scPeaks_chromVAR_Buenrostro_2018.rds&quot;))
fragment_counts</code></pre>
<pre><code>class: RangedSummarizedExperiment 
dim: 237450 2034 
metadata(0):
assays(1): counts
rownames: NULL
rowData names(0):
colnames(2034): BM1077-CLP-Frozen-160106-13 BM1077-CLP-Frozen-160106-14
  ... singles-PB1022-mono-160128-95 singles-PB1022-mono-160128-96
colData names(2): celltype depth</code></pre>
<pre class="r"><code>counts &lt;- assay(fragment_counts)
counts &lt;- t(counts)

peaks &lt;- as.data.frame(peaks)[,1:3]
colnames(peaks) &lt;- c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;)
peak_names &lt;- paste(peaks$chr, peaks$start, peaks$end, sep = &quot;_&quot;)
colnames(counts) &lt;- peak_names</code></pre>
<p>Filter peaks using <code>filterPeaks</code> in <code>chromVAR</code></p>
<pre class="r"><code>idx_peaks_filtered &lt;- filterPeaks(fragment_counts, min_fragments_per_peak = 1, non_overlapping = TRUE, ix_return = TRUE)
peaks &lt;- peaks[idx_peaks_filtered,]
counts &lt;- counts[,idx_peaks_filtered]
cat(length(idx_peaks_filtered), &quot;peaks after filtering.  \n&quot;)</code></pre>
<pre><code>228965 peaks after filtering.  </code></pre>
<p>Plot the distribution of filtered fragment counts.</p>
<pre class="r"><code>y &lt;- table(summary(counts)$x)
x &lt;- names(y)
y &lt;- as.numeric(y)
plot(x,y,pch = 20,log = &quot;y&quot;,cex = 0.65,
     xlab = &quot;number of fragments mapping to peak&quot;,
     ylab = &quot;number of peaks&quot;)
lines(x,y)</code></pre>
<p><img src="figure/process_data_Buenrostro2018.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Binarize counts</p>
<pre class="r"><code>binarized_counts &lt;- as.matrix((counts &gt; 0) + 0)
binarized_counts &lt;- Matrix(binarized_counts, sparse = TRUE) 
dim(binarized_counts)</code></pre>
<pre><code>[1]   2034 228965</code></pre>
<pre class="r"><code>saveRDS(counts, file.path(data.dir, &quot;counts_scPeaks_Buenrostro_2018.rds&quot;))
saveRDS(binarized_counts, file.path(data.dir, &quot;binarized_counts_scPeaks_Buenrostro_2018.rds&quot;))

save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018_scPeaks.RData&quot;))

counts &lt;- binarized_counts
save(list = c(&quot;samples&quot;, &quot;peaks&quot;, &quot;counts&quot;), 
     file = file.path(data.dir, &quot;Buenrostro_2018_binarized_scPeaks.RData&quot;))</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/chromVAR/&quot;

load(file.path(data.dir, &quot;Buenrostro_2018_binarized_scPeaks.RData&quot;))
cat(sprintf(&quot;Loaded %d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))</code></pre>
<pre><code>Loaded 2034 x 228965 counts matrix.</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of samples (cells): %d\n&quot;,nrow(counts)))</code></pre>
<pre><code>Number of samples (cells): 2034</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))</code></pre>
<pre><code>Number of peaks: 228965</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))</code></pre>
<pre><code>Proportion of counts that are non-zero: 2.6%.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    tools     stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg19_1.4.0 BSgenome_1.52.0                  
 [3] rtracklayer_1.44.0                Biostrings_2.52.0                
 [5] XVector_0.24.0                    JASPAR2016_1.12.0                
 [7] SummarizedExperiment_1.14.1       DelayedArray_0.10.0              
 [9] BiocParallel_1.18.0               matrixStats_0.57.0               
[11] Biobase_2.42.0                    GenomicRanges_1.36.0             
[13] GenomeInfoDb_1.20.0               IRanges_2.18.1                   
[15] S4Vectors_0.22.1                  BiocGenerics_0.30.0              
[17] motifmatchr_1.4.0                 chromVAR_1.4.1                   
[19] data.table_1.12.8                 readr_1.3.1                      
[21] Matrix_1.2-18                     workflowr_1.6.2                  

loaded via a namespace (and not attached):
 [1] bitops_1.0-6                fs_1.3.1                   
 [3] DirichletMultinomial_1.26.0 TFBSTools_1.22.0           
 [5] bit64_0.9-7                 httr_1.4.1                 
 [7] rprojroot_1.3-2             backports_1.1.10           
 [9] R6_2.5.0                    DT_0.13                    
[11] lazyeval_0.2.2              seqLogo_1.50.0             
[13] DBI_1.1.0                   colorspace_1.4-1           
[15] tidyselect_1.1.0            bit_1.1-14                 
[17] compiler_3.6.1              git2r_0.27.1               
[19] plotly_4.9.0                caTools_1.17.1.2           
[21] scales_1.1.1                stringr_1.4.0              
[23] digest_0.6.27               Rsamtools_2.0.0            
[25] rmarkdown_2.1               R.utils_2.9.2              
[27] pkgconfig_2.0.3             htmltools_0.4.0            
[29] fastmap_1.0.1               htmlwidgets_1.5.1          
[31] rlang_0.4.8                 VGAM_1.1-1                 
[33] RSQLite_2.1.1               shiny_1.4.0.2              
[35] jsonlite_1.6                gtools_3.8.1               
[37] dplyr_0.8.5                 R.oo_1.23.0                
[39] RCurl_1.98-1.1              magrittr_1.5               
[41] GO.db_3.8.2                 GenomeInfoDbData_1.2.1     
[43] Rcpp_1.0.5                  munsell_0.5.0              
[45] lifecycle_0.2.0             R.methodsS3_1.7.1          
[47] stringi_1.4.6               whisker_0.4                
[49] yaml_2.2.0                  zlibbioc_1.30.0            
[51] plyr_1.8.6                  grid_3.6.1                 
[53] blob_1.2.0                  promises_1.1.0             
[55] crayon_1.3.4                miniUI_0.1.1.1             
[57] CNEr_1.20.0                 lattice_0.20-38            
[59] splines_3.6.1               annotate_1.62.0            
[61] KEGGREST_1.24.0             hms_0.5.3                  
[63] knitr_1.28                  pillar_1.4.6               
[65] reshape2_1.4.3              TFMPvalue_0.0.8            
[67] XML_3.98-1.20               glue_1.4.2                 
[69] evaluate_0.14               png_0.1-7                  
[71] vctrs_0.3.4                 httpuv_1.5.3.1             
[73] tidyr_1.1.0                 gtable_0.3.0               
[75] poweRlaw_0.70.2             purrr_0.3.4                
[77] assertthat_0.2.1            ggplot2_3.3.2              
[79] xfun_0.14                   mime_0.9                   
[81] xtable_1.8-4                later_1.0.0                
[83] viridisLite_0.3.0           tibble_3.0.4               
[85] AnnotationDbi_1.46.0        GenomicAlignments_1.20.1   
[87] memoise_1.1.0               ellipsis_0.3.1             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
