<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2020-08-17" />

<title>Topic modeling analysis for Cusanovich and Hill 2018 sci-ATAC-seq datasets</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Topic modeling analysis for Cusanovich and Hill 2018 sci-ATAC-seq datasets</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">2020-08-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong23b953f"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 23b953f </a>
</p>
</div>
<div id="strongRepositoryversionstrong23b953f" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 23b953f. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/scATACseq_analysis_Lareau2019.Rmd
    Untracked:  analysis/scATACseq_data_Cusanovich2018.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/references.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/scATACseq_analysis_Cusanovich2018.Rmd</code>) and HTML (<code>docs/scATACseq_analysis_Cusanovich2018.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
23b953f
</td>
<td>
kevinlkx
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“analysis/scATACseq_analysis_Cusanovich2018.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="cusanovich-and-hill-et-al.-2018-dataset" class="section level2">
<h2>Cusanovich and Hill, et al. 2018 dataset</h2>
<p>Reference: Cusanovich, D., Hill, A., Aghamirzaie, D., Daza, R., Pliner, H., Berletch, J., Filippova, G., Huang, X., Christiansen, L., DeWitt, W., Lee, C., Regalado, S., Read, D., Steemers, F., Disteche, C., Trapnell, C., Shendure, J. (2018). A Single-Cell Atlas of In Vivo Mammalian Chromatin Accessibility Cell 174(5), 1 35. <a href="https://dx.doi.org/10.1016/j.cell.2018.06.052" class="uri">https://dx.doi.org/10.1016/j.cell.2018.06.052</a></p>
<p>Data were downloaded from the website: <a href="https://atlas.gs.washington.edu/mouse-atac/data/" class="uri">https://atlas.gs.washington.edu/mouse-atac/data/</a>. They also provided detail descriptions about these datasets.</p>
<p>RCC directory: <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/</code></p>
<p>The authors included nice tutorials on how to get started with analysis of sci-ATAC-seq data: <a href="http://atlas.gs.washington.edu/mouse-atac/docs/" class="uri">http://atlas.gs.washington.edu/mouse-atac/docs/</a></p>
<p>R/python scripts referenced in this tutorial are available on: <a href="https://github.com/shendurelab/mouse-atac" class="uri">https://github.com/shendurelab/mouse-atac</a> and downloaded to <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/code_from_authors/mouse-atac</code> on RCC.</p>
<p>Many of the initial steps of processing raw sci-ATAC-seq libraries used for this study are similar to their previous work of sci-ATAC-seq on Drosophila melanogaster embryos at 3 different stages of development. “Cusanovich, D., Reddington, J., Garfield, D. et al. The cis-regulatory dynamics of embryonic development at single-cell resolution. Nature 555, 538–542 (2018). <a href="https://doi.org/10.1038/nature25981" class="uri">https://doi.org/10.1038/nature25981</a>”. Code and documentation on processing sequencing data can be found on the <a href="https://github.com/shendurelab/fly-atac">Fly ATAC Github</a>. Documentation on various downstream steps can be found in the <a href="http://atlas.gs.washington.edu/fly-atac/">Fly ATAC documentation</a>.</p>
</div>
<div id="matrix-files" class="section level2">
<h2>Matrix files</h2>
<ul>
<li>Binarized peak by cell matrix: <code>atac_matrix.binary.qc_filtered.rds</code> (only QC filtered cells are included).</li>
</ul>
<pre class="r"><code>library(Matrix)

## atac_matrix.binary.qc_filtered.rds: binarized peak by cell matrix in RDS format.
binarized_matrix &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds&quot;)
dim(binarized_matrix)</code></pre>
<pre><code>[1] 436206  81173</code></pre>
<ul>
<li>The subset of peaks used as input to TFIDF: <code>atac_matrix.tfidf.qc_filtered.peaks.txt</code>.</li>
</ul>
<pre class="r"><code>peaks_tfidf &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.tfidf.qc_filtered.peaks.txt&quot;, header = FALSE, stringsAsFactors = TRUE)

cat(nrow(peaks_tfidf), &quot;peaks used as input to TFIDF.&quot;)</code></pre>
<pre><code>167013 peaks used as input to TFIDF.</code></pre>
<ul>
<li>TFIDF normalized peak by cell matrix: <code>atac_matrix.tfidf.qc_filtered.rds</code>. This dataset has rare peaks filtered out and is then normalized with TFIDF to allow for input to PCA/TSNE (only QC filtered cells are included).</li>
</ul>
<pre class="r"><code>library(Matrix)

## atac_matrix.tfidf.qc_filtered.rds: TFIDF normalized peak by cell matrix in RDS format.
atac_matrix.tfidf &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.tfidf.qc_filtered.rds&quot;)
dim(atac_matrix.tfidf)</code></pre>
<pre><code>[1] 167013  81173</code></pre>
</div>
<div id="metadata" class="section level2">
<h2>Metadata</h2>
<ul>
<li><code>cell_metadata.txt</code>: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments. Columns in this data:
<ul>
<li>cell: cell barcode (combined and corrected)</li>
<li>tissue: the tissue that this cell originated from</li>
<li>tissue_replicate: same as tissue, but each replicate has a unique id</li>
<li>cluster: cluster assignment in initial t-SNE</li>
<li>subset_cluster: cluster assignment in iterative t-SNE space</li>
<li>tsne_1: t-SNE1 coordinate in initial t-SNE</li>
<li>tsne_2: t-SNE2 coordinate in initial t-SNE</li>
<li>subset_tsne1: t-SNE1 coordinate in iterative t-SNE</li>
<li>subset_tsne2: t-SNE2 coordinate in iterative t-SNE</li>
<li>id: combined ID for major + iterative cluster assignment</li>
<li>cell_label: assigned cell type</li>
</ul></li>
</ul>
<pre class="r"><code>cell_metadata &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(cell_metadata)</code></pre>
<pre><code>[1] 81173    11</code></pre>
<pre class="r"><code>cell_metadata[1:3,]</code></pre>
<pre><code>                                  cell     tissue tissue.replicate cluster
1 AGCGATAGAACGAATTCGCCTCCGACGGCAGGACGT       Lung      Lung2_62216      10
2 AGCGATAGAACGAATTCGTTGGTAGTCGATAGAGGC     Spleen     Spleen_62016      10
3 AGCGATAGAACGCGCAGAAAGCTAGGTTAGGCGAAG BoneMarrow BoneMarrow_62016      10
  subset_cluster   tsne_1   tsne_2 subset_tsne1 subset_tsne2
1              1 16.25804 17.05674    -14.44873    -3.100071
2              1 15.28149 16.10859    -15.23808     1.123518
3              1 17.03914 14.69750    -16.88165     5.244815
                     id                cell_label
1 clusters_10.cluster_1 Hematopoietic progenitors
2 clusters_10.cluster_1 Hematopoietic progenitors
3 clusters_10.cluster_1 Hematopoietic progenitors</code></pre>
<ul>
<li><code>cell_metadata.tissue_freq_filtered.txt</code>: Same as <code>cell_metadata.txt</code>, but removes cells in each tissue belonging to a cell_label that accounts for less than 0.5% of the cells in that tissue. These very low frequency labels are often not cell types expected their respective tissues and could be due to slight imperfections in clustering, for example. Provided matrices would need to be subsetted to match this set of cells if using this metadata.</li>
</ul>
<pre class="r"><code>cell_metadata &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.tissue_freq_filtered.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(cell_metadata)</code></pre>
<pre><code>[1] 80254    11</code></pre>
<pre class="r"><code>cell_metadata[1:3,]</code></pre>
<pre><code>                                  cell     tissue tissue.replicate cluster
1 AGCGATAGAACGAATTCGCCTCCGACGGCAGGACGT       Lung      Lung2_62216      10
2 AGCGATAGAACGAATTCGTTGGTAGTCGATAGAGGC     Spleen     Spleen_62016      10
3 AGCGATAGAACGCGCAGAAAGCTAGGTTAGGCGAAG BoneMarrow BoneMarrow_62016      10
  subset_cluster   tsne_1   tsne_2 subset_tsne1 subset_tsne2
1              1 16.25804 17.05674    -14.44873    -3.100071
2              1 15.28149 16.10859    -15.23808     1.123518
3              1 17.03914 14.69750    -16.88165     5.244815
                     id                cell_label
1 clusters_10.cluster_1 Hematopoietic progenitors
2 clusters_10.cluster_1 Hematopoietic progenitors
3 clusters_10.cluster_1 Hematopoietic progenitors</code></pre>
<ul>
<li><p><code>cell_type_assignments.xlsx</code>: Excel document with three tabs expected cell types, cell type markers, and Cell type assignments that contain a pairs of tissues and expected cell types, a list of positive markers for each cell type, and the table of cell type assignments with extra details about assignment criteria when applicable, respectively. This is meant to document justification for cell type assignments provided in <code>cell_metadata.txt</code> above.</p></li>
<li><p><code>peak_promoter_intersections.txt</code>: Metadata for peaks-intersected TSS pairs in TSV format.</p></li>
</ul>
<pre class="r"><code>peak_promoter_intersections &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/peak_promoter_intersections.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(peak_promoter_intersections)</code></pre>
<pre><code>[1] 62545     9</code></pre>
<pre class="r"><code>peak_promoter_intersections[1:3,]</code></pre>
<pre><code>               peak_id peak_chr peak_start peak_end           ensembl_id
1 chr1_3206330_3206546     chr1    3206330  3206546 ENSMUSG00000051951.5
2 chr1_3456171_3457239     chr1    3456171  3457239 ENSMUSG00000089699.1
3 chr1_3660494_3662750     chr1    3660494  3662750 ENSMUSG00000051951.5
  gene_short_name ensembl_transcript_id              biotype strand
1            Xkr4  ENSMUST00000162897.1 processed_transcript      -
2          Gm1992  ENSMUST00000161581.1            antisense      +
3            Xkr4  ENSMUST00000070533.4       protein_coding      -</code></pre>
</div>
<div id="dimensionality-reduction-in-the-original-paper" class="section level2">
<h2>Dimensionality Reduction in the original paper</h2>
<p>They used the LSI (Latent Semantic Indexing) appraoch for dimensionality reduction: first, transform the data using the frequency-inverse document frequency transformation (TF-IDF), and then use singular value decomposition (SVD) on the TF-IDF matrix to generate a lower dimensional representation of the data. Introduction to LSI from <a href="https://en.wikipedia.org/wiki/Latent_semantic_analysis">wikipedia</a>.</p>
<p>TF-IDF: first weight all the sites for individual cells by the total number of sites accessible in that cell (<code>term frequency</code>); then multiply these weighted values by log(1 + the inverse frequency of each site across all cells), the <code>inverse document frequency</code>. Introduction to TF-IDF from <a href="https://en.wikipedia.org/wiki/Tf–idf">wikipedia</a>.</p>
<p>This representation was then used as input for the T-SNE (t-distributed Stochastic Neighbor Embedding) analysis using <code>Rtsne</code> package in R.</p>
<p>The code below uses an example function <code>atac_dim_reduction</code> in <a href="https://github.com/shendurelab/mouse-atac/blob/master/dim_reduction/dim_reduction.R">dim_reduction.R</a> from the authors’ Github page that given a matrix will do TFIDF, PCA, and t-SNE and return the resulting PCA and TSNE coordinates. Note that this function takes the binarized matrix and a site_frequency_threshold argument (default 0.03 or site observed in at least 3% of cells).</p>
<pre class="r"><code>source(&#39;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/code_from_authors/mouse-atac/dim_reduction/dim_reduction.R&#39;)
binarized_matrix &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds&quot;)
# This function outputs a list with two items pca_coords and tsne_coords, which contain the PCA and t-SNE coordinates as dataframes where the cell IDs are included as the rownames.
results.dim_reduction &lt;- atac_dim_reduction(binarized_matrix, site_frequency_threshold=0.02)</code></pre>
<div id="tf-idf-transformation" class="section level3">
<h3>TF-IDF transformation</h3>
<pre class="r"><code>library(Matrix)
library(irlba)
dir_results &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/results/&quot;
dir.create(dir_results, showWarnings = FALSE, recursive = TRUE)

site_frequency_threshold &lt;- 0.02
atac_matrix &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds&quot;)

## select sites/peaks observed in at least (site_frequency_threshold)% of cells
num_cells_ncounted &lt;- Matrix::rowSums(atac_matrix)
threshold &lt;- ncol(atac_matrix) * site_frequency_threshold

ncounts &lt;- atac_matrix[num_cells_ncounted &gt;= threshold,]

## Normalize the data with TF-IDF
nfreqs &lt;- t(t(ncounts) / Matrix::colSums(ncounts)) # term frequency
tf_idf_counts &lt;- nfreqs * log(1 + ncol(ncounts) / Matrix::rowSums(ncounts)) # term frequency * inverse document frequency

saveRDS(tf_idf_counts, paste0(dir_results, &quot;/atac_matrix.tfidf.qc_filtered_freq&quot;, site_frequency_threshold, &quot;.rds&quot;))</code></pre>
</div>
<div id="lsi-tf-idf-and-svd-on-atac-seq-matrix-only-qc-filtered-cells-are-included" class="section level3">
<h3>LSI (TF-IDF and SVD) on ATAC-seq matrix (only QC filtered cells are included)</h3>
<pre class="r"><code>## LSI on ATAC-seq matrix
# atac_matrix: a peak x cell scATAC-seq matrix
# n_PCs: number of PCs (singular vectors) included
# site_frequency_threshold: site observed in at least % of cells
# adapted from the atac_dim_reduction function in https://github.com/shendurelab/mouse-atac/blob/master/dim_reduction/dim_reduction.R
LSI_atac &lt;- function(atac_matrix, n_PCs = 50, site_frequency_threshold = 0.03){
  library(Matrix)
  library(irlba)
  
  ## select sites/peaks observed in at least (site_frequency_threshold)% of cells
  num_cells_ncounted &lt;- Matrix::rowSums(atac_matrix)
  threshold &lt;- ncol(atac_matrix) * site_frequency_threshold
  
  ncounts &lt;- atac_matrix[num_cells_ncounted &gt;= threshold,]
  
  ## Normalize the data with TF-IDF
  nfreqs &lt;- t(t(ncounts) / Matrix::colSums(ncounts)) # term frequency
  tf_idf_counts &lt;- nfreqs * log(1 + ncol(ncounts) / Matrix::rowSums(ncounts)) # term frequency * inverse document frequency
  
  ## Do SVD
  set.seed(0)
  # use IRLBA algorithm to compute a partial SVD
  # IRLBA: Fast Truncated Singular Value Decomposition and Principal Components Analysis for Large Dense and Sparse Matrices
  SVD &lt;- irlba(tf_idf_counts, n_PCs, n_PCs, maxit=1000) 
  d_diag &lt;- matrix(0, nrow=length(SVD$d), ncol=length(SVD$d))
  diag(d_diag) &lt;- SVD$d
  SVD_vd &lt;- t(d_diag %*% t(SVD$v))
  rownames(SVD_vd) &lt;- colnames(atac_matrix)
  colnames(SVD_vd) &lt;- paste0(&#39;pca_&#39;, 1:ncol(SVD_vd))
  # return SVD_vd: cell x PC matrix
  return(SVD_vd)
}</code></pre>
<pre class="r"><code>atac_LSI &lt;- LSI_atac(atac_matrix, 50, 0.02)</code></pre>
</div>
</div>
<div id="topic-modeling-of-the-sci-atac-seq-matrix" class="section level2">
<h2>Topic modeling of the sci-ATAC-seq matrix</h2>
<ul>
<li>Binarized peak by cell matrix: <code>atac_matrix.binary.qc_filtered.rds</code> (only QC filtered cells are included).</li>
</ul>
<pre class="r"><code>library(Matrix)

## atac_matrix.binary.qc_filtered.rds: binarized peak by cell matrix in RDS format.
binarized_matrix &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds&quot;)
dim(binarized_matrix)</code></pre>
<pre><code>[1] 436206  81173</code></pre>
<pre class="r"><code>counts &lt;- t(binarized_matrix)

cat(sprintf(&quot;Number of cells (samples): %d\n&quot;,nrow(counts)))</code></pre>
<pre><code>Number of cells (samples): 81173</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))</code></pre>
<pre><code>Number of peaks: 436206</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))</code></pre>
<pre><code>Proportion of counts that are non-zero: 1.2%.</code></pre>
<pre class="r"><code># 100*nnzero(counts)/prod(dim(counts))</code></pre>
<ul>
<li><code>cell_metadata.txt</code>: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments. Columns in this data:
<ul>
<li>cell: cell barcode (combined and corrected)</li>
<li>tissue: the tissue that this cell originated from</li>
<li>tissue_replicate: same as tissue, but each replicate has a unique id</li>
<li>cluster: cluster assignment in initial t-SNE</li>
<li>subset_cluster: cluster assignment in iterative t-SNE space</li>
<li>tsne_1: t-SNE1 coordinate in initial t-SNE</li>
<li>tsne_2: t-SNE2 coordinate in initial t-SNE</li>
<li>subset_tsne1: t-SNE1 coordinate in iterative t-SNE</li>
<li>subset_tsne2: t-SNE2 coordinate in iterative t-SNE</li>
<li>id: combined ID for major + iterative cluster assignment</li>
<li>cell_label: assigned cell type</li>
</ul></li>
</ul>
<pre class="r"><code>cell_metadata &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(cell_metadata)</code></pre>
<pre><code>[1] 81173    11</code></pre>
<pre class="r"><code>cat(sprintf(&quot;Number of cells (samples): %d\n&quot;,nrow(cell_metadata)))</code></pre>
<pre><code>Number of cells (samples): 81173</code></pre>
<pre class="r"><code>print(cell_metadata[1:3,])</code></pre>
<pre><code>                                  cell     tissue tissue.replicate cluster
1 AGCGATAGAACGAATTCGCCTCCGACGGCAGGACGT       Lung      Lung2_62216      10
2 AGCGATAGAACGAATTCGTTGGTAGTCGATAGAGGC     Spleen     Spleen_62016      10
3 AGCGATAGAACGCGCAGAAAGCTAGGTTAGGCGAAG BoneMarrow BoneMarrow_62016      10
  subset_cluster   tsne_1   tsne_2 subset_tsne1 subset_tsne2
1              1 16.25804 17.05674    -14.44873    -3.100071
2              1 15.28149 16.10859    -15.23808     1.123518
3              1 17.03914 14.69750    -16.88165     5.244815
                     id                cell_label
1 clusters_10.cluster_1 Hematopoietic progenitors
2 clusters_10.cluster_1 Hematopoietic progenitors
3 clusters_10.cluster_1 Hematopoietic progenitors</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Matrix_1.2-15   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6    lattice_0.20-38 rprojroot_1.3-2 digest_0.6.25  
 [5] later_1.0.0     grid_3.5.1      R6_2.4.1        backports_1.1.7
 [9] git2r_0.27.1    magrittr_1.5    evaluate_0.14   stringi_1.4.6  
[13] rlang_0.4.6     fs_1.3.1        promises_1.1.0  whisker_0.4    
[17] rmarkdown_2.1   tools_3.5.1     stringr_1.4.0   glue_1.4.1     
[21] httpuv_1.5.3.1  xfun_0.14       yaml_2.2.0      compiler_3.5.1 
[25] htmltools_0.4.0 knitr_1.28     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
