<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Process data for Cusanovich 2018 sci-ATAC-seq dataset</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Process data for Cusanovich 2018 sci-ATAC-seq dataset</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee5ff431d02014329f45e486cd32da28d480951a7targetblanke5ff431a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/e5ff431d02014329f45e486cd32da28d480951a7" target="_blank">e5ff431</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstreee5ff431d02014329f45e486cd32da28d480951a7targetblanke5ff431a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/e5ff431d02014329f45e486cd32da28d480951a7" target="_blank">e5ff431</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/site_libs/

Untracked files:
    Untracked:  analysis/motif_analysis_Buenrostro2018_v2.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  paper/
    Untracked:  scripts/DA_Buenrostro2018_filteredpeaks.sbatch
    Untracked:  scripts/analysis_Cusanovich2018_filteredpeaks.sh
    Untracked:  scripts/fit_poisson_nmf_Buenrostro2018_filtered_HeLabExample.R
    Untracked:  scripts/postfit_Buenrostro2018_filteredpeaks.sbatch
    Untracked:  scripts/postfit_Buenrostro2018_v2.sbatch

Unstaged changes:
    Modified:   analysis/assess_fits_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/clusters_Cusanovich2018_k13.Rmd
    Modified:   analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/motif_analysis_Cusanovich2018.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   scripts/DA_analysis_Buenrostro2018_vsnull.R
    Modified:   scripts/analysis_Buenrostro2018_filteredpeaks.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/process_data_Cusanovich2018.Rmd</code>) and HTML (<code>docs/process_data_Cusanovich2018.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/e5ff431d02014329f45e486cd32da28d480951a7/analysis/process_data_Cusanovich2018.Rmd" target="_blank">e5ff431</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-08-02
</td>
<td>
filtered peaks for data by tissues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/c4ead99b374eb437500d2b18a36425c6827a2813/docs/process_data_Cusanovich2018.html" target="_blank">c4ead99</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/7044d98df33142f0ec4b3145a4e8cd5118b7d20c/docs/process_data_Cusanovich2018.html" target="_blank">7044d98</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c08dcfc166eb14e0192cf0a18c9303aa471ffd15/analysis/process_data_Cusanovich2018.Rmd" target="_blank">c08dcfc</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-02-24
</td>
<td>
wflow_publish(&quot;analysis/process_data_Cusanovich2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/5587790a9be2a0dde5df6b9472c9d69b22263974/docs/process_data_Cusanovich2018.html" target="_blank">5587790</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/0e6a0a4c3d15f4f2d782560e754b070c9055a41d/analysis/process_data_Cusanovich2018.Rmd" target="_blank">0e6a0a4</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-02-01
</td>
<td>
minor change on the output name for endothelial cells
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/6d68febbd36b09dfa87302753b44936bfcea3ee0/docs/process_data_Cusanovich2018.html" target="_blank">6d68feb</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/c1e87ecc7279ccfd46c679d0c52dc2212a965322/analysis/process_data_Cusanovich2018.Rmd" target="_blank">c1e87ec</a>
</td>
<td>
kevinlkx
</td>
<td>
2021-01-29
</td>
<td>
save data by tissues separately
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/0440e358ae1df7319ca266039ed3d4d0cac59c2a/docs/process_data_Cusanovich2018.html" target="_blank">0440e35</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/7b8b36f0b5f79d7d4e4c67738105568a4fe5d0e9/analysis/process_data_Cusanovich2018.Rmd" target="_blank">7b8b36f</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-09-10
</td>
<td>
wflow_publish(&quot;analysis/process_data_Cusanovich2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/edbd049ded58124265af8878c25a0ece8bc37860/docs/process_data_Cusanovich2018.html" target="_blank">edbd049</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/b8367612b7033c79d3d75ab977d302d253ae4077/analysis/process_data_Cusanovich2018.Rmd" target="_blank">b836761</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-24
</td>
<td>
wflow_publish(&quot;analysis/process_data_Cusanovich2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/ceef630525f86fc358bbf728f2b71920fae80c4f/docs/process_data_Cusanovich2018.html" target="_blank">ceef630</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/d0bc60c9bf5c71a2a7d6b5ec9559b467c92da91f/analysis/process_data_Cusanovich2018.Rmd" target="_blank">d0bc60c</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-21
</td>
<td>
wflow_publish(&quot;analysis/process_data_Cusanovich2018.Rmd&quot;)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/02895086808a1b38679072fd57e83f5416d94ad6/analysis/process_data_Cusanovich2018.Rmd" target="_blank">0289508</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-21
</td>
<td>
wflow_rename(&quot;analysis/scATACseq_analysis_Cusanovich2018.Rmd&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/scATACseq-topics/02895086808a1b38679072fd57e83f5416d94ad6/docs/process_data_Cusanovich2018.html" target="_blank">0289508</a>
</td>
<td>
kevinlkx
</td>
<td>
2020-08-21
</td>
<td>
wflow_rename(&quot;analysis/scATACseq_analysis_Cusanovich2018.Rmd&quot;,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(Matrix)
library(tools)</code></pre>
<div id="about-cusanovich-2018-dataset" class="section level2">
<h2>About Cusanovich 2018 dataset</h2>
<p>Reference: Cusanovich, D., Hill, A., Aghamirzaie, D., Daza, R., Pliner, H., Berletch, J., Filippova, G., Huang, X., Christiansen, L., DeWitt, W., Lee, C., Regalado, S., Read, D., Steemers, F., Disteche, C., Trapnell, C., Shendure, J. (2018). A Single-Cell Atlas of In Vivo Mammalian Chromatin Accessibility Cell 174(5), 1 35. <a href="https://dx.doi.org/10.1016/j.cell.2018.06.052" class="uri">https://dx.doi.org/10.1016/j.cell.2018.06.052</a></p>
<p>Data were downloaded from the website: <a href="https://atlas.gs.washington.edu/mouse-atac/data/" class="uri">https://atlas.gs.washington.edu/mouse-atac/data/</a>. They also provided detail descriptions about these datasets.</p>
<p>RCC directory: <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/</code></p>
<p>The authors included nice tutorials on how to get started with analysis of sci-ATAC-seq data: <a href="http://atlas.gs.washington.edu/mouse-atac/docs/" class="uri">http://atlas.gs.washington.edu/mouse-atac/docs/</a></p>
<p>R/python scripts referenced in this tutorial are available on: <a href="https://github.com/shendurelab/mouse-atac" class="uri">https://github.com/shendurelab/mouse-atac</a> and downloaded to <code>/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/code_from_authors/mouse-atac</code> on RCC.</p>
<p>Many of the initial steps of processing raw sci-ATAC-seq libraries used for this study are similar to their previous work of sci-ATAC-seq on Drosophila melanogaster embryos at 3 different stages of development. &quot;Cusanovich, D., Reddington, J., Garfield, D. et al. The cis-regulatory dynamics of embryonic development at single-cell resolution. Nature 555, 538–542 (2018). <a href="https://doi.org/10.1038/nature25981" class="uri">https://doi.org/10.1038/nature25981</a>&quot;. Code and documentation on processing sequencing data can be found on the <a href="https://github.com/shendurelab/fly-atac">Fly ATAC Github</a>. Documentation on various downstream steps can be found in the <a href="http://atlas.gs.washington.edu/fly-atac/">Fly ATAC documentation</a>.</p>
</div>
<div id="matrix-files" class="section level2">
<h2>Matrix files</h2>
<ul>
<li><p>Binarized peak by cell matrix: <code>atac_matrix.binary.qc_filtered.rds</code> (only QC filtered cells are included).</p></li>
<li><p>The subset of peaks used as input to TFIDF: <code>atac_matrix.tfidf.qc_filtered.peaks.txt</code>.</p></li>
<li><p>TFIDF normalized peak by cell matrix: <code>atac_matrix.tfidf.qc_filtered.rds</code>. This dataset has rare peaks filtered out and is then normalized with TFIDF to allow for input to PCA/TSNE (only QC filtered cells are included).</p></li>
</ul>
</div>
<div id="metadata" class="section level2">
<h2>Metadata</h2>
<ul>
<li><code>cell_metadata.txt</code>: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments. Columns in this data:</li>
<li>cell: cell barcode (combined and corrected)</li>
<li>tissue: the tissue that this cell originated from</li>
<li>tissue_replicate: same as tissue, but each replicate has a unique id</li>
<li>cluster: cluster assignment in initial t-SNE</li>
<li>subset_cluster: cluster assignment in iterative t-SNE space</li>
<li>tsne_1: t-SNE1 coordinate in initial t-SNE</li>
<li>tsne_2: t-SNE2 coordinate in initial t-SNE</li>
<li>subset_tsne1: t-SNE1 coordinate in iterative t-SNE</li>
<li>subset_tsne2: t-SNE2 coordinate in iterative t-SNE</li>
<li>id: combined ID for major + iterative cluster assignment</li>
<li>cell_label: assigned cell type</li>
</ul>
<pre class="r"><code>cell_metadata &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(cell_metadata)

cell_metadata[1:3,]</code></pre>
<ul>
<li><code>cell_metadata.tissue_freq_filtered.txt</code>: Same as <code>cell_metadata.txt</code>, but removes cells in each tissue belonging to a cell_label that accounts for less than 0.5% of the cells in that tissue. These very low frequency labels are often not cell types expected their respective tissues and could be due to slight imperfections in clustering, for example. Provided matrices would need to be subsetted to match this set of cells if using this metadata.</li>
</ul>
<pre class="r"><code>cell_metadata &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.tissue_freq_filtered.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(cell_metadata)

cell_metadata[1:3,]</code></pre>
<ul>
<li><p><code>cell_type_assignments.xlsx</code>: Excel document with three tabs expected cell types, cell type markers, and Cell type assignments that contain a pairs of tissues and expected cell types, a list of positive markers for each cell type, and the table of cell type assignments with extra details about assignment criteria when applicable, respectively. This is meant to document justification for cell type assignments provided in <code>cell_metadata.txt</code> above.</p></li>
<li><p><code>peak_promoter_intersections.txt</code>: Metadata for peaks-intersected TSS pairs in TSV format.</p></li>
</ul>
<pre class="r"><code>peak_promoter_intersections &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/peak_promoter_intersections.txt&quot;, header = TRUE, stringsAsFactors = TRUE, sep = &quot;\t&quot;)

dim(peak_promoter_intersections)

peak_promoter_intersections[1:3,]</code></pre>
</div>
<div id="prepare-data-for-topic-modeling" class="section level2">
<h2>Prepare data for topic modeling</h2>
<ul>
<li>Binarized peak by cell matrix: <code>atac_matrix.binary.qc_filtered.rds</code> (only QC filtered cells are included).</li>
</ul>
<pre class="r"><code>## atac_matrix.binary.qc_filtered.rds: binarized peak by cell matrix in RDS format.
binarized_matrix &lt;- readRDS(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds&quot;)
dim(binarized_matrix)
counts &lt;- t(binarized_matrix)

## all peaks exist in all of the cells
length(which(colSums(counts &gt; 0) == 0))
length(which(rowSums(counts &gt; 0) == 0))

## all peaks are accessible in at least 40 cells.
range(colSums(counts))

cat(sprintf(&quot;Number of samples (cells): %d\n&quot;,nrow(counts)))
cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))
cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))</code></pre>
<ul>
<li><code>cell_metadata.txt</code>: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments.</li>
</ul>
<pre class="r"><code>samples &lt;- read.table(&quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt&quot;, header = TRUE, stringsAsFactors = FALSE, sep = &quot;\t&quot;)
cat(sprintf(&quot;Number of samples: %d\n&quot;,nrow(samples)))
print(samples[1:3,])</code></pre>
<p>This study measured single cell chromatin accessibility in 13 different tissues in mice:</p>
<pre class="r"><code>table(samples$tissue)</code></pre>
<p>Cells were labeled into the cell types:</p>
<pre class="r"><code>table(samples$cell_label)</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
dir.create(data.dir, showWarnings = FALSE, recursive = TRUE)

# saveRDS(counts, file.path(data.dir, &quot;counts_Cusanovich_2018.rds&quot;))

save(list = c(&quot;samples&quot;,&quot;counts&quot;), file = file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))
resaveRdaFiles(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))</code></pre>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Cusanovich_2018.RData&quot;))

cat(sprintf(&quot;Loaded %d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))

cat(sprintf(&quot;Number of samples (cells): %d\n&quot;,nrow(counts)))
cat(sprintf(&quot;Number of peaks: %d\n&quot;,ncol(counts)))
cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))

range(colSums(counts))</code></pre>
</div>
<div id="save-data-separately-by-tissues" class="section level2">
<h2>Save data separately by tissues</h2>
<pre class="r"><code>all_counts &lt;- counts
all_samples &lt;- samples

tissues &lt;- unique(all_samples$tissue)
print(tissues)</code></pre>
<pre class="r"><code>for ( tissue in tissues ) {
  cat(tissue, &quot;\n&quot;)
  rows &lt;- which(all_samples$tissue == tissue)
  cat(sprintf(&quot;%d cells in %s.\n&quot;,length(rows),tissue))
  counts &lt;- all_counts[rows,]
  samples &lt;- all_samples[rows,]
  dim(counts)
  if(any(colSums(counts) &lt; 20)){
    # Filtered out peaks with accessbility in fewer than 20 cells
    cat(&quot;Filter out peaks with accessbility in fewer than 20 cells...\n&quot;)
    i      &lt;- which(colSums(counts) &gt;= 20)
    counts &lt;- counts[,i]
    cat(sprintf(&quot;After filtering, we have %d rows and %d columns left. \n&quot;,nrow(counts),ncol(counts)))
    save(list = c(&quot;samples&quot;,&quot;counts&quot;), file = paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_&quot;, tissue,&quot;.RData&quot;))
  resaveRdaFiles(paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_&quot;, tissue,&quot;.RData&quot;))
  }else{
    cat(&quot;All peaks with accessbility in more than 20 cells. No filtering needed.&quot;)
  }
}</code></pre>
<p>Heart, kidney and lung tissues combined.</p>
<pre class="r"><code>rows &lt;- which(all_samples$tissue %in% c(&quot;Heart&quot;, &quot;Kidney&quot;, &quot;Lung&quot;))
cat(sprintf(&quot;%d cells in %s.\n&quot;,length(rows),tissue))
cat(sprintf(&quot;%d cells in heart, kidney and lung.\n&quot;,length(rows)))
counts &lt;- all_counts[rows,]
samples &lt;- all_samples[rows,]
save(list = c(&quot;samples&quot;,&quot;counts&quot;), file = paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_HeartKidneyLung.RData&quot;))
resaveRdaFiles(paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_HeartKidneyLung.RData&quot;)))</code></pre>
<p>Endothelial cells.</p>
<pre class="r"><code>rows &lt;- grep(&quot;Endothelial&quot;,all_samples$cell_label)
cat(sprintf(&quot;%d endothelial cells. \n&quot;,length(rows)))
counts &lt;- all_counts[rows,]
samples &lt;- all_samples[rows,]
save(list = c(&quot;samples&quot;,&quot;counts&quot;), file = paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_Endothelial.RData&quot;))
resaveRdaFiles(paste0(data.dir, &quot;/&quot;, &quot;Cusanovich_2018_Endothelial.RData&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
