<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />


<title>Structure plots and differential analysis of Buenrostro et al (2018) scATAC-seq data using fastTopics</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scATACseq-topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/scATACseq-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Structure plots and differential analysis of Buenrostro <em>et al</em> (2018) scATAC-seq data using fastTopics</h1>
<h4 class="author">Kaixuan Luo</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>scATACseq-topics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200729code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200729)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200729code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200729)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree181b83b6698804e1c65b11b5a7585f72149a93fatargetblank181b83ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/scATACseq-topics/tree/181b83b6698804e1c65b11b5a7585f72149a93fa" target="_blank">181b83b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabscATACseqtopicstree181b83b6698804e1c65b11b5a7585f72149a93fatargetblank181b83ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/scATACseq-topics/tree/181b83b6698804e1c65b11b5a7585f72149a93fa" target="_blank">181b83b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .tmp.motifs
    Untracked:  analysis/motif_analysis_Buenrostro2018_v2.Rmd
    Untracked:  output/clustering-Cusanovich2018.rds
    Untracked:  paper/
    Untracked:  scripts/DA_Buenrostro2018_filteredpeaks.sbatch
    Untracked:  scripts/postfit_Buenrostro2018_filteredpeaks.sbatch
    Untracked:  scripts/postfit_Buenrostro2018_v2.sbatch

Unstaged changes:
    Modified:   analysis/assess_fits_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/clusters_Cusanovich2018_k13.Rmd
    Modified:   analysis/gene_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/gene_analysis_Cusanovich2018.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/motif_analysis_Buenrostro2018_Chen2019pipeline.Rmd
    Modified:   analysis/motif_analysis_Cusanovich2018.Rmd
    Modified:   analysis/plots_Cusanovich2018.Rmd
    Modified:   scripts/DA_analysis_Buenrostro2018_vsnull.R
    Modified:   scripts/analysis_Buenrostro2018_filteredpeaks.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd</code>) and HTML (<code>docs/structureplot_DA_Buenrostro2018_filteredpeaks_k10.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/scATACseq-topics/blob/181b83b6698804e1c65b11b5a7585f72149a93fa/analysis/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd" target="_blank">181b83b</a>
</td>
<td>
kevinlkx
</td>
<td>
2022-03-29
</td>
<td>
added structure plots for results with k = 8 and 9
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we explore the structure in the Buenrostro <em>et al</em> (2018) scATAC-seq data inferred from the multinomial topic model.</p>
<p>Load packages and some functions used in this analysis.</p>
<pre class="r"><code>library(fastTopics)
library(Matrix)
library(dplyr)
library(ggplot2)
library(cowplot)
library(plyr)
library(dplyr)
library(RColorBrewer)
library(DT)
library(reshape)
source(&quot;code/plots.R&quot;)</code></pre>
<div id="load-the-data" class="section level3">
<h3>Load the data</h3>
<p>Data downloaded from original paper.</p>
<p>Original data</p>
<pre class="r"><code>data.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data/&quot;
load(file.path(data.dir, &quot;Buenrostro_2018_binarized.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
samples$cell &lt;- rownames(samples)
samples$label &lt;- as.factor(samples$label)
# 2034 x 465536 counts matrix.</code></pre>
<p>Filtered out peaks with accessbility in fewer than 20 cells</p>
<pre class="r"><code>i      &lt;- which(colSums(counts) &gt;= 20)
peaks  &lt;- peaks[i,]
counts &lt;- counts[,i]
cat(sprintf(&quot;After filtering, we have %d rows and %d columns left. \n&quot;,nrow(counts),ncol(counts)))</code></pre>
<p>Load data after filtering peaks with accessbility in fewer than 20 cells</p>
<pre class="r"><code>res.dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018/binarized/filtered_peaks&quot;
load(file.path(res.dir, &quot;Buenrostro_2018_binarized_filtered.RData&quot;))
cat(sprintf(&quot;%d x %d counts matrix.\n&quot;,nrow(counts),ncol(counts)))
samples$cell &lt;- rownames(samples)
samples$label &lt;- as.factor(samples$label)
# 2034 x 126719 counts matrix.</code></pre>
</div>
<div id="topic-model-fit" class="section level3">
<h3>Topic model fit</h3>
<div id="k-8" class="section level4">
<h4>K = 8</h4>
<p>Load the K = 8 topic model fit</p>
<pre class="r"><code>fit &lt;- readRDS(file.path(res.dir, &quot;/fit-Buenrostro2018-binarized-filtered-scd-ex-k=8.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)</code></pre>
<p>Structure plot</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;darkorange&quot;,&quot;limegreen&quot;,&quot;magenta&quot;,&quot;gold&quot;,&quot;skyblue&quot;,
                  &quot;darkblue&quot;,&quot;dodgerblue&quot;,&quot;darkmagenta&quot;)

set.seed(1)
# labels &lt;- factor(samples$label, levels = c(&quot;HSC&quot;, &quot;MPP&quot;, &quot;CMP&quot;, &quot;GMP&quot;, &quot;mono&quot;, &quot;MEP&quot;, &quot;LMPP&quot;, &quot;CLP&quot;, &quot;pDC&quot;, &quot;UNK&quot;))

labels &lt;- factor(samples$label, c(&quot;mono&quot;,&quot;pDC&quot;,&quot;MEP&quot;,&quot;HSC&quot;,&quot;MPP&quot;,&quot;CLP&quot;,
                                 &quot;LMPP&quot;,&quot;CMP&quot;,&quot;GMP&quot;,&quot;UNK&quot;))
structure_plot(fit,grouping = labels,colors = topic_colors,
               # topics = 1:10,
               gap = 20,perplexity = 50,verbose = FALSE)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/structure-plot-k8-1.png" width="840" style="display: block; margin: auto;" /></p>
</div>
<div id="k-9" class="section level4">
<h4>K = 9</h4>
<p>Load the K = 9 topic model fit.</p>
<pre class="r"><code>fit &lt;- readRDS(file.path(res.dir, &quot;/fit-Buenrostro2018-binarized-filtered-scd-ex-k=9.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)</code></pre>
<p>Structure plot</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;darkorange&quot;,&quot;limegreen&quot;,&quot;magenta&quot;,&quot;gold&quot;,&quot;skyblue&quot;,
                  &quot;darkblue&quot;,&quot;dodgerblue&quot;,&quot;darkmagenta&quot;,&quot;red&quot;)

set.seed(1)
# labels &lt;- factor(samples$label, levels = c(&quot;HSC&quot;, &quot;MPP&quot;, &quot;CMP&quot;, &quot;GMP&quot;, &quot;mono&quot;, &quot;MEP&quot;, &quot;LMPP&quot;, &quot;CLP&quot;, &quot;pDC&quot;, &quot;UNK&quot;))

labels &lt;- factor(samples$label, c(&quot;mono&quot;,&quot;pDC&quot;,&quot;MEP&quot;,&quot;HSC&quot;,&quot;MPP&quot;,&quot;CLP&quot;,
                                 &quot;LMPP&quot;,&quot;CMP&quot;,&quot;GMP&quot;,&quot;UNK&quot;))
structure_plot(fit,grouping = labels,colors = topic_colors,
               # topics = 1:10,
               gap = 20,perplexity = 50,verbose = FALSE)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/structure-plot-k9-1.png" width="840" style="display: block; margin: auto;" /></p>
</div>
<div id="k-10" class="section level4">
<h4>K = 10</h4>
<p>Load the K = 10 topic model fit.</p>
<pre class="r"><code>fit &lt;- readRDS(file.path(res.dir, &quot;/fit-Buenrostro2018-binarized-filtered-scd-ex-k=10.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)</code></pre>
<p>Structure plot</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;darkorange&quot;,&quot;limegreen&quot;,&quot;magenta&quot;,&quot;gold&quot;,&quot;skyblue&quot;,
                  &quot;darkblue&quot;,&quot;dodgerblue&quot;,&quot;darkmagenta&quot;,&quot;red&quot;,&quot;olivedrab&quot;)

set.seed(1)
# labels &lt;- factor(samples$label, levels = c(&quot;HSC&quot;, &quot;MPP&quot;, &quot;CMP&quot;, &quot;GMP&quot;, &quot;mono&quot;, &quot;MEP&quot;, &quot;LMPP&quot;, &quot;CLP&quot;, &quot;pDC&quot;, &quot;UNK&quot;))

labels &lt;- factor(samples$label, c(&quot;mono&quot;,&quot;pDC&quot;,&quot;MEP&quot;,&quot;HSC&quot;,&quot;MPP&quot;,&quot;CLP&quot;,
                                 &quot;LMPP&quot;,&quot;CMP&quot;,&quot;GMP&quot;,&quot;UNK&quot;))
structure_plot(fit,grouping = labels,colors = topic_colors,
               # topics = 1:10,
               gap = 20,perplexity = 50,verbose = FALSE)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/structure-plot-k10-1.png" width="840" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="k-means-clustering-on-topic-proportions" class="section level3">
<h3>K-means clustering on topic proportions</h3>
<p>Define clusters using k-means, and then create structure plot based on the clusters from k-means.</p>
<p>k-means clustering (using 12 clusters) on topic proportions</p>
<pre class="r"><code>set.seed(1)
clusters &lt;- factor(kmeans(fit$L,centers = 12,iter.max = 100)$cluster)
summary(clusters)

structure_plot(fit,grouping = clusters,colors = topic_colors,
               gap = 20,perplexity = 50,verbose = FALSE)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/structure-plot-kmeans-12clusters-1.png" width="840" style="display: block; margin: auto;" /></p>
<pre><code>#   1   2   3   4   5   6   7   8   9  10  11  12 
# 262 244 125 132 150 179 180 153  89  72 257 191</code></pre>
</div>
<div id="differential-accessibility-da-analysis-on-topic-modeling-result-with-k-10" class="section level3">
<h3>Differential accessibility (DA) analysis on topic modeling result with <span class="math inline">\(K = 10\)</span></h3>
<p>DA analysis results <strong>without shrinkage</strong> (10000 MCMC iterations)</p>
<pre class="r"><code>DA_dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018/binarized/filtered_peaks/DAanalysis-Buenrostro2018-k=10&quot;

DA_res &lt;- readRDS(file.path(DA_dir, &quot;DA_regions_topics_vsnull_noshrinkage_10000iters.rds&quot;))
summary(DA_res)
#          Length  Class  Mode   
# ar       1267190 -none- numeric
# est      1267190 -none- numeric
# postmean 1267190 -none- numeric
# lower    1267190 -none- numeric
# upper    1267190 -none- numeric
# z        1267190 -none- numeric
# lpval    1267190 -none- numeric
# svalue         1 -none- numeric
# lfsr           1 -none- numeric
# F        1267190 -none- numeric
# f0        126719 -none- numeric</code></pre>
<p>Number of regions selected at different p-value cutoffs:</p>
<pre class="r"><code>sig_regions &lt;- matrix(NA, nrow = 10, ncol = 3)
colnames(sig_regions) &lt;- c(&quot;p &lt; 0.01&quot;, &quot;p &lt; 0.05&quot;, &quot;p &lt; 0.1&quot;)
rownames(sig_regions) &lt;- paste(&quot;topic&quot;, 1:nrow(sig_regions))

for(k in 1:10){
  lpval &lt;- DA_res$lpval[,k]
  pval &lt;- 10^(-lpval)
  sig_regions[k, ] &lt;- c(length(which(pval &lt; 0.01)), length(which(pval &lt; 0.05)), length(which(pval &lt; 0.1)))
}

sig_regions
#          p &lt; 0.01 p &lt; 0.05 p &lt; 0.1
# topic 1     15065    21923   28375
# topic 2     12633    20427   27623
# topic 3     16774    23671   29901
# topic 4     15825    20964   25905
# topic 5     15603    21209   24833
# topic 6      6986     9405   11438
# topic 7     14769    20177   24667
# topic 8     12989    20187   26041
# topic 9     15614    20703   24928
# topic 10     9455    14334   17737</code></pre>
<p>Volcano plots for the regions</p>
<pre class="r"><code>plots &lt;- vector(&quot;list&quot;,10)
names(plots) &lt;- 1:10

for (k in 1:10)
  plots[[k]] &lt;- volcano_plot(DA_res, k, labels = rep(&quot;&quot;,nrow(DA_res$z)))
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
# lfsr is not available, probably because &quot;shrink.method&quot; was not set to &quot;ash&quot;; lfsr in plot should be ignored
do.call(plot_grid,plots)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/region-volcano-plots-no-shrinkage-1.png" width="1200" style="display: block; margin: auto;" /></p>
<p>DA analysis results <strong>with &quot;ash&quot; shrinkage</strong> (10000 MCMC iterations)</p>
<pre class="r"><code>DA_dir &lt;- &quot;/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018/binarized/filtered_peaks/DAanalysis-Buenrostro2018-k=10&quot;

DA_res &lt;- readRDS(file.path(DA_dir, &quot;DA_regions_topics_vsnull_ash_10000iters.rds&quot;))
summary(DA_res)

dim(DA_res$z)
#          Length  Class  Mode   
# ar       1267190 -none- numeric
# est      1267190 -none- numeric
# postmean 1267190 -none- numeric
# lower    1267190 -none- numeric
# upper    1267190 -none- numeric
# z        1267190 -none- numeric
# lfsr     1267190 -none- numeric
# lpval          1 -none- numeric
# svalue   1267190 -none- numeric
# ash            3 ash    list   
# F        1267190 -none- numeric
# f0        126719 -none- numeric
# [1] 126719     10</code></pre>
<p>Number of regions selected at different lfsr cutoffs:</p>
<pre class="r"><code>sig_regions &lt;- matrix(NA, nrow = 10, ncol = 3)
colnames(sig_regions) &lt;- c(&quot;lfsr &lt; 0.01&quot;, &quot;lfsr &lt; 0.05&quot;, &quot;lfsr &lt; 0.1&quot;)
rownames(sig_regions) &lt;- paste(&quot;topic&quot;, 1:nrow(sig_regions))

for(k in 1:10){
  lfsr &lt;- DA_res$lfsr[,k]
  sig_regions[k, ] &lt;- c(length(which(lfsr &lt; 0.01)), length(which(lfsr &lt; 0.05)), length(which(lfsr &lt; 0.1)))
}

sig_regions
#          lfsr &lt; 0.01 lfsr &lt; 0.05 lfsr &lt; 0.1
# topic 1        15066       21034      25515
# topic 2        11164       18536      24600
# topic 3        15139       20590      25533
# topic 4        14210       19177      23482
# topic 5        15240       22228      26977
# topic 6         6892        9588      11650
# topic 7        13505       19528      24046
# topic 8         9983       17102      22741
# topic 9        13028       18154      22778
# topic 10        6605       10974      14385</code></pre>
<p>Volcano plots for the regions</p>
<pre class="r"><code>plots &lt;- vector(&quot;list&quot;,10)
names(plots) &lt;- 1:10

for (k in 1:10)
  plots[[k]] &lt;- volcano_plot(DA_res, k, labels = rep(&quot;&quot;,nrow(DA_res$z)))
do.call(plot_grid,plots)</code></pre>
<p><img src="figure/structureplot_DA_Buenrostro2018_filteredpeaks_k10.Rmd/region-volcano-plots-ash-1.png" width="1200" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.0.4 (2021-02-15)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#  [1] reshape_0.8.8      DT_0.21            RColorBrewer_1.1-2 plyr_1.8.6        
#  [5] cowplot_1.1.1      ggplot2_3.3.5      dplyr_1.0.8        Matrix_1.4-1      
#  [9] fastTopics_0.6-97  workflowr_1.7.0   
# 
# loaded via a namespace (and not attached):
#  [1] mcmc_0.9-7         fs_1.5.2           progress_1.2.2     httr_1.4.2        
#  [5] rprojroot_2.0.2    tools_4.0.4        bslib_0.3.1        utf8_1.2.2        
#  [9] R6_2.5.1           irlba_2.3.5        uwot_0.1.11        DBI_1.1.2         
# [13] lazyeval_0.2.2     colorspace_2.0-3   withr_2.5.0        tidyselect_1.1.2  
# [17] prettyunits_1.1.1  processx_3.5.2     compiler_4.0.4     git2r_0.30.1      
# [21] cli_3.2.0          quantreg_5.88      SparseM_1.81       plotly_4.10.0     
# [25] labeling_0.4.2     sass_0.4.1         scales_1.1.1       SQUAREM_2021.1    
# [29] quadprog_1.5-8     callr_3.7.0        pbapply_1.5-0      mixsqp_0.3-43     
# [33] stringr_1.4.0      digest_0.6.29      rmarkdown_2.13     MCMCpack_1.6-1    
# [37] pkgconfig_2.0.3    htmltools_0.5.2    highr_0.9          fastmap_1.1.0     
# [41] invgamma_1.1       htmlwidgets_1.5.4  rlang_1.0.2        rstudioapi_0.13   
# [45] farver_2.1.0       jquerylib_0.1.4    generics_0.1.2     jsonlite_1.8.0    
# [49] magrittr_2.0.2     Rcpp_1.0.8.3       munsell_0.5.0      fansi_1.0.3       
# [53] lifecycle_1.0.1    stringi_1.7.6      whisker_0.4        yaml_2.3.5        
# [57] MASS_7.3-56        Rtsne_0.15         grid_4.0.4         parallel_4.0.4    
# [61] promises_1.2.0.1   ggrepel_0.9.1      crayon_1.5.0       lattice_0.20-45   
# [65] hms_1.1.1          knitr_1.37         ps_1.6.0           pillar_1.7.0      
# [69] glue_1.6.2         evaluate_0.14      getPass_0.2-2      data.table_1.14.2 
# [73] RcppParallel_5.1.5 vctrs_0.3.8        httpuv_1.6.5       MatrixModels_0.5-0
# [77] gtable_0.3.0       purrr_0.3.4        tidyr_1.2.0        assertthat_0.2.1  
# [81] ashr_2.2-54        xfun_0.30          coda_0.19-4        later_1.3.0       
# [85] viridisLite_0.4.0  truncnorm_1.0-8    tibble_3.1.6       ellipsis_0.3.2</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
