---
title: "Exploring Cicero co-assessibility data in kidney data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

The aim of this short analysis is to get a better understanding of the
Cicero co-accessibility data, and how (and whether) these data can be
used to connect chromatin accessibility peaks to genes in order to
identify "driving genes" for topics estimated from single-cell
ATAC-seq data. As an illustration, here we focus on the Cicero data
for a single gene, *Slc12a1*, that was highlighted in
[Cusanovich *et al*, 2018][cusanovich-2018-cell] in connection with
the "loop of henle" cell type (see Fig. 5 of that paper, and see also
[Park *et al*, 2018][park-2018-science]).

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below.

```{r load-pkgs, message=FALSE}
library(fastTopics)
library(ggplot2)
library(cowplot)
library(ashr)
```

Load the base-pair positions of the genes for the mm9 Mouse Genome
Assembly.

```{r load-genes}
load("data/mm9_seq_gene.RData")
```

Load the Cicero co-accessibility data, including the "gene activity
scores", for gene *Slc12a1.* (These data were downloaded from the
[Mouse sci-ATAC-seq Atlas website][mouse-atlas] then prepared using
the `extract_slc12a1_data.R` script.)

```{r load-cicero}
load("data/Cusanovich_2018/processed_data/slc12a1_data.RData")
cicero <- transform(cicero,
                    Peak1 = as.character(Peak1),
                    Peak2 = as.character(Peak2))
```

Load the $K = 10$ topic model fit, and the results of the DE analysis
using this topic model (without the adaptive shrinkage step).

```{r load-topics}
fit <- readRDS(file.path("output/Cusanovich_2018/tissues",
                         "fit-Cusanovich2018-Kidney-scd-ex-k=10.rds"))$fit
fit <- poisson2multinom(fit)
load(file.path("output/Cusanovich_2018/tissues",
	       "de-cusanovich2018-kidney-k=10-noshrink.RData"))
```

From the Structure plots [here](cusanovich2018_kidney_k10.html), topic
8 appears to capture Loop of Henle (LoH) cells, so in the remainder 
we focus on topic 8.

```{r which-topic}
k <- 8
```

Get the base-pair positions of the peaks.

```{r get-positions}
feature_names <- rownames(de$postmean)
out           <- strsplit(feature_names,"_")
positions     <- data.frame(chr   = sapply(out,"[[",1),
                            start = sapply(out,"[[",2),
                            end   = sapply(out,"[[",3),
                            name  = feature_names,
                            stringsAsFactors = FALSE)
positions <- transform(positions,
                       start = as.numeric(start),
                       end   = as.numeric(end))
```

Before examining the co-accessibility data in detail, this first plot
confirms that *Slc12a1* is highly relevant to topic 8, the LoH topic.
It is a simple scatterplot showing the *Slc12a1* gene activity score
and topic proportion for each cell. The dashed blue line shows the
"best fit" line between the two quantities. (Note that the gene
activity scores are shown on the log-scale.)

```{r scores-vs-loadings-scatterplot, fig.height=3, fig.width=3}
pdat <- data.frame(loading = fit$L[,k],score = log10(1 + scores))
b <- coef(lm(score ~ loading,data = pdat))
ggplot(pdat,aes(x = loading,y = score)) +
  geom_point() +
  geom_abline(intercept = b["(Intercept)"],slope = b["loading"],
              color = "dodgerblue",size = 1,linetype = "dashed") +
  labs(x = "topic proportion",y = "gene activity score") +
  theme_cowplot()
```

Having confirmed the strong relationship between *Slc12a1* and topic
8, let's examine more closely the `de_analysis` results for the peaks
near *Slc12a1*. Let's use a window of 400 kb near the transcribed
region of *Slc12a1* (124.97--125.06 Mb on chromosome 2).

```{r get-peaks-near-slc12a1}
d <- 4e5
seq_gene <- subset(seq_gene,feature_name == "Slc12a1")
rows <- which(with(positions,chr == paste0("chr",seq_gene$chromosome) &
                   start > seq_gene$chr_start - d &
                   end < seq_gene$chr_stop + d))
peaks <- c(cicero$Peak1,cicero$Peak2)
```

These next two plots show the *p*-values (top) and the LFC estimates
for topic 8 and for the `r length(rows)` peaks near *Slc12a1*. The
genes that are identified by Cicero as being relevant to *Slc12a1* are
highlighted in magenta.

```{r de-slc12a1, fig.height=5, fig.width=5}
pdat <- data.frame(start      = positions[rows,"start"]/1e5,
                   postmean   = de$postmean[rows,k],
                   z          = de$z[rows,k],
                   lpval      = de$lpval[rows,k],
                   cicero_hit = is.element(positions[rows,"name"],peaks))
p1 <- ggplot(pdat,aes(x = start,y = lpval,color = cicero_hit)) +
  geom_point() +
  scale_color_manual(values = c("royalblue","magenta")) +
  labs(x = "position on chromosome 2 (kb)",
       y = "-log10 p-value") +
  theme_cowplot()
p2 <- ggplot(pdat,aes(x = start,y = postmean,color = cicero_hit)) +
  geom_point() +
  geom_hline(yintercept = 0,color = "black",linetype = "dashed") +
  scale_color_manual(values = c("royalblue","magenta")) +
  labs(x = "position on chromosome 2 (kb)",
       y = "LFC estimate") +
  theme_cowplot()
plot_grid(p1,p2,nrow = 2,ncol = 1)
```

It is interesting that most of the Cicero-identified peaks show good
evidence for being more accessible in topic 8. But it is also
interesting that many other peaks near the gene also show good
evidence for being more accessible in topic 8.

To improve the LFC estimates, a simple thing we can do is run ash
separately on the subset of peaks that are near *Slc12a1*. By
focussing on this small subset of peaks, ash should recognize that
there is a strong signal near the gene, and not shrink the estimates
too strongly.

```{r run-ashr}
pdat <- transform(pdat,se = postmean/z)
fit <- ash(pdat$postmean,pdat$se,mixcompdist = "normal",method = "shrink")
```

Indeed, ash preserves the strongest signals, and shrinks the others to
zero, or close to zero.  It is also interesting that the effects of
all the Cicero-identified peaks remain after the adaptive shrinkage
step:

```{r shrinkage-scatterplot, fig.height=3, fig.width=4}
pdat$ashmean <- fit$result$PosteriorMean
ggplot(pdat,aes(x = postmean,y = ashmean,color = cicero_hit)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dashed") +
  scale_color_manual(values = c("royalblue","magenta")) +
  labs(x = "original LFC estimate",y = "ash LFC estimate") +
  theme_cowplot()
```

[mouse-atlas]: https://atlas.gs.washington.edu/mouse-atac/
[cusanovich-2018-cell]: https://doi.org/10.1016/j.cell.2018.06.052
[park-2018-science]: https://doi.org/10.1126/science.aar2131
