---
title: "Exploring Cicero co-assessibility data in kidney data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

The aim of this short analysis is to get a better understanding of the
Cicero co-accessibility data, and how (and whether) these data can be
used to connect chromatin accessibility peaks to genes in order to
identify "driving genes" for topics estimated from single-cell
ATAC-seq data. As an illustration, here we focus on the Cicero data
for a single gene, *Slc12a1*, that was highlighted in
[Cusanovich *et al*, 2018][cusanovich-2018-cell] in connection with
the "loop of henle" cell type (see Fig. 5 of that paper, and see also
[Park *et al*, 2018][park-2018-science]).

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below.

```{r load-pkgs, message=FALSE}
library(fastTopics)
library(ggplot2)
library(cowplot)
library(ashr)
```

Load the base-pair positions of the genes for the mm9 Mouse Genome
Assembly.

```{r load-genes}
load("data/mm9_seq_gene.RData")
```

Load the Cicero co-accessibility data, including the "gene activity
scores", for gene *Slc12a1.* (These data were downloaded from the
[Mouse sci-ATAC-seq Atlas website][mouse-atlas] then prepared using
the `extract_slc12a1_data.R` script.)

```{r load-cicero}
load("data/Cusanovich_2018/processed_data/slc12a1_data.RData")
cicero <- transform(cicero,
                    Peak1 = as.character(Peak1),
                    Peak2 = as.character(Peak2))
```

Load the $K = 10$ topic model fit, and the results of the DE analysis
using this topic model (without the adaptive shrinkage step).

```{r load-topics}
fit <- readRDS(file.path("output/Cusanovich_2018/tissues",
                         "fit-Cusanovich2018-Kidney-scd-ex-k=10.rds"))$fit
fit <- poisson2multinom(fit)
load(file.path("output/Cusanovich_2018/tissues",
	       "de-cusanovich2018-kidney-k=10-noshrink.RData"))
```

From the Structure plots [here](cusanovich2018_kidney_k10.html), topic
8 appears to capture Loop of Henle (LoH) cells, so in the remainder 
we focus on topic 8.

```{r which-topic}
k <- 8
```

Get the base-pair positions of the peaks.

```{r get-positions}
feature_names <- rownames(de$postmean)
out           <- strsplit(feature_names,"_")
positions     <- data.frame(chr   = sapply(out,"[[",1),
                            start = sapply(out,"[[",2),
                            end   = sapply(out,"[[",3),
                            name  = feature_names,
                            stringsAsFactors = FALSE)
positions <- transform(positions,
                       start = as.numeric(start),
                       end   = as.numeric(end))
```

Before examining the co-accessibility data in detail, this first plot
confirms that *Slc12a1* is highly relevant to topic 8, the LoH topic.
It is a simple scatterplot showing the *Slc12a1* gene activity score
and topic proportion for each cell. (Note that the gene activity
scores are shown on the log-scale.)

```{r scores-vs-loadings-scatterplot, fig.height=3, fig.width=3}
pdat <- data.frame(loading = fit$L[,k],score = log10(1 + scores))
b <- coef(lm(score ~ loading,data = pdat))
ggplot(pdat,aes(x = loading,y = score)) +
  geom_point() +
  geom_abline(intercept = b["(Intercept)"],slope = b["loading"],
              color = "dodgerblue",size = 1,linetype = "dashed") +
  labs(x = "topic proportion",y = "gene activity score") +
  theme_cowplot()
```

[mouse-atlas]: https://atlas.gs.washington.edu/mouse-atac/
[cusanovich-2018-cell]: https://doi.org/10.1016/j.cell.2018.06.052
[park-2018-science]: https://doi.org/10.1126/science.aar2131
