---
title: "Gene analysis using topic modeling and DA results (v2) for Buenrostro *et al* (2018) scATAC-seq result (data processed using Chen 2019 pipeline)"
author: Kaixuan Luo
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: false
---

Here we perform TF motif and gene analysis
for the Buenrostro *et al* (2018) scATAC-seq result inferred 
from the multinomial topic model with $k = 11$.

We use binarized scPeaks and scATAC-seq data was processed using Chen *et al* (2019) pipeline.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = FALSE,results = "hold",
                      fig.align = "center",dpi = 120)
```

## Load packages and some functions used in this analysis
```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape)
library(pathways)

source("code/plots.R")

```

## Load data and topic model results
Load the **binarized** data and the $k = 11$ Poisson NMF fit results 
```{r load-data}
data.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/"
load(file.path(data.dir, "Buenrostro_2018_binarized_counts.RData"))
cat(sprintf("%d x %d counts matrix.\n",nrow(counts),ncol(counts)))
```

```{r load-fit}
fit.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_Chen2019pipeline/binarized/"
fit <- readRDS(file.path(fit.dir, "/fit-Buenrostro2018-binarized-scd-ex-k=11.rds"))$fit
fit <- poisson2multinom(fit)
```

## Visualize by Structure plot grouped by cell labels.
```{r structure-plot, fig.width=7, fig.height=2.5, message=FALSE}
set.seed(10)
colors_topics <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c",
                   "#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","#b15928",
                   "gray")
samples$label <- as.factor(samples$label)

p.structure <- structure_plot(fit,
                     grouping = samples[, "label"],n = Inf,gap = 40,
                     perplexity = 50,colors = colors_topics,
                     num_threads = 6,verbose = FALSE)

print(p.structure)
```

## Differential accessbility analysis of the ATAC-seq regions for the topics

Load results from differential accessbility analysis for the topics
```{r}
out.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_Chen2019pipeline/binarized/postfit_v2"
cat(sprintf("Load results from %s \n", out.dir))
DA_res <- readRDS(file.path(out.dir, paste0("DAanalysis-Buenrostro2018-k=11/DA_regions_topics_10000iters.rds")))
```

**Volcano plot of the regions**
```{r region-volcano-plots, fig.width=8, fig.height=6}
m <- ncol(DA_res$z)
plots <- vector("list",m)
names(plots) <- colnames(DA_res$z)
for (k in 1:m) {
  plots[[k]] <- volcano_plot(DA_res, k = k,labels = rep("",nrow(DA_res$z)))
}
do.call(plot_grid, plots)
```

## Gene score analysis

Set output directorry
```{r set-dir}
out.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_Chen2019pipeline/binarized/postfit_v2"

fig.dir <- "output/plotly/Buenrostro_2018_Chen2019pipeline_v2/"
dir.create(fig.dir, showWarnings = F, recursive = T)
```


### TSS model version 1

Gene scores were computed using TSS based method as in Lareau *et al* Nature Biotech, 2019 as well as the `model 21` in `archR` paper.
This model weights chromatin accessibility around gene promoters by using bi-directional exponential decays from the TSS.

* TSS model, normalized by the l2 norm of weights, as in Stouffer's z-score method
* use abs(z) scores

```{r top-genes-tss-1}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-TSS-absZ-l2")

cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result.rds"))

genes <- genescore_res$genes
gene_scores <- genescore_res$Z
gene_logFC <- genescore_res$logFC

topics <- colnames(gene_scores)
top_genes <- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) <- topics

for (k in topics){
  top_genes[,k] <- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = "Top 10 genes by abs(gene z-scores)")

```


* Volcano plots of gene scores

**all topics**
```{r gene-volcano-plots-tss, fig.width=10, fig.height=6, message=FALSE, warning=FALSE}
genescore_volcano_plot(genescore_res, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)
```

**topic 1 and topic 4 examples**
```{r gene-volcano-plot-tss-1, fig.width=8, fig.height=3, message=FALSE, warning=FALSE}
p.volcano.1 <- genescore_volcano_plot(genescore_res, k = 1, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

p.volcano.4 <- genescore_volcano_plot(genescore_res, k=4, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

plot_grid(p.volcano.1, p.volcano.4)
```

### TSS model version 2

Gene scores were computed using TSS based method as in Lareau *et al* Nature Biotech, 2019 as well as the `model 21` in `archR` paper.
This model weights chromatin accessibility around gene promoters by using bi-directional exponential decays from the TSS.

* TSS model, normalized by the sum of weights
* use abs(z) scores

```{r top-genes-tss-2}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-TSS-absZ-sum")

cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result.rds"))

genes <- genescore_res$genes
gene_scores <- genescore_res$Z
gene_logFC <- genescore_res$logFC

topics <- colnames(gene_scores)
top_genes <- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) <- topics

for (k in topics){
  top_genes[,k] <- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = "Top 10 genes by abs(gene z-scores)")

```


* Volcano plots of gene scores

**all topics**
```{r gene-volcano-plots-tss-2, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
genescore_volcano_plot(genescore_res, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)
```

**topic 1 and topic 4 examples**
```{r gene-volcano-plot-tss-2, fig.width=8, fig.height=3, message=FALSE, warning=FALSE}
p.volcano.1 <- genescore_volcano_plot(genescore_res, k = 1, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

p.volcano.4 <- genescore_volcano_plot(genescore_res, k=4, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

plot_grid(p.volcano.1, p.volcano.4)
```

### Gene body model version 1

Gene scores were computed using the gene score model (model 42) in the `archR` paper with some modifications.
This model uses bi-directional exponential decays from the gene TSS (extended upstream by 5 kb by default)
and the gene transcription termination site (TTS).
Note: the current version of the function does not account for neighboring gene boundaries.

* Gene body model, normalized by the l2 norm of weights, as in Stouffer's z-score method
* use abs(z) scores

```{r top-genes-genebody-1}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-genebody-Z-l2")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result.rds"))

genes <- genescore_res$genes
gene_scores <- genescore_res$Z
gene_logFC <- genescore_res$logFC

topics <- colnames(gene_scores)
top_genes <- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) <- topics

for (k in topics){
  top_genes[,k] <- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = "Top 10 genes by abs(gene z-scores)")

```


* Volcano plots of gene scores

**all topics**
```{r gene-volcano-plots-genebody-1, fig.width=10, fig.height=6, message=FALSE, warning=FALSE}
genescore_volcano_plot(genescore_res, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)
```


**topic 1 and topic 4 examples**
```{r gene-volcano-plot-genebody-1, fig.width=8, fig.height=3, message=FALSE, warning=FALSE}
p.volcano.1 <- genescore_volcano_plot(genescore_res, k = 1, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

p.volcano.4 <- genescore_volcano_plot(genescore_res, k=4, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

plot_grid(p.volcano.1, p.volcano.4)
```

### Gene body model version 2

Gene scores were computed using the gene score model (model 42) in the `archR` paper with some modifications.
This model uses bi-directional exponential decays from the gene TSS (extended upstream by 5 kb by default)
and the gene transcription termination site (TTS).
Note: the current version of the function does not account for neighboring gene boundaries.

* Gene body model, normalized by the sum of weights
* use abs(z) scores

```{r top-genes-genebody-2}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-genebody-Z-sum")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result.rds"))

genes <- genescore_res$genes
gene_scores <- genescore_res$Z
gene_logFC <- genescore_res$logFC

topics <- colnames(gene_scores)
top_genes <- data.frame(matrix(nrow=10, ncol = ncol(gene_scores)))
colnames(top_genes) <- topics

for (k in topics){
  top_genes[,k] <- genes$SYMBOL[head(order(abs(gene_scores[,k]), decreasing=TRUE), 10)]
}

DT::datatable(data.frame(rank = 1:10, top_genes), rownames = F, caption = "Top 10 genes by abs(gene z-scores)")

```


* Volcano plots of gene scores

**all topics**
```{r gene-volcano-plots-genebody-2, fig.width=10, fig.height=6, message=FALSE, warning=FALSE}
genescore_volcano_plot(genescore_res, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)
```

**topic 1 and topic 4 examples**
```{r gene-volcano-plot-genebody-2, fig.width=8, fig.height=3, message=FALSE, warning=FALSE}
p.volcano.1 <- genescore_volcano_plot(genescore_res, k = 1, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

p.volcano.4 <- genescore_volcano_plot(genescore_res, k=4, label_above_quantile = 0.99,
                       labels = genescore_res$genes$SYMBOL, max.overlaps = 20,
                       subsample_below_quantile = 0.5, subsample_rate = 0.1)

plot_grid(p.volcano.1, p.volcano.4)
```

## Gene-set enrichment analysis (GSEA)

Loading gene set data
```{r load-gene-sets}
cat("Loading human gene set data.\n")
data(gene_sets_human)
gene_sets <- gene_sets_human$gene_sets
gene_set_info <- gene_sets_human$gene_set_info
```


* TSS model, normalized by the l2 norm of weights, as in Stouffer's z-score method
* use abs(z) scores
```{r top-pathways-tss-1}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-TSS-absZ-l2")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
gsea_res <- readRDS(file.path(gene.dir, "gsea_result.rds"))

top_pathways_up <- top_pathways_down <- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) <- colnames(top_pathways_down) <- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic <- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up <- gsea_topic[gsea_topic$ES > 0,]
  top_IDs_up <- as.character(gsea_up[head(order(gsea_up$pval), 10), "pathway"])
  top_IDs_up <- gene_set_info[match(top_IDs_up, gene_set_info$id),c("name", "id")]
  top_pathways_up[,k] <- paste0(top_IDs_up$name, "(", top_IDs_up$id, ")")
  
  gsea_down <- gsea_topic[gsea_topic$ES < 0,]
  top_IDs_down <- as.character(gsea_down[head(order(gsea_down$pval), 10), "pathway"])
  top_IDs_down <- gene_set_info[match(top_IDs_down, gene_set_info$id),c("name", "id")]
  top_pathways_down[,k] <- paste0(top_IDs_down$name, "(", top_IDs_down$id, ")")
  
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = "Top 10 pathways enriched at the top of the gene rank list.")

```

* Gene body model, normalized by the l2 norm of weights, as in Stouffer's z-score method. 
```{r top-pathways-genebody-1}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-genebody-absZ-l2")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
gsea_res <- readRDS(file.path(gene.dir, "gsea_result.rds"))

top_pathways_up <- top_pathways_down <- data.frame(matrix(nrow=10, ncol = ncol(gsea_res$pval)))
colnames(top_pathways_up) <- colnames(top_pathways_down) <- colnames(gsea_res$pval)

for (k in 1:ncol(gsea_res$pval)){
  gsea_topic <- data.frame(pathway = rownames(gsea_res$pval),  
                           pval = gsea_res$pval[,k],
                           log2err = gsea_res$log2err[,k],
                           ES = gsea_res$ES[,k],
                           NES = gsea_res$NES[,k])
  gsea_up <- gsea_topic[gsea_topic$ES > 0,]
  top_IDs_up <- as.character(gsea_up[head(order(gsea_up$pval), 10), "pathway"])
  top_IDs_up <- gene_set_info[match(top_IDs_up, gene_set_info$id),c("name", "id")]
  top_pathways_up[,k] <- paste0(top_IDs_up$name, "(", top_IDs_up$id, ")")
  
  gsea_down <- gsea_topic[gsea_topic$ES < 0,]
  top_IDs_down <- as.character(gsea_down[head(order(gsea_down$pval), 10), "pathway"])
  top_IDs_down <- gene_set_info[match(top_IDs_down, gene_set_info$id),c("name", "id")]
  top_pathways_down[,k] <- paste0(top_IDs_down$name, "(", top_IDs_down$id, ")")
  
}

DT::datatable(data.frame(rank = 1:10, top_pathways_up), rownames = F,
              caption = "Top 10 pathways enriched at the top of the gene rank list.")

```
