---
title: "Motif analysis using topic modeling results for Buenrostro *et al* (2018) scATAC-seq result (data processed using chromVAR processed counts on peaks called from Chen 2019, filtering regions with accessible in > 1% samples)"
author: Kaixuan Luo
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: false
---

Here we perform TF motif analysis
for the Buenrostro *et al* (2018) scATAC-seq result inferred 
from the multinomial topic model with $k = 11$.

scATAC-seq peaks were called using Chen *et al* (2019) pipeline, 
and we counted the reads in peaks using chromVAR, and further filtering peaks
requiring peaks to be accessible in > 1% samples.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = FALSE,results = "hold",
                      fig.align = "center",dpi = 120)
```

## Load packages and some functions used in this analysis
```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape2)
library(Logolas)
library(grid)
source("code/motif_analysis.R")
source("code/plots.R")
```

## Load data and topic model results
Load the **binarized** data and the $k = 11$ Poisson NMF fit results 
from Buenrostro *et al* 2018 dataset using binarized scPeaks 
based on chromVAR processed counts with peaks accessible in at least 1 sample.

```{r load-data}
data.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/data/Buenrostro_2018/processed_data_Chen2019pipeline/chromVAR/"
load(file.path(data.dir, "Buenrostro_2018_binarized_scPeaks_filtered.RData"))
cat(sprintf("%d x %d counts matrix.\n",nrow(counts),ncol(counts)))
```

```{r load-fit}
fit.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized/"
fit <- readRDS(file.path(fit.dir, "/fit-Buenrostro2018-binarized-scd-ex-k=11.rds"))$fit
fit_multinom <- poisson2multinom(fit)
```

## Visualize by Structure plot grouped by cell labels.
```{r structure-plot, fig.width=7, fig.height=2.5, message=FALSE}
set.seed(10)
colors_topics <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c",
                   "#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","#b15928",
                   "gray")
samples$label <- as.factor(samples$label)

p.structure <- structure_plot(fit_multinom,
                     grouping = samples[, "label"],n = Inf,gap = 40,
                     perplexity = 50,topics = 1:11,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)

print(p.structure)
```

## Differential accessbility analysis of the ATAC-seq regions for the topics

Load results from differential accessbility analysis for the topics
```{r}
out.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Buenrostro_2018_chromVAR_scPeaks_filtered/binarized/"
cat(sprintf("Load results from %s \n", out.dir))
```

```{r diff_count_topics,message=FALSE}
diffcount_file <- file.path(out.dir, "diffcount-Buenrostro2018-11topics.rds")
if(file.exists(diffcount_file)){
  cat("Load precomputed differential accessbility statistics.\n")
  diff_count_topics <- readRDS(diffcount_file)
}else{
  cat("Computing differential accessbility statistics from topic model.\n")
  timing <- system.time(diff_count_topics <- diff_count_analysis(fit,counts))
  cat(sprintf("Computation took %0.2f seconds.\n",timing["elapsed"]))
  cat("Saving results.\n")
  saveRDS(diff_count_topics, diffcount_file)
}

```

Distribution of z-scores
```{r hist-zscore, fig.width=7, fig.height=5}
zscore_topics <-  melt(diff_count_topics$Z)
colnames(zscore_topics) <- c("region", "topic", "zscore")
levels(zscore_topics$topic) <- colnames(diff_count_topics$Z)

z.quantile.99 <- apply(abs(diff_count_topics$Z), 2, quantile, 0.99)
cat("z-score 99% quantile: \n")
print(z.quantile.99)

p.hist.zscores <- ggplot(zscore_topics, aes(x=zscore)) + 
  geom_histogram(binwidth=1, color="black", fill="white") + 
  coord_cartesian(xlim = c(-10, 30)) + theme_cowplot(font_size = 10) +
  facet_wrap(~ topic, ncol=4)

print(p.hist.zscores)
```

Volcano plot of the regions

**topic 5 example**
```{r region-volcano-plot, fig.width=5, fig.height=4}
volcano_plot(diff_count_topics,k = 5,label_above_quantile = Inf, 
             subsample_below_quantile = 0.7, subsample_rate = 0.1)
```

## Motif enrichment analysis using [HOMER](http://homer.ucsd.edu/homer/ngs/peakMotifs.html)

* Details about HOMER motif analysis:
    - http://homer.ucsd.edu/homer/motif/
    - http://homer.ucsd.edu/homer/ngs/peakMotifs.html

* Motif enrichment result using regions with z-score above 99% quantile.

Compile Homer results across topics
```{r compile-homer-motif-results}
homer.dir <- paste0(out.dir, "/motifanalysis-Buenrostro2018-k=11-quantile/HOMER/")
cat(sprintf("Directory of motif analysis result: %s \n", homer.dir))
homer_res_topics <- readRDS(file.path(homer.dir, "/homer_knownResults.rds"))
selected_regions <- readRDS(file.path(homer.dir, "/selected_regions.rds"))

# Compile Homer results (pvalue and ranking) across topics
motif_res <- compile_homer_motif_res(homer_res_topics)
saveRDS(motif_res, paste0(homer.dir, "/homer_motif_enrichment_results.rds"))
cat("compiled homer motif results are saved in", paste0(homer.dir, "/homer_motif_enrichment_results.rds"))

motif_table <- data.frame(motif = gsub("/.*", "", rownames(motif_res$mlog10P)),
                          round(motif_res$mlog10P,2))
DT::datatable(motif_table, rownames = F, caption = "Motif enrichment (-log10P)")
```

### Top 10 motifs in each topic
```{r top-motifs}
cat("Number of regions selected for each topic: \n")
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

colnames_homer <- c("motif_name", "consensus", "P", "log10P", "Padj",  "num_target", "percent_target", "num_bg", "percent_bg")

top_motifs <- data.frame(matrix(nrow=10, ncol = length(homer_res_topics)))
colnames(top_motifs) <- names(homer_res_topics)
for (k in 1:length(homer_res_topics)){
  homer_res <- homer_res_topics[[k]]
  colnames(homer_res) <- colnames_homer
  homer_res <- homer_res %>% separate(motif_name, c("motif", "origin", "database"), "/")
  top_motifs[,k] <- head(homer_res$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F, caption = "Top 10 motifs enriched in each topic.")

```

### Heatmap of motif enrichment across topics

Heatmap of motif enrichment -log10(p-value). 
Order motifs by hierarchical clustering.
```{r motif-heatmap-pvalue, fig.width=7, fig.height=9}
create_motif_enrichment_heatmap(motif_res, enrichment = "-log10(p-value)", 
                                cluster_motifs = TRUE, cluster_topics = FALSE, motif_filter = 50, 
                                enrichment_range = c(0,100), method_cluster = "average", 
                                font.size.motifs = 5, font.size.topics = 9)

```

Heatmap of motif enrichment z-score.
Order motifs by hierarchical clustering.
```{r motif-heatmap-zscore, fig.width=7, fig.height=11}
create_motif_enrichment_heatmap(motif_res, enrichment = "z-score",
                                cluster_motifs = TRUE, cluster_topics = FALSE, motif_filter = 10, 
                                enrichment_range = c(-20,20), method_cluster = "average", 
                                font.size.motifs = 5, font.size.topics = 9)

```

### Scatterplots of motif enrichment

```{r motif-enrichment-ranking-plot-1, warning=FALSE}
# Plot enrichment (-log10 p-value) and ranking of the motifs

plots <- vector("list", ncol(motif_res$mlog10P))
names(plots) <- colnames(motif_res$mlog10P)

for( i in 1:length(plots)){
  plots[[i]] <- create_motif_enrichment_ranking_plot(motif_res, k = i, 
                                                     max.overlaps = 20, subsample = FALSE)
}

# do.call(plot_grid,plots)
```


* Plot motif enrichment (-log10 p-value) and the ranking

**Topic 5 example**
```{r motif-enrichment-ranking-plot-2, fig.width=5, fig.height=4, warning=FALSE}
print(plots[[5]])
```

```{r motif-enrichment-plot-1, warning=FALSE}
# Plot motif enrichment (-log10 p-value) in each topic vs other topics

plots <- vector("list", ncol(motif_res$mlog10P))
names(plots) <- colnames(motif_res$mlog10P)

for( i in 1:length(homer_res_topics)){
  plots[[i]] <- create_motif_enrichment_plot(motif_res, k = i, 
                                             max.overlaps = 20, subsample = TRUE)
}

# do.call(plot_grid,plots)

```

* Plot motif enrichment (-log10 p-value) in **topic 5** vs other topics

```{r motif-enrichment-plot-2, fig.width=5, fig.height=4, warning=FALSE}
print(plots[[5]])
```

### Motif enrichment vs gene score

Load pre-computed gene scores
```{r load-gene-scores}
gene.dir <- paste0(out.dir, "/geneanalysis-Buenrostro2018-k=11-TSS-l2")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result_topics.rds"))

genes <- genescore_res$genes
gene_scores <- genescore_res$Z
rownames(gene_scores) <- genes$SYMBOL
gene_logFC <- genescore_res$beta
rownames(gene_logFC) <- genes$SYMBOL

```

Get TF genes 
```{r get-TF-genes}
motif_names <- gsub("\\s*\\(.*", "", motif_res$motifs$motif)
gene_names <- genes$SYMBOL

TF_genes <- intersect(toupper(motif_names), toupper(gene_names))
cat(sprintf("%s TF genes mapped between motif names and gene symbol. \n", length(TF_genes)))

```

Compute correlation between motif enrichment z-score and gene score:

**Topic 5 example**

* Compute motif enrichment z-scores from the motif enrichment p-values
* Plot motif enrichment (-log10 p-value) and correlation between motif enrichment z-scores and gene scores
* Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment z-score and gene scores

```{r motif-cor-plot-1, fig.width=5, fig.height=4,  warning=FALSE}

motif_gene_mapping <- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 5, cor.motif = "z-score")

motif_gene_mapping <- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_zscore, decreasing = T)),]
rownames(motif_gene_mapping) <- 1:nrow(motif_gene_mapping)

cat("Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n")
print(head(motif_gene_mapping[,c("motif","motif_mlog10P", "gene_score", "cor_zscore")], 10))

```

* Plot motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores
* Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment (-log10 p-value) and gene scores

```{r motif-cor-plot-2, fig.width=5, fig.height=4, warning=FALSE}

motif_gene_mapping <- create_motif_gene_cor_scatterplot(motif_res, gene_scores, motif_names, gene_names, TF_genes, 
                                                        k = 5, cor.motif = "-log10(p-value)")

motif_gene_mapping <- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_mlog10P, decreasing = T)),]
rownames(motif_gene_mapping) <- 1:nrow(motif_gene_mapping)

cat("Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n")
print(head(motif_gene_mapping[,c("motif","motif_mlog10P", "gene_score", "cor_mlog10P")], 10))
```

**GATA family**
```{r select-GATA-genes}
GATA_genes <- grep("^GATA\\d*$", TF_genes, ignore.case=T, value=T)
```

Plot GATA motifs in topic 5
```{r plot-GATA-logos-1, fig.width=5, fig.height=2, warning=FALSE, message=FALSE}
k = 5
selected_motifs <- rownames(motif_res$motifs)[match(toupper(GATA_genes), toupper(motif_names))]
motif.dir <- paste0(homer.dir, "/homer_result_topic_", k, "/knownResults/")

for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = "both")
}

```

* Plot motif enrichment (-log10 p-value) and gene scores
```{r motif-gene-plot-1, fig.width=8, fig.height=6, warning=FALSE}
plots <- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = "-log10(p-value)", 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)
```

* Plot motif enrichment (z-score) and gene scores
```{r motif-gene-plot-2, fig.width=8, fig.height=6, warning=FALSE}

plots <- create_motif_gene_scatterplot(motif_res, gene_scores, 
                                       motif_names, gene_names, 
                                       selected_genes = GATA_genes,
                                       y = "z-score", 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)

```
