---
title: "Motif analysis using topic modeling and DA results (v2) for Cusanovich *et al* (2018) scATAC-seq data"
author: Kaixuan Luo
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: false
---

Here we perform TF motif analysis
for the Cusanovich *et al* (2018) scATAC-seq result inferred 
from the multinomial topic model with $k = 13$.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = FALSE,results = "hold",
                      fig.align = "center",dpi = 120)
```

## Load packages and some functions used in this analysis
```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)
library(cowplot)
library(plotly)
library(htmlwidgets)
library(DT)
library(reshape2)
library(Logolas)
library(grid)
source("code/motif_analysis.R")
source("code/plots.R")
```


## Load data and topic model results
Load the data and the $k = 13$ Poisson NMF fit results.
```{r load-data}
data.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/"
load(file.path(data.dir, "Cusanovich_2018.RData"))
cat(sprintf("%d x %d counts matrix.\n",nrow(counts),ncol(counts)))
rm(counts)
```

```{r load-fit}
fit.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018"
fit <- readRDS(file.path(fit.dir, "/fit-Cusanovich2018-scd-ex-k=13.rds"))$fit
fit <- poisson2multinom(fit)
```

## Visualize by Structure plot grouped by tissues
```{r structure-plot-tissue, fig.width=7, fig.height=2.5, message=FALSE}
set.seed(10)
colors_topics <- c("#a6cee3","#1f78b4","#b2df8a","#33a02c","#fb9a99","#e31a1c",
                   "#fdbf6f","#ff7f00","#cab2d6","#6a3d9a","#ffff99","#b15928",
                   "gray")
rows <- sample(nrow(fit$L),4000)
samples$tissue <- as.factor(samples$tissue)

p.structure <- structure_plot(select(fit,loadings = rows),
                     grouping = samples[rows, "tissue"], n = Inf,gap = 40,
                     perplexity = 50,colors = colors_topics,
                     num_threads = 4,verbose = FALSE)

print(p.structure)
```

## Differential accessbility analysis of the ATAC-seq regions for the topics

Load results from differential accessbility analysis for the topics
```{r load-DA-res}
out.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/output/Cusanovich_2018/postfit_v2"
cat(sprintf("Load results from %s \n", out.dir))
DA_res <- readRDS(file.path(out.dir, paste0("DAanalysis-Cusanovich2018-k=13/DA_regions_topics_10000iters.rds")))
```

**Volcano plot of the regions**

Topic 1 and topic 4 examples
```{r region-volcano-plot, fig.width=8, fig.height=3}
p.volcano.1 <- volcano_plot(DA_res,k = 1, labels = rep("",nrow(DA_res$z)))

p.volcano.4 <- volcano_plot(DA_res,k = 4, labels = rep("",nrow(DA_res$z)))

plot_grid(p.volcano.1, p.volcano.4)
```


## Motif enrichment analysis using [HOMER](http://homer.ucsd.edu/homer/ngs/peakMotifs.html)

* Details about HOMER motif analysis:
- http://homer.ucsd.edu/homer/motif/
- http://homer.ucsd.edu/homer/ngs/peakMotifs.html

* Motif enrichment result using regions with z-score above 99% quantile.

Compile Homer results across topics
```{r compile-homer-motif-results}
homer.dir <- paste0(out.dir, "/motifanalysis-Cusanovich2018-k=13-quantile/HOMER")
cat(sprintf("Directory of motif analysis result: %s \n", homer.dir))
homer_res_topics <- readRDS(file.path(homer.dir, "/homer_knownResults.rds"))
selected_regions <- readRDS(file.path(homer.dir, "/selected_regions.rds"))

# Compile Homer results (pvalue and ranking) across topics
motif_res <- compile_homer_motif_res(homer_res_topics)
saveRDS(motif_res, paste0(homer.dir, "/homer_motif_enrichment_results.rds"))
cat("compiled homer motif results are saved in", paste0(homer.dir, "/homer_motif_enrichment_results.rds"))

motif_table <- data.frame(motif = gsub("/.*", "", rownames(motif_res$mlog10P)),
                          round(motif_res$mlog10P,2))
DT::datatable(motif_table, rownames = F, caption = "Motif enrichment (-log10P)")
```


### Top 10 motifs in each topic
```{r top-motifs}
cat("Number of regions selected for each topic: \n")
print(mapply(nrow, selected_regions[1:(length(selected_regions)-1)]))

colnames_homer <- c("motif_name", "consensus", "P", "log10P", "Padj",  "num_target", "percent_target", "num_bg", "percent_bg")

top_motifs <- data.frame(matrix(nrow=10, ncol = length(homer_res_topics)))
colnames(top_motifs) <- names(homer_res_topics)
for (k in 1:length(homer_res_topics)){
  homer_res <- homer_res_topics[[k]]
  colnames(homer_res) <- colnames_homer
  homer_res <- homer_res %>% separate(motif_name, c("motif", "origin", "database"), "/")
  top_motifs[,k] <- head(homer_res$motif, 10)
}

DT::datatable(data.frame(rank = 1:10, top_motifs), rownames = F, caption = "Top 10 motifs enriched in each topic.")

```

### Heatmap of motif enrichment across topics

Heatmap of motif enrichment -log10(p-value). 
```{r motif-heatmap-pvalue, fig.width=7, fig.height=12}
create_motif_enrichment_heatmap(motif_res, enrichment = "-log10(p-value)", 
                                cluster_motifs = TRUE, cluster_topics = TRUE, motif_filter = 10, horizontal = FALSE,
                                enrichment_range = c(0,100), method_cluster = "average", font.size.motifs = 4, font.size.topics = 9)

```

### Scatterplots of motif enrichment

Plot motif enrichment (-log10 p-value) and the ranking
```{r motif-enrichment-ranking-plot-1, fig.width=10, fig.height=7, warning=FALSE}
# Plot enrichment (-log10 p-value) and ranking of the motifs

plots <- vector("list", ncol(motif_res$mlog10P))
names(plots) <- colnames(motif_res$mlog10P)

for( i in 1:length(plots)){
  plots[[i]] <- create_motif_enrichment_ranking_plot(motif_res, k = i, 
                                                     max.overlaps = 20, subsample = FALSE)
}

do.call(plot_grid,plots)
```

**Topic 1 example**
```{r motif-enrichment-ranking-plot-2, fig.width=5, fig.height=4, warning=FALSE}
print(plots[[1]])
```

### Motif enrichment vs gene score

Load pre-computed gene scores
```{r load-gene-scores}
gene.dir <- paste0(out.dir, "/geneanalysis-Cusanovich2018-k=13-TSS-absZ-l2")
cat(sprintf("Directory of gene analysis result: %s \n", gene.dir))
genescore_res <- readRDS(file.path(gene.dir, "genescore_result.rds"))

```

Get TF genes 
```{r get-TF-genes}
motif_names <- motif_res$motifs$motif
gene_names <- genescore_res$genes$SYMBOL
common_genes <- intersect(toupper(motif_names), toupper(gene_names))
cat(sprintf("%s TF genes mapped between motif names and gene symbol. \n", length(common_genes)))

motif_gene_table <- data.frame(motif = motif_names[match(common_genes, toupper(motif_names))], 
                                      gene = gene_names[match(common_genes, toupper(gene_names))])

```

Compute correlation between motif enrichment z-score and gene score:

**Topic 1 example**

* Compute motif enrichment z-scores from the motif enrichment p-values
* Plot motif enrichment (-log10 p-value) and correlation between motif enrichment z-scores and gene scores
* Rank motifs by motif enrichment (-log10 p-value) and correlation between motif enrichment z-score and gene scores

```{r motif-cor-plot-1, fig.width=5, fig.height=4,  warning=FALSE}

motif_gene_mapping <- create_motif_gene_cor_scatterplot(motif_res, genescore_res, motif_gene_table, 
                                                        k = 1, cor.motif = "z-score")

motif_gene_mapping <- motif_gene_mapping[with(motif_gene_mapping, order(motif_mlog10P*cor_zscore, decreasing = T)),]
rownames(motif_gene_mapping) <- 1:nrow(motif_gene_mapping)

cat("Top 10 motifs by motif enrichment (-log10 p-value) and correlation to gene scores: \n")
print(head(motif_gene_mapping[,c("motif","motif_mlog10P", "gene_score", "cor_zscore")], 10))

```

**GATA family**
```{r select-GATA-genes}
motif_names <- motif_res$motifs$motif
gene_names <- genescore_res$genes$SYMBOL
TF_motifs <- sort(unique(grep("^GATA\\d*$", motif_names, ignore.case=T, value=T)))
TF_genes <- sort(unique(grep("^GATA\\d*$", gene_names, ignore.case=T, value=T)))
common_genes <- intersect(toupper(TF_motifs), toupper(TF_genes))

motif_gene_table <- data.frame(motif = TF_motifs[match(common_genes, toupper(TF_motifs))], 
                                      gene = TF_genes[match(common_genes, toupper(TF_genes))])
print(motif_gene_table)
```

Plot GATA motifs in topic 1
```{r plot-GATA-logos, eval=FALSE, fig.width=5, fig.height=1.5,  message=FALSE, warning=FALSE}
# Plot GATA motifs in topic 1
k = 1
motif_order <- order(motif_res$mlog10P[,k], decreasing = T)
motifs <- rownames(motif_res$motifs[motif_order,])
motif_names <- motif_res$motifs[motif_order, "motif"]
selected_motifs <- unique(motifs[match(toupper(motif_gene_table$motif), toupper(motif_names))])
motif.dir <- paste0(homer.dir, "/homer_result_topic_", k, "/knownResults/")

for (i in 1:length(selected_motifs)){
  plot_motif_logo(homer_res_topics, selected_motifs[i], k, motif.dir, type = "both")
}

```

* Plot motif enrichment (-log10 p-value) and gene scores
```{r motif-gene-plot-GATA-1, fig.width=8, fig.height=5, warning=FALSE}
plots <- create_motif_gene_scatterplot(motif_res, genescore_res, 
                                       motif_gene_table,
                                       k = 1, 
                                       y = "-log10(p-value)", 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)
```

* Plot motif enrichment (zscore) and gene scores
```{r motif-gene-plot-GATA-2, fig.width=8, fig.height=5, warning=FALSE}

plots <- create_motif_gene_scatterplot(motif_res, genescore_res, 
                                       motif_gene_table,
                                       k = 1, 
                                       y = "z-score", 
                                       colors = colors_topics,
                                       max.overlaps = 10)

do.call(plot_grid,plots)

```

