---
title: "Process data for Cusanovich 2018 sci-ATAC-seq dataset"
author: "Kaixuan Luo"
date: "2020-08-17"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## About Cusanovich 2018 dataset

Reference: Cusanovich, D., Hill, A., Aghamirzaie, D., Daza, R., Pliner, H., Berletch, J., Filippova, G., Huang, X., Christiansen, L., DeWitt, W., Lee, C., Regalado, S., Read, D., Steemers, F., Disteche, C., Trapnell, C., Shendure, J. (2018). A Single-Cell Atlas of In Vivo Mammalian Chromatin Accessibility Cell 174(5), 1 35. https://dx.doi.org/10.1016/j.cell.2018.06.052

Data were downloaded from the website: https://atlas.gs.washington.edu/mouse-atac/data/. They also provided detail descriptions about these datasets.

RCC directory: `/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/`

The authors included nice tutorials on how to get started with analysis of sci-ATAC-seq data: http://atlas.gs.washington.edu/mouse-atac/docs/

R/python scripts referenced in this tutorial are available on: https://github.com/shendurelab/mouse-atac and downloaded to `/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/code_from_authors/mouse-atac` on RCC. 

Many of the initial steps of processing raw sci-ATAC-seq libraries used for this study are similar to their previous work of sci-ATAC-seq on Drosophila melanogaster embryos at 3 different stages of development. "Cusanovich, D., Reddington, J., Garfield, D. et al. The cis-regulatory dynamics of embryonic development at single-cell resolution. Nature 555, 538â€“542 (2018). https://doi.org/10.1038/nature25981". Code and documentation on processing sequencing data can be found on the [Fly ATAC Github](https://github.com/shendurelab/fly-atac). Documentation on various downstream steps can be found in the [Fly ATAC documentation](http://atlas.gs.washington.edu/fly-atac/).

## Matrix files

* Binarized peak by cell matrix: `atac_matrix.binary.qc_filtered.rds` (only QC filtered cells are included).

* The subset of peaks used as input to TFIDF: `atac_matrix.tfidf.qc_filtered.peaks.txt`.

* TFIDF normalized peak by cell matrix: `atac_matrix.tfidf.qc_filtered.rds`. This dataset has rare peaks filtered out and is then normalized with TFIDF to allow for input to PCA/TSNE (only QC filtered cells are included).

## Metadata
+ `cell_metadata.txt`: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments. Columns in this data: 
    - cell: cell barcode (combined and corrected)
    - tissue: the tissue that this cell originated from
    - tissue_replicate: same as tissue, but each replicate has a unique id
    - cluster: cluster assignment in initial t-SNE
    - subset_cluster: cluster assignment in iterative t-SNE space
    - tsne_1: t-SNE1 coordinate in initial t-SNE
    - tsne_2: t-SNE2 coordinate in initial t-SNE
    - subset_tsne1: t-SNE1 coordinate in iterative t-SNE
    - subset_tsne2: t-SNE2 coordinate in iterative t-SNE
    - id: combined ID for major + iterative cluster assignment
    - cell_label: assigned cell type

```{r}
cell_metadata <- read.table("/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt", header = TRUE, stringsAsFactors = TRUE, sep = "\t")

dim(cell_metadata)

cell_metadata[1:3,]
```

+ `cell_metadata.tissue_freq_filtered.txt`: Same as `cell_metadata.txt`, but removes cells in each tissue belonging to a cell_label that accounts for less than 0.5% of the cells in that tissue. These very low frequency labels are often not cell types expected their respective tissues and could be due to slight imperfections in clustering, for example. Provided matrices would need to be subsetted to match this set of cells if using this metadata.

```{r}
cell_metadata <- read.table("/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.tissue_freq_filtered.txt", header = TRUE, stringsAsFactors = TRUE, sep = "\t")

dim(cell_metadata)

cell_metadata[1:3,]
```

+ `cell_type_assignments.xlsx`: Excel document with three tabs expected cell types, cell type markers, and Cell type assignments that contain a pairs of tissues and expected cell types, a list of positive markers for each cell type, and the table of cell type assignments with extra details about assignment criteria when applicable, respectively. This is meant to document justification for cell type assignments provided in `cell_metadata.txt` above.

+ `peak_promoter_intersections.txt`: Metadata for peaks-intersected TSS pairs in TSV format.

```{r}
peak_promoter_intersections <- read.table("/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/peak_promoter_intersections.txt", header = TRUE, stringsAsFactors = TRUE, sep = "\t")

dim(peak_promoter_intersections)

peak_promoter_intersections[1:3,]
```


## Prepare data for topic modeling

* Binarized peak by cell matrix: `atac_matrix.binary.qc_filtered.rds` (only QC filtered cells are included).
```{r}
library(Matrix)
library(tools)

## atac_matrix.binary.qc_filtered.rds: binarized peak by cell matrix in RDS format.
binarized_matrix <- readRDS("/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/ATAC_matrices/atac_matrix.binary.qc_filtered.rds")
dim(binarized_matrix)
counts <- t(binarized_matrix)

cat(sprintf("Number of samples (cells): %d\n",nrow(counts)))
cat(sprintf("Number of peaks: %d\n",ncol(counts)))
cat(sprintf("Proportion of counts that are non-zero: %0.1f%%.\n",
            100*mean(counts > 0)))

# 100*nnzero(counts)/prod(dim(counts))

```

* `cell_metadata.txt`: Metadata for cells in TSV format, including several features such as TSNE coordinates, cluster assignments, and cell type assignments. 
```{r}
samples <- read.table("/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/metadata/cell_metadata.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")

dim(samples)

cat(sprintf("Number of samples: %d\n",nrow(samples)))

print(samples[1:3,])

```

This study measured single cell chromatin accessibility in 13 different tissues in mice:
```{r}
table(samples$tissue)
```

Cells were labeled into the cell types:
```{r}
table(samples$cell_label)
```

```{r}
data.dir <- "/project2/mstephens/kevinluo/scATACseq-topics/data/Cusanovich_2018/processed_data/"
dir.create(data.dir, showWarnings = FALSE, recursive = TRUE)

saveRDS(counts, file.path(data.dir, "counts_Cusanovich_2018.rds"))

save(list = c("samples","counts"), file = file.path(data.dir, "Cusanovich_2018.RData"))

```
